<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.475">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Steven Roberts">
<meta name="dcterms.date" content="2024-05-14">
<meta name="description" content="to DESeq">

<title>Tumbling Oysters - HiSat</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../../site_libs/bootstrap/bootstrap-dark.min.css" rel="stylesheet" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>

  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<meta name="twitter:title" content="Tumbling Oysters - HiSat">
<meta name="twitter:description" content="to DESeq">
<meta name="twitter:image" content="http://gannet.fish.washington.edu/seashell/snaps/Monosnap_bestblogever_-_main_-_RStudio_2024-05-15_20-23-56.png">
<meta name="twitter:card" content="summary_large_image">
</head>

<body class="nav-fixed fullcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Tumbling Oysters</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../about.html">
 <span class="menu-text">Me Elsewhere</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/sr320"><i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com/sr320"><i class="bi bi-twitter" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
              <div class="quarto-toggle-container">
                  <a href="" class="quarto-color-scheme-toggle nav-link" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
              </div>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <div class="quarto-title-block"><div><h1 class="title">HiSat</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
                  <div>
        <div class="description">
          to DESeq
        </div>
      </div>
                          <div class="quarto-categories">
                <div class="quarto-category">R</div>
                <div class="quarto-category">hisat</div>
              </div>
                  </div>
  </div>
    
  <div class="quarto-title-meta-author">
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-heading">Affiliation</div>
    
      <div class="quarto-title-meta-contents">
      Steven Roberts <a href="https://orcid.org/0000-0001-8302-1138" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a>
    </div>
      <div class="quarto-title-meta-contents">
          <p class="affiliation">
              <a href="https://robertslab.info">
              Professor, UW - School of Aquatic and Fishery Sciences
              </a>
            </p>
        </div>
      </div>

  <div class="quarto-title-meta">

        
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">May 14, 2024</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">




<ul>
<li><a href="#1-run-hisat-on-rna-seq" id="toc-1-run-hisat-on-rna-seq">1 Run HiSat on RNA-seq</a>
<ul>
<li><a href="#11-grab-trimmed-rna-seq-reads" id="toc-11-grab-trimmed-rna-seq-reads">1.1 Grab Trimmed RNA-seq Reads</a></li>
<li><a href="#12-genome" id="toc-12-genome">1.2 Genome</a></li>
<li><a href="#13-hisat" id="toc-13-hisat">1.3 HiSat</a>
<ul>
<li><a href="#131-performance-tuning" id="toc-131-performance-tuning">1.3.1 Performance tuning</a></li>
</ul></li>
<li><a href="#14-convert-to-bam" id="toc-14-convert-to-bam">1.4 convert to bam</a></li>
<li><a href="#15-looking-at-bams" id="toc-15-looking-at-bams">1.5 Looking at Bams</a>
<ul>
<li><a href="#151-an-alternate-faster-differential-expression-analysis-workflow" id="toc-151-an-alternate-faster-differential-expression-analysis-workflow">1.5.1 An alternate, faster differential expression analysis workflow</a></li>
</ul></li>
</ul></li>
<li><a href="#2-stringtie" id="toc-2-stringtie">2 StringTie</a></li>
</ul>
<section id="run-hisat-on-rna-seq" class="level1">
<h1>1 Run HiSat on RNA-seq</h1>
<p>Will end up with 5 sorted bam files.</p>
<section id="grab-trimmed-rna-seq-reads" class="level2">
<h2 class="anchored" data-anchor-id="grab-trimmed-rna-seq-reads">1.1 Grab Trimmed RNA-seq Reads</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> ../data/fastq/ </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>RNA-ACR-140-S1-TP2_R1_001.fastp-trim.20230519.fastq.gz
RNA-ACR-140-S1-TP2_R2_001.fastp-trim.20230519.fastq.gz
RNA-ACR-145-S1-TP2_R1_001.fastp-trim.20230519.fastq.gz
RNA-ACR-145-S1-TP2_R2_001.fastp-trim.20230519.fastq.gz
RNA-ACR-150-S1-TP2_R1_001.fastp-trim.20230519.fastq.gz
RNA-ACR-150-S1-TP2_R2_001.fastp-trim.20230519.fastq.gz
RNA-ACR-173-S1-TP2_R1_001.fastp-trim.20230519.fastq.gz
RNA-ACR-173-S1-TP2_R2_001.fastp-trim.20230519.fastq.gz
RNA-ACR-178-S1-TP2_R1_001.fastp-trim.20230519.fastq.gz
RNA-ACR-178-S1-TP2_R2_001.fastp-trim.20230519.fastq.gz</code></pre>
</section>
<section id="genome" class="level2">
<h2 class="anchored" data-anchor-id="genome">1.2 Genome</h2>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> ../data/GCF_013753865.1_Amil_v2.1_genomic.fna</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>../data/GCF_013753865.1_Amil_v2.1_genomic.fna</code></pre>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> ../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gtf</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="fu">wc</span> <span class="at">-l</span> ../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gtf</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="fu">grep</span> <span class="at">-v</span> <span class="st">'^#'</span> ../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gtf <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-f3</span> <span class="kw">|</span> <span class="fu">sort</span> <span class="kw">|</span> <span class="fu">uniq</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="fu">grep</span> <span class="at">-c</span> <span class="st">"transcript"</span> ../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gtf</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="fu">grep</span> <span class="at">-c</span> <span class="st">"gene"</span> ../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gtf</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>#gtf-version 2.2
#!genome-build Amil_v2.1
#!genome-build-accession NCBI_Assembly:GCF_013753865.1
#!annotation-source NCBI Acropora millepora Annotation Release 101
NC_058066.1 Gnomon  gene    1962    23221   .   -   .   gene_id "LOC114963522"; transcript_id ""; db_xref "GeneID:114963522"; gbkey "Gene"; gene "LOC114963522"; gene_biotype "lncRNA"; 
NC_058066.1 Gnomon  transcript  1962    23221   .   -   .   gene_id "LOC114963522"; transcript_id "XR_003825913.2"; db_xref "GeneID:114963522"; gbkey "ncRNA"; gene "LOC114963522"; model_evidence "Supporting evidence includes similarity to: 100% coverage of the annotated genomic feature by RNAseq alignments, including 2 samples with support for all annotated introns"; product "uncharacterized LOC114963522"; transcript_biotype "lnc_RNA"; 
NC_058066.1 Gnomon  exon    23085   23221   .   -   .   gene_id "LOC114963522"; transcript_id "XR_003825913.2"; db_xref "GeneID:114963522"; gene "LOC114963522"; model_evidence "Supporting evidence includes similarity to: 100% coverage of the annotated genomic feature by RNAseq alignments, including 2 samples with support for all annotated introns"; product "uncharacterized LOC114963522"; transcript_biotype "lnc_RNA"; exon_number "1"; 
NC_058066.1 Gnomon  exon    21001   21093   .   -   .   gene_id "LOC114963522"; transcript_id "XR_003825913.2"; db_xref "GeneID:114963522"; gene "LOC114963522"; model_evidence "Supporting evidence includes similarity to: 100% coverage of the annotated genomic feature by RNAseq alignments, including 2 samples with support for all annotated introns"; product "uncharacterized LOC114963522"; transcript_biotype "lnc_RNA"; exon_number "2"; 
NC_058066.1 Gnomon  exon    18711   18775   .   -   .   gene_id "LOC114963522"; transcript_id "XR_003825913.2"; db_xref "GeneID:114963522"; gene "LOC114963522"; model_evidence "Supporting evidence includes similarity to: 100% coverage of the annotated genomic feature by RNAseq alignments, including 2 samples with support for all annotated introns"; product "uncharacterized LOC114963522"; transcript_biotype "lnc_RNA"; exon_number "3"; 
NC_058066.1 Gnomon  exon    1962    2119    .   -   .   gene_id "LOC114963522"; transcript_id "XR_003825913.2"; db_xref "GeneID:114963522"; gene "LOC114963522"; model_evidence "Supporting evidence includes similarity to: 100% coverage of the annotated genomic feature by RNAseq alignments, including 2 samples with support for all annotated introns"; product "uncharacterized LOC114963522"; transcript_biotype "lnc_RNA"; exon_number "4"; 
874259 ../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gtf
CDS
exon
gene
start_codon
stop_codon
transcript
874254
874254</code></pre>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> ../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gff</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="fu">wc</span> <span class="at">-l</span> ../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gff</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="fu">grep</span> <span class="at">-v</span> <span class="st">'^#'</span> ../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gff <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-f3</span> <span class="kw">|</span> <span class="fu">sort</span> <span class="kw">|</span> <span class="fu">uniq</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="fu">grep</span> <span class="at">-c</span> <span class="st">"transcript"</span> ../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gff</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="fu">grep</span> <span class="at">-c</span> <span class="st">"gene"</span> ../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gff</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>##gff-version 3
#!gff-spec-version 1.21
#!processor NCBI annotwriter
#!genome-build Amil_v2.1
#!genome-build-accession NCBI_Assembly:GCF_013753865.1
#!annotation-source NCBI Acropora millepora Annotation Release 101
##sequence-region NC_058066.1 1 39361238
##species https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id=45264
NC_058066.1 RefSeq  region  1   39361238    .   +   .   ID=NC_058066.1:1..39361238;Dbxref=taxon:45264;Name=1;chromosome=1;collection-date=2017;country=Indonesia;gbkey=Src;genome=chromosome;isolate=JS-1;isolation-source=Whole tissue;mol_type=genomic DNA;tissue-type=Adult tissue
NC_058066.1 Gnomon  gene    1962    23221   .   -   .   ID=gene-LOC114963522;Dbxref=GeneID:114963522;Name=LOC114963522;gbkey=Gene;gene=LOC114963522;gene_biotype=lncRNA
828216 ../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gff
cDNA_match
CDS
exon
gene
guide_RNA
lnc_RNA
mRNA
pseudogene
region
rRNA
snoRNA
snRNA
transcript
tRNA
431975
803260</code></pre>
</section>
<section id="hisat" class="level2">
<h2 class="anchored" data-anchor-id="hisat">1.3 HiSat</h2>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="ex">/home/shared/hisat2-2.2.1/hisat2_extract_exons.py</span> <span class="dt">\</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gtf <span class="dt">\</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span> ../output/15-Apul-hisat/m_exon.tab</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> ../output/15-Apul-hisat/m_exon.tab</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="fu">wc</span> <span class="at">-l</span> ../output/15-Apul-hisat/m_exon.tab</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>NC_058066.1 1961    2118    -
NC_058066.1 15360   15663   +
NC_058066.1 18710   18774   -
NC_058066.1 21000   21092   -
NC_058066.1 22158   22442   +
NC_058066.1 23084   23220   -
NC_058066.1 24770   25066   +
NC_058066.1 26652   26912   +
NC_058066.1 27071   27358   +
NC_058066.1 27658   27951   +
228010 ../output/15-Apul-hisat/m_exon.tab</code></pre>
<div class="sourceCode" id="cb12"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="ex">/home/shared/hisat2-2.2.1/hisat2_extract_splice_sites.py</span> <span class="dt">\</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gtf <span class="dt">\</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span> ../output/15-Apul-hisat/m_splice_sites.tab</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb13"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> ../output/15-Apul-hisat/m_splice_sites.tab</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>NC_058066.1 2118    18710   -
NC_058066.1 15663   22158   +
NC_058066.1 18774   21000   -
NC_058066.1 21092   23084   -
NC_058066.1 22442   24770   +
NC_058066.1 25066   26652   +
NC_058066.1 26912   27071   +
NC_058066.1 27358   27658   +
NC_058066.1 27951   28247   +
NC_058066.1 28534   29196   +</code></pre>
<div class="sourceCode" id="cb15"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="ex">/home/shared/hisat2-2.2.1/hisat2-build</span> <span class="dt">\</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>../data/GCF_013753865.1_Amil_v2.1_genomic.fna <span class="dt">\</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>../output/15-Apul-hisat/GCF_013753865.1_Amil_v2.1.index <span class="dt">\</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>--exon ../output/15-Apul-hisat/m_exon.tab <span class="dt">\</span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>--ss ../output/15-Apul-hisat/m_splice_sites.tab <span class="dt">\</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>-p 40 <span class="dt">\</span></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gtf <span class="dt">\</span></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>2<span class="op">&gt;</span> ../output/15-Apul-hisat/hisat2-build_stats.txt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb16"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">tail</span> ../output/15-Apul-hisat/hisat2-build_stats.txt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>    sideSz: 128
    sideGbwtSz: 104
    sideGbwtLen: 208
    numSides: 2299750
    numLines: 2299750
    gbwtTotLen: 294368000
    gbwtTotSz: 294368000
    reverse: 0
    linearFM: No
Total time for call to driver() for forward index: 00:07:58</code></pre>
<section id="performance-tuning" class="level3">
<h3 class="anchored" data-anchor-id="performance-tuning">1.3.1 Performance tuning</h3>
<p>If your computer has multiple processors/cores, use -p The -p option causes HISAT2 to launch a specified number of parallel search threads. Each thread runs on a different processor/core and all threads find alignments in parallel, increasing alignment throughput by approximately a multiple of the number of threads (though in practice, speedup is somewhat worse than linear).</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">find</span> ../data/fastq/<span class="pp">*</span>R2_001.fastp-trim.20230519.fastq.gz <span class="dt">\</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span> <span class="fu">xargs</span> basename <span class="at">-s</span> <span class="at">-S1-TP2_R2_001.fastp-trim.20230519.fastq.gz</span> <span class="kw">|</span> <span class="fu">xargs</span> <span class="at">-I{}</span> <span class="dt">\</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>/home/shared/hisat2-2.2.1/hisat2 <span class="dt">\</span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>-x ../output/15-Apul-hisat/GCF_013753865.1_Amil_v2.1.index <span class="dt">\</span></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>--dta <span class="dt">\</span></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>-p 20 <span class="dt">\</span></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>-1 ../data/fastq/{}-S1-TP2_R1_001.fastp-trim.20230519.fastq.gz <span class="dt">\</span></span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>-2 ../data/fastq/{}-S1-TP2_R2_001.fastp-trim.20230519.fastq.gz <span class="dt">\</span></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>-S ../output/15-Apul-hisat/{}.sam <span class="dt">\</span></span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>2<span class="op">&gt;</span> ../output/15-Apul-hisat/hisat.out</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb19"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> ../output/15-Apul-hisat/hisat.out</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>47710408 reads; of these:
  47710408 (100.00%) were paired; of these:
    27825914 (58.32%) aligned concordantly 0 times
    18559152 (38.90%) aligned concordantly exactly 1 time
    1325342 (2.78%) aligned concordantly &gt;1 times
    ----
    27825914 pairs aligned concordantly 0 times; of these:
      1201636 (4.32%) aligned discordantly 1 time
    ----
    26624278 pairs aligned 0 times concordantly or discordantly; of these:
      53248556 mates make up the pairs; of these:
        42533390 (79.88%) aligned 0 times
        9692346 (18.20%) aligned exactly 1 time
        1022820 (1.92%) aligned &gt;1 times
55.43% overall alignment rate
42864294 reads; of these:
  42864294 (100.00%) were paired; of these:
    24356414 (56.82%) aligned concordantly 0 times
    17135863 (39.98%) aligned concordantly exactly 1 time
    1372017 (3.20%) aligned concordantly &gt;1 times
    ----
    24356414 pairs aligned concordantly 0 times; of these:
      1142678 (4.69%) aligned discordantly 1 time
    ----
    23213736 pairs aligned 0 times concordantly or discordantly; of these:
      46427472 mates make up the pairs; of these:
        36609836 (78.85%) aligned 0 times
        8675847 (18.69%) aligned exactly 1 time
        1141789 (2.46%) aligned &gt;1 times
57.30% overall alignment rate
43712298 reads; of these:
  43712298 (100.00%) were paired; of these:
    30967520 (70.84%) aligned concordantly 0 times
    11666917 (26.69%) aligned concordantly exactly 1 time
    1077861 (2.47%) aligned concordantly &gt;1 times
    ----
    30967520 pairs aligned concordantly 0 times; of these:
      791338 (2.56%) aligned discordantly 1 time
    ----
    30176182 pairs aligned 0 times concordantly or discordantly; of these:
      60352364 mates make up the pairs; of these:
        52153613 (86.42%) aligned 0 times
        7011301 (11.62%) aligned exactly 1 time
        1187450 (1.97%) aligned &gt;1 times
40.34% overall alignment rate
47501524 reads; of these:
  47501524 (100.00%) were paired; of these:
    28603496 (60.22%) aligned concordantly 0 times
    17627744 (37.11%) aligned concordantly exactly 1 time
    1270284 (2.67%) aligned concordantly &gt;1 times
    ----
    28603496 pairs aligned concordantly 0 times; of these:
      1201709 (4.20%) aligned discordantly 1 time
    ----
    27401787 pairs aligned 0 times concordantly or discordantly; of these:
      54803574 mates make up the pairs; of these:
        44045470 (80.37%) aligned 0 times
        9614980 (17.54%) aligned exactly 1 time
        1143124 (2.09%) aligned &gt;1 times
53.64% overall alignment rate
42677752 reads; of these:
  42677752 (100.00%) were paired; of these:
    26422177 (61.91%) aligned concordantly 0 times
    15005422 (35.16%) aligned concordantly exactly 1 time
    1250153 (2.93%) aligned concordantly &gt;1 times
    ----
    26422177 pairs aligned concordantly 0 times; of these:
      1009838 (3.82%) aligned discordantly 1 time
    ----
    25412339 pairs aligned 0 times concordantly or discordantly; of these:
      50824678 mates make up the pairs; of these:
        39698309 (78.11%) aligned 0 times
        9739502 (19.16%) aligned exactly 1 time
        1386867 (2.73%) aligned &gt;1 times
53.49% overall alignment rate</code></pre>
</section>
</section>
<section id="convert-to-bam" class="level2">
<h2 class="anchored" data-anchor-id="convert-to-bam">1.4 convert to bam</h2>
<div class="sourceCode" id="cb21"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> samfile <span class="kw">in</span> ../output/15-Apul-hisat/<span class="pp">*</span>.sam<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>  <span class="va">bamfile</span><span class="op">=</span><span class="st">"</span><span class="va">${samfile</span><span class="op">%</span>.sam<span class="va">}</span><span class="st">.bam"</span></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>  <span class="va">sorted_bamfile</span><span class="op">=</span><span class="st">"</span><span class="va">${samfile</span><span class="op">%</span>.sam<span class="va">}</span><span class="st">.sorted.bam"</span></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>  <span class="ex">/home/shared/samtools-1.12/samtools</span> view <span class="at">-bS</span> <span class="at">-@</span> 20 <span class="st">"</span><span class="va">$samfile</span><span class="st">"</span> <span class="op">&gt;</span> <span class="st">"</span><span class="va">$bamfile</span><span class="st">"</span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>  <span class="ex">/home/shared/samtools-1.12/samtools</span> sort <span class="at">-@</span> 20 <span class="st">"</span><span class="va">$bamfile</span><span class="st">"</span> <span class="at">-o</span> <span class="st">"</span><span class="va">$sorted_bamfile</span><span class="st">"</span></span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>  <span class="ex">/home/shared/samtools-1.12/samtools</span> index <span class="at">-@</span> 20 <span class="st">"</span><span class="va">$sorted_bamfile</span><span class="st">"</span></span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="looking-at-bams" class="level2">
<h2 class="anchored" data-anchor-id="looking-at-bams">1.5 Looking at Bams</h2>
<p><span class="math display">\[igv\]</span></p>
<hr>
<section id="differential-expression-analysis" class="level4">
<h4 class="anchored" data-anchor-id="differential-expression-analysis">1.5.0.1 <strong>Differential expression analysis</strong></h4>
<p>Together with <a href="http://ccb.jhu.edu/software/hisat2">HISAT</a> and <a href="http://biorxiv.org/content/early/2014/09/05/003665">Ballgown</a>, StringTie can be used for estimating differential expression across multiple RNA-Seq samples and generating plots and differential expression tables as described in our <a href="http://www.nature.com/nprot/journal/v11/n9/full/nprot.2016.095.html">protocol paper</a>.<img src="https://ccb.jhu.edu/software/stringtie/DE_pipeline.png" alt="DE workflow"><br>
Our recommended workflow includes the following steps:</p>
<ol type="1">
<li><p>for each RNA-Seq sample, map the reads to the genome with <a href="http://ccb.jhu.edu/software/hisat2">HISAT2</a> using the –dta option. It is highly recommended to use the reference annotation information when mapping the reads, which can be either embedded in the genome index (built with the –ss and –exon options, see <a href="http://ccb.jhu.edu/software/hisat2/manual.shtml#the-hisat2-build-indexer">HISAT2 manual</a>), or provided separately at run time (using the –known-splicesite-infile option of <a href="http://ccb.jhu.edu/software/hisat2/manual.shtml#running-hisat2">HISAT2</a>). The SAM output of each HISAT2 run must be sorted and converted to BAM using samtools as explained above.</p></li>
<li><p>for each RNA-Seq sample, run StringTie to assemble the read alignments obtained in the previous step; it is recommended to run StringTie with the -G option if the reference annotation is available.</p></li>
<li><p>run StringTie with <strong>–merge</strong> in order to generate a non-redundant set of transcripts observed in any of the RNA-Seq samples assembled previously. The stringtie –merge mode takes as input a list of all the assembled transcripts files (in GTF format) previously obtained for each sample, as well as a reference annotation file (-G option) if available.</p></li>
<li><p>for each RNA-Seq sample, run StringTie using the -B/-b and -e options in order to estimate transcript abundances and generate read coverage tables for Ballgown. The -e option is not required but recommended for this run in order to produce more accurate abundance estimations of the input transcripts. Each StringTie run in this step will take as input the sorted read alignments (BAM file) obtained in step 1 for the corresponding sample and the -G option with the merged transcripts (GTF file) generated by stringtie –merge in step 3. Please note that this is the only case where the -G option is not used with a reference annotation, but with the global, merged set of transcripts as observed across all samples. (This step is the equivalent of the <em>Tablemaker</em> step described in the original <a href="https://github.com/alyssafrazee/ballgown">Ballgown pipeline</a>.)</p></li>
<li><p>Ballgown can now be used to load the coverage tables generated in the previous step and perform various statistical analyses for differential expression, generate plots etc.</p></li>
</ol>
<!--# using faster -->
</section>
<section id="an-alternate-faster-differential-expression-analysis-workflow" class="level3">
<h3 class="anchored" data-anchor-id="an-alternate-faster-differential-expression-analysis-workflow">1.5.1 An alternate, faster differential expression analysis workflow</h3>
<p>can be pursued if there is no interest in novel isoforms (i.e.&nbsp;assembled transcripts present in the samples but missing from the reference annotation), or if only a well known set of transcripts of interest are targeted by the analysis. This simplified protocol has only 3 steps (depicted below) as it bypasses the individual assembly of each RNA-Seq sample and the <em>“transcript merge”</em> step.<img src="https://ccb.jhu.edu/software/stringtie/DE_pipeline_refonly.png" alt="DE workflow">This simplified workflow attempts to directly estimate and analyze the expression of a known set of transcripts as given in the <em>reference annotation</em> file.</p>
<p>The R package <a href="https://bioconductor.org/packages/devel/bioc/html/IsoformSwitchAnalyzeR.html"><strong>IsoformSwitchAnalyzeR</strong></a> can be used to assign gene names to transcripts assembled by StringTie, which can be particularly helpful in cases where StringTie could not perform this assignment unambigiously.<br>
The <code>importIsoformExpression() + importRdata()</code> function of the package can be used to import the expression and annotation data into R. During this import the package will attempt to clean up and recover isoform annotations where possible. From the resulting <code>switchAnalyzeRlist</code> object, <em>IsoformSwitchAnalyzeR</em> can detect isoform switches along with predicted functional consequences. The <code>extractGeneExpression()</code> function can be used to get a gene expression (read count) matrix for analysis with other tools.<br>
More information and code examples can be found <a href="https://bioconductor.org/packages/devel/bioc/vignettes/IsoformSwitchAnalyzeR/inst/doc/IsoformSwitchAnalyzeR.html#examples-visualizations">here</a>.</p>
</section>
</section>
</section>
<section id="stringtie" class="level1">
<h1>2 StringTie</h1>
<p>StringTie uses the sorted BAM files to assemble transcripts for each sample, outputting them as GTF files. And then merges all individual GTF assemblies into a single merged GTF file. This step extracts transcript information and merges GTFs from all samples–an important step in creating a canonical list of across all samples included in the pipeline.</p>
<table class="table">
<colgroup>
<col style="width: 4%">
<col style="width: 95%">
</colgroup>
<tbody>
<tr class="odd">
<td style="text-align: left;">-b &lt;path&gt;</td>
<td style="text-align: left;">Just like -B this option enables the output of *.ctab files for Ballgown, but these files will be created in the provided directory &lt;path&gt; instead of the directory specified by the -o option. Note: adding the -e option is recommended with the -B/-b options, unless novel transcripts are still wanted in the StringTie GTF output.</td>
</tr>
</tbody>
</table>
<table class="table">
<tbody>
<tr class="odd">
<td style="text-align: left;"><br>
-e</td>
<td style="text-align: left;">this option directs StringTie to operate in <a href="https://ccb.jhu.edu/software/stringtie/index.shtml?t=manual#emode">expression estimation mode</a>; this limits the processing of read alignments to estimating the coverage of the transcripts given with the -G option (hence this option requires -G).</td>
</tr>
</tbody>
</table>
<table class="table">
<tbody>
<tr class="odd">
<td style="text-align: left;">-G &lt;ref_ann.gff&gt;</td>
<td style="text-align: left;">Use a <a href="https://ccb.jhu.edu/software/stringtie/index.shtml?t=manual#refann">reference annotation file</a> (in <a href="https://ccb.jhu.edu/software/stringtie/gff.shtml">GTF or GFF3 format</a>) to guide the assembly process. The output will include expressed reference transcripts as well as any novel transcripts that are assembled. This option is required by options -B, -b, -e, -C (see below).</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb22"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">find</span> ../output/15-Apul-hisat/<span class="pp">*</span>sorted.bam <span class="dt">\</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span> <span class="fu">xargs</span> basename <span class="at">-s</span> .sorted.bam <span class="kw">|</span> <span class="fu">xargs</span> <span class="at">-I{}</span> <span class="dt">\</span></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>/home/shared/stringtie-2.2.1.Linux_x86_64/stringtie <span class="dt">\</span></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>-p 20 <span class="dt">\</span></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>-eB <span class="dt">\</span></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>-G ../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gff <span class="dt">\</span></span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>-o ../output/15-Apul-hisat/{}.gtf <span class="dt">\</span></span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>../output/15-Apul-hisat/{}.sorted.bam</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb23"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wc</span> <span class="at">-l</span> ../output/15-Apul-hisat/RNA<span class="pp">*</span>.gtf</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> ../output/15-Apul-hisat/RNA<span class="pp">*</span>.gtf</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a><span class="co">#head ../output/15-Apul-hisat/RNA*.gtf</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>   445208 ../output/15-Apul-hisat/RNA-ACR-140.gtf
   445208 ../output/15-Apul-hisat/RNA-ACR-145.gtf
   445208 ../output/15-Apul-hisat/RNA-ACR-150.gtf
   445208 ../output/15-Apul-hisat/RNA-ACR-173.gtf
   445208 ../output/15-Apul-hisat/RNA-ACR-178.gtf
  2226040 total
../output/15-Apul-hisat/RNA-ACR-140.gtf
../output/15-Apul-hisat/RNA-ACR-145.gtf
../output/15-Apul-hisat/RNA-ACR-150.gtf
../output/15-Apul-hisat/RNA-ACR-173.gtf
../output/15-Apul-hisat/RNA-ACR-178.gtf</code></pre>
<p>The R package&nbsp;<a href="https://bioconductor.org/packages/devel/bioc/html/IsoformSwitchAnalyzeR.html"><strong>IsoformSwitchAnalyzeR</strong></a>&nbsp;can be used to assign gene names to transcripts assembled by StringTie, which can be particularly helpful in cases where StringTie could not perform this assignment unambigiously.<br>
The&nbsp;<code>importIsoformExpression() + importRdata()</code>&nbsp;function of the package can be used to import the expression and annotation data into R. During this import the package will attempt to clean up and recover isoform annotations where possible. From the resulting&nbsp;<code>switchAnalyzeRlist</code>&nbsp;object,&nbsp;<em>IsoformSwitchAnalyzeR</em>&nbsp;can detect isoform switches along with predicted functional consequences. The&nbsp;<code>extractGeneExpression()</code>&nbsp;function can be used to get a gene expression (read count) matrix for analysis with other tools.<br>
More information and code examples can be found&nbsp;<a href="https://bioconductor.org/packages/devel/bioc/vignettes/IsoformSwitchAnalyzeR/inst/doc/IsoformSwitchAnalyzeR.html#examples-visualizations">here</a>.</p>
<hr>
<section id="using-stringtie-with-deseq2-and-edger" class="level3">
<h3 class="anchored" data-anchor-id="using-stringtie-with-deseq2-and-edger">2.0.1&nbsp;<strong>Using StringTie with DESeq2 and edgeR</strong></h3>
<p><a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html">DESeq2</a>&nbsp;and&nbsp;<a href="https://bioconductor.org/packages/release/bioc/html/edgeR.html">edgeR</a>&nbsp;are two popular&nbsp;<a href="https://www.bioconductor.org/">Bioconductor</a>&nbsp;packages for analyzing differential expression, which take as input a matrix of read counts mapped to particular genomic features (e.g., genes). We provide a Python script (<a href="https://ccb.jhu.edu/software/stringtie/dl/prepDE.py">prepDE.py</a>, or the Python 3 version:<a href="https://ccb.jhu.edu/software/stringtie/dl/prepDE.py3">prepDE.py3</a>&nbsp;) that can be used to extract this read count information directly from the files generated by StringTie (run with the -e parameter).</p>
<p><em>prepDE.py</em>&nbsp;derives hypothetical read counts for each transcript from the coverage values estimated by StringTie for each transcript, by using this simple formula:&nbsp;<em>reads_per_transcript = coverage * transcript_len / read_len</em></p>
<p>There are two ways to provide input to the prepDE.py script:</p>
<ul>
<li><p>one option is to provide a path to a directory containing all sample sub-directories, with the same structure as the&nbsp;<em>ballgown</em>directory in the&nbsp;<a href="http://www.nature.com/nprot/journal/v11/n9/full/nprot.2016.095.html">StringTie protocol paper</a>&nbsp;in preparation for Ballgown. By default (no -i option), the script is going to look in the current directory for all sub-directories having .gtf files in them, as in this example:</p>
<pre><code>    ./sample1/sample1.gtf
    ./sample2/sample2.gtf
    ./sample3/sample3.gtf</code></pre></li>
<li><p>Alternatively, one can provide a text file listing sample IDs and their respective paths (<a href="https://ccb.jhu.edu/software/stringtie/dl/sample_lst.txt">sample_lst.txt</a>).</p></li>
</ul>
<p>Usage: prepDE.py&nbsp;options<br>
generates two CSV files containing the count matrices for genes and transcripts, using the coverage values found in the output of stringtie -e</p>
<p>Options:</p>
<hr>
<p>-h, –help show this help message and exit</p>
<p>-i INPUT, a folder containing all sample sub-directories, or a –input=INPUT, text file with sample ID and path to its GTF file on –in=INPUT each line&nbsp;default:.</p>
<p>-g G where to output the gene count matrix&nbsp;default:genecountmatrix.csv</p>
<p>-t T where to output the transcript count matrix&nbsp;default:transcriptcountmatrix.csv</p>
<p>-l LENGTH, the average read length&nbsp;default:75–length=LENGTH</p>
<p>-p PATTERN, a regular expression that selects the sample –pattern=PATTERN subdirectories</p>
<p>-c, –cluster whether to cluster genes that overlap with different gene IDs, ignoring ones with geneID pattern (see below)</p>
<p>-s STRING, if a different prefix is used for geneIDs assigned by –string=STRING StringTie&nbsp;default:MSTRG</p>
<p>-k KEY, –key=KEY if clustering, what prefix to use for geneIDs assigned by this script&nbsp;default:prepG</p>
<p>–legend=LEGEND if clustering, where to output the legend file mapping transcripts to assigned geneIDs&nbsp;default:legend.csv——————– ——————————————————</p>
<p>These count matrices (CSV files) can then be imported into R for use by DESeq2 and edgeR (using the&nbsp;<em>DESeqDataSetFromMatrix</em>&nbsp;and&nbsp;<em>DGEList</em>&nbsp;functions, respectively).</p>
<section id="protocol-using-stringtie-with-deseq2" class="level5">
<h5 class="anchored" data-anchor-id="protocol-using-stringtie-with-deseq2">2.0.1.0.1&nbsp;<strong>Protocol: Using StringTie with DESeq2</strong></h5>
<p>Given a list of GTFs, which were re-estimated upon merging, users can follow the below protocol to use DESeq2 for differential expression analysis. To generate GTFs from raw reads follow the&nbsp;<a href="http://www.nature.com/nprot/journal/v11/n9/full/nprot.2016.095.html">StringTie protocol paper</a>&nbsp;(up to the Ballgown step).</p>
<p>As described above, prepDE.py either accepts a .txt (<a href="https://ccb.jhu.edu/software/stringtie/dl/sample_lst.txt">sample_lst.txt</a>) file listing the sample IDs and the GTFs’ paths or by default expects a ballgown directory produced by StringTie run with the -B option. The following steps leads us through generating count matrices for genes and transcripts, importing this data into DESeq2, and conducting some basic analysis.</p>
<ol type="1">
<li><p>Generate count matrices using prepDE.py</p>
<ol type="1">
<li><p>Assuming default ballgown directory structure produced by StringTie<br>
$ python prepDE.py</p></li>
<li><p>List of GTFs sample IDs and paths (<a href="https://ccb.jhu.edu/software/stringtie/dl/sample_lst.txt">sample_lst.txt</a>)<br>
$ python prepDE.py -i sample_lst.txt</p></li>
</ol></li>
<li><p><a href="https://cran.r-project.org/">Install R</a>&nbsp;and DESeq2. Upon installing R, install DESeq2 on R:<br>
<code>&gt; source("https://bioconductor.org/biocLite.R")&gt; biocLite("DESeq2")</code></p></li>
<li><p>Import DESeq2 library in R<br>
<code>&gt; library("DESeq2")</code></p></li>
<li><p>Load gene(/transcript) count matrix and labels<br>
<code>&gt; countData &lt;- as.matrix(read.csv("gene_count_matrix.csv", row.names="gene_id"))&gt; colData &lt;- read.csv(PHENO_DATA, sep="\t", row.names=1)</code><br>
Note: The PHENO_DATA file contains information on each sample, e.g., sex or population. The exact way to import this depends on the format of the file.</p></li>
<li><p>Check all sample IDs in colData are also in CountData and match their orders<br>
<code>&gt; all(rownames(colData) %in% colnames(countData))</code>1TRUE<br>
<code>&gt; countData &lt;- countData[, rownames(colData)]&gt; all(rownames(colData) == colnames(countData))</code>1TRUE</p></li>
<li><p>Create a DESeqDataSet from count matrix and labels<br>
<code>&gt; dds &lt;- DESeqDataSetFromMatrix(countData = countData,         colData = colData, design = ~ CHOOSE_FEATURE)</code></p></li>
<li><p>Run the default analysis for DESeq2 and generate results table<br>
<code>&gt; dds &lt;- DESeq(dds)&gt; res &lt;- results(dds)</code></p></li>
<li><p>Sort by adjusted p-value and display<br>
<code>&gt; (resOrdered &lt;- res[order(res$padj), ])</code></p></li>
</ol>
<hr>
<pre><code>python /home/shared/stringtie-2.2.1.Linux_x86_64/prepDE.py </code></pre>
<pre><code>python /home/shared/stringtie-2.2.1.Linux_x86_64/prepDE.py \
-i ../data/list01.txt \
-g ../output/15-Apul-hisat/gene_count_matrix.csv \
-t ../output/15-Apul-hisat/transcript_count_matrix.csv</code></pre>
</section>
</section>
</section>
<section id="deseq" class="level1">
<h1>3 DESeq</h1>
<pre><code>library(DESeq2)</code></pre>
<pre><code># Load gene(/transcript) count matrix and labels
countData &lt;- as.matrix(read.csv("../output/15-Apul-hisat/gene_count_matrix.csv", row.names="gene_id"))
colData &lt;- read.csv("../data/conditions.txt", sep="\t", row.names=1)

# Note: The PHENO_DATA file contains information on each sample, e.g., sex or population.
# The exact way to import this depends on the format of the file.

# Check all sample IDs in colData are also in CountData and match their orders
all(rownames(colData) %in% colnames(countData)) # This should return TRUE</code></pre>
<pre><code>[1] TRUE</code></pre>
<pre><code>countData &lt;- countData[, rownames(colData)]
all(rownames(colData) == colnames(countData)) # This should also return TRUE</code></pre>
<pre><code>[1] TRUE</code></pre>
<pre><code># Create a DESeqDataSet from count matrix and labels
dds &lt;- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData, design = ~ Condition)

# Run the default analysis for DESeq2 and generate results table
dds &lt;- DESeq(dds)
deseq2.res &lt;- results(dds)

# Sort by adjusted p-value and display
resOrdered &lt;- deseq2.res[order(deseq2.res$padj), ]</code></pre>
<pre><code>vsd &lt;- vst(dds, blind = FALSE)
plotPCA(vsd, intgroup = "Condition")</code></pre>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><a href="https://github.com/urol-e5/deep-dive/blob/main/D-Apul/code/15-Apul-HiSat_files/figure-gfm/unnamed-chunk-20-1.png"><img src="https://github.com/urol-e5/deep-dive/raw/main/D-Apul/code/15-Apul-HiSat_files/figure-gfm/unnamed-chunk-20-1.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
<pre><code>tmp &lt;- deseq2.res
# The main plot
plot(tmp$baseMean, tmp$log2FoldChange, pch=20, cex=0.45, ylim=c(-3, 3), log="x", col="darkgray",
     main="DEG Dessication  (pval &lt;= 0.05)",
     xlab="mean of normalized counts",
     ylab="Log2 Fold Change")
# Getting the significant points and plotting them again so they're a different color
tmp.sig &lt;- deseq2.res[!is.na(deseq2.res$padj) &amp; deseq2.res$padj &lt;= 0.05, ]
points(tmp.sig$baseMean, tmp.sig$log2FoldChange, pch=20, cex=0.45, col="red")
# 2 FC lines
abline(h=c(-1,1), col="blue")</code></pre>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><a href="https://github.com/urol-e5/deep-dive/blob/main/D-Apul/code/15-Apul-HiSat_files/figure-gfm/unnamed-chunk-21-1.png"><img src="https://github.com/urol-e5/deep-dive/raw/main/D-Apul/code/15-Apul-HiSat_files/figure-gfm/unnamed-chunk-21-1.png" class="img-fluid figure-img"></a></p>
</figure>
</div>


<!-- -->

</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  let localAlternateSentinel = 'alternate';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<pre class="markdown" data-shortcodes="false"><code>---
title: "HiSat"
description: "to DESeq"
date: 05-14-24
author:
  - name: Steven Roberts
    url: 
    orcid: 0000-0001-8302-1138
    affiliation: Professor, UW - School of Aquatic and Fishery Sciences
    affiliation-url: https://robertslab.info

categories: [R, hisat] # self-defined categories
#citation: 
  #url:  # self-defined
image: http://gannet.fish.washington.edu/seashell/snaps/Monosnap_bestblogever_-_main_-_RStudio_2024-05-15_20-23-56.png # finding a good image
draft: false # setting this to `true` will prevent your post from appearing on your listing page until you're ready!

format:
  html:
    code-fold: true
    code-tools: true
    code-copy: true
    highlight-style: github
    code-overflow: wrap
---


-   &lt;a href="#1-run-hisat-on-rna-seq" id="toc-1-run-hisat-on-rna-seq"&gt;1 Run HiSat on RNA-seq&lt;/a&gt;
    -   &lt;a href="#11-grab-trimmed-rna-seq-reads"
        id="toc-11-grab-trimmed-rna-seq-reads"&gt;1.1 Grab Trimmed RNA-seq Reads&lt;/a&gt;
    -   &lt;a href="#12-genome" id="toc-12-genome"&gt;1.2 Genome&lt;/a&gt;
    -   &lt;a href="#13-hisat" id="toc-13-hisat"&gt;1.3 HiSat&lt;/a&gt;
        -   &lt;a href="#131-performance-tuning" id="toc-131-performance-tuning"&gt;1.3.1 Performance tuning&lt;/a&gt;
    -   &lt;a href="#14-convert-to-bam" id="toc-14-convert-to-bam"&gt;1.4 convert to bam&lt;/a&gt;
    -   &lt;a href="#15-looking-at-bams" id="toc-15-looking-at-bams"&gt;1.5 Looking at Bams&lt;/a&gt;
        -   &lt;a
            href="#151-an-alternate-faster-differential-expression-analysis-workflow"
            id="toc-151-an-alternate-faster-differential-expression-analysis-workflow"&gt;1.5.1 An alternate, faster differential expression analysis workflow&lt;/a&gt;
-   &lt;a href="#2-stringtie" id="toc-2-stringtie"&gt;2 StringTie&lt;/a&gt;

# 1 Run HiSat on RNA-seq

Will end up with 5 sorted bam files.

## 1.1 Grab Trimmed RNA-seq Reads

``` bash

ls ../data/fastq/ 
```

```         
RNA-ACR-140-S1-TP2_R1_001.fastp-trim.20230519.fastq.gz
RNA-ACR-140-S1-TP2_R2_001.fastp-trim.20230519.fastq.gz
RNA-ACR-145-S1-TP2_R1_001.fastp-trim.20230519.fastq.gz
RNA-ACR-145-S1-TP2_R2_001.fastp-trim.20230519.fastq.gz
RNA-ACR-150-S1-TP2_R1_001.fastp-trim.20230519.fastq.gz
RNA-ACR-150-S1-TP2_R2_001.fastp-trim.20230519.fastq.gz
RNA-ACR-173-S1-TP2_R1_001.fastp-trim.20230519.fastq.gz
RNA-ACR-173-S1-TP2_R2_001.fastp-trim.20230519.fastq.gz
RNA-ACR-178-S1-TP2_R1_001.fastp-trim.20230519.fastq.gz
RNA-ACR-178-S1-TP2_R2_001.fastp-trim.20230519.fastq.gz
```

## 1.2 Genome

``` bash
ls ../data/GCF_013753865.1_Amil_v2.1_genomic.fna
```

```         
../data/GCF_013753865.1_Amil_v2.1_genomic.fna
```

``` bash
head ../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gtf

wc -l ../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gtf

grep -v '^#' ../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gtf | cut -f3 | sort | uniq

grep -c "transcript" ../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gtf

grep -c "gene" ../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gtf
```

```         
#gtf-version 2.2
#!genome-build Amil_v2.1
#!genome-build-accession NCBI_Assembly:GCF_013753865.1
#!annotation-source NCBI Acropora millepora Annotation Release 101
NC_058066.1 Gnomon  gene    1962    23221   .   -   .   gene_id "LOC114963522"; transcript_id ""; db_xref "GeneID:114963522"; gbkey "Gene"; gene "LOC114963522"; gene_biotype "lncRNA"; 
NC_058066.1 Gnomon  transcript  1962    23221   .   -   .   gene_id "LOC114963522"; transcript_id "XR_003825913.2"; db_xref "GeneID:114963522"; gbkey "ncRNA"; gene "LOC114963522"; model_evidence "Supporting evidence includes similarity to: 100% coverage of the annotated genomic feature by RNAseq alignments, including 2 samples with support for all annotated introns"; product "uncharacterized LOC114963522"; transcript_biotype "lnc_RNA"; 
NC_058066.1 Gnomon  exon    23085   23221   .   -   .   gene_id "LOC114963522"; transcript_id "XR_003825913.2"; db_xref "GeneID:114963522"; gene "LOC114963522"; model_evidence "Supporting evidence includes similarity to: 100% coverage of the annotated genomic feature by RNAseq alignments, including 2 samples with support for all annotated introns"; product "uncharacterized LOC114963522"; transcript_biotype "lnc_RNA"; exon_number "1"; 
NC_058066.1 Gnomon  exon    21001   21093   .   -   .   gene_id "LOC114963522"; transcript_id "XR_003825913.2"; db_xref "GeneID:114963522"; gene "LOC114963522"; model_evidence "Supporting evidence includes similarity to: 100% coverage of the annotated genomic feature by RNAseq alignments, including 2 samples with support for all annotated introns"; product "uncharacterized LOC114963522"; transcript_biotype "lnc_RNA"; exon_number "2"; 
NC_058066.1 Gnomon  exon    18711   18775   .   -   .   gene_id "LOC114963522"; transcript_id "XR_003825913.2"; db_xref "GeneID:114963522"; gene "LOC114963522"; model_evidence "Supporting evidence includes similarity to: 100% coverage of the annotated genomic feature by RNAseq alignments, including 2 samples with support for all annotated introns"; product "uncharacterized LOC114963522"; transcript_biotype "lnc_RNA"; exon_number "3"; 
NC_058066.1 Gnomon  exon    1962    2119    .   -   .   gene_id "LOC114963522"; transcript_id "XR_003825913.2"; db_xref "GeneID:114963522"; gene "LOC114963522"; model_evidence "Supporting evidence includes similarity to: 100% coverage of the annotated genomic feature by RNAseq alignments, including 2 samples with support for all annotated introns"; product "uncharacterized LOC114963522"; transcript_biotype "lnc_RNA"; exon_number "4"; 
874259 ../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gtf
CDS
exon
gene
start_codon
stop_codon
transcript
874254
874254
```

``` bash
head ../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gff

wc -l ../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gff

grep -v '^#' ../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gff | cut -f3 | sort | uniq

grep -c "transcript" ../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gff

grep -c "gene" ../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gff
```

```         
##gff-version 3
#!gff-spec-version 1.21
#!processor NCBI annotwriter
#!genome-build Amil_v2.1
#!genome-build-accession NCBI_Assembly:GCF_013753865.1
#!annotation-source NCBI Acropora millepora Annotation Release 101
##sequence-region NC_058066.1 1 39361238
##species https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id=45264
NC_058066.1 RefSeq  region  1   39361238    .   +   .   ID=NC_058066.1:1..39361238;Dbxref=taxon:45264;Name=1;chromosome=1;collection-date=2017;country=Indonesia;gbkey=Src;genome=chromosome;isolate=JS-1;isolation-source=Whole tissue;mol_type=genomic DNA;tissue-type=Adult tissue
NC_058066.1 Gnomon  gene    1962    23221   .   -   .   ID=gene-LOC114963522;Dbxref=GeneID:114963522;Name=LOC114963522;gbkey=Gene;gene=LOC114963522;gene_biotype=lncRNA
828216 ../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gff
cDNA_match
CDS
exon
gene
guide_RNA
lnc_RNA
mRNA
pseudogene
region
rRNA
snoRNA
snRNA
transcript
tRNA
431975
803260
```

## 1.3 HiSat

``` bash
/home/shared/hisat2-2.2.1/hisat2_extract_exons.py \
../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gtf \
&gt; ../output/15-Apul-hisat/m_exon.tab
```

``` bash
head ../output/15-Apul-hisat/m_exon.tab
wc -l ../output/15-Apul-hisat/m_exon.tab
```

```         
NC_058066.1 1961    2118    -
NC_058066.1 15360   15663   +
NC_058066.1 18710   18774   -
NC_058066.1 21000   21092   -
NC_058066.1 22158   22442   +
NC_058066.1 23084   23220   -
NC_058066.1 24770   25066   +
NC_058066.1 26652   26912   +
NC_058066.1 27071   27358   +
NC_058066.1 27658   27951   +
228010 ../output/15-Apul-hisat/m_exon.tab
```

``` bash

/home/shared/hisat2-2.2.1/hisat2_extract_splice_sites.py \
../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gtf \
&gt; ../output/15-Apul-hisat/m_splice_sites.tab
```

``` bash
head ../output/15-Apul-hisat/m_splice_sites.tab
```

```         
NC_058066.1 2118    18710   -
NC_058066.1 15663   22158   +
NC_058066.1 18774   21000   -
NC_058066.1 21092   23084   -
NC_058066.1 22442   24770   +
NC_058066.1 25066   26652   +
NC_058066.1 26912   27071   +
NC_058066.1 27358   27658   +
NC_058066.1 27951   28247   +
NC_058066.1 28534   29196   +
```

``` bash
/home/shared/hisat2-2.2.1/hisat2-build \
../data/GCF_013753865.1_Amil_v2.1_genomic.fna \
../output/15-Apul-hisat/GCF_013753865.1_Amil_v2.1.index \
--exon ../output/15-Apul-hisat/m_exon.tab \
--ss ../output/15-Apul-hisat/m_splice_sites.tab \
-p 40 \
../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gtf \
2&gt; ../output/15-Apul-hisat/hisat2-build_stats.txt
```

``` bash
tail ../output/15-Apul-hisat/hisat2-build_stats.txt
```

```         
    sideSz: 128
    sideGbwtSz: 104
    sideGbwtLen: 208
    numSides: 2299750
    numLines: 2299750
    gbwtTotLen: 294368000
    gbwtTotSz: 294368000
    reverse: 0
    linearFM: No
Total time for call to driver() for forward index: 00:07:58
```

### 1.3.1 Performance tuning

If your computer has multiple processors/cores, use -p The -p option causes HISAT2 to launch a specified number of parallel search threads. Each thread runs on a different processor/core and all threads find alignments in parallel, increasing alignment throughput by approximately a multiple of the number of threads (though in practice, speedup is somewhat worse than linear).

``` bash
find ../data/fastq/*R2_001.fastp-trim.20230519.fastq.gz \
| xargs basename -s -S1-TP2_R2_001.fastp-trim.20230519.fastq.gz | xargs -I{} \
/home/shared/hisat2-2.2.1/hisat2 \
-x ../output/15-Apul-hisat/GCF_013753865.1_Amil_v2.1.index \
--dta \
-p 20 \
-1 ../data/fastq/{}-S1-TP2_R1_001.fastp-trim.20230519.fastq.gz \
-2 ../data/fastq/{}-S1-TP2_R2_001.fastp-trim.20230519.fastq.gz \
-S ../output/15-Apul-hisat/{}.sam \
2&gt; ../output/15-Apul-hisat/hisat.out
```

``` bash
cat ../output/15-Apul-hisat/hisat.out
```

```         
47710408 reads; of these:
  47710408 (100.00%) were paired; of these:
    27825914 (58.32%) aligned concordantly 0 times
    18559152 (38.90%) aligned concordantly exactly 1 time
    1325342 (2.78%) aligned concordantly &gt;1 times
    ----
    27825914 pairs aligned concordantly 0 times; of these:
      1201636 (4.32%) aligned discordantly 1 time
    ----
    26624278 pairs aligned 0 times concordantly or discordantly; of these:
      53248556 mates make up the pairs; of these:
        42533390 (79.88%) aligned 0 times
        9692346 (18.20%) aligned exactly 1 time
        1022820 (1.92%) aligned &gt;1 times
55.43% overall alignment rate
42864294 reads; of these:
  42864294 (100.00%) were paired; of these:
    24356414 (56.82%) aligned concordantly 0 times
    17135863 (39.98%) aligned concordantly exactly 1 time
    1372017 (3.20%) aligned concordantly &gt;1 times
    ----
    24356414 pairs aligned concordantly 0 times; of these:
      1142678 (4.69%) aligned discordantly 1 time
    ----
    23213736 pairs aligned 0 times concordantly or discordantly; of these:
      46427472 mates make up the pairs; of these:
        36609836 (78.85%) aligned 0 times
        8675847 (18.69%) aligned exactly 1 time
        1141789 (2.46%) aligned &gt;1 times
57.30% overall alignment rate
43712298 reads; of these:
  43712298 (100.00%) were paired; of these:
    30967520 (70.84%) aligned concordantly 0 times
    11666917 (26.69%) aligned concordantly exactly 1 time
    1077861 (2.47%) aligned concordantly &gt;1 times
    ----
    30967520 pairs aligned concordantly 0 times; of these:
      791338 (2.56%) aligned discordantly 1 time
    ----
    30176182 pairs aligned 0 times concordantly or discordantly; of these:
      60352364 mates make up the pairs; of these:
        52153613 (86.42%) aligned 0 times
        7011301 (11.62%) aligned exactly 1 time
        1187450 (1.97%) aligned &gt;1 times
40.34% overall alignment rate
47501524 reads; of these:
  47501524 (100.00%) were paired; of these:
    28603496 (60.22%) aligned concordantly 0 times
    17627744 (37.11%) aligned concordantly exactly 1 time
    1270284 (2.67%) aligned concordantly &gt;1 times
    ----
    28603496 pairs aligned concordantly 0 times; of these:
      1201709 (4.20%) aligned discordantly 1 time
    ----
    27401787 pairs aligned 0 times concordantly or discordantly; of these:
      54803574 mates make up the pairs; of these:
        44045470 (80.37%) aligned 0 times
        9614980 (17.54%) aligned exactly 1 time
        1143124 (2.09%) aligned &gt;1 times
53.64% overall alignment rate
42677752 reads; of these:
  42677752 (100.00%) were paired; of these:
    26422177 (61.91%) aligned concordantly 0 times
    15005422 (35.16%) aligned concordantly exactly 1 time
    1250153 (2.93%) aligned concordantly &gt;1 times
    ----
    26422177 pairs aligned concordantly 0 times; of these:
      1009838 (3.82%) aligned discordantly 1 time
    ----
    25412339 pairs aligned 0 times concordantly or discordantly; of these:
      50824678 mates make up the pairs; of these:
        39698309 (78.11%) aligned 0 times
        9739502 (19.16%) aligned exactly 1 time
        1386867 (2.73%) aligned &gt;1 times
53.49% overall alignment rate
```

## 1.4 convert to bam

``` bash
for samfile in ../output/15-Apul-hisat/*.sam; do
  bamfile="${samfile%.sam}.bam"
  sorted_bamfile="${samfile%.sam}.sorted.bam"
  /home/shared/samtools-1.12/samtools view -bS -@ 20 "$samfile" &gt; "$bamfile"
  /home/shared/samtools-1.12/samtools sort -@ 20 "$bamfile" -o "$sorted_bamfile"
  /home/shared/samtools-1.12/samtools index -@ 20 "$sorted_bamfile"
done
```

## 1.5 Looking at Bams

$$igv$$

------------------------------------------------------------------------

#### 1.5.0.1 **Differential expression analysis**

Together with [HISAT](http://ccb.jhu.edu/software/hisat2) and [Ballgown](http://biorxiv.org/content/early/2014/09/05/003665), StringTie can be used for estimating differential expression across multiple RNA-Seq samples and generating plots and differential expression tables as described in our [protocol paper](http://www.nature.com/nprot/journal/v11/n9/full/nprot.2016.095.html).&lt;img src="https://ccb.jhu.edu/software/stringtie/DE_pipeline.png" alt="DE workflow"/&gt;\
Our recommended workflow includes the following steps:

1.  for each RNA-Seq sample, map the reads to the genome with [HISAT2](http://ccb.jhu.edu/software/hisat2) using the --dta option. It is highly recommended to use the reference annotation information when mapping the reads, which can be either embedded in the genome index (built with the --ss and --exon options, see [HISAT2 manual](http://ccb.jhu.edu/software/hisat2/manual.shtml#the-hisat2-build-indexer)), or provided separately at run time (using the --known-splicesite-infile option of [HISAT2](http://ccb.jhu.edu/software/hisat2/manual.shtml#running-hisat2)). The SAM output of each HISAT2 run must be sorted and converted to BAM using samtools as explained above.

2.  for each RNA-Seq sample, run StringTie to assemble the read alignments obtained in the previous step; it is recommended to run StringTie with the -G option if the reference annotation is available.

3.  run StringTie with **--merge** in order to generate a non-redundant set of transcripts observed in any of the RNA-Seq samples assembled previously. The stringtie --merge mode takes as input a list of all the assembled transcripts files (in GTF format) previously obtained for each sample, as well as a reference annotation file (-G option) if available.

4.  for each RNA-Seq sample, run StringTie using the -B/-b and -e options in order to estimate transcript abundances and generate read coverage tables for Ballgown. The -e option is not required but recommended for this run in order to produce more accurate abundance estimations of the input transcripts. Each StringTie run in this step will take as input the sorted read alignments (BAM file) obtained in step 1 for the corresponding sample and the -G option with the merged transcripts (GTF file) generated by stringtie --merge in step 3. Please note that this is the only case where the -G option is not used with a reference annotation, but with the global, merged set of transcripts as observed across all samples. (This step is the equivalent of the *Tablemaker* step described in the original [Ballgown pipeline](https://github.com/alyssafrazee/ballgown).)

5.  Ballgown can now be used to load the coverage tables generated in the previous step and perform various statistical analyses for differential expression, generate plots etc.

&lt;!--# using faster --&gt;

### 1.5.1 An alternate, faster differential expression analysis workflow

can be pursued if there is no interest in novel isoforms (i.e.&nbsp;assembled transcripts present in the samples but missing from the reference annotation), or if only a well known set of transcripts of interest are targeted by the analysis. This simplified protocol has only 3 steps (depicted below) as it bypasses the individual assembly of each RNA-Seq sample and the *"transcript merge"* step.&lt;img src="https://ccb.jhu.edu/software/stringtie/DE_pipeline_refonly.png" alt="DE workflow"/&gt;This simplified workflow attempts to directly estimate and analyze the expression of a known set of transcripts as given in the *reference annotation* file.

The R package [**IsoformSwitchAnalyzeR**](https://bioconductor.org/packages/devel/bioc/html/IsoformSwitchAnalyzeR.html) can be used to assign gene names to transcripts assembled by StringTie, which can be particularly helpful in cases where StringTie could not perform this assignment unambigiously.\
The `importIsoformExpression() + importRdata()` function of the package can be used to import the expression and annotation data into R. During this import the package will attempt to clean up and recover isoform annotations where possible. From the resulting `switchAnalyzeRlist` object, *IsoformSwitchAnalyzeR* can detect isoform switches along with predicted functional consequences. The `extractGeneExpression()` function can be used to get a gene expression (read count) matrix for analysis with other tools.\
More information and code examples can be found [here](https://bioconductor.org/packages/devel/bioc/vignettes/IsoformSwitchAnalyzeR/inst/doc/IsoformSwitchAnalyzeR.html#examples-visualizations).

# 2 StringTie

StringTie uses the sorted BAM files to assemble transcripts for each sample, outputting them as GTF files. And then merges all individual GTF assemblies into a single merged GTF file. This step extracts transcript information and merges GTFs from all samples--an important step in creating a canonical list of across all samples included in the pipeline.

+:------------+:--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| -b \&lt;path\&gt; | Just like -B this option enables the output of \*.ctab files for Ballgown, but these files will be created in the provided directory \&lt;path\&gt; instead of the directory specified by the -o option. Note: adding the -e option is recommended with the -B/-b options, unless novel transcripts are still wanted in the StringTie GTF output. |
+-------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+

+:--------+:---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| \       | this option directs StringTie to operate in [expression estimation mode](https://ccb.jhu.edu/software/stringtie/index.shtml?t=manual#emode); this limits the processing of read alignments to estimating the coverage of the transcripts given with the -G option (hence this option requires -G). |
| -e      |                                                                                                                                                                                                                                                                                                    |
+---------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+

+:-------------------+:---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| -G \&lt;ref_ann.gff\&gt; | Use a [reference annotation file](https://ccb.jhu.edu/software/stringtie/index.shtml?t=manual#refann) (in [GTF or GFF3 format](https://ccb.jhu.edu/software/stringtie/gff.shtml)) to guide the assembly process. The output will include expressed reference transcripts as well as any novel transcripts that are assembled. This option is required by options -B, -b, -e, -C (see below). |
+--------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+

``` bash
find ../output/15-Apul-hisat/*sorted.bam \
| xargs basename -s .sorted.bam | xargs -I{} \
/home/shared/stringtie-2.2.1.Linux_x86_64/stringtie \
-p 20 \
-eB \
-G ../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gff \
-o ../output/15-Apul-hisat/{}.gtf \
../output/15-Apul-hisat/{}.sorted.bam
```

``` bash
wc -l ../output/15-Apul-hisat/RNA*.gtf
ls ../output/15-Apul-hisat/RNA*.gtf
#head ../output/15-Apul-hisat/RNA*.gtf
```

```         
   445208 ../output/15-Apul-hisat/RNA-ACR-140.gtf
   445208 ../output/15-Apul-hisat/RNA-ACR-145.gtf
   445208 ../output/15-Apul-hisat/RNA-ACR-150.gtf
   445208 ../output/15-Apul-hisat/RNA-ACR-173.gtf
   445208 ../output/15-Apul-hisat/RNA-ACR-178.gtf
  2226040 total
../output/15-Apul-hisat/RNA-ACR-140.gtf
../output/15-Apul-hisat/RNA-ACR-145.gtf
../output/15-Apul-hisat/RNA-ACR-150.gtf
../output/15-Apul-hisat/RNA-ACR-173.gtf
../output/15-Apul-hisat/RNA-ACR-178.gtf
```

The R package&nbsp;[**IsoformSwitchAnalyzeR**](https://bioconductor.org/packages/devel/bioc/html/IsoformSwitchAnalyzeR.html)&nbsp;can be used to assign gene names to transcripts assembled by StringTie, which can be particularly helpful in cases where StringTie could not perform this assignment unambigiously.\
The&nbsp;`importIsoformExpression() + importRdata()`&nbsp;function of the package can be used to import the expression and annotation data into R. During this import the package will attempt to clean up and recover isoform annotations where possible. From the resulting&nbsp;`switchAnalyzeRlist`&nbsp;object,&nbsp;*IsoformSwitchAnalyzeR*&nbsp;can detect isoform switches along with predicted functional consequences. The&nbsp;`extractGeneExpression()`&nbsp;function can be used to get a gene expression (read count) matrix for analysis with other tools.\
More information and code examples can be found&nbsp;[here](https://bioconductor.org/packages/devel/bioc/vignettes/IsoformSwitchAnalyzeR/inst/doc/IsoformSwitchAnalyzeR.html#examples-visualizations).

------------------------------------------------------------------------

### 2.0.1&nbsp;**Using StringTie with DESeq2 and edgeR**

[DESeq2](https://bioconductor.org/packages/release/bioc/html/DESeq2.html)&nbsp;and&nbsp;[edgeR](https://bioconductor.org/packages/release/bioc/html/edgeR.html)&nbsp;are two popular&nbsp;[Bioconductor](https://www.bioconductor.org/)&nbsp;packages for analyzing differential expression, which take as input a matrix of read counts mapped to particular genomic features (e.g., genes). We provide a Python script ([prepDE.py](https://ccb.jhu.edu/software/stringtie/dl/prepDE.py), or the Python 3 version:[prepDE.py3](https://ccb.jhu.edu/software/stringtie/dl/prepDE.py3)&nbsp;) that can be used to extract this read count information directly from the files generated by StringTie (run with the -e parameter).

*prepDE.py*&nbsp;derives hypothetical read counts for each transcript from the coverage values estimated by StringTie for each transcript, by using this simple formula:&nbsp;*reads_per_transcript = coverage \* transcript_len / read_len*

There are two ways to provide input to the prepDE.py script:

-   one option is to provide a path to a directory containing all sample sub-directories, with the same structure as the&nbsp;*ballgown*directory in the&nbsp;[StringTie protocol paper](http://www.nature.com/nprot/journal/v11/n9/full/nprot.2016.095.html)&nbsp;in preparation for Ballgown. By default (no -i option), the script is going to look in the current directory for all sub-directories having .gtf files in them, as in this example:

    ```         
        ./sample1/sample1.gtf
        ./sample2/sample2.gtf
        ./sample3/sample3.gtf
    ```

-   Alternatively, one can provide a text file listing sample IDs and their respective paths ([sample_lst.txt](https://ccb.jhu.edu/software/stringtie/dl/sample_lst.txt)).

Usage: prepDE.py&nbsp;options\
generates two CSV files containing the count matrices for genes and transcripts, using the coverage values found in the output of stringtie -e

Options:

------------------------------------------------------------------------

-h, --help show this help message and exit

-i INPUT, a folder containing all sample sub-directories, or a --input=INPUT, text file with sample ID and path to its GTF file on --in=INPUT each line&nbsp;default:.

-g G where to output the gene count matrix&nbsp;default:genecountmatrix.csv

-t T where to output the transcript count matrix&nbsp;default:transcriptcountmatrix.csv

-l LENGTH, the average read length&nbsp;default:75--length=LENGTH

-p PATTERN, a regular expression that selects the sample --pattern=PATTERN subdirectories

-c, --cluster whether to cluster genes that overlap with different gene IDs, ignoring ones with geneID pattern (see below)

-s STRING, if a different prefix is used for geneIDs assigned by --string=STRING StringTie&nbsp;default:MSTRG

-k KEY, --key=KEY if clustering, what prefix to use for geneIDs assigned by this script&nbsp;default:prepG

--legend=LEGEND if clustering, where to output the legend file mapping transcripts to assigned geneIDs&nbsp;default:legend.csv——————– ——————————————————

These count matrices (CSV files) can then be imported into R for use by DESeq2 and edgeR (using the&nbsp;*DESeqDataSetFromMatrix*&nbsp;and&nbsp;*DGEList*&nbsp;functions, respectively).

##### 2.0.1.0.1&nbsp;**Protocol: Using StringTie with DESeq2**

Given a list of GTFs, which were re-estimated upon merging, users can follow the below protocol to use DESeq2 for differential expression analysis. To generate GTFs from raw reads follow the&nbsp;[StringTie protocol paper](http://www.nature.com/nprot/journal/v11/n9/full/nprot.2016.095.html)&nbsp;(up to the Ballgown step).

As described above, prepDE.py either accepts a .txt ([sample_lst.txt](https://ccb.jhu.edu/software/stringtie/dl/sample_lst.txt)) file listing the sample IDs and the GTFs’ paths or by default expects a ballgown directory produced by StringTie run with the -B option. The following steps leads us through generating count matrices for genes and transcripts, importing this data into DESeq2, and conducting some basic analysis.

1.  Generate count matrices using prepDE.py

    1.  Assuming default ballgown directory structure produced by StringTie\
        \$ python prepDE.py

    2.  List of GTFs sample IDs and paths ([sample_lst.txt](https://ccb.jhu.edu/software/stringtie/dl/sample_lst.txt))\
        \$ python prepDE.py -i sample_lst.txt

2.  [Install R](https://cran.r-project.org/)&nbsp;and DESeq2. Upon installing R, install DESeq2 on R:\
    `&gt; source("https://bioconductor.org/biocLite.R")&gt; biocLite("DESeq2")`

3.  Import DESeq2 library in R\
    `&gt; library("DESeq2")`

4.  Load gene(/transcript) count matrix and labels\
    `&gt; countData &lt;- as.matrix(read.csv("gene_count_matrix.csv", row.names="gene_id"))&gt; colData &lt;- read.csv(PHENO_DATA, sep="\t", row.names=1)`\
    Note: The PHENO_DATA file contains information on each sample, e.g., sex or population. The exact way to import this depends on the format of the file.

5.  Check all sample IDs in colData are also in CountData and match their orders\
    `&gt; all(rownames(colData) %in% colnames(countData))`1TRUE\
    `&gt; countData &lt;- countData[, rownames(colData)]&gt; all(rownames(colData) == colnames(countData))`1TRUE

6.  Create a DESeqDataSet from count matrix and labels\
    `&gt; dds &lt;- DESeqDataSetFromMatrix(countData = countData,         colData = colData, design = ~ CHOOSE_FEATURE)`

7.  Run the default analysis for DESeq2 and generate results table\
    `&gt; dds &lt;- DESeq(dds)&gt; res &lt;- results(dds)`

8.  Sort by adjusted p-value and display\
    `&gt; (resOrdered &lt;- res[order(res$padj), ])`

------------------------------------------------------------------------

```         
python /home/shared/stringtie-2.2.1.Linux_x86_64/prepDE.py 
```

```         
python /home/shared/stringtie-2.2.1.Linux_x86_64/prepDE.py \
-i ../data/list01.txt \
-g ../output/15-Apul-hisat/gene_count_matrix.csv \
-t ../output/15-Apul-hisat/transcript_count_matrix.csv
```

# 3 DESeq

```         
library(DESeq2)
```

```         
# Load gene(/transcript) count matrix and labels
countData &lt;- as.matrix(read.csv("../output/15-Apul-hisat/gene_count_matrix.csv", row.names="gene_id"))
colData &lt;- read.csv("../data/conditions.txt", sep="\t", row.names=1)

# Note: The PHENO_DATA file contains information on each sample, e.g., sex or population.
# The exact way to import this depends on the format of the file.

# Check all sample IDs in colData are also in CountData and match their orders
all(rownames(colData) %in% colnames(countData)) # This should return TRUE
```

```         
[1] TRUE
```

```         
countData &lt;- countData[, rownames(colData)]
all(rownames(colData) == colnames(countData)) # This should also return TRUE
```

```         
[1] TRUE
```

```         
# Create a DESeqDataSet from count matrix and labels
dds &lt;- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData, design = ~ Condition)

# Run the default analysis for DESeq2 and generate results table
dds &lt;- DESeq(dds)
deseq2.res &lt;- results(dds)

# Sort by adjusted p-value and display
resOrdered &lt;- deseq2.res[order(deseq2.res$padj), ]
```

```         
vsd &lt;- vst(dds, blind = FALSE)
plotPCA(vsd, intgroup = "Condition")
```

[![](https://github.com/urol-e5/deep-dive/raw/main/D-Apul/code/15-Apul-HiSat_files/figure-gfm/unnamed-chunk-20-1.png)](https://github.com/urol-e5/deep-dive/blob/main/D-Apul/code/15-Apul-HiSat_files/figure-gfm/unnamed-chunk-20-1.png)

```         
tmp &lt;- deseq2.res
# The main plot
plot(tmp$baseMean, tmp$log2FoldChange, pch=20, cex=0.45, ylim=c(-3, 3), log="x", col="darkgray",
     main="DEG Dessication  (pval &lt;= 0.05)",
     xlab="mean of normalized counts",
     ylab="Log2 Fold Change")
# Getting the significant points and plotting them again so they're a different color
tmp.sig &lt;- deseq2.res[!is.na(deseq2.res$padj) &amp; deseq2.res$padj &lt;= 0.05, ]
points(tmp.sig$baseMean, tmp.sig$log2FoldChange, pch=20, cex=0.45, col="red")
# 2 FC lines
abline(h=c(-1,1), col="blue")
```

[![](https://github.com/urol-e5/deep-dive/raw/main/D-Apul/code/15-Apul-HiSat_files/figure-gfm/unnamed-chunk-21-1.png)](https://github.com/urol-e5/deep-dive/blob/main/D-Apul/code/15-Apul-HiSat_files/figure-gfm/unnamed-chunk-21-1.png)</code></pre>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>