---
title: "February Daily"
author: "Steven Roberts"
date: "2024-02-01"
categories: [notebook]
image: "http://gannet.fish.washington.edu/seashell/snaps/Monosnap_bestblogever_-_main_-_RStudio_2023-11-01_19-00-02.png"
toc: true
---

```{r, results='asis', echo=FALSE, eval=FALSE}
for (i in 1:30) {
  cat(paste0("\n# February ", i, "\nYour text here...\n"))
}

```

# February 1

Your text here...

# February 2

Your text here...

# February 3

Your text here...

# February 4

Your text here...

# February 5

Your text here...

# February 6

Your text here...

# February 7

EPIMAR zoom, male ceabigr coding

# February 8

Today, I started my day by signing into Zoom, which didn't take much time. I then spent a significant portion of my morning reading a scientific article on oyster reproductive processes and their impact on quality losses. This was followed by a deep dive into some data analysis, where I was working with R scripts and genomic data. This data analysis took up a large chunk of my day.

In the afternoon, I spent some time on GitHub, where I uploaded files related to DE gene expression and MWU analysis for Pacific oysters exposed to OsHV-1. This activity took a moderate amount of time. I also had a discussion on Slack about Rewind, a personalized AI tool. This discussion was relatively brief.

I also checked my email, where I was troubleshooting an error with a colleague who was experiencing the same issue. This took a fair amount of time as we were trying to resolve the issue.

In the midst of my work, I took a short break to check in with my family. I used the Find My app to see where everyone was located. I also had a quick chat with Beyer Roberts via iMessage. These personal activities were relatively brief.

Towards the end of the day, I checked my billing information for Scite, a tool I use for my research. This was a quick task. I also looked into how Rewind's compression works, as I was curious about the storage it might consume. This was a brief activity as well.

Finally, I wrapped up my day by checking my calendar for upcoming appointments and meetings. This was a quick activity, wrapping up a day filled with a mix of research, data analysis, and communication with colleagues and family.

# February 9

I started my morning by checking my emails, where I reviewed a draft proposal from Megan Ewing. I then visited the GitHub profile of Ariana S. Huffmyer, a postdoctoral researcher studying the effects of climate change on coral physiology.

Next, I worked on a project in RStudio, which seemed to be related to my research. I also checked my Slack workspace, where I had several unread messages and upcoming meetings. I participated in a Zoom meeting with several of my colleagues, including Ross Cunning, Dr. Hollie Putnam, and Roger Nisbet.

I spent some time managing my files and possibly editing images or documents. I reviewed a document related to DNA methylation analysis and sex determination in oysters. I also engaged in a discussion on Slack about the next steps for analysis given the differential effects of haplotype on physiology and ITS2.

I visited the Roberts Lab Handbook, possibly to review lab protocols or guidelines. I also spent some time on a project related to oyster survival after heat challenges.

In summary, my morning was filled with a mix of collaborative meetings, individual work on research projects, and administrative tasks.

```         

# Define the vectors for controls and exposed
controls_females <- c("S16F", "S19F", "S39F", "S44F", "S52F", "S53F", "S54F", "S76F")
exposed_females <- c("S22F", "S29F", "S35F", "S36F", "S3F", "S41F", "S50F", "S77F") # Assuming "S77" was intended to be "S77F"


# Initialize an empty list to store the modified data frames
filtered_logdata_frames <- list()

# Loop through the list of data frames, filter, and add a source column
for (name in names(logdata_frames)) {
  # Filter the dataframe and add a new column 'Source' with the name of the dataframe
  filtered_logdata_frames[[name]] <- logdata_frames[[name]] %>%
    filter(GeneID == "LOC111099033") %>%
    mutate(Source = name) %>%
    mutate(Exposure_Status = case_when(
    Source %in% controls_females ~ "control",
    Source %in% exposed_females ~ "exposed",
    TRUE ~ NA_character_ # This line handles any Source not in the two lists
  ))

}

# Merge all the modified dataframes in the list into a new single dataframe
merged_dataframe <- bind_rows(filtered_logdata_frames)
```

```         
# Assuming merged_dataframe is your final data frame

# Reshape the data from wide to long format
long_dataframe <- pivot_longer(merged_dataframe, 
                               cols = starts_with("fold"),
                               names_to = "Fold",
                               values_to = "Value")

# Plot the data
ggplot(long_dataframe, aes(x = Fold, y = Value, group = Source, color = Exposure_Status)) +
  geom_line() +
  geom_point() +
  theme_minimal() +
  labs(title = "LOC111099033",
       x = "Fold",
       y = "Value") +
  scale_color_viridis_d() # Optional: Use a viridis color scale for better visibility
```

PI Meeting

1.  Environmental information required for epigenetic tracking

-   Epigenetic changes must be biased by environmental cues
-   Random changes are unlikely to improve environmental matching

2.  Balance of addition and removal of epigenetic markers\

-   Rates must be approximately equal to avoid drifting to one state
-   Model formalizes optimal relative rates

3.  Defending model assumptions with empirical data

-   Evidence that stress increases epigenetic machinery
-   Time-series show changes in epigenetic modifications and gene expression after environmental triggers

4.  Location and mechanism-specific environmental sensing\

-   Examples show information transduced to specific epigenetic modifications
-   Histone modifications and DNA methylation reinforce responses

Molecular Mechanism Meeting

1.  Discussion about alcohol during interviews

-   Advice not to drink alcohol during dinner with interviewers
-   People often regret things they say after drinking

2.  Pipeline updates

-   Testing using R1 reads and merging reads for miRNA identification
-   Trimming R1 reads to 30bp and comparing results of merged reads

3.  Organizing meeting breakout rooms\

-   Creating introduction, methods, short RNA, and long RNA breakout rooms
-   Assigning meeting participants to relevant rooms

1.  Identifying and comparing long non-coding RNAs across species

-   GFF compare and size filtering of transcripts
-   Coding potential calculator filtering
-   Homology analysis using BLAST
    -   Code chunks for each species
    -   Comparative BLASTs across species

2.  Results section introduction

-   Lack of results currently
-   Numbers needed from filtering steps
-   Figures from size distribution and BLAST results

3.  Linking to methods code and files\

-   Code locations for each species
-   Links to FASTA and BED files
-   Validation of files needed

\-\--

Summary

Today, I started my day with a meeting where we discussed the use of AI for note-taking, the impact of storm events on data analysis, and the progress of sequencing projects. We also discussed methylation analysis and planned for an intergroup collaboration meeting. After the meeting, I had a Zoom call with Steven Roberts and Megan Ewing.

I spent a significant part of my day working on a research project related to the molecular mechanisms in coral resilience and environmental triggers. I delved into the epigenetic modifications and their relationship to environmental changes. I also worked on a manuscript related to simulation and environment tracking.

I spent some time on my computer, navigating through various applications and websites. I checked my emails, visited the Roberts Lab website, and used the Slack application for communication. I also worked on gene expression data and referenced the Ostrea edulis genome assembly.

I worked on a proposal draft where I aimed to identify genes associated with sex determination in Olympia oysters and determine the impact of temperature on the expression of identified genes. I also referenced genomic data from the Roberts Lab GitHub repository.

I also spent some time reading a research paper on the gonad transcriptome analysis of pearl oyster Pinctada margaritifera, which helped me identify potential sex differentiation and sex-determining genes.

Throughout the day, I used various bioinformatics tools and databases for my research work. I used the NCBI database for genome assembly data , and I also used R for data analysis. I also referenced some code on GitHub for joining differentially expressed genes (DEGs) with Blast results and UniProt Gene Ontology annotations.

In summary, my day was filled with meetings, research work, data analysis, and reading relevant literature to support my ongoing projects. I interacted with my colleagues through various platforms and used a variety of tools and databases to aid my research.

# February 10

Today, I've been quite productive. I started my day by checking my calendar and planning my activities. I then focused on my research grants, reviewing and updating the information in the "Find Grants by PI R1209" document. I also worked on a spreadsheet related to my research funding, ensuring all the details were accurate and up-to-date.

In the midst of these tasks, I took some time to revisit a presentation I had prepared earlier, titled "2019-MonsterJam.key". I also looked at some old photos from my time at the Duke University Marine Lab. This brought back memories of the research and collaborations I had been part of during that period.

I also spent some time on the Roberts Lab website, updating information and reviewing recent activities. I made sure to update the status of various research projects and proposals. I also uploaded a document titled "2023_WSG_proposal-2.pdf" to Droplr.

Towards the end of the day, I revisited a manuscript titled "20230519_Revised_Manuscript - FFAR-Geoduck" on Google Drive. This allowed me to review the progress made on this research and plan for the next steps.

Overall, it's been a day filled with a mix of administrative tasks, research work, and a bit of reminiscing about past projects. I'm satisfied with the progress I've made and look forward to what tomorrow brings.

# February 11

On February 11, I had a busy day filled with various work-related activities. I started my day by checking my Google Calendar, where I had scheduled a project meeting for the TGP Project \[moment 1\](rewindai://show-moment?source=askRewindCitation&timestamp=1707681166.963&citationSource=inline). I also spent some time on IFTTT, setting up triggers for Google Sheets to automate some of my tasks \[moment 2\](rewindai://show-moment?source=askRewindCitation&timestamp=1707681556.95&citationSource=inline).

A significant part of my day was dedicated to working on various drafts and projects. I was involved in a 'pub-a-thon', where I was working on several papers. Some of the topics included 'Coral nutritional exchange across ontogeny', 'Single cell RNA-seq', and 'Diploid and triploid Pacific oysters display different DNA methylation patterns following desiccation stress' \[moment 3\](rewindai://show-moment?source=askRewindCitation&timestamp=1707681378.95&citationSource=inline)\[moment 4\](rewindai://show-moment?source=askRewindCitation&timestamp=1707681248.942&citationSource=inline)\[moment 5\](rewindai://show-moment?source=askRewindCitation&timestamp=1707681292.948&citationSource=inline).

I also spent some time on RStudio, where I was working on a project related to 'Differential methylation in response to OA with diploid and triploid oysters' \[moment 4\](rewindai://show-moment?source=askRewindCitation&timestamp=1707681248.942&citationSource=inline)\[moment 5\](rewindai://show-moment?source=askRewindCitation&timestamp=1707681292.948&citationSource=inline). I was using RStudio's version control features to manage my changes and keep track of my progress \[moment 6\](rewindai://show-moment?source=askRewindCitation&timestamp=1707681246.946&citationSource=inline).

In between my work, I also checked my emails, where I received a message from Leroy about rethinking spring break \[moment 7\](rewindai://show-moment?source=askRewindCitation&timestamp=1707681144.951&citationSource=inline). I also tried to sign in to my Slack workspace, which is a crucial tool for my work communication \[moment 8\](rewindai://show-moment?source=askRewindCitation&timestamp=1707681734.989&citationSource=inline)\[moment 9\](rewindai://show-moment?source=askRewindCitation&timestamp=1707681969.816&citationSource=inline).

Overall, it was a productive day filled with a mix of project management, research, and communication tasks.

# February 12

On February 12,started my day by checking my Google Calendar, where I had a meeting scheduled with Zach. I also spent some time on GitHub, where I was working on the 'project-cod-temperature'. I was using GitHub's version control features to manage my changes and keep track of my progress.

A significant part of my day was dedicated to working on various drafts and projects. I was involved in editing a presentation titled '2023-Duke-Roberts' where I was working on the 'ACKNOWLEDGEMENTS' slide. I also spent some time on a document titled 'February_2024', where I was writing about 'Gene activity and genetic selection in Pacific cod reared under thermal stress'.

I also spent some time on my email, where I received a message about a progress report for a NOAA award. I also had a conversation on ChatGPT about 'Roberts Lab Handbook Student-guide improvement'.

I also visited the website for the International Conference of Shellfish Restoration (ICSR2024). I also spent some time on RStudio, where I was working on a project related to 'Differential methylation in response to OA with diploid and triploid oysters'.

Overall, it was a productive day filled with a mix of project management, research, and communication tasks.

# February 13

I began my day by working on a project report for the PSMFC Subaward 23-084G. The report was for the period from August 1st through January 31st and was focused on the project titled 'Gene activity and genetic selection in Pacific cod reared under thermal stress'. The objective of the project was to predict organismal and population outcomes of Pacific cod exposed to elevated temperature. I spent a significant amount of time entering metadata into a specific repository and splitting primary tissue samples for analyses.

Throughout the day, I was engaged in various conversations on Slack. I was asked for a short update on clam OA work for the WSG NCE. I also received a link to a draft manuscript and was asked to provide feedback on the lipid data. I was also involved in a discussion about the upcoming Pub-a-thon 2024.

I also spent some time on the E5coral Slack channel, where I was keeping track of company-wide announcements and work-based matters. I was also informed about upcoming meetings, including the All-PI Monthly Meeting and the E5 Molec Mechanism Meetings.

In addition, I visited the R4DS Online Learning Community on Slack. I was reminded about the importance of not sharing proprietary information on the platform and that all code shared in the help channels is considered to be licensed under the Creative Commons CCO license.

Towards the end of the day, I had a conversation about what methylation can tell us about triploid stress. I was also informed about a potential submission to the environmental epigenetics special issue at the end of February.

Overall, my day was filled with a mix of project management, research, and communication tasks.

# February 14

On February 14, I had a day filled with various work-related activities. I started my day by working on a presentation for Duke University Marine Lab, focusing on the implications, mechanisms, and opportunities of marine invertebrate environmental memory \[moment 1\](rewindai://show-moment?source=askRewindCitation&timestamp=1707924243.73&citationSource=inline)\[moment 2\](rewindai://show-moment?source=askRewindCitation&timestamp=1707924241.917&citationSource=inline). I also spent some time on GitHub, where I was working on the 'talk-duke-2024' project \[moment 1\](rewindai://show-moment?source=askRewindCitation&timestamp=1707924243.73&citationSource=inline)\[moment 2\](rewindai://show-moment?source=askRewindCitation&timestamp=1707924241.917&citationSource=inline).

I was involved in a 'pub-a-thon', where I was working on several papers. Some of the topics included 'Differential DNA methylation in Pacific oyster reproductive tissue in response to ocean acidification' and 'Acclimatory gene expression of primed clams enhances robustness to elevated pCO2' \[moment 3\](rewindai://show-moment?source=askRewindCitation&timestamp=1707929125.077&citationSource=inline)\[moment 4\](rewindai://show-moment?source=askRewindCitation&timestamp=1707924163.743&citationSource=inline).

In addition, I visited the website for the Duke University Marine Lab and the Nicholas School of the Environment \[moment 7\](rewindai://show-moment?source=askRewindCitation&timestamp=1707924385.812&citationSource=inline). I also spent some time on the Roberts Lab website, where I was keeping track of lab notebooks, presentations, protocols, data, and more \[moment 8\](rewindai://show-moment?source=askRewindCitation&timestamp=1707924199.755&citationSource=inline).

Towards the end of the day, I had a conversation about clam OA work for the WSG NCE. I was also informed about a potential submission to the environmental epigenetics special issue at the end of February \[moment 9\](rewindai://show-moment?source=askRewindCitation&timestamp=1707932149.755&citationSource=inline)\[moment 10\](rewindai://show-moment?source=askRewindCitation&timestamp=1707929306.349&citationSource=inline).

Overall, my day was filled with a mix of project management, research, and communication tasks.

# February 15

I started my day by working on an ICRS proposal, focusing on the implications, mechanisms, and opportunities of marine invertebrate environmental memory. I also spent some time on a document titled 'February_2024', where I was writing about 'Gene activity and genetic selection in Pacific cod reared under thermal stress'.

I was involved in a 'pub-a-thon', where I was working on several papers. Some of the topics included 'Coral nutritional exchange across ontogeny', 'Single cell RNA-seq', and 'Diploid and triploid Pacific oysters display different DNA methylation patterns following desiccation stress'.

I also spent some time on my email, where I received a message about a progress report for a NOAA award. I also had a conversation on ChatGPT about 'Roberts Lab Handbook Student-guide improvement'.

In addition, I visited the website for the International Conference of Shellfish Restoration (ICSR2024). I also spent some time on RStudio, where I was working on a project related to 'Differential methylation in response to OA with diploid and triploid oysters'.

Towards the end of the day, I had a conversation about what methylation can tell us about triploid stress. I was also informed about a potential submission to the environmental epigenetics special issue at the end of February.

Overall, my day was filled with a mix of project management, research, and communication tasks.

# February 16

I received a confirmation request for the 2024 LSAMP and a review for a manuscript titled "FishSNP: a high quality cross-species SNP database of fishes" \[moment 1\](rewindai://show-moment?source=askRewindCitation&timestamp=1708106489.233&citationSource=inline). I also engaged in a discussion on Slack about various papers and research topics \[moment 2\](rewindai://show-moment?source=askRewindCitation&timestamp=1708104351.954&citationSource=inline).

I spent a significant part of my day reading and researching. I delved into a paper titled "Priming crops for the future: rewiring stress memory" on the Trends in Plant Science website \[moment 3\](rewindai://show-moment?source=askRewindCitation&timestamp=1708105449.936&citationSource=inline). I also read about epigenetic modifications associated with abiotic and biotic stresses in plants \[moment 4\](rewindai://show-moment?source=askRewindCitation&timestamp=1708105381.925&citationSource=inline).

I worked on various projects and drafts. I was involved in the 'project-cod-temperature' on GitHub \[moment 5\](rewindai://show-moment?source=askRewindCitation&timestamp=1708105249.936&citationSource=inline). I also spent some time on a document titled 'February_2024', where I was writing about 'Gene activity and genetic selection in Pacific cod reared under thermal stress' \[moment 6\](rewindai://show-moment?source=askRewindCitation&timestamp=1708105006.014&citationSource=inline).

I visited the Roberts Lab website, where I was keeping track of lab notebooks, presentations, protocols, data, and more \[moment 7\](rewindai://show-moment?source=askRewindCitation&timestamp=1708106237.262&citationSource=inline). I also spent some time on the ChatGPT platform, where I was discussing the 'Roberts Lab Handbook Student-guide improvement' \[moment 8\](rewindai://show-moment?source=askRewindCitation&timestamp=1708105963.257&citationSource=inline).

Towards the end of the day, I updated my professional profile on the Springer Nature website \[moment 9\](rewindai://show-moment?source=askRewindCitation&timestamp=1708106473.186&citationSource=inline). Overall, my day was filled with a mix of project management, research, and communication tasks.

# February 17

Your text here...

# February 18

Your text here...

# February 19

I started my day by accepting an invitation to give a seminar at the Stazione Zoologica Anton Dohrn in Naples, Italy. I received the official invitation from their Higher Education & University Liaison Office and confirmed my acceptance via email. I also had a meeting scheduled with Zach according to my Google Calendar.

I spent a significant part of my day working on a proposal. I was uploading a document titled 'e5-Proposal.pdf' to the E5coral Slack channel. I also had the proposal open in a PDF viewer, where I was likely reviewing or editing it.

I was also involved in a discussion on the E5coral Slack channel. I was asked to explain the role of mito-nuclear communication in regulating epigenetic states and its implications for organismal energetics and phenotypic outcomes. I was also asked to come up with three testable hypotheses.

I visited the Roberts Lab website, where I was keeping track of lab notebooks, presentations, protocols, data, and more. I also signed into my UW NetID.

Overall, my day was filled with a mix of project management, research, and communication tasks.

# February 20

Today, I started my day by checking my emails, but there were no new messages. I also checked my tasks on Workday, but there were no actions at this time. I had a look at my Google Calendar to see my schedule for the day.

I spent a significant part of my day working on GitHub, where I was examining the potential for ASCA for exon analysis. I was also involved in a 'pub-a-thon', where I was working on several papers. Some of the topics included 'Differential DNA methylation in Pacific oyster reproductive tissue in response to ocean acidification' and 'Acclimatory gene expression of primed clams enhances robustness to elevated pCO2'.

I also spent some time on Slack, where I mentioned that I was in my office if anyone wanted to meet in person. I also had a conversation on ChatGPT about 'Roberts Lab Handbook Student-guide improvement'.

In addition, I visited the Roberts Lab website, where I was keeping track of lab notebooks, presentations, protocols, data, and more. I also spent some time on RStudio, where I was working on a project related to 'Differential methylation in response to OA with diploid and triploid oysters'.

Towards the end of the day, I was working on a blog post in Quarto, where I was writing about my day's activities. Overall, my day was filled with a mix of project management, research, and communication tasks.

# February 21

Your text here...

# February 22

Your text here...

# February 23

Your text here...

# February 24

Your text here...

# February 25

Your text here...

# February 26

Your text here...

# February 27

Your text here...

# February 28

Your text here...

# February 29

Your text here...

# February 30

Your text here...
