<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.553">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Steven Roberts">
<meta name="dcterms.date" content="2024-05-21">
<meta name="description" content="for cod">

<title>Tumbling Oysters - Differentially Expressed Genes</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../../site_libs/bootstrap/bootstrap-dark.min.css" rel="stylesheet" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<meta name="twitter:title" content="Tumbling Oysters - Differentially Expressed Genes">
<meta name="twitter:description" content="for cod">
<meta name="twitter:image" content="http://gannet.fish.washington.edu/seashell/snaps/Monosnap_project-cod-temperature__RStudio_Server_2024-05-22_16-07-31.png">
<meta name="twitter:card" content="summary_large_image">
</head>

<body class="nav-fixed fullcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top quarto-banner">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Tumbling Oysters</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../about.html"> 
<span class="menu-text">Me Elsewhere</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/sr320"> <i class="bi bi-github" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com/sr320"> <i class="bi bi-twitter" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
          <div class="quarto-navbar-tools">
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <div class="quarto-title-block"><div><h1 class="title">Differentially Expressed Genes</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
                  <div>
        <div class="description">
          for cod
        </div>
      </div>
                          <div class="quarto-categories">
                <div class="quarto-category">R</div>
                <div class="quarto-category">datasets</div>
                <div class="quarto-category">cod</div>
              </div>
                  </div>
  </div>
    
  <div class="quarto-title-meta-author">
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-heading">Affiliation</div>
    
      <div class="quarto-title-meta-contents">
      <p class="author">Steven Roberts <a href="https://orcid.org/0000-0001-8302-1138" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
    </div>
    <div class="quarto-title-meta-contents">
          <p class="affiliation">
              <a href="https://robertslab.info">
              Professor, UW - School of Aquatic and Fishery Sciences
              </a>
            </p>
        </div>
    </div>

  <div class="quarto-title-meta">

        
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">May 21, 2024</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">





<ul>
<li><a href="#1-differentially-expressed-genes" id="toc-1-differentially-expressed-genes">1 Differentially Expressed Genes</a></li>
<li><a href="#2-reads" id="toc-2-reads">2 Reads</a></li>
<li><a href="#3-genome" id="toc-3-genome">3 Genome</a></li>
<li><a href="#4-hisat" id="toc-4-hisat">4 Hisat</a></li>
<li><a href="#5-stringtie" id="toc-5-stringtie">5 Stringtie</a></li>
<li><a href="#6-count-matrix" id="toc-6-count-matrix">6 Count Matrix</a></li>
<li><a href="#7-deseq2" id="toc-7-deseq2">7 DEseq2</a></li>
</ul>
<section id="differentially-expressed-genes" class="level1">
<h1>1 Differentially Expressed Genes</h1>
<p>Alt splice test run</p>
</section>
<section id="reads" class="level1">
<h1>2 Reads</h1>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> ../data/reads/<span class="pp">*</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>../data/reads/100.trimmed.R1.fastq.gz
../data/reads/100.trimmed.R2.fastq.gz
../data/reads/107.trimmed.R1.fastq.gz
../data/reads/107.trimmed.R2.fastq.gz
../data/reads/108.trimmed.R1.fastq.gz
../data/reads/108.trimmed.R2.fastq.gz
../data/reads/109.trimmed.R1.fastq.gz
../data/reads/109.trimmed.R2.fastq.gz
../data/reads/10.trimmed.R1.fastq.gz
../data/reads/10.trimmed.R2.fastq.gz
../data/reads/110.trimmed.R1.fastq.gz
../data/reads/110.trimmed.R2.fastq.gz
../data/reads/116.trimmed.R1.fastq.gz
../data/reads/116.trimmed.R2.fastq.gz
../data/reads/11.trimmed.R1.fastq.gz
../data/reads/11.trimmed.R2.fastq.gz
../data/reads/12.trimmed.R1.fastq.gz
../data/reads/12.trimmed.R2.fastq.gz
../data/reads/13.trimmed.R1.fastq.gz
../data/reads/13.trimmed.R2.fastq.gz
../data/reads/18.trimmed.R1.fastq.gz
../data/reads/18.trimmed.R2.fastq.gz
../data/reads/19.trimmed.R1.fastq.gz
../data/reads/19.trimmed.R2.fastq.gz
../data/reads/1.trimmed.R1.fastq.gz
../data/reads/1.trimmed.R2.fastq.gz
../data/reads/20.trimmed.R1.fastq.gz
../data/reads/20.trimmed.R2.fastq.gz
../data/reads/21.trimmed.R1.fastq.gz
../data/reads/21.trimmed.R2.fastq.gz
../data/reads/28.trimmed.R1.fastq.gz
../data/reads/28.trimmed.R2.fastq.gz
../data/reads/29.trimmed.R1.fastq.gz
../data/reads/29.trimmed.R2.fastq.gz
../data/reads/2.trimmed.R1.fastq.gz
../data/reads/2.trimmed.R2.fastq.gz
../data/reads/30.trimmed.R1.fastq.gz
../data/reads/30.trimmed.R2.fastq.gz
../data/reads/31.trimmed.R1.fastq.gz
../data/reads/31.trimmed.R2.fastq.gz
../data/reads/36.trimmed.R1.fastq.gz
../data/reads/36.trimmed.R2.fastq.gz
../data/reads/3.trimmed.R1.fastq.gz
../data/reads/3.trimmed.R2.fastq.gz
../data/reads/4.trimmed.R1.fastq.gz
../data/reads/4.trimmed.R2.fastq.gz
../data/reads/5.trimmed.R1.fastq.gz
../data/reads/5.trimmed.R2.fastq.gz
../data/reads/78.trimmed.R1.fastq.gz
../data/reads/78.trimmed.R2.fastq.gz
../data/reads/79.trimmed.R1.fastq.gz
../data/reads/79.trimmed.R2.fastq.gz
../data/reads/80.trimmed.R1.fastq.gz
../data/reads/80.trimmed.R2.fastq.gz
../data/reads/83.trimmed.R1.fastq.gz
../data/reads/83.trimmed.R2.fastq.gz
../data/reads/88.trimmed.R1.fastq.gz
../data/reads/88.trimmed.R2.fastq.gz
../data/reads/90.trimmed.R1.fastq.gz
../data/reads/90.trimmed.R2.fastq.gz
../data/reads/91.trimmed.R1.fastq.gz
../data/reads/91.trimmed.R2.fastq.gz
../data/reads/92.trimmed.R1.fastq.gz
../data/reads/92.trimmed.R2.fastq.gz
../data/reads/94.trimmed.R1.fastq.gz
../data/reads/94.trimmed.R2.fastq.gz
../data/reads/97.trimmed.R1.fastq.gz
../data/reads/97.trimmed.R2.fastq.gz
../data/reads/98.trimmed.R1.fastq.gz
../data/reads/98.trimmed.R2.fastq.gz
../data/reads/99.trimmed.R1.fastq.gz
../data/reads/99.trimmed.R2.fastq.gz
../data/reads/splice-test-files.txt</code></pre>
<p>Need to find treatment file…</p>
</section>
<section id="genome" class="level1">
<h1>3 Genome</h1>
<p><a href="https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_031168955.1/" class="uri">https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_031168955.1/</a></p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="http://gannet.fish.washington.edu/seashell/snaps/Monosnap_Gadus_macrocephalus_genome_assembly_ASM3116895v1_-_NCBI_-_NLM_2024-05-15_09-20-59.png" class="img-fluid figure-img"></p>
<figcaption>genome</figcaption>
</figure>
</div>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> ../data</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="ex">/home/shared/datasets</span> download genome accession GCF_031168955.1 <span class="at">--include</span> gff3,gtf,rna,cds,protein,genome,seq-report</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> ../data </span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="fu">unzip</span> ncbi_dataset.zip</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> /home/shared/8TB_HDD_03/sr320/github/project-cod-temperature/data/ncbi_dataset/data/GCF_031168955.1</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>cds_from_genomic.fna
GCF_031168955.1_ASM3116895v1_genomic.fna
genomic.gff
genomic.gtf
protein.faa
rna.fna
sequence_report.jsonl</code></pre>
</section>
<section id="hisat" class="level1">
<h1>4 Hisat</h1>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="ex">/home/shared/hisat2-2.2.1/hisat2_extract_exons.py</span> <span class="dt">\</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>../data/ncbi_dataset/data/GCF_031168955.1/genomic.gtf <span class="dt">\</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span> ../output/10-hisat-deseq2/m_exon.tab</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb8"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="ex">/home/shared/hisat2-2.2.1/hisat2_extract_splice_sites.py</span> <span class="dt">\</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>../data/ncbi_dataset/data/GCF_031168955.1/genomic.gtf <span class="dt">\</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span> ../output/10-hisat-deseq2/m_spice_sites.tab</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"10-hisat-deseq2/GCF*"</span> <span class="op">&gt;&gt;</span> ../output/.gitignore</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="ex">/home/shared/hisat2-2.2.1/hisat2-build</span> <span class="dt">\</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>../data/ncbi_dataset/data/GCF_031168955.1/GCF_031168955.1_ASM3116895v1_genomic.fna <span class="dt">\</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>../output/10-hisat-deseq2/GCF_031168955.1.index <span class="dt">\</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>--exon ../output/10-hisat-deseq2/m_exon.tab <span class="dt">\</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>--ss ../output/10-hisat-deseq2/m_spice_sites.tab <span class="dt">\</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>-p 20 <span class="dt">\</span></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>../data/ncbi_dataset/data/GCF_031168955.1/genomic.gtf <span class="dt">\</span></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>2<span class="op">&gt;</span> ../output/10-hisat-deseq2/hisat2-build_stats.txt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb11"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"10-hisat-deseq2/*sam"</span> <span class="op">&gt;&gt;</span> ../output/.gitignore</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb12"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">find</span> ../data/reads/<span class="pp">*</span>.trimmed.R1.fastq.gz <span class="dt">\</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span> <span class="fu">xargs</span> basename <span class="at">-s</span> .trimmed.R1.fastq.gz <span class="kw">|</span> <span class="fu">xargs</span> <span class="at">-I{}</span> <span class="dt">\</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>/home/shared/hisat2-2.2.1/hisat2 <span class="dt">\</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>-x ../output/10-hisat-deseq2/GCF_031168955.1.index <span class="dt">\</span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>--dta <span class="dt">\</span></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>-p 20 <span class="dt">\</span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>-1 ../data/reads/{}.trimmed.R1.fastq.gz <span class="dt">\</span></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>-2 ../data/reads/{}.trimmed.R2.fastq.gz <span class="dt">\</span></span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>-S ../output/10-hisat-deseq2/{}.sam <span class="dt">\</span></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>2<span class="op">&gt;</span> ../output/10-hisat-deseq2/hisat.out</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb13"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"10-hisat-deseq2/*bam"</span> <span class="op">&gt;&gt;</span> ../output/.gitignore</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"10-hisat-deseq2/*bam*"</span> <span class="op">&gt;&gt;</span> ../output/.gitignore</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb14"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> samfile <span class="kw">in</span> ../output/10-hisat-deseq2/<span class="pp">*</span>.sam<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>  <span class="va">bamfile</span><span class="op">=</span><span class="st">"</span><span class="va">${samfile</span><span class="op">%</span>.sam<span class="va">}</span><span class="st">.bam"</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  <span class="va">sorted_bamfile</span><span class="op">=</span><span class="st">"</span><span class="va">${samfile</span><span class="op">%</span>.sam<span class="va">}</span><span class="st">.sorted.bam"</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>  <span class="ex">/home/shared/samtools-1.12/samtools</span> view <span class="at">-bS</span> <span class="at">-@</span> 20 <span class="st">"</span><span class="va">$samfile</span><span class="st">"</span> <span class="op">&gt;</span> <span class="st">"</span><span class="va">$bamfile</span><span class="st">"</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>  <span class="ex">/home/shared/samtools-1.12/samtools</span> sort <span class="at">-@</span> 20 <span class="st">"</span><span class="va">$bamfile</span><span class="st">"</span> <span class="at">-o</span> <span class="st">"</span><span class="va">$sorted_bamfile</span><span class="st">"</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>  <span class="ex">/home/shared/samtools-1.12/samtools</span> index <span class="at">-@</span> 20 <span class="st">"</span><span class="va">$sorted_bamfile</span><span class="st">"</span></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb15"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span> ../output/10-hisat-deseq2/<span class="pp">*</span>sam</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb16"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> ../output/10-hisat-deseq2/<span class="pp">*</span>sorted.bam <span class="kw">|</span> <span class="fu">wc</span> <span class="at">-l</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb17"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">tail</span> ../output/10-hisat-deseq2/hisat.out</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>    ----
    3794910 pairs aligned concordantly 0 times; of these:
      146293 (3.85%) aligned discordantly 1 time
    ----
    3648617 pairs aligned 0 times concordantly or discordantly; of these:
      7297234 mates make up the pairs; of these:
        5640288 (77.29%) aligned 0 times
        1498025 (20.53%) aligned exactly 1 time
        158921 (2.18%) aligned &gt;1 times
82.81% overall alignment rate</code></pre>
<div class="sourceCode" id="cb19"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> ../output/10-hisat-deseq2/hisat.out <span class="dt">\</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span> <span class="fu">grep</span> <span class="st">"overall alignment rate"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>87.53% overall alignment rate
86.32% overall alignment rate
81.37% overall alignment rate
86.39% overall alignment rate
83.54% overall alignment rate
87.67% overall alignment rate
86.69% overall alignment rate
86.56% overall alignment rate
84.58% overall alignment rate
81.26% overall alignment rate
83.03% overall alignment rate
86.29% overall alignment rate
84.82% overall alignment rate
82.29% overall alignment rate
83.16% overall alignment rate
86.14% overall alignment rate
81.04% overall alignment rate
87.28% overall alignment rate
84.89% overall alignment rate
87.76% overall alignment rate
88.45% overall alignment rate
83.13% overall alignment rate
83.21% overall alignment rate
84.99% overall alignment rate
86.33% overall alignment rate
86.47% overall alignment rate
84.37% overall alignment rate
87.60% overall alignment rate
75.31% overall alignment rate
81.73% overall alignment rate
84.47% overall alignment rate
80.84% overall alignment rate
86.72% overall alignment rate
87.60% overall alignment rate
81.89% overall alignment rate
82.81% overall alignment rate</code></pre>
</section>
<section id="stringtie" class="level1">
<h1>5 Stringtie</h1>
<div class="sourceCode" id="cb21"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"10-hisat-deseq2/*gtf"</span> <span class="op">&gt;&gt;</span> ../output/.gitignore</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb22"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">find</span> ../output/10-hisat-deseq2/<span class="pp">*</span>sorted.bam <span class="dt">\</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span> <span class="fu">xargs</span> basename <span class="at">-s</span> .sorted.bam <span class="kw">|</span> <span class="fu">xargs</span> <span class="at">-I{}</span> <span class="dt">\</span></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>/home/shared/stringtie-2.2.1.Linux_x86_64/stringtie <span class="dt">\</span></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>-p 36 <span class="dt">\</span></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>-eB <span class="dt">\</span></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>-G ../data/ncbi_dataset/data/GCF_031168955.1/genomic.gff <span class="dt">\</span></span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>-o ../output/10-hisat-deseq2/{}.gtf <span class="dt">\</span></span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>../output/10-hisat-deseq2/{}.sorted.bam</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="count-matrix" class="level1">
<h1>6 Count Matrix</h1>
<p>list format</p>
<pre><code>RNA-ACR-140 ../output/15-Apul-hisat/RNA-ACR-140.gtf
RNA-ACR-145 ../output/15-Apul-hisat/RNA-ACR-145.gtf
RNA-ACR-173 ../output/15-Apul-hisat/RNA-ACR-173.gtf
RNA-ACR-178 ../output/15-Apul-hisat/RNA-ACR-178.gtf</code></pre>
<div class="sourceCode" id="cb24"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> ../output/10-hisat-deseq2/<span class="pp">*</span>gtf</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>../output/10-hisat-deseq2/100.gtf
../output/10-hisat-deseq2/107.gtf
../output/10-hisat-deseq2/108.gtf
../output/10-hisat-deseq2/109.gtf
../output/10-hisat-deseq2/10.gtf
../output/10-hisat-deseq2/110.gtf
../output/10-hisat-deseq2/116.gtf
../output/10-hisat-deseq2/11.gtf
../output/10-hisat-deseq2/12.gtf
../output/10-hisat-deseq2/13.gtf
../output/10-hisat-deseq2/18.gtf
../output/10-hisat-deseq2/19.gtf
../output/10-hisat-deseq2/1.gtf
../output/10-hisat-deseq2/20.gtf
../output/10-hisat-deseq2/21.gtf
../output/10-hisat-deseq2/28.gtf
../output/10-hisat-deseq2/29.gtf
../output/10-hisat-deseq2/2.gtf
../output/10-hisat-deseq2/30.gtf
../output/10-hisat-deseq2/31.gtf
../output/10-hisat-deseq2/36.gtf
../output/10-hisat-deseq2/3.gtf
../output/10-hisat-deseq2/4.gtf
../output/10-hisat-deseq2/5.gtf
../output/10-hisat-deseq2/78.gtf
../output/10-hisat-deseq2/79.gtf
../output/10-hisat-deseq2/80.gtf
../output/10-hisat-deseq2/83.gtf
../output/10-hisat-deseq2/88.gtf
../output/10-hisat-deseq2/90.gtf
../output/10-hisat-deseq2/91.gtf
../output/10-hisat-deseq2/92.gtf
../output/10-hisat-deseq2/94.gtf
../output/10-hisat-deseq2/97.gtf
../output/10-hisat-deseq2/98.gtf
../output/10-hisat-deseq2/99.gtf</code></pre>
<div class="sourceCode" id="cb26"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> ../data/list01.txt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>1   ../output/10-hisat-deseq2/1.gtf
2   ../output/10-hisat-deseq2/2.gtf
3   ../output/10-hisat-deseq2/3.gtf
4   ../output/10-hisat-deseq2/4.gtf
5   ../output/10-hisat-deseq2/5.gtf
10  ../output/10-hisat-deseq2/10.gtf
11  ../output/10-hisat-deseq2/11.gtf
12  ../output/10-hisat-deseq2/12.gtf
13  ../output/10-hisat-deseq2/13.gtf
18  ../output/10-hisat-deseq2/18.gtf</code></pre>
<div class="sourceCode" id="cb28"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="ex">python</span> /home/shared/stringtie-2.2.1.Linux_x86_64/prepDE.py <span class="dt">\</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>-i ../data/list01.txt <span class="dt">\</span></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>-g ../output/10-hisat-deseq2/gene_count_matrix.csv <span class="dt">\</span></span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>-t ../output/10-hisat-deseq2/transcript_count_matrix.csv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb29"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> ../output/10-hisat-deseq2/<span class="pp">*</span>matrix.csv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>==&gt; ../output/10-hisat-deseq2/gene_count_matrix.csv &lt;==
gene_id,1,10,100,107,108,109,11,110,116,12,13,18,19,2,20,21,28,29,3,30,31,36,4,5,78,79,80,83,88,90,91,92,94,97,98,99
gene-LOC132462341|LOC132462341,360,464,391,346,691,408,436,509,366,373,432,385,330,288,307,346,293,347,452,452,984,230,469,400,472,311,368,312,551,631,605,366,339,517,577,235
gene-abce1|abce1,694,325,276,77,409,196,281,284,327,310,393,386,106,290,252,326,345,254,363,353,922,160,588,285,178,119,197,134,128,103,315,204,301,56,49,260
gene-si:dkey-6i22.5|si:dkey-6i22.5,0,10,13,11,49,42,14,36,0,14,9,0,0,14,3,21,26,9,39,20,10,0,0,10,48,41,60,39,49,54,20,27,7,22,63,12
gene-ube2v1|ube2v1,10,22,15,19,74,28,63,35,30,31,25,42,17,14,21,60,61,14,29,37,68,37,0,30,10,13,56,32,45,55,27,38,60,25,18,50
gene-cldn15la|cldn15la,0,9,0,0,18,0,0,0,0,27,0,0,0,0,0,0,0,0,4,35,0,0,0,4,0,0,0,0,8,32,0,0,14,0,0,0
gene-muc15|muc15,0,40,40,29,8,0,38,0,16,15,0,0,0,0,22,22,69,22,0,15,0,34,55,51,0,30,19,31,0,0,42,34,0,0,26,56
gene-pcloa|pcloa,0,3,0,0,41,7,4,0,3,0,0,0,27,0,0,0,0,4,0,10,0,23,0,0,0,0,4,4,0,0,4,0,0,22,101,0
gene-LOC132472829|LOC132472829,402,550,293,182,1082,213,234,149,444,360,531,242,269,194,359,383,395,392,408,489,310,352,168,403,462,287,310,113,457,321,396,234,455,279,229,208
gene-ifi35|ifi35,3696,2492,2615,2184,2702,2579,3631,3158,2121,2673,2893,2210,2324,2384,2561,2654,2887,2250,3143,2830,1297,1981,2971,2651,2669,2671,1790,2173,2603,3260,2186,2451,2464,2768,1935,1991

==&gt; ../output/10-hisat-deseq2/transcript_count_matrix.csv &lt;==
transcript_id,1,10,100,107,108,109,11,110,116,12,13,18,19,2,20,21,28,29,3,30,31,36,4,5,78,79,80,83,88,90,91,92,94,97,98,99
rna-XM_060037252.1,20,918,84,11,385,70,54,82,131,72,52,8,26,41,33,62,30,12,45,85,377,59,58,57,96,19,55,94,92,125,54,34,69,36,16,83
rna-XM_060070002.1,0,0,0,0,0,12,0,0,0,14,14,0,0,0,0,0,0,3,0,0,0,0,0,0,0,0,0,0,0,0,13,0,0,0,0,7
rna-XM_060056394.1,0,0,0,1,0,0,336,0,37,0,0,0,86,0,0,0,0,0,395,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1
rna-XM_060072836.1,0,0,14,4,0,0,5,0,15,0,0,0,4,0,0,5,0,0,0,11,0,0,5,0,0,12,0,0,12,3,10,0,9,0,0,0
rna-XM_060062883.1,719,38,380,279,652,462,461,381,432,0,831,479,641,171,593,82,520,625,366,533,640,311,36,876,365,322,351,342,430,324,450,410,540,336,513,370
rna-XM_060057326.1,1072,1577,801,1056,1420,909,1523,1101,951,1459,2162,1170,1983,1317,1264,1398,2075,1055,1303,1820,1496,934,966,1509,1132,928,845,948,1930,2047,1039,781,979,2259,2747,740
rna-XM_060051796.1,0,1,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,3,0,0,0,0,0,0,0,4,0,2,0,0,0,0,0,0,0,2
rna-XM_060076321.1,0,0,0,0,16,0,0,0,0,0,0,4,58,0,0,0,0,0,0,12,0,0,0,0,0,0,0,0,0,0,0,0,5,13,80,0
rna-XM_060064475.1,6,1771,834,10,33,0,1752,997,470,1572,0,0,16,713,0,0,0,5,0,679,292,656,0,1051,6,1326,0,0,0,0,592,448,0,0,0,5</code></pre>
</section>
<section id="deseq2" class="level1">
<h1>7 DEseq2</h1>
<p>need <code>conditions.txt</code> file in this format</p>
<pre><code>SampleID    Condition
RNA.ACR.140 control
RNA.ACR.145 control
RNA.ACR.173 treated
RNA.ACR.178 treated</code></pre>
<div class="sourceCode" id="cb32"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> ../data/conditions.txt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>SampleID    Condition
X1  16
X2  16
X3  16
X4  16
X5  16
X10 16
X11 16
X12 16
X13 16</code></pre>
<div class="sourceCode" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load gene(/transcript) count matrix and labels</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>countData <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">read.csv</span>(<span class="st">"../output/10-hisat-deseq2/gene_count_matrix.csv"</span>, <span class="at">row.names=</span><span class="st">"gene_id"</span>))</span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>colData <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../data/conditions.txt"</span>, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Note: The PHENO_DATA file contains information on each sample, e.g., sex or population.</span></span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a><span class="co"># The exact way to import this depends on the format of the file.</span></span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Check all sample IDs in colData are also in CountData and match their orders</span></span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">rownames</span>(colData) <span class="sc">%in%</span> <span class="fu">colnames</span>(countData)) <span class="co"># This should return TRUE</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>[1] TRUE</code></pre>
<div class="sourceCode" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>countData <span class="ot">&lt;-</span> countData[, <span class="fu">rownames</span>(colData)]</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">rownames</span>(colData) <span class="sc">==</span> <span class="fu">colnames</span>(countData)) <span class="co"># This should also return TRUE</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>[1] TRUE</code></pre>
<div class="sourceCode" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a DESeqDataSet from count matrix and labels</span></span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> countData,</span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> colData, <span class="at">design =</span> <span class="sc">~</span> Condition)</span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Run the default analysis for DESeq2 and generate results table</span></span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a>deseq2.res <span class="ot">&lt;-</span> <span class="fu">results</span>(dds)</span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-10"><a href="#cb39-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Sort by adjusted p-value and display</span></span>
<span id="cb39-11"><a href="#cb39-11" aria-hidden="true" tabindex="-1"></a>resOrdered <span class="ot">&lt;-</span> deseq2.res[<span class="fu">order</span>(deseq2.res<span class="sc">$</span>padj), ]</span>
<span id="cb39-12"><a href="#cb39-12" aria-hidden="true" tabindex="-1"></a>vsd <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds, <span class="at">blind =</span> <span class="cn">FALSE</span>)</span>
<span id="cb39-13"><a href="#cb39-13" aria-hidden="true" tabindex="-1"></a><span class="fu">plotPCA</span>(vsd, <span class="at">intgroup =</span> <span class="st">"Condition"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="http://gannet.fish.washington.edu/seashell/snaps/Monosnap_project-cod-temperature__RStudio_Server_2024-05-22_16-10-07.png" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Select top 50 differentially expressed genes</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">results</span>(dds)</span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>res_ordered <span class="ot">&lt;-</span> res[<span class="fu">order</span>(res<span class="sc">$</span>padj), ]</span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a>top_genes <span class="ot">&lt;-</span> <span class="fu">row.names</span>(res_ordered)[<span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>]</span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract counts and normalize</span></span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a>counts <span class="ot">&lt;-</span> <span class="fu">counts</span>(dds, <span class="at">normalized =</span> <span class="cn">TRUE</span>)</span>
<span id="cb40-8"><a href="#cb40-8" aria-hidden="true" tabindex="-1"></a>counts_top <span class="ot">&lt;-</span> counts[top_genes, ]</span>
<span id="cb40-9"><a href="#cb40-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-10"><a href="#cb40-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Log-transform counts</span></span>
<span id="cb40-11"><a href="#cb40-11" aria-hidden="true" tabindex="-1"></a>log_counts_top <span class="ot">&lt;-</span> <span class="fu">log2</span>(counts_top <span class="sc">+</span> <span class="dv">1</span>)</span>
<span id="cb40-12"><a href="#cb40-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-13"><a href="#cb40-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate heatmap</span></span>
<span id="cb40-14"><a href="#cb40-14" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(log_counts_top, <span class="at">scale =</span> <span class="st">"row"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="http://gannet.fish.washington.edu/seashell/snaps/Monosnap_project-cod-temperature__RStudio_Server_2024-05-22_16-09-54.png" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Count number of hits with adjusted p-value less then 0.05</span></span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(res[<span class="sc">!</span><span class="fu">is.na</span>(deseq2.res<span class="sc">$</span>padj) <span class="sc">&amp;</span> deseq2.res<span class="sc">$</span>padj <span class="sc">&lt;=</span> <span class="fl">0.05</span>, ])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>[1] 5061    6</code></pre>
<div class="sourceCode" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>tmp <span class="ot">&lt;-</span> deseq2.res</span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a><span class="co"># The main plot</span></span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(tmp<span class="sc">$</span>baseMean, tmp<span class="sc">$</span>log2FoldChange, <span class="at">pch=</span><span class="dv">20</span>, <span class="at">cex=</span><span class="fl">0.45</span>, <span class="at">ylim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">3</span>, <span class="dv">3</span>), <span class="at">log=</span><span class="st">"x"</span>, <span class="at">col=</span><span class="st">"darkgray"</span>,</span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a>     <span class="at">main=</span><span class="st">"DEG Dessication  (pval &lt;= 0.05)"</span>,</span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab=</span><span class="st">"mean of normalized counts"</span>,</span>
<span id="cb43-6"><a href="#cb43-6" aria-hidden="true" tabindex="-1"></a>     <span class="at">ylab=</span><span class="st">"Log2 Fold Change"</span>)</span>
<span id="cb43-7"><a href="#cb43-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Getting the significant points and plotting them again so they're a different color</span></span>
<span id="cb43-8"><a href="#cb43-8" aria-hidden="true" tabindex="-1"></a>tmp.sig <span class="ot">&lt;-</span> deseq2.res[<span class="sc">!</span><span class="fu">is.na</span>(deseq2.res<span class="sc">$</span>padj) <span class="sc">&amp;</span> deseq2.res<span class="sc">$</span>padj <span class="sc">&lt;=</span> <span class="fl">0.05</span>, ]</span>
<span id="cb43-9"><a href="#cb43-9" aria-hidden="true" tabindex="-1"></a><span class="fu">points</span>(tmp.sig<span class="sc">$</span>baseMean, tmp.sig<span class="sc">$</span>log2FoldChange, <span class="at">pch=</span><span class="dv">20</span>, <span class="at">cex=</span><span class="fl">0.45</span>, <span class="at">col=</span><span class="st">"red"</span>)</span>
<span id="cb43-10"><a href="#cb43-10" aria-hidden="true" tabindex="-1"></a><span class="co"># 2 FC lines</span></span>
<span id="cb43-11"><a href="#cb43-11" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">1</span>), <span class="at">col=</span><span class="st">"blue"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="http://gannet.fish.washington.edu/seashell/snaps/Monosnap_project-cod-temperature__RStudio_Server_2024-05-22_16-09-37.png" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(tmp.sig, <span class="st">"../output/10-hisat-deseq2/DEGlist.tab"</span>, <span class="at">sep =</span> <span class="st">'</span><span class="sc">\t</span><span class="st">'</span>, <span class="at">row.names =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>deglist <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../output/10-hisat-deseq2/DEGlist.tab"</span>, <span class="at">sep =</span> <span class="st">'</span><span class="sc">\t</span><span class="st">'</span>, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>deglist<span class="sc">$</span>RowName <span class="ot">&lt;-</span> <span class="fu">rownames</span>(deglist)</span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a>deglist2 <span class="ot">&lt;-</span> deglist[, <span class="fu">c</span>(<span class="st">"RowName"</span>, <span class="st">"pvalue"</span>)] <span class="co"># Optionally, reorder the columns</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(deglist)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>                                     baseMean log2FoldChange      lfcSE
gene-abce1|abce1                    262.70027     0.10595278 0.02812204
gene-si:dkey-6i22.5|si:dkey-6i22.5   23.18182    -0.27199275 0.07237460
gene-LOC132463139|LOC132463139      264.89473     0.29450501 0.05965925
gene-prkaa1|prkaa1                 3214.61286    -0.12678273 0.03404263
gene-snx27a|snx27a                 1220.23705    -0.06307107 0.01399906
gene-LOC132467924|LOC132467924      197.52277     0.09450712 0.03203037
                                        stat       pvalue         padj
gene-abce1|abce1                    3.767607 1.648201e-04 1.695796e-03
gene-si:dkey-6i22.5|si:dkey-6i22.5 -3.758124 1.711919e-04 1.744561e-03
gene-LOC132463139|LOC132463139      4.936451 7.955683e-07 2.120046e-05
gene-prkaa1|prkaa1                 -3.724235 1.959086e-04 1.933956e-03
gene-snx27a|snx27a                 -4.505379 6.625450e-06 1.259621e-04
gene-LOC132467924|LOC132467924      2.950547 3.172118e-03 1.713888e-02
                                                              RowName
gene-abce1|abce1                                     gene-abce1|abce1
gene-si:dkey-6i22.5|si:dkey-6i22.5 gene-si:dkey-6i22.5|si:dkey-6i22.5
gene-LOC132463139|LOC132463139         gene-LOC132463139|LOC132463139
gene-prkaa1|prkaa1                                 gene-prkaa1|prkaa1
gene-snx27a|snx27a                                 gene-snx27a|snx27a
gene-LOC132467924|LOC132467924         gene-LOC132467924|LOC132467924</code></pre>


<!-- -->

</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
    const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
    const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
    let newTheme = '';
    if(darkModeDefault) {
      newTheme = isAlternate ? baseTheme : alternateTheme;
    } else {
      newTheme = isAlternate ? alternateTheme : baseTheme;
    }
    const changeGiscusTheme = () => {
      // From: https://github.com/giscus/giscus/issues/336
      const sendMessage = (message) => {
        const iframe = document.querySelector('iframe.giscus-frame');
        if (!iframe) return;
        iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
      }
      sendMessage({
        setConfig: {
          theme: newTheme
        }
      });
    }
    const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
    if (isGiscussLoaded) {
      changeGiscusTheme();
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  const darkModeDefault = true;
  let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
    toggleGiscusIfUsed(toAlternate, darkModeDefault);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb48" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Differentially Expressed Genes"</span></span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a><span class="an">description:</span><span class="co"> "for cod"</span></span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a><span class="an">categories:</span><span class="co"> [R, datasets, cod] # self-defined categories</span></span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a><span class="co">#citation: </span></span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> 05-21-2024</span></span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span></span>
<span id="cb48-9"><a href="#cb48-9" aria-hidden="true" tabindex="-1"></a><span class="co">  - name: Steven Roberts</span></span>
<span id="cb48-10"><a href="#cb48-10" aria-hidden="true" tabindex="-1"></a><span class="co">    url: </span></span>
<span id="cb48-11"><a href="#cb48-11" aria-hidden="true" tabindex="-1"></a><span class="co">    orcid: 0000-0001-8302-1138</span></span>
<span id="cb48-12"><a href="#cb48-12" aria-hidden="true" tabindex="-1"></a><span class="co">    affiliation: Professor, UW - School of Aquatic and Fishery Sciences</span></span>
<span id="cb48-13"><a href="#cb48-13" aria-hidden="true" tabindex="-1"></a><span class="co">    affiliation-url: https://robertslab.info</span></span>
<span id="cb48-14"><a href="#cb48-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-15"><a href="#cb48-15" aria-hidden="true" tabindex="-1"></a><span class="co">  #url:  # self-defined</span></span>
<span id="cb48-16"><a href="#cb48-16" aria-hidden="true" tabindex="-1"></a><span class="an">image:</span><span class="co"> http://gannet.fish.washington.edu/seashell/snaps/Monosnap_project-cod-temperature__RStudio_Server_2024-05-22_16-07-31.png # finding a good image</span></span>
<span id="cb48-17"><a href="#cb48-17" aria-hidden="true" tabindex="-1"></a><span class="an">draft:</span><span class="co"> false # setting this to `true` will prevent your post from appearing on your listing page until you're ready!</span></span>
<span id="cb48-18"><a href="#cb48-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-19"><a href="#cb48-19" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb48-20"><a href="#cb48-20" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb48-21"><a href="#cb48-21" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: true</span></span>
<span id="cb48-22"><a href="#cb48-22" aria-hidden="true" tabindex="-1"></a><span class="co">    code-tools: true</span></span>
<span id="cb48-23"><a href="#cb48-23" aria-hidden="true" tabindex="-1"></a><span class="co">    code-copy: true</span></span>
<span id="cb48-24"><a href="#cb48-24" aria-hidden="true" tabindex="-1"></a><span class="co">    highlight-style: github</span></span>
<span id="cb48-25"><a href="#cb48-25" aria-hidden="true" tabindex="-1"></a><span class="co">    code-overflow: wrap</span></span>
<span id="cb48-26"><a href="#cb48-26" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb48-27"><a href="#cb48-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-28"><a href="#cb48-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-29"><a href="#cb48-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-30"><a href="#cb48-30" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>&lt;a href="#1-differentially-expressed-genes"</span>
<span id="cb48-31"><a href="#cb48-31" aria-hidden="true" tabindex="-1"></a>  id="toc-1-differentially-expressed-genes"&gt;1 Differentially Expressed</span>
<span id="cb48-32"><a href="#cb48-32" aria-hidden="true" tabindex="-1"></a>  Genes&lt;/a&gt;</span>
<span id="cb48-33"><a href="#cb48-33" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>&lt;a href="#2-reads" id="toc-2-reads"&gt;2 Reads&lt;/a&gt;</span>
<span id="cb48-34"><a href="#cb48-34" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>&lt;a href="#3-genome" id="toc-3-genome"&gt;3 Genome&lt;/a&gt;</span>
<span id="cb48-35"><a href="#cb48-35" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>&lt;a href="#4-hisat" id="toc-4-hisat"&gt;4 Hisat&lt;/a&gt;</span>
<span id="cb48-36"><a href="#cb48-36" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>&lt;a href="#5-stringtie" id="toc-5-stringtie"&gt;5 Stringtie&lt;/a&gt;</span>
<span id="cb48-37"><a href="#cb48-37" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>&lt;a href="#6-count-matrix" id="toc-6-count-matrix"&gt;6 Count Matrix&lt;/a&gt;</span>
<span id="cb48-38"><a href="#cb48-38" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>&lt;a href="#7-deseq2" id="toc-7-deseq2"&gt;7 DEseq2&lt;/a&gt;</span>
<span id="cb48-39"><a href="#cb48-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-40"><a href="#cb48-40" aria-hidden="true" tabindex="-1"></a><span class="fu"># 1 Differentially Expressed Genes</span></span>
<span id="cb48-41"><a href="#cb48-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-42"><a href="#cb48-42" aria-hidden="true" tabindex="-1"></a>Alt splice test run</span>
<span id="cb48-43"><a href="#cb48-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-44"><a href="#cb48-44" aria-hidden="true" tabindex="-1"></a><span class="fu"># 2 Reads</span></span>
<span id="cb48-45"><a href="#cb48-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-46"><a href="#cb48-46" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb48-47"><a href="#cb48-47" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> ../data/reads/<span class="pp">*</span></span>
<span id="cb48-48"><a href="#cb48-48" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb48-49"><a href="#cb48-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-50"><a href="#cb48-50" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/100.trimmed.R1.fastq.gz</span></span>
<span id="cb48-51"><a href="#cb48-51" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/100.trimmed.R2.fastq.gz</span></span>
<span id="cb48-52"><a href="#cb48-52" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/107.trimmed.R1.fastq.gz</span></span>
<span id="cb48-53"><a href="#cb48-53" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/107.trimmed.R2.fastq.gz</span></span>
<span id="cb48-54"><a href="#cb48-54" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/108.trimmed.R1.fastq.gz</span></span>
<span id="cb48-55"><a href="#cb48-55" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/108.trimmed.R2.fastq.gz</span></span>
<span id="cb48-56"><a href="#cb48-56" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/109.trimmed.R1.fastq.gz</span></span>
<span id="cb48-57"><a href="#cb48-57" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/109.trimmed.R2.fastq.gz</span></span>
<span id="cb48-58"><a href="#cb48-58" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/10.trimmed.R1.fastq.gz</span></span>
<span id="cb48-59"><a href="#cb48-59" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/10.trimmed.R2.fastq.gz</span></span>
<span id="cb48-60"><a href="#cb48-60" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/110.trimmed.R1.fastq.gz</span></span>
<span id="cb48-61"><a href="#cb48-61" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/110.trimmed.R2.fastq.gz</span></span>
<span id="cb48-62"><a href="#cb48-62" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/116.trimmed.R1.fastq.gz</span></span>
<span id="cb48-63"><a href="#cb48-63" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/116.trimmed.R2.fastq.gz</span></span>
<span id="cb48-64"><a href="#cb48-64" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/11.trimmed.R1.fastq.gz</span></span>
<span id="cb48-65"><a href="#cb48-65" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/11.trimmed.R2.fastq.gz</span></span>
<span id="cb48-66"><a href="#cb48-66" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/12.trimmed.R1.fastq.gz</span></span>
<span id="cb48-67"><a href="#cb48-67" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/12.trimmed.R2.fastq.gz</span></span>
<span id="cb48-68"><a href="#cb48-68" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/13.trimmed.R1.fastq.gz</span></span>
<span id="cb48-69"><a href="#cb48-69" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/13.trimmed.R2.fastq.gz</span></span>
<span id="cb48-70"><a href="#cb48-70" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/18.trimmed.R1.fastq.gz</span></span>
<span id="cb48-71"><a href="#cb48-71" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/18.trimmed.R2.fastq.gz</span></span>
<span id="cb48-72"><a href="#cb48-72" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/19.trimmed.R1.fastq.gz</span></span>
<span id="cb48-73"><a href="#cb48-73" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/19.trimmed.R2.fastq.gz</span></span>
<span id="cb48-74"><a href="#cb48-74" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/1.trimmed.R1.fastq.gz</span></span>
<span id="cb48-75"><a href="#cb48-75" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/1.trimmed.R2.fastq.gz</span></span>
<span id="cb48-76"><a href="#cb48-76" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/20.trimmed.R1.fastq.gz</span></span>
<span id="cb48-77"><a href="#cb48-77" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/20.trimmed.R2.fastq.gz</span></span>
<span id="cb48-78"><a href="#cb48-78" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/21.trimmed.R1.fastq.gz</span></span>
<span id="cb48-79"><a href="#cb48-79" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/21.trimmed.R2.fastq.gz</span></span>
<span id="cb48-80"><a href="#cb48-80" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/28.trimmed.R1.fastq.gz</span></span>
<span id="cb48-81"><a href="#cb48-81" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/28.trimmed.R2.fastq.gz</span></span>
<span id="cb48-82"><a href="#cb48-82" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/29.trimmed.R1.fastq.gz</span></span>
<span id="cb48-83"><a href="#cb48-83" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/29.trimmed.R2.fastq.gz</span></span>
<span id="cb48-84"><a href="#cb48-84" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/2.trimmed.R1.fastq.gz</span></span>
<span id="cb48-85"><a href="#cb48-85" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/2.trimmed.R2.fastq.gz</span></span>
<span id="cb48-86"><a href="#cb48-86" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/30.trimmed.R1.fastq.gz</span></span>
<span id="cb48-87"><a href="#cb48-87" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/30.trimmed.R2.fastq.gz</span></span>
<span id="cb48-88"><a href="#cb48-88" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/31.trimmed.R1.fastq.gz</span></span>
<span id="cb48-89"><a href="#cb48-89" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/31.trimmed.R2.fastq.gz</span></span>
<span id="cb48-90"><a href="#cb48-90" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/36.trimmed.R1.fastq.gz</span></span>
<span id="cb48-91"><a href="#cb48-91" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/36.trimmed.R2.fastq.gz</span></span>
<span id="cb48-92"><a href="#cb48-92" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/3.trimmed.R1.fastq.gz</span></span>
<span id="cb48-93"><a href="#cb48-93" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/3.trimmed.R2.fastq.gz</span></span>
<span id="cb48-94"><a href="#cb48-94" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/4.trimmed.R1.fastq.gz</span></span>
<span id="cb48-95"><a href="#cb48-95" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/4.trimmed.R2.fastq.gz</span></span>
<span id="cb48-96"><a href="#cb48-96" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/5.trimmed.R1.fastq.gz</span></span>
<span id="cb48-97"><a href="#cb48-97" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/5.trimmed.R2.fastq.gz</span></span>
<span id="cb48-98"><a href="#cb48-98" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/78.trimmed.R1.fastq.gz</span></span>
<span id="cb48-99"><a href="#cb48-99" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/78.trimmed.R2.fastq.gz</span></span>
<span id="cb48-100"><a href="#cb48-100" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/79.trimmed.R1.fastq.gz</span></span>
<span id="cb48-101"><a href="#cb48-101" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/79.trimmed.R2.fastq.gz</span></span>
<span id="cb48-102"><a href="#cb48-102" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/80.trimmed.R1.fastq.gz</span></span>
<span id="cb48-103"><a href="#cb48-103" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/80.trimmed.R2.fastq.gz</span></span>
<span id="cb48-104"><a href="#cb48-104" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/83.trimmed.R1.fastq.gz</span></span>
<span id="cb48-105"><a href="#cb48-105" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/83.trimmed.R2.fastq.gz</span></span>
<span id="cb48-106"><a href="#cb48-106" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/88.trimmed.R1.fastq.gz</span></span>
<span id="cb48-107"><a href="#cb48-107" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/88.trimmed.R2.fastq.gz</span></span>
<span id="cb48-108"><a href="#cb48-108" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/90.trimmed.R1.fastq.gz</span></span>
<span id="cb48-109"><a href="#cb48-109" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/90.trimmed.R2.fastq.gz</span></span>
<span id="cb48-110"><a href="#cb48-110" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/91.trimmed.R1.fastq.gz</span></span>
<span id="cb48-111"><a href="#cb48-111" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/91.trimmed.R2.fastq.gz</span></span>
<span id="cb48-112"><a href="#cb48-112" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/92.trimmed.R1.fastq.gz</span></span>
<span id="cb48-113"><a href="#cb48-113" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/92.trimmed.R2.fastq.gz</span></span>
<span id="cb48-114"><a href="#cb48-114" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/94.trimmed.R1.fastq.gz</span></span>
<span id="cb48-115"><a href="#cb48-115" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/94.trimmed.R2.fastq.gz</span></span>
<span id="cb48-116"><a href="#cb48-116" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/97.trimmed.R1.fastq.gz</span></span>
<span id="cb48-117"><a href="#cb48-117" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/97.trimmed.R2.fastq.gz</span></span>
<span id="cb48-118"><a href="#cb48-118" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/98.trimmed.R1.fastq.gz</span></span>
<span id="cb48-119"><a href="#cb48-119" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/98.trimmed.R2.fastq.gz</span></span>
<span id="cb48-120"><a href="#cb48-120" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/99.trimmed.R1.fastq.gz</span></span>
<span id="cb48-121"><a href="#cb48-121" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/99.trimmed.R2.fastq.gz</span></span>
<span id="cb48-122"><a href="#cb48-122" aria-hidden="true" tabindex="-1"></a><span class="in">    ../data/reads/splice-test-files.txt</span></span>
<span id="cb48-123"><a href="#cb48-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-124"><a href="#cb48-124" aria-hidden="true" tabindex="-1"></a>Need to find treatment file…</span>
<span id="cb48-125"><a href="#cb48-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-126"><a href="#cb48-126" aria-hidden="true" tabindex="-1"></a><span class="fu"># 3 Genome</span></span>
<span id="cb48-127"><a href="#cb48-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-128"><a href="#cb48-128" aria-hidden="true" tabindex="-1"></a><span class="ot">&lt;https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_031168955.1/&gt;</span></span>
<span id="cb48-129"><a href="#cb48-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-130"><a href="#cb48-130" aria-hidden="true" tabindex="-1"></a><span class="al">![genome](http://gannet.fish.washington.edu/seashell/snaps/Monosnap_Gadus_macrocephalus_genome_assembly_ASM3116895v1_-_NCBI_-_NLM_2024-05-15_09-20-59.png)</span></span>
<span id="cb48-131"><a href="#cb48-131" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-132"><a href="#cb48-132" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb48-133"><a href="#cb48-133" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> ../data</span>
<span id="cb48-134"><a href="#cb48-134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-135"><a href="#cb48-135" aria-hidden="true" tabindex="-1"></a><span class="ex">/home/shared/datasets</span> download genome accession GCF_031168955.1 <span class="at">--include</span> gff3,gtf,rna,cds,protein,genome,seq-report</span>
<span id="cb48-136"><a href="#cb48-136" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb48-137"><a href="#cb48-137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-138"><a href="#cb48-138" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb48-139"><a href="#cb48-139" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> ../data </span>
<span id="cb48-140"><a href="#cb48-140" aria-hidden="true" tabindex="-1"></a><span class="fu">unzip</span> ncbi_dataset.zip</span>
<span id="cb48-141"><a href="#cb48-141" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb48-142"><a href="#cb48-142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-143"><a href="#cb48-143" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb48-144"><a href="#cb48-144" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> /home/shared/8TB_HDD_03/sr320/github/project-cod-temperature/data/ncbi_dataset/data/GCF_031168955.1</span>
<span id="cb48-145"><a href="#cb48-145" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb48-146"><a href="#cb48-146" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-147"><a href="#cb48-147" aria-hidden="true" tabindex="-1"></a><span class="in">    cds_from_genomic.fna</span></span>
<span id="cb48-148"><a href="#cb48-148" aria-hidden="true" tabindex="-1"></a><span class="in">    GCF_031168955.1_ASM3116895v1_genomic.fna</span></span>
<span id="cb48-149"><a href="#cb48-149" aria-hidden="true" tabindex="-1"></a><span class="in">    genomic.gff</span></span>
<span id="cb48-150"><a href="#cb48-150" aria-hidden="true" tabindex="-1"></a><span class="in">    genomic.gtf</span></span>
<span id="cb48-151"><a href="#cb48-151" aria-hidden="true" tabindex="-1"></a><span class="in">    protein.faa</span></span>
<span id="cb48-152"><a href="#cb48-152" aria-hidden="true" tabindex="-1"></a><span class="in">    rna.fna</span></span>
<span id="cb48-153"><a href="#cb48-153" aria-hidden="true" tabindex="-1"></a><span class="in">    sequence_report.jsonl</span></span>
<span id="cb48-154"><a href="#cb48-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-155"><a href="#cb48-155" aria-hidden="true" tabindex="-1"></a><span class="fu"># 4 Hisat</span></span>
<span id="cb48-156"><a href="#cb48-156" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-157"><a href="#cb48-157" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb48-158"><a href="#cb48-158" aria-hidden="true" tabindex="-1"></a><span class="ex">/home/shared/hisat2-2.2.1/hisat2_extract_exons.py</span> <span class="dt">\</span></span>
<span id="cb48-159"><a href="#cb48-159" aria-hidden="true" tabindex="-1"></a>../data/ncbi_dataset/data/GCF_031168955.1/genomic.gtf <span class="dt">\</span></span>
<span id="cb48-160"><a href="#cb48-160" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span> ../output/10-hisat-deseq2/m_exon.tab</span>
<span id="cb48-161"><a href="#cb48-161" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb48-162"><a href="#cb48-162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-163"><a href="#cb48-163" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb48-164"><a href="#cb48-164" aria-hidden="true" tabindex="-1"></a><span class="ex">/home/shared/hisat2-2.2.1/hisat2_extract_splice_sites.py</span> <span class="dt">\</span></span>
<span id="cb48-165"><a href="#cb48-165" aria-hidden="true" tabindex="-1"></a>../data/ncbi_dataset/data/GCF_031168955.1/genomic.gtf <span class="dt">\</span></span>
<span id="cb48-166"><a href="#cb48-166" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span> ../output/10-hisat-deseq2/m_spice_sites.tab</span>
<span id="cb48-167"><a href="#cb48-167" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb48-168"><a href="#cb48-168" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-169"><a href="#cb48-169" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb48-170"><a href="#cb48-170" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"10-hisat-deseq2/GCF*"</span> <span class="op">&gt;&gt;</span> ../output/.gitignore</span>
<span id="cb48-171"><a href="#cb48-171" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb48-172"><a href="#cb48-172" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-173"><a href="#cb48-173" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb48-174"><a href="#cb48-174" aria-hidden="true" tabindex="-1"></a><span class="ex">/home/shared/hisat2-2.2.1/hisat2-build</span> <span class="dt">\</span></span>
<span id="cb48-175"><a href="#cb48-175" aria-hidden="true" tabindex="-1"></a>../data/ncbi_dataset/data/GCF_031168955.1/GCF_031168955.1_ASM3116895v1_genomic.fna <span class="dt">\</span></span>
<span id="cb48-176"><a href="#cb48-176" aria-hidden="true" tabindex="-1"></a>../output/10-hisat-deseq2/GCF_031168955.1.index <span class="dt">\</span></span>
<span id="cb48-177"><a href="#cb48-177" aria-hidden="true" tabindex="-1"></a>--exon ../output/10-hisat-deseq2/m_exon.tab <span class="dt">\</span></span>
<span id="cb48-178"><a href="#cb48-178" aria-hidden="true" tabindex="-1"></a>--ss ../output/10-hisat-deseq2/m_spice_sites.tab <span class="dt">\</span></span>
<span id="cb48-179"><a href="#cb48-179" aria-hidden="true" tabindex="-1"></a>-p 20 <span class="dt">\</span></span>
<span id="cb48-180"><a href="#cb48-180" aria-hidden="true" tabindex="-1"></a>../data/ncbi_dataset/data/GCF_031168955.1/genomic.gtf <span class="dt">\</span></span>
<span id="cb48-181"><a href="#cb48-181" aria-hidden="true" tabindex="-1"></a>2<span class="op">&gt;</span> ../output/10-hisat-deseq2/hisat2-build_stats.txt</span>
<span id="cb48-182"><a href="#cb48-182" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb48-183"><a href="#cb48-183" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-184"><a href="#cb48-184" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb48-185"><a href="#cb48-185" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"10-hisat-deseq2/*sam"</span> <span class="op">&gt;&gt;</span> ../output/.gitignore</span>
<span id="cb48-186"><a href="#cb48-186" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb48-187"><a href="#cb48-187" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-188"><a href="#cb48-188" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb48-189"><a href="#cb48-189" aria-hidden="true" tabindex="-1"></a><span class="fu">find</span> ../data/reads/<span class="pp">*</span>.trimmed.R1.fastq.gz <span class="dt">\</span></span>
<span id="cb48-190"><a href="#cb48-190" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span> <span class="fu">xargs</span> basename <span class="at">-s</span> .trimmed.R1.fastq.gz <span class="kw">|</span> <span class="fu">xargs</span> <span class="at">-I{}</span> <span class="dt">\</span></span>
<span id="cb48-191"><a href="#cb48-191" aria-hidden="true" tabindex="-1"></a>/home/shared/hisat2-2.2.1/hisat2 <span class="dt">\</span></span>
<span id="cb48-192"><a href="#cb48-192" aria-hidden="true" tabindex="-1"></a>-x ../output/10-hisat-deseq2/GCF_031168955.1.index <span class="dt">\</span></span>
<span id="cb48-193"><a href="#cb48-193" aria-hidden="true" tabindex="-1"></a>--dta <span class="dt">\</span></span>
<span id="cb48-194"><a href="#cb48-194" aria-hidden="true" tabindex="-1"></a>-p 20 <span class="dt">\</span></span>
<span id="cb48-195"><a href="#cb48-195" aria-hidden="true" tabindex="-1"></a>-1 ../data/reads/{}.trimmed.R1.fastq.gz <span class="dt">\</span></span>
<span id="cb48-196"><a href="#cb48-196" aria-hidden="true" tabindex="-1"></a>-2 ../data/reads/{}.trimmed.R2.fastq.gz <span class="dt">\</span></span>
<span id="cb48-197"><a href="#cb48-197" aria-hidden="true" tabindex="-1"></a>-S ../output/10-hisat-deseq2/{}.sam <span class="dt">\</span></span>
<span id="cb48-198"><a href="#cb48-198" aria-hidden="true" tabindex="-1"></a>2<span class="op">&gt;</span> ../output/10-hisat-deseq2/hisat.out</span>
<span id="cb48-199"><a href="#cb48-199" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb48-200"><a href="#cb48-200" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-201"><a href="#cb48-201" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb48-202"><a href="#cb48-202" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"10-hisat-deseq2/*bam"</span> <span class="op">&gt;&gt;</span> ../output/.gitignore</span>
<span id="cb48-203"><a href="#cb48-203" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"10-hisat-deseq2/*bam*"</span> <span class="op">&gt;&gt;</span> ../output/.gitignore</span>
<span id="cb48-204"><a href="#cb48-204" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb48-205"><a href="#cb48-205" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-206"><a href="#cb48-206" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb48-207"><a href="#cb48-207" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> samfile <span class="kw">in</span> ../output/10-hisat-deseq2/<span class="pp">*</span>.sam<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb48-208"><a href="#cb48-208" aria-hidden="true" tabindex="-1"></a>  <span class="va">bamfile</span><span class="op">=</span><span class="st">"</span><span class="va">${samfile</span><span class="op">%</span>.sam<span class="va">}</span><span class="st">.bam"</span></span>
<span id="cb48-209"><a href="#cb48-209" aria-hidden="true" tabindex="-1"></a>  <span class="va">sorted_bamfile</span><span class="op">=</span><span class="st">"</span><span class="va">${samfile</span><span class="op">%</span>.sam<span class="va">}</span><span class="st">.sorted.bam"</span></span>
<span id="cb48-210"><a href="#cb48-210" aria-hidden="true" tabindex="-1"></a>  <span class="ex">/home/shared/samtools-1.12/samtools</span> view <span class="at">-bS</span> <span class="at">-@</span> 20 <span class="st">"</span><span class="va">$samfile</span><span class="st">"</span> <span class="op">&gt;</span> <span class="st">"</span><span class="va">$bamfile</span><span class="st">"</span></span>
<span id="cb48-211"><a href="#cb48-211" aria-hidden="true" tabindex="-1"></a>  <span class="ex">/home/shared/samtools-1.12/samtools</span> sort <span class="at">-@</span> 20 <span class="st">"</span><span class="va">$bamfile</span><span class="st">"</span> <span class="at">-o</span> <span class="st">"</span><span class="va">$sorted_bamfile</span><span class="st">"</span></span>
<span id="cb48-212"><a href="#cb48-212" aria-hidden="true" tabindex="-1"></a>  <span class="ex">/home/shared/samtools-1.12/samtools</span> index <span class="at">-@</span> 20 <span class="st">"</span><span class="va">$sorted_bamfile</span><span class="st">"</span></span>
<span id="cb48-213"><a href="#cb48-213" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb48-214"><a href="#cb48-214" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb48-215"><a href="#cb48-215" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-216"><a href="#cb48-216" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb48-217"><a href="#cb48-217" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span> ../output/10-hisat-deseq2/<span class="pp">*</span>sam</span>
<span id="cb48-218"><a href="#cb48-218" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb48-219"><a href="#cb48-219" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-220"><a href="#cb48-220" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb48-221"><a href="#cb48-221" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> ../output/10-hisat-deseq2/<span class="pp">*</span>sorted.bam <span class="kw">|</span> <span class="fu">wc</span> <span class="at">-l</span></span>
<span id="cb48-222"><a href="#cb48-222" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb48-223"><a href="#cb48-223" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-224"><a href="#cb48-224" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb48-225"><a href="#cb48-225" aria-hidden="true" tabindex="-1"></a><span class="fu">tail</span> ../output/10-hisat-deseq2/hisat.out</span>
<span id="cb48-226"><a href="#cb48-226" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb48-227"><a href="#cb48-227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-228"><a href="#cb48-228" aria-hidden="true" tabindex="-1"></a><span class="in">        ----</span></span>
<span id="cb48-229"><a href="#cb48-229" aria-hidden="true" tabindex="-1"></a><span class="in">        3794910 pairs aligned concordantly 0 times; of these:</span></span>
<span id="cb48-230"><a href="#cb48-230" aria-hidden="true" tabindex="-1"></a><span class="in">          146293 (3.85%) aligned discordantly 1 time</span></span>
<span id="cb48-231"><a href="#cb48-231" aria-hidden="true" tabindex="-1"></a><span class="in">        ----</span></span>
<span id="cb48-232"><a href="#cb48-232" aria-hidden="true" tabindex="-1"></a><span class="in">        3648617 pairs aligned 0 times concordantly or discordantly; of these:</span></span>
<span id="cb48-233"><a href="#cb48-233" aria-hidden="true" tabindex="-1"></a><span class="in">          7297234 mates make up the pairs; of these:</span></span>
<span id="cb48-234"><a href="#cb48-234" aria-hidden="true" tabindex="-1"></a><span class="in">            5640288 (77.29%) aligned 0 times</span></span>
<span id="cb48-235"><a href="#cb48-235" aria-hidden="true" tabindex="-1"></a><span class="in">            1498025 (20.53%) aligned exactly 1 time</span></span>
<span id="cb48-236"><a href="#cb48-236" aria-hidden="true" tabindex="-1"></a><span class="in">            158921 (2.18%) aligned &gt;1 times</span></span>
<span id="cb48-237"><a href="#cb48-237" aria-hidden="true" tabindex="-1"></a><span class="in">    82.81% overall alignment rate</span></span>
<span id="cb48-238"><a href="#cb48-238" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-239"><a href="#cb48-239" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb48-240"><a href="#cb48-240" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> ../output/10-hisat-deseq2/hisat.out <span class="dt">\</span></span>
<span id="cb48-241"><a href="#cb48-241" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span> <span class="fu">grep</span> <span class="st">"overall alignment rate"</span></span>
<span id="cb48-242"><a href="#cb48-242" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb48-243"><a href="#cb48-243" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-244"><a href="#cb48-244" aria-hidden="true" tabindex="-1"></a><span class="in">    87.53% overall alignment rate</span></span>
<span id="cb48-245"><a href="#cb48-245" aria-hidden="true" tabindex="-1"></a><span class="in">    86.32% overall alignment rate</span></span>
<span id="cb48-246"><a href="#cb48-246" aria-hidden="true" tabindex="-1"></a><span class="in">    81.37% overall alignment rate</span></span>
<span id="cb48-247"><a href="#cb48-247" aria-hidden="true" tabindex="-1"></a><span class="in">    86.39% overall alignment rate</span></span>
<span id="cb48-248"><a href="#cb48-248" aria-hidden="true" tabindex="-1"></a><span class="in">    83.54% overall alignment rate</span></span>
<span id="cb48-249"><a href="#cb48-249" aria-hidden="true" tabindex="-1"></a><span class="in">    87.67% overall alignment rate</span></span>
<span id="cb48-250"><a href="#cb48-250" aria-hidden="true" tabindex="-1"></a><span class="in">    86.69% overall alignment rate</span></span>
<span id="cb48-251"><a href="#cb48-251" aria-hidden="true" tabindex="-1"></a><span class="in">    86.56% overall alignment rate</span></span>
<span id="cb48-252"><a href="#cb48-252" aria-hidden="true" tabindex="-1"></a><span class="in">    84.58% overall alignment rate</span></span>
<span id="cb48-253"><a href="#cb48-253" aria-hidden="true" tabindex="-1"></a><span class="in">    81.26% overall alignment rate</span></span>
<span id="cb48-254"><a href="#cb48-254" aria-hidden="true" tabindex="-1"></a><span class="in">    83.03% overall alignment rate</span></span>
<span id="cb48-255"><a href="#cb48-255" aria-hidden="true" tabindex="-1"></a><span class="in">    86.29% overall alignment rate</span></span>
<span id="cb48-256"><a href="#cb48-256" aria-hidden="true" tabindex="-1"></a><span class="in">    84.82% overall alignment rate</span></span>
<span id="cb48-257"><a href="#cb48-257" aria-hidden="true" tabindex="-1"></a><span class="in">    82.29% overall alignment rate</span></span>
<span id="cb48-258"><a href="#cb48-258" aria-hidden="true" tabindex="-1"></a><span class="in">    83.16% overall alignment rate</span></span>
<span id="cb48-259"><a href="#cb48-259" aria-hidden="true" tabindex="-1"></a><span class="in">    86.14% overall alignment rate</span></span>
<span id="cb48-260"><a href="#cb48-260" aria-hidden="true" tabindex="-1"></a><span class="in">    81.04% overall alignment rate</span></span>
<span id="cb48-261"><a href="#cb48-261" aria-hidden="true" tabindex="-1"></a><span class="in">    87.28% overall alignment rate</span></span>
<span id="cb48-262"><a href="#cb48-262" aria-hidden="true" tabindex="-1"></a><span class="in">    84.89% overall alignment rate</span></span>
<span id="cb48-263"><a href="#cb48-263" aria-hidden="true" tabindex="-1"></a><span class="in">    87.76% overall alignment rate</span></span>
<span id="cb48-264"><a href="#cb48-264" aria-hidden="true" tabindex="-1"></a><span class="in">    88.45% overall alignment rate</span></span>
<span id="cb48-265"><a href="#cb48-265" aria-hidden="true" tabindex="-1"></a><span class="in">    83.13% overall alignment rate</span></span>
<span id="cb48-266"><a href="#cb48-266" aria-hidden="true" tabindex="-1"></a><span class="in">    83.21% overall alignment rate</span></span>
<span id="cb48-267"><a href="#cb48-267" aria-hidden="true" tabindex="-1"></a><span class="in">    84.99% overall alignment rate</span></span>
<span id="cb48-268"><a href="#cb48-268" aria-hidden="true" tabindex="-1"></a><span class="in">    86.33% overall alignment rate</span></span>
<span id="cb48-269"><a href="#cb48-269" aria-hidden="true" tabindex="-1"></a><span class="in">    86.47% overall alignment rate</span></span>
<span id="cb48-270"><a href="#cb48-270" aria-hidden="true" tabindex="-1"></a><span class="in">    84.37% overall alignment rate</span></span>
<span id="cb48-271"><a href="#cb48-271" aria-hidden="true" tabindex="-1"></a><span class="in">    87.60% overall alignment rate</span></span>
<span id="cb48-272"><a href="#cb48-272" aria-hidden="true" tabindex="-1"></a><span class="in">    75.31% overall alignment rate</span></span>
<span id="cb48-273"><a href="#cb48-273" aria-hidden="true" tabindex="-1"></a><span class="in">    81.73% overall alignment rate</span></span>
<span id="cb48-274"><a href="#cb48-274" aria-hidden="true" tabindex="-1"></a><span class="in">    84.47% overall alignment rate</span></span>
<span id="cb48-275"><a href="#cb48-275" aria-hidden="true" tabindex="-1"></a><span class="in">    80.84% overall alignment rate</span></span>
<span id="cb48-276"><a href="#cb48-276" aria-hidden="true" tabindex="-1"></a><span class="in">    86.72% overall alignment rate</span></span>
<span id="cb48-277"><a href="#cb48-277" aria-hidden="true" tabindex="-1"></a><span class="in">    87.60% overall alignment rate</span></span>
<span id="cb48-278"><a href="#cb48-278" aria-hidden="true" tabindex="-1"></a><span class="in">    81.89% overall alignment rate</span></span>
<span id="cb48-279"><a href="#cb48-279" aria-hidden="true" tabindex="-1"></a><span class="in">    82.81% overall alignment rate</span></span>
<span id="cb48-280"><a href="#cb48-280" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-281"><a href="#cb48-281" aria-hidden="true" tabindex="-1"></a><span class="fu"># 5 Stringtie</span></span>
<span id="cb48-282"><a href="#cb48-282" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-283"><a href="#cb48-283" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb48-284"><a href="#cb48-284" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"10-hisat-deseq2/*gtf"</span> <span class="op">&gt;&gt;</span> ../output/.gitignore</span>
<span id="cb48-285"><a href="#cb48-285" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb48-286"><a href="#cb48-286" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-287"><a href="#cb48-287" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb48-288"><a href="#cb48-288" aria-hidden="true" tabindex="-1"></a><span class="fu">find</span> ../output/10-hisat-deseq2/<span class="pp">*</span>sorted.bam <span class="dt">\</span></span>
<span id="cb48-289"><a href="#cb48-289" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span> <span class="fu">xargs</span> basename <span class="at">-s</span> .sorted.bam <span class="kw">|</span> <span class="fu">xargs</span> <span class="at">-I{}</span> <span class="dt">\</span></span>
<span id="cb48-290"><a href="#cb48-290" aria-hidden="true" tabindex="-1"></a>/home/shared/stringtie-2.2.1.Linux_x86_64/stringtie <span class="dt">\</span></span>
<span id="cb48-291"><a href="#cb48-291" aria-hidden="true" tabindex="-1"></a>-p 36 <span class="dt">\</span></span>
<span id="cb48-292"><a href="#cb48-292" aria-hidden="true" tabindex="-1"></a>-eB <span class="dt">\</span></span>
<span id="cb48-293"><a href="#cb48-293" aria-hidden="true" tabindex="-1"></a>-G ../data/ncbi_dataset/data/GCF_031168955.1/genomic.gff <span class="dt">\</span></span>
<span id="cb48-294"><a href="#cb48-294" aria-hidden="true" tabindex="-1"></a>-o ../output/10-hisat-deseq2/{}.gtf <span class="dt">\</span></span>
<span id="cb48-295"><a href="#cb48-295" aria-hidden="true" tabindex="-1"></a>../output/10-hisat-deseq2/{}.sorted.bam</span>
<span id="cb48-296"><a href="#cb48-296" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb48-297"><a href="#cb48-297" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-298"><a href="#cb48-298" aria-hidden="true" tabindex="-1"></a><span class="fu"># 6 Count Matrix</span></span>
<span id="cb48-299"><a href="#cb48-299" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-300"><a href="#cb48-300" aria-hidden="true" tabindex="-1"></a>list format</span>
<span id="cb48-301"><a href="#cb48-301" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-302"><a href="#cb48-302" aria-hidden="true" tabindex="-1"></a><span class="in">    RNA-ACR-140 ../output/15-Apul-hisat/RNA-ACR-140.gtf</span></span>
<span id="cb48-303"><a href="#cb48-303" aria-hidden="true" tabindex="-1"></a><span class="in">    RNA-ACR-145 ../output/15-Apul-hisat/RNA-ACR-145.gtf</span></span>
<span id="cb48-304"><a href="#cb48-304" aria-hidden="true" tabindex="-1"></a><span class="in">    RNA-ACR-173 ../output/15-Apul-hisat/RNA-ACR-173.gtf</span></span>
<span id="cb48-305"><a href="#cb48-305" aria-hidden="true" tabindex="-1"></a><span class="in">    RNA-ACR-178 ../output/15-Apul-hisat/RNA-ACR-178.gtf</span></span>
<span id="cb48-306"><a href="#cb48-306" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-307"><a href="#cb48-307" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb48-308"><a href="#cb48-308" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> ../output/10-hisat-deseq2/<span class="pp">*</span>gtf</span>
<span id="cb48-309"><a href="#cb48-309" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb48-310"><a href="#cb48-310" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-311"><a href="#cb48-311" aria-hidden="true" tabindex="-1"></a><span class="in">    ../output/10-hisat-deseq2/100.gtf</span></span>
<span id="cb48-312"><a href="#cb48-312" aria-hidden="true" tabindex="-1"></a><span class="in">    ../output/10-hisat-deseq2/107.gtf</span></span>
<span id="cb48-313"><a href="#cb48-313" aria-hidden="true" tabindex="-1"></a><span class="in">    ../output/10-hisat-deseq2/108.gtf</span></span>
<span id="cb48-314"><a href="#cb48-314" aria-hidden="true" tabindex="-1"></a><span class="in">    ../output/10-hisat-deseq2/109.gtf</span></span>
<span id="cb48-315"><a href="#cb48-315" aria-hidden="true" tabindex="-1"></a><span class="in">    ../output/10-hisat-deseq2/10.gtf</span></span>
<span id="cb48-316"><a href="#cb48-316" aria-hidden="true" tabindex="-1"></a><span class="in">    ../output/10-hisat-deseq2/110.gtf</span></span>
<span id="cb48-317"><a href="#cb48-317" aria-hidden="true" tabindex="-1"></a><span class="in">    ../output/10-hisat-deseq2/116.gtf</span></span>
<span id="cb48-318"><a href="#cb48-318" aria-hidden="true" tabindex="-1"></a><span class="in">    ../output/10-hisat-deseq2/11.gtf</span></span>
<span id="cb48-319"><a href="#cb48-319" aria-hidden="true" tabindex="-1"></a><span class="in">    ../output/10-hisat-deseq2/12.gtf</span></span>
<span id="cb48-320"><a href="#cb48-320" aria-hidden="true" tabindex="-1"></a><span class="in">    ../output/10-hisat-deseq2/13.gtf</span></span>
<span id="cb48-321"><a href="#cb48-321" aria-hidden="true" tabindex="-1"></a><span class="in">    ../output/10-hisat-deseq2/18.gtf</span></span>
<span id="cb48-322"><a href="#cb48-322" aria-hidden="true" tabindex="-1"></a><span class="in">    ../output/10-hisat-deseq2/19.gtf</span></span>
<span id="cb48-323"><a href="#cb48-323" aria-hidden="true" tabindex="-1"></a><span class="in">    ../output/10-hisat-deseq2/1.gtf</span></span>
<span id="cb48-324"><a href="#cb48-324" aria-hidden="true" tabindex="-1"></a><span class="in">    ../output/10-hisat-deseq2/20.gtf</span></span>
<span id="cb48-325"><a href="#cb48-325" aria-hidden="true" tabindex="-1"></a><span class="in">    ../output/10-hisat-deseq2/21.gtf</span></span>
<span id="cb48-326"><a href="#cb48-326" aria-hidden="true" tabindex="-1"></a><span class="in">    ../output/10-hisat-deseq2/28.gtf</span></span>
<span id="cb48-327"><a href="#cb48-327" aria-hidden="true" tabindex="-1"></a><span class="in">    ../output/10-hisat-deseq2/29.gtf</span></span>
<span id="cb48-328"><a href="#cb48-328" aria-hidden="true" tabindex="-1"></a><span class="in">    ../output/10-hisat-deseq2/2.gtf</span></span>
<span id="cb48-329"><a href="#cb48-329" aria-hidden="true" tabindex="-1"></a><span class="in">    ../output/10-hisat-deseq2/30.gtf</span></span>
<span id="cb48-330"><a href="#cb48-330" aria-hidden="true" tabindex="-1"></a><span class="in">    ../output/10-hisat-deseq2/31.gtf</span></span>
<span id="cb48-331"><a href="#cb48-331" aria-hidden="true" tabindex="-1"></a><span class="in">    ../output/10-hisat-deseq2/36.gtf</span></span>
<span id="cb48-332"><a href="#cb48-332" aria-hidden="true" tabindex="-1"></a><span class="in">    ../output/10-hisat-deseq2/3.gtf</span></span>
<span id="cb48-333"><a href="#cb48-333" aria-hidden="true" tabindex="-1"></a><span class="in">    ../output/10-hisat-deseq2/4.gtf</span></span>
<span id="cb48-334"><a href="#cb48-334" aria-hidden="true" tabindex="-1"></a><span class="in">    ../output/10-hisat-deseq2/5.gtf</span></span>
<span id="cb48-335"><a href="#cb48-335" aria-hidden="true" tabindex="-1"></a><span class="in">    ../output/10-hisat-deseq2/78.gtf</span></span>
<span id="cb48-336"><a href="#cb48-336" aria-hidden="true" tabindex="-1"></a><span class="in">    ../output/10-hisat-deseq2/79.gtf</span></span>
<span id="cb48-337"><a href="#cb48-337" aria-hidden="true" tabindex="-1"></a><span class="in">    ../output/10-hisat-deseq2/80.gtf</span></span>
<span id="cb48-338"><a href="#cb48-338" aria-hidden="true" tabindex="-1"></a><span class="in">    ../output/10-hisat-deseq2/83.gtf</span></span>
<span id="cb48-339"><a href="#cb48-339" aria-hidden="true" tabindex="-1"></a><span class="in">    ../output/10-hisat-deseq2/88.gtf</span></span>
<span id="cb48-340"><a href="#cb48-340" aria-hidden="true" tabindex="-1"></a><span class="in">    ../output/10-hisat-deseq2/90.gtf</span></span>
<span id="cb48-341"><a href="#cb48-341" aria-hidden="true" tabindex="-1"></a><span class="in">    ../output/10-hisat-deseq2/91.gtf</span></span>
<span id="cb48-342"><a href="#cb48-342" aria-hidden="true" tabindex="-1"></a><span class="in">    ../output/10-hisat-deseq2/92.gtf</span></span>
<span id="cb48-343"><a href="#cb48-343" aria-hidden="true" tabindex="-1"></a><span class="in">    ../output/10-hisat-deseq2/94.gtf</span></span>
<span id="cb48-344"><a href="#cb48-344" aria-hidden="true" tabindex="-1"></a><span class="in">    ../output/10-hisat-deseq2/97.gtf</span></span>
<span id="cb48-345"><a href="#cb48-345" aria-hidden="true" tabindex="-1"></a><span class="in">    ../output/10-hisat-deseq2/98.gtf</span></span>
<span id="cb48-346"><a href="#cb48-346" aria-hidden="true" tabindex="-1"></a><span class="in">    ../output/10-hisat-deseq2/99.gtf</span></span>
<span id="cb48-347"><a href="#cb48-347" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-348"><a href="#cb48-348" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb48-349"><a href="#cb48-349" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> ../data/list01.txt</span>
<span id="cb48-350"><a href="#cb48-350" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb48-351"><a href="#cb48-351" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-352"><a href="#cb48-352" aria-hidden="true" tabindex="-1"></a><span class="in">    1   ../output/10-hisat-deseq2/1.gtf</span></span>
<span id="cb48-353"><a href="#cb48-353" aria-hidden="true" tabindex="-1"></a><span class="in">    2   ../output/10-hisat-deseq2/2.gtf</span></span>
<span id="cb48-354"><a href="#cb48-354" aria-hidden="true" tabindex="-1"></a><span class="in">    3   ../output/10-hisat-deseq2/3.gtf</span></span>
<span id="cb48-355"><a href="#cb48-355" aria-hidden="true" tabindex="-1"></a><span class="in">    4   ../output/10-hisat-deseq2/4.gtf</span></span>
<span id="cb48-356"><a href="#cb48-356" aria-hidden="true" tabindex="-1"></a><span class="in">    5   ../output/10-hisat-deseq2/5.gtf</span></span>
<span id="cb48-357"><a href="#cb48-357" aria-hidden="true" tabindex="-1"></a><span class="in">    10  ../output/10-hisat-deseq2/10.gtf</span></span>
<span id="cb48-358"><a href="#cb48-358" aria-hidden="true" tabindex="-1"></a><span class="in">    11  ../output/10-hisat-deseq2/11.gtf</span></span>
<span id="cb48-359"><a href="#cb48-359" aria-hidden="true" tabindex="-1"></a><span class="in">    12  ../output/10-hisat-deseq2/12.gtf</span></span>
<span id="cb48-360"><a href="#cb48-360" aria-hidden="true" tabindex="-1"></a><span class="in">    13  ../output/10-hisat-deseq2/13.gtf</span></span>
<span id="cb48-361"><a href="#cb48-361" aria-hidden="true" tabindex="-1"></a><span class="in">    18  ../output/10-hisat-deseq2/18.gtf</span></span>
<span id="cb48-362"><a href="#cb48-362" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-363"><a href="#cb48-363" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb48-364"><a href="#cb48-364" aria-hidden="true" tabindex="-1"></a><span class="ex">python</span> /home/shared/stringtie-2.2.1.Linux_x86_64/prepDE.py <span class="dt">\</span></span>
<span id="cb48-365"><a href="#cb48-365" aria-hidden="true" tabindex="-1"></a>-i ../data/list01.txt <span class="dt">\</span></span>
<span id="cb48-366"><a href="#cb48-366" aria-hidden="true" tabindex="-1"></a>-g ../output/10-hisat-deseq2/gene_count_matrix.csv <span class="dt">\</span></span>
<span id="cb48-367"><a href="#cb48-367" aria-hidden="true" tabindex="-1"></a>-t ../output/10-hisat-deseq2/transcript_count_matrix.csv</span>
<span id="cb48-368"><a href="#cb48-368" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb48-369"><a href="#cb48-369" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-370"><a href="#cb48-370" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb48-371"><a href="#cb48-371" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> ../output/10-hisat-deseq2/<span class="pp">*</span>matrix.csv</span>
<span id="cb48-372"><a href="#cb48-372" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb48-373"><a href="#cb48-373" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-374"><a href="#cb48-374" aria-hidden="true" tabindex="-1"></a><span class="in">    ==&gt; ../output/10-hisat-deseq2/gene_count_matrix.csv &lt;==</span></span>
<span id="cb48-375"><a href="#cb48-375" aria-hidden="true" tabindex="-1"></a><span class="in">    gene_id,1,10,100,107,108,109,11,110,116,12,13,18,19,2,20,21,28,29,3,30,31,36,4,5,78,79,80,83,88,90,91,92,94,97,98,99</span></span>
<span id="cb48-376"><a href="#cb48-376" aria-hidden="true" tabindex="-1"></a><span class="in">    gene-LOC132462341|LOC132462341,360,464,391,346,691,408,436,509,366,373,432,385,330,288,307,346,293,347,452,452,984,230,469,400,472,311,368,312,551,631,605,366,339,517,577,235</span></span>
<span id="cb48-377"><a href="#cb48-377" aria-hidden="true" tabindex="-1"></a><span class="in">    gene-abce1|abce1,694,325,276,77,409,196,281,284,327,310,393,386,106,290,252,326,345,254,363,353,922,160,588,285,178,119,197,134,128,103,315,204,301,56,49,260</span></span>
<span id="cb48-378"><a href="#cb48-378" aria-hidden="true" tabindex="-1"></a><span class="in">    gene-si:dkey-6i22.5|si:dkey-6i22.5,0,10,13,11,49,42,14,36,0,14,9,0,0,14,3,21,26,9,39,20,10,0,0,10,48,41,60,39,49,54,20,27,7,22,63,12</span></span>
<span id="cb48-379"><a href="#cb48-379" aria-hidden="true" tabindex="-1"></a><span class="in">    gene-ube2v1|ube2v1,10,22,15,19,74,28,63,35,30,31,25,42,17,14,21,60,61,14,29,37,68,37,0,30,10,13,56,32,45,55,27,38,60,25,18,50</span></span>
<span id="cb48-380"><a href="#cb48-380" aria-hidden="true" tabindex="-1"></a><span class="in">    gene-cldn15la|cldn15la,0,9,0,0,18,0,0,0,0,27,0,0,0,0,0,0,0,0,4,35,0,0,0,4,0,0,0,0,8,32,0,0,14,0,0,0</span></span>
<span id="cb48-381"><a href="#cb48-381" aria-hidden="true" tabindex="-1"></a><span class="in">    gene-muc15|muc15,0,40,40,29,8,0,38,0,16,15,0,0,0,0,22,22,69,22,0,15,0,34,55,51,0,30,19,31,0,0,42,34,0,0,26,56</span></span>
<span id="cb48-382"><a href="#cb48-382" aria-hidden="true" tabindex="-1"></a><span class="in">    gene-pcloa|pcloa,0,3,0,0,41,7,4,0,3,0,0,0,27,0,0,0,0,4,0,10,0,23,0,0,0,0,4,4,0,0,4,0,0,22,101,0</span></span>
<span id="cb48-383"><a href="#cb48-383" aria-hidden="true" tabindex="-1"></a><span class="in">    gene-LOC132472829|LOC132472829,402,550,293,182,1082,213,234,149,444,360,531,242,269,194,359,383,395,392,408,489,310,352,168,403,462,287,310,113,457,321,396,234,455,279,229,208</span></span>
<span id="cb48-384"><a href="#cb48-384" aria-hidden="true" tabindex="-1"></a><span class="in">    gene-ifi35|ifi35,3696,2492,2615,2184,2702,2579,3631,3158,2121,2673,2893,2210,2324,2384,2561,2654,2887,2250,3143,2830,1297,1981,2971,2651,2669,2671,1790,2173,2603,3260,2186,2451,2464,2768,1935,1991</span></span>
<span id="cb48-385"><a href="#cb48-385" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-386"><a href="#cb48-386" aria-hidden="true" tabindex="-1"></a><span class="in">    ==&gt; ../output/10-hisat-deseq2/transcript_count_matrix.csv &lt;==</span></span>
<span id="cb48-387"><a href="#cb48-387" aria-hidden="true" tabindex="-1"></a><span class="in">    transcript_id,1,10,100,107,108,109,11,110,116,12,13,18,19,2,20,21,28,29,3,30,31,36,4,5,78,79,80,83,88,90,91,92,94,97,98,99</span></span>
<span id="cb48-388"><a href="#cb48-388" aria-hidden="true" tabindex="-1"></a><span class="in">    rna-XM_060037252.1,20,918,84,11,385,70,54,82,131,72,52,8,26,41,33,62,30,12,45,85,377,59,58,57,96,19,55,94,92,125,54,34,69,36,16,83</span></span>
<span id="cb48-389"><a href="#cb48-389" aria-hidden="true" tabindex="-1"></a><span class="in">    rna-XM_060070002.1,0,0,0,0,0,12,0,0,0,14,14,0,0,0,0,0,0,3,0,0,0,0,0,0,0,0,0,0,0,0,13,0,0,0,0,7</span></span>
<span id="cb48-390"><a href="#cb48-390" aria-hidden="true" tabindex="-1"></a><span class="in">    rna-XM_060056394.1,0,0,0,1,0,0,336,0,37,0,0,0,86,0,0,0,0,0,395,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1</span></span>
<span id="cb48-391"><a href="#cb48-391" aria-hidden="true" tabindex="-1"></a><span class="in">    rna-XM_060072836.1,0,0,14,4,0,0,5,0,15,0,0,0,4,0,0,5,0,0,0,11,0,0,5,0,0,12,0,0,12,3,10,0,9,0,0,0</span></span>
<span id="cb48-392"><a href="#cb48-392" aria-hidden="true" tabindex="-1"></a><span class="in">    rna-XM_060062883.1,719,38,380,279,652,462,461,381,432,0,831,479,641,171,593,82,520,625,366,533,640,311,36,876,365,322,351,342,430,324,450,410,540,336,513,370</span></span>
<span id="cb48-393"><a href="#cb48-393" aria-hidden="true" tabindex="-1"></a><span class="in">    rna-XM_060057326.1,1072,1577,801,1056,1420,909,1523,1101,951,1459,2162,1170,1983,1317,1264,1398,2075,1055,1303,1820,1496,934,966,1509,1132,928,845,948,1930,2047,1039,781,979,2259,2747,740</span></span>
<span id="cb48-394"><a href="#cb48-394" aria-hidden="true" tabindex="-1"></a><span class="in">    rna-XM_060051796.1,0,1,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,3,0,0,0,0,0,0,0,4,0,2,0,0,0,0,0,0,0,2</span></span>
<span id="cb48-395"><a href="#cb48-395" aria-hidden="true" tabindex="-1"></a><span class="in">    rna-XM_060076321.1,0,0,0,0,16,0,0,0,0,0,0,4,58,0,0,0,0,0,0,12,0,0,0,0,0,0,0,0,0,0,0,0,5,13,80,0</span></span>
<span id="cb48-396"><a href="#cb48-396" aria-hidden="true" tabindex="-1"></a><span class="in">    rna-XM_060064475.1,6,1771,834,10,33,0,1752,997,470,1572,0,0,16,713,0,0,0,5,0,679,292,656,0,1051,6,1326,0,0,0,0,592,448,0,0,0,5</span></span>
<span id="cb48-397"><a href="#cb48-397" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-398"><a href="#cb48-398" aria-hidden="true" tabindex="-1"></a><span class="fu"># 7 DEseq2</span></span>
<span id="cb48-399"><a href="#cb48-399" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-400"><a href="#cb48-400" aria-hidden="true" tabindex="-1"></a>need <span class="in">`conditions.txt`</span> file in this format</span>
<span id="cb48-401"><a href="#cb48-401" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-402"><a href="#cb48-402" aria-hidden="true" tabindex="-1"></a><span class="in">    SampleID    Condition</span></span>
<span id="cb48-403"><a href="#cb48-403" aria-hidden="true" tabindex="-1"></a><span class="in">    RNA.ACR.140 control</span></span>
<span id="cb48-404"><a href="#cb48-404" aria-hidden="true" tabindex="-1"></a><span class="in">    RNA.ACR.145 control</span></span>
<span id="cb48-405"><a href="#cb48-405" aria-hidden="true" tabindex="-1"></a><span class="in">    RNA.ACR.173 treated</span></span>
<span id="cb48-406"><a href="#cb48-406" aria-hidden="true" tabindex="-1"></a><span class="in">    RNA.ACR.178 treated</span></span>
<span id="cb48-407"><a href="#cb48-407" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-408"><a href="#cb48-408" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb48-409"><a href="#cb48-409" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> ../data/conditions.txt</span>
<span id="cb48-410"><a href="#cb48-410" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb48-411"><a href="#cb48-411" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-412"><a href="#cb48-412" aria-hidden="true" tabindex="-1"></a><span class="in">    SampleID    Condition</span></span>
<span id="cb48-413"><a href="#cb48-413" aria-hidden="true" tabindex="-1"></a><span class="in">    X1  16</span></span>
<span id="cb48-414"><a href="#cb48-414" aria-hidden="true" tabindex="-1"></a><span class="in">    X2  16</span></span>
<span id="cb48-415"><a href="#cb48-415" aria-hidden="true" tabindex="-1"></a><span class="in">    X3  16</span></span>
<span id="cb48-416"><a href="#cb48-416" aria-hidden="true" tabindex="-1"></a><span class="in">    X4  16</span></span>
<span id="cb48-417"><a href="#cb48-417" aria-hidden="true" tabindex="-1"></a><span class="in">    X5  16</span></span>
<span id="cb48-418"><a href="#cb48-418" aria-hidden="true" tabindex="-1"></a><span class="in">    X10 16</span></span>
<span id="cb48-419"><a href="#cb48-419" aria-hidden="true" tabindex="-1"></a><span class="in">    X11 16</span></span>
<span id="cb48-420"><a href="#cb48-420" aria-hidden="true" tabindex="-1"></a><span class="in">    X12 16</span></span>
<span id="cb48-421"><a href="#cb48-421" aria-hidden="true" tabindex="-1"></a><span class="in">    X13 16</span></span>
<span id="cb48-422"><a href="#cb48-422" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-423"><a href="#cb48-423" aria-hidden="true" tabindex="-1"></a><span class="in">``` r</span></span>
<span id="cb48-424"><a href="#cb48-424" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb48-425"><a href="#cb48-425" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb48-426"><a href="#cb48-426" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-427"><a href="#cb48-427" aria-hidden="true" tabindex="-1"></a><span class="in">``` r</span></span>
<span id="cb48-428"><a href="#cb48-428" aria-hidden="true" tabindex="-1"></a><span class="co"># Load gene(/transcript) count matrix and labels</span></span>
<span id="cb48-429"><a href="#cb48-429" aria-hidden="true" tabindex="-1"></a>countData <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">read.csv</span>(<span class="st">"../output/10-hisat-deseq2/gene_count_matrix.csv"</span>, <span class="at">row.names=</span><span class="st">"gene_id"</span>))</span>
<span id="cb48-430"><a href="#cb48-430" aria-hidden="true" tabindex="-1"></a>colData <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../data/conditions.txt"</span>, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb48-431"><a href="#cb48-431" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-432"><a href="#cb48-432" aria-hidden="true" tabindex="-1"></a><span class="co"># Note: The PHENO_DATA file contains information on each sample, e.g., sex or population.</span></span>
<span id="cb48-433"><a href="#cb48-433" aria-hidden="true" tabindex="-1"></a><span class="co"># The exact way to import this depends on the format of the file.</span></span>
<span id="cb48-434"><a href="#cb48-434" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-435"><a href="#cb48-435" aria-hidden="true" tabindex="-1"></a><span class="co"># Check all sample IDs in colData are also in CountData and match their orders</span></span>
<span id="cb48-436"><a href="#cb48-436" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">rownames</span>(colData) <span class="sc">%in%</span> <span class="fu">colnames</span>(countData)) <span class="co"># This should return TRUE</span></span>
<span id="cb48-437"><a href="#cb48-437" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb48-438"><a href="#cb48-438" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-439"><a href="#cb48-439" aria-hidden="true" tabindex="-1"></a><span class="in">    [1] TRUE</span></span>
<span id="cb48-440"><a href="#cb48-440" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-441"><a href="#cb48-441" aria-hidden="true" tabindex="-1"></a><span class="in">``` r</span></span>
<span id="cb48-442"><a href="#cb48-442" aria-hidden="true" tabindex="-1"></a>countData <span class="ot">&lt;-</span> countData[, <span class="fu">rownames</span>(colData)]</span>
<span id="cb48-443"><a href="#cb48-443" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">rownames</span>(colData) <span class="sc">==</span> <span class="fu">colnames</span>(countData)) <span class="co"># This should also return TRUE</span></span>
<span id="cb48-444"><a href="#cb48-444" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb48-445"><a href="#cb48-445" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-446"><a href="#cb48-446" aria-hidden="true" tabindex="-1"></a><span class="in">    [1] TRUE</span></span>
<span id="cb48-447"><a href="#cb48-447" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-448"><a href="#cb48-448" aria-hidden="true" tabindex="-1"></a><span class="in">``` r</span></span>
<span id="cb48-449"><a href="#cb48-449" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a DESeqDataSet from count matrix and labels</span></span>
<span id="cb48-450"><a href="#cb48-450" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> countData,</span>
<span id="cb48-451"><a href="#cb48-451" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> colData, <span class="at">design =</span> <span class="sc">~</span> Condition)</span>
<span id="cb48-452"><a href="#cb48-452" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-453"><a href="#cb48-453" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-454"><a href="#cb48-454" aria-hidden="true" tabindex="-1"></a><span class="co"># Run the default analysis for DESeq2 and generate results table</span></span>
<span id="cb48-455"><a href="#cb48-455" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span>
<span id="cb48-456"><a href="#cb48-456" aria-hidden="true" tabindex="-1"></a>deseq2.res <span class="ot">&lt;-</span> <span class="fu">results</span>(dds)</span>
<span id="cb48-457"><a href="#cb48-457" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-458"><a href="#cb48-458" aria-hidden="true" tabindex="-1"></a><span class="co"># Sort by adjusted p-value and display</span></span>
<span id="cb48-459"><a href="#cb48-459" aria-hidden="true" tabindex="-1"></a>resOrdered <span class="ot">&lt;-</span> deseq2.res[<span class="fu">order</span>(deseq2.res<span class="sc">$</span>padj), ]</span>
<span id="cb48-460"><a href="#cb48-460" aria-hidden="true" tabindex="-1"></a>vsd <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds, <span class="at">blind =</span> <span class="cn">FALSE</span>)</span>
<span id="cb48-461"><a href="#cb48-461" aria-hidden="true" tabindex="-1"></a><span class="fu">plotPCA</span>(vsd, <span class="at">intgroup =</span> <span class="st">"Condition"</span>)</span>
<span id="cb48-462"><a href="#cb48-462" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb48-463"><a href="#cb48-463" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-464"><a href="#cb48-464" aria-hidden="true" tabindex="-1"></a>&lt;img src="http://gannet.fish.washington.edu/seashell/snaps/Monosnap_project-cod-temperature__RStudio_Server_2024-05-22_16-10-07.png" style="display: block; margin: auto;" /&gt;</span>
<span id="cb48-465"><a href="#cb48-465" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-466"><a href="#cb48-466" aria-hidden="true" tabindex="-1"></a><span class="in">``` r</span></span>
<span id="cb48-467"><a href="#cb48-467" aria-hidden="true" tabindex="-1"></a><span class="co"># Select top 50 differentially expressed genes</span></span>
<span id="cb48-468"><a href="#cb48-468" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">results</span>(dds)</span>
<span id="cb48-469"><a href="#cb48-469" aria-hidden="true" tabindex="-1"></a>res_ordered <span class="ot">&lt;-</span> res[<span class="fu">order</span>(res<span class="sc">$</span>padj), ]</span>
<span id="cb48-470"><a href="#cb48-470" aria-hidden="true" tabindex="-1"></a>top_genes <span class="ot">&lt;-</span> <span class="fu">row.names</span>(res_ordered)[<span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>]</span>
<span id="cb48-471"><a href="#cb48-471" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-472"><a href="#cb48-472" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract counts and normalize</span></span>
<span id="cb48-473"><a href="#cb48-473" aria-hidden="true" tabindex="-1"></a>counts <span class="ot">&lt;-</span> <span class="fu">counts</span>(dds, <span class="at">normalized =</span> <span class="cn">TRUE</span>)</span>
<span id="cb48-474"><a href="#cb48-474" aria-hidden="true" tabindex="-1"></a>counts_top <span class="ot">&lt;-</span> counts[top_genes, ]</span>
<span id="cb48-475"><a href="#cb48-475" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-476"><a href="#cb48-476" aria-hidden="true" tabindex="-1"></a><span class="co"># Log-transform counts</span></span>
<span id="cb48-477"><a href="#cb48-477" aria-hidden="true" tabindex="-1"></a>log_counts_top <span class="ot">&lt;-</span> <span class="fu">log2</span>(counts_top <span class="sc">+</span> <span class="dv">1</span>)</span>
<span id="cb48-478"><a href="#cb48-478" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-479"><a href="#cb48-479" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate heatmap</span></span>
<span id="cb48-480"><a href="#cb48-480" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(log_counts_top, <span class="at">scale =</span> <span class="st">"row"</span>)</span>
<span id="cb48-481"><a href="#cb48-481" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb48-482"><a href="#cb48-482" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-483"><a href="#cb48-483" aria-hidden="true" tabindex="-1"></a>&lt;img src="http://gannet.fish.washington.edu/seashell/snaps/Monosnap_project-cod-temperature__RStudio_Server_2024-05-22_16-09-54.png" style="display: block; margin: auto;" /&gt;</span>
<span id="cb48-484"><a href="#cb48-484" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-485"><a href="#cb48-485" aria-hidden="true" tabindex="-1"></a><span class="in">``` r</span></span>
<span id="cb48-486"><a href="#cb48-486" aria-hidden="true" tabindex="-1"></a><span class="co"># Count number of hits with adjusted p-value less then 0.05</span></span>
<span id="cb48-487"><a href="#cb48-487" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(res[<span class="sc">!</span><span class="fu">is.na</span>(deseq2.res<span class="sc">$</span>padj) <span class="sc">&amp;</span> deseq2.res<span class="sc">$</span>padj <span class="sc">&lt;=</span> <span class="fl">0.05</span>, ])</span>
<span id="cb48-488"><a href="#cb48-488" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb48-489"><a href="#cb48-489" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-490"><a href="#cb48-490" aria-hidden="true" tabindex="-1"></a><span class="in">    [1] 5061    6</span></span>
<span id="cb48-491"><a href="#cb48-491" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-492"><a href="#cb48-492" aria-hidden="true" tabindex="-1"></a><span class="in">``` r</span></span>
<span id="cb48-493"><a href="#cb48-493" aria-hidden="true" tabindex="-1"></a>tmp <span class="ot">&lt;-</span> deseq2.res</span>
<span id="cb48-494"><a href="#cb48-494" aria-hidden="true" tabindex="-1"></a><span class="co"># The main plot</span></span>
<span id="cb48-495"><a href="#cb48-495" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(tmp<span class="sc">$</span>baseMean, tmp<span class="sc">$</span>log2FoldChange, <span class="at">pch=</span><span class="dv">20</span>, <span class="at">cex=</span><span class="fl">0.45</span>, <span class="at">ylim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">3</span>, <span class="dv">3</span>), <span class="at">log=</span><span class="st">"x"</span>, <span class="at">col=</span><span class="st">"darkgray"</span>,</span>
<span id="cb48-496"><a href="#cb48-496" aria-hidden="true" tabindex="-1"></a>     <span class="at">main=</span><span class="st">"DEG Dessication  (pval &lt;= 0.05)"</span>,</span>
<span id="cb48-497"><a href="#cb48-497" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab=</span><span class="st">"mean of normalized counts"</span>,</span>
<span id="cb48-498"><a href="#cb48-498" aria-hidden="true" tabindex="-1"></a>     <span class="at">ylab=</span><span class="st">"Log2 Fold Change"</span>)</span>
<span id="cb48-499"><a href="#cb48-499" aria-hidden="true" tabindex="-1"></a><span class="co"># Getting the significant points and plotting them again so they're a different color</span></span>
<span id="cb48-500"><a href="#cb48-500" aria-hidden="true" tabindex="-1"></a>tmp.sig <span class="ot">&lt;-</span> deseq2.res[<span class="sc">!</span><span class="fu">is.na</span>(deseq2.res<span class="sc">$</span>padj) <span class="sc">&amp;</span> deseq2.res<span class="sc">$</span>padj <span class="sc">&lt;=</span> <span class="fl">0.05</span>, ]</span>
<span id="cb48-501"><a href="#cb48-501" aria-hidden="true" tabindex="-1"></a><span class="fu">points</span>(tmp.sig<span class="sc">$</span>baseMean, tmp.sig<span class="sc">$</span>log2FoldChange, <span class="at">pch=</span><span class="dv">20</span>, <span class="at">cex=</span><span class="fl">0.45</span>, <span class="at">col=</span><span class="st">"red"</span>)</span>
<span id="cb48-502"><a href="#cb48-502" aria-hidden="true" tabindex="-1"></a><span class="co"># 2 FC lines</span></span>
<span id="cb48-503"><a href="#cb48-503" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">1</span>), <span class="at">col=</span><span class="st">"blue"</span>)</span>
<span id="cb48-504"><a href="#cb48-504" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb48-505"><a href="#cb48-505" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-506"><a href="#cb48-506" aria-hidden="true" tabindex="-1"></a>&lt;img src="http://gannet.fish.washington.edu/seashell/snaps/Monosnap_project-cod-temperature__RStudio_Server_2024-05-22_16-09-37.png" style="display: block; margin: auto;" /&gt;</span>
<span id="cb48-507"><a href="#cb48-507" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-508"><a href="#cb48-508" aria-hidden="true" tabindex="-1"></a><span class="in">``` r</span></span>
<span id="cb48-509"><a href="#cb48-509" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(tmp.sig, <span class="st">"../output/10-hisat-deseq2/DEGlist.tab"</span>, <span class="at">sep =</span> <span class="st">'</span><span class="sc">\t</span><span class="st">'</span>, <span class="at">row.names =</span> T)</span>
<span id="cb48-510"><a href="#cb48-510" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb48-511"><a href="#cb48-511" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-512"><a href="#cb48-512" aria-hidden="true" tabindex="-1"></a><span class="in">``` r</span></span>
<span id="cb48-513"><a href="#cb48-513" aria-hidden="true" tabindex="-1"></a>deglist <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../output/10-hisat-deseq2/DEGlist.tab"</span>, <span class="at">sep =</span> <span class="st">'</span><span class="sc">\t</span><span class="st">'</span>, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb48-514"><a href="#cb48-514" aria-hidden="true" tabindex="-1"></a>deglist<span class="sc">$</span>RowName <span class="ot">&lt;-</span> <span class="fu">rownames</span>(deglist)</span>
<span id="cb48-515"><a href="#cb48-515" aria-hidden="true" tabindex="-1"></a>deglist2 <span class="ot">&lt;-</span> deglist[, <span class="fu">c</span>(<span class="st">"RowName"</span>, <span class="st">"pvalue"</span>)] <span class="co"># Optionally, reorder the columns</span></span>
<span id="cb48-516"><a href="#cb48-516" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb48-517"><a href="#cb48-517" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-518"><a href="#cb48-518" aria-hidden="true" tabindex="-1"></a><span class="in">``` r</span></span>
<span id="cb48-519"><a href="#cb48-519" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(deglist)</span>
<span id="cb48-520"><a href="#cb48-520" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb48-521"><a href="#cb48-521" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-522"><a href="#cb48-522" aria-hidden="true" tabindex="-1"></a><span class="in">                                         baseMean log2FoldChange      lfcSE</span></span>
<span id="cb48-523"><a href="#cb48-523" aria-hidden="true" tabindex="-1"></a><span class="in">    gene-abce1|abce1                    262.70027     0.10595278 0.02812204</span></span>
<span id="cb48-524"><a href="#cb48-524" aria-hidden="true" tabindex="-1"></a><span class="in">    gene-si:dkey-6i22.5|si:dkey-6i22.5   23.18182    -0.27199275 0.07237460</span></span>
<span id="cb48-525"><a href="#cb48-525" aria-hidden="true" tabindex="-1"></a><span class="in">    gene-LOC132463139|LOC132463139      264.89473     0.29450501 0.05965925</span></span>
<span id="cb48-526"><a href="#cb48-526" aria-hidden="true" tabindex="-1"></a><span class="in">    gene-prkaa1|prkaa1                 3214.61286    -0.12678273 0.03404263</span></span>
<span id="cb48-527"><a href="#cb48-527" aria-hidden="true" tabindex="-1"></a><span class="in">    gene-snx27a|snx27a                 1220.23705    -0.06307107 0.01399906</span></span>
<span id="cb48-528"><a href="#cb48-528" aria-hidden="true" tabindex="-1"></a><span class="in">    gene-LOC132467924|LOC132467924      197.52277     0.09450712 0.03203037</span></span>
<span id="cb48-529"><a href="#cb48-529" aria-hidden="true" tabindex="-1"></a><span class="in">                                            stat       pvalue         padj</span></span>
<span id="cb48-530"><a href="#cb48-530" aria-hidden="true" tabindex="-1"></a><span class="in">    gene-abce1|abce1                    3.767607 1.648201e-04 1.695796e-03</span></span>
<span id="cb48-531"><a href="#cb48-531" aria-hidden="true" tabindex="-1"></a><span class="in">    gene-si:dkey-6i22.5|si:dkey-6i22.5 -3.758124 1.711919e-04 1.744561e-03</span></span>
<span id="cb48-532"><a href="#cb48-532" aria-hidden="true" tabindex="-1"></a><span class="in">    gene-LOC132463139|LOC132463139      4.936451 7.955683e-07 2.120046e-05</span></span>
<span id="cb48-533"><a href="#cb48-533" aria-hidden="true" tabindex="-1"></a><span class="in">    gene-prkaa1|prkaa1                 -3.724235 1.959086e-04 1.933956e-03</span></span>
<span id="cb48-534"><a href="#cb48-534" aria-hidden="true" tabindex="-1"></a><span class="in">    gene-snx27a|snx27a                 -4.505379 6.625450e-06 1.259621e-04</span></span>
<span id="cb48-535"><a href="#cb48-535" aria-hidden="true" tabindex="-1"></a><span class="in">    gene-LOC132467924|LOC132467924      2.950547 3.172118e-03 1.713888e-02</span></span>
<span id="cb48-536"><a href="#cb48-536" aria-hidden="true" tabindex="-1"></a><span class="in">                                                                  RowName</span></span>
<span id="cb48-537"><a href="#cb48-537" aria-hidden="true" tabindex="-1"></a><span class="in">    gene-abce1|abce1                                     gene-abce1|abce1</span></span>
<span id="cb48-538"><a href="#cb48-538" aria-hidden="true" tabindex="-1"></a><span class="in">    gene-si:dkey-6i22.5|si:dkey-6i22.5 gene-si:dkey-6i22.5|si:dkey-6i22.5</span></span>
<span id="cb48-539"><a href="#cb48-539" aria-hidden="true" tabindex="-1"></a><span class="in">    gene-LOC132463139|LOC132463139         gene-LOC132463139|LOC132463139</span></span>
<span id="cb48-540"><a href="#cb48-540" aria-hidden="true" tabindex="-1"></a><span class="in">    gene-prkaa1|prkaa1                                 gene-prkaa1|prkaa1</span></span>
<span id="cb48-541"><a href="#cb48-541" aria-hidden="true" tabindex="-1"></a><span class="in">    gene-snx27a|snx27a                                 gene-snx27a|snx27a</span></span>
<span id="cb48-542"><a href="#cb48-542" aria-hidden="true" tabindex="-1"></a><span class="in">    gene-LOC132467924|LOC132467924         gene-LOC132467924|LOC132467924</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




</body></html>