<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.40">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Steven Roberts">
<meta name="dcterms.date" content="2025-12-01">
<meta name="description" content="Contra">

<title>Context dependent DNA methylation :: miRNA regulation of gene expression – Tumbling Oysters</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-86daaaaad7353f9cc0c554efc1dd6d94.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-dark-f8dc6eab18fde03278982b0b35885446.css" rel="stylesheet" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap-0b0971fe3c507452e01567f62a81532a.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../../site_libs/bootstrap/bootstrap-dark-e76cdc2a28bf56c993f4e6ffb14f81c0.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="nav-fixed fullcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top quarto-banner">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Tumbling Oysters</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/sr320"> <i class="bi bi-github" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <div class="quarto-title-block"><div><h1 class="title">Context dependent DNA methylation :: miRNA regulation of gene expression</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
                  <div>
        <div class="description">
          Contra
        </div>
      </div>
                          <div class="quarto-categories">
                <div class="quarto-category">e5</div>
                <div class="quarto-category">coral</div>
              </div>
                  </div>
  </div>
    
  <div class="quarto-title-meta-author">
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-heading">Affiliation</div>
    
      <div class="quarto-title-meta-contents">
      <p class="author">Steven Roberts <a href="https://orcid.org/0000-0001-8302-1138" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
    </div>
    <div class="quarto-title-meta-contents">
          <p class="affiliation">
              <a href="https://robertslab.info">
              Professor, UW - School of Aquatic and Fishery Sciences
              </a>
            </p>
        </div>
    </div>

  <div class="quarto-title-meta">

        
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">December 1, 2025</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">





<section id="read-in-cross-species-methylation-mirna-comparison-data" class="level1">
<h1>Read in Cross-Species Methylation miRNA Comparison Data</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Read CSV from remote URL</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>biomin_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"https://gannet.fish.washington.edu/v1_web/owlshell/bu-github/ConTra/output/biomin_comparison_20251130_052506/cross_species_methylation_mirna_comparison.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 611 Columns: 15
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr  (1): og_id
dbl (11): n_species, apul_context_strength, apul_improvement, apul_r2, peve_...
lgl  (3): apul_context_dependent, peve_context_dependent, ptua_context_depen...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Display structure of the data</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">glimpse</span>(biomin_data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Rows: 611
Columns: 15
$ og_id                  &lt;chr&gt; "OG_01155", "OG_02619", "OG_06119", "OG_08793",…
$ n_species              &lt;dbl&gt; 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3,…
$ apul_context_strength  &lt;dbl&gt; 1.5210053, 1.4686122, 0.9717545, 0.8087638, 0.9…
$ apul_context_dependent &lt;lgl&gt; FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE…
$ apul_improvement       &lt;dbl&gt; 4.542280e-02, 4.080240e-04, 6.730879e-03, 7.044…
$ apul_r2                &lt;dbl&gt; 0.5495437, 0.5965734, 0.5632711, 0.6156382, 0.5…
$ peve_context_strength  &lt;dbl&gt; 0.9962671, 1.0244916, 1.2526347, 1.4507635, 1.1…
$ peve_context_dependent &lt;lgl&gt; FALSE, FALSE, TRUE, FALSE, FALSE, FALSE, FALSE,…
$ peve_improvement       &lt;dbl&gt; 2.785907e-02, 4.279548e-02, 6.041180e-02, 3.231…
$ peve_r2                &lt;dbl&gt; 0.5239709, 0.4887706, 0.7519803, 0.5892956, 0.6…
$ ptua_context_strength  &lt;dbl&gt; 1.4149060, 1.3928000, 1.4860590, 1.4495109, 1.4…
$ ptua_context_dependent &lt;lgl&gt; FALSE, TRUE, TRUE, FALSE, FALSE, FALSE, TRUE, F…
$ ptua_improvement       &lt;dbl&gt; 3.367814e-02, 6.301667e-02, 1.140166e-01, 4.800…
$ ptua_r2                &lt;dbl&gt; 0.5576365, 0.4958976, 0.5536735, 0.6081574, 0.4…
$ mean_context_strength  &lt;dbl&gt; 1.310726, 1.295301, 1.236816, 1.236346, 1.19955…</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># View column names to identify context dependent columns</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(biomin_data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "og_id"                  "n_species"              "apul_context_strength" 
 [4] "apul_context_dependent" "apul_improvement"       "apul_r2"               
 [7] "peve_context_strength"  "peve_context_dependent" "peve_improvement"      
[10] "peve_r2"                "ptua_context_strength"  "ptua_context_dependent"
[13] "ptua_improvement"       "ptua_r2"                "mean_context_strength" </code></pre>
</div>
</div>
</section>
<section id="summary-of-context-dependent-groups" class="level1">
<h1>Summary of Context Dependent Groups</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify columns that contain "context" in their name (case insensitive)</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>context_cols <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="st">"context"</span>, <span class="fu">names</span>(biomin_data), <span class="at">ignore.case =</span> <span class="cn">TRUE</span>, <span class="at">value =</span> <span class="cn">TRUE</span>)</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Context-related columns found:</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Context-related columns found:</code></pre>
</div>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(context_cols)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "apul_context_strength"  "apul_context_dependent" "peve_context_strength" 
[4] "peve_context_dependent" "ptua_context_strength"  "ptua_context_dependent"
[7] "mean_context_strength" </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter for rows where any context dependent column is TRUE</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="co"># First, let's see unique values in context columns</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">length</span>(context_cols) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (col <span class="cf">in</span> context_cols) {</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">Unique values in"</span>, col, <span class="st">":</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">unique</span>(biomin_data[[col]]))</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Unique values in apul_context_strength :
  [1] 1.5210053 1.4686122 0.9717545 0.8087638 0.9392760 1.1867517 1.1661343
  [8] 1.5266324 0.9790248 1.6806886 1.1407917 0.9352201 1.0162652 1.0759084
 [15] 1.0906613 0.7956819 1.2331926 1.0647917 1.3480270 1.0065100 0.6788373
 [22] 1.7699775 0.9817814 0.6953154 0.9697093 0.9795832 1.1846948 0.8948526
 [29] 0.8376585 0.7537790 0.9734574 1.2986062 1.3042339 1.2566377 1.2073803
 [36] 1.1719138 0.4838400 0.9116153 1.0712626 1.0302731 0.7004548 1.1107593
 [43] 0.6135444 0.9840409 0.8025398 0.8977149 0.9465684 0.3889291 0.8141095
 [50] 0.6951846 0.6790790 1.0387031 1.0217601 0.8537525 0.6037977 0.7654756
 [57] 0.9779878 1.2961206 0.7238753 0.7272980 0.9720048 0.5653736 0.9874203
 [64] 0.9261418 0.6736686 0.8024550 1.2647949 0.9824010 0.6881109 0.3984062
 [71] 0.8853583 0.9122248 1.0299710 0.9420400 0.8114961 1.1269004 0.8736419
 [78] 0.8922733 0.2592512 1.0534686 0.6377086 0.7193908 0.9016007 0.8919931
 [85] 0.5131795 1.0572084 0.9190864 0.6425073 0.9476628 0.9079487 0.9888151
 [92] 0.6163162 0.7210294 0.5506604 0.3311742 0.7299102 0.6722439 0.4242897
 [99] 0.3994733 0.5870028 1.2700770        NA 1.3787355 1.5612844 1.5826942
[106] 1.1830914 0.9200262 1.4490784 1.8053371 1.0123960 1.1363415 1.1501350
[113] 1.3385444 1.2854709 1.1493544 1.0207283 1.3098951 1.1911210 0.8752388
[120] 1.0066910 1.3683771 1.1880021 1.5110779 0.8778843 1.1780774 1.1268855
[127] 1.2006001 1.2360252 1.0489726 0.8279252 1.1145611 0.8548954 0.9624495
[134] 1.1860685 1.4929435 1.1703841 1.0182194 1.5617765 1.2675733 1.2915712
[141] 1.0185594 0.8157635 1.1920148 1.2129086 1.3030838 0.7302221 1.0549104
[148] 1.1700095 1.4198918 1.1943181 1.0296594 1.1618123 0.9681725 1.0928932
[155] 0.8833417 1.1268010 1.0813868 0.9619526 0.8739806 0.8528617 1.3603219
[162] 1.1271233 1.2460202 0.9868753 1.0457793 0.9559422 1.1463036 0.9443693
[169] 1.2890331 0.8460410 0.8616859 0.6904839 1.6004786 0.5267972 0.7319011
[176] 1.0486275 1.1499568 1.1971106 0.9933572 0.9060864 0.8549193 0.8589213
[183] 0.6600287 1.1974981 0.7560685 0.6584202 1.0042000 0.8793598 0.8490334
[190] 1.2014146 1.2491064 1.1129278 1.1947807 1.0846427 1.2458097 1.3920571
[197] 0.9489826 0.9241358 0.6221756 0.8703770 0.7863923 0.5062372 0.7492466
[204] 0.7512697 0.9140403 0.5068823 1.0500980 0.6439405 0.6937298 0.7409128
[211] 0.9435256 0.5997723 0.8153285 0.6686174 0.7933766 0.5650183 0.6202492
[218] 1.0458144 0.8489848 0.9733974 0.9932778 0.8413606 1.1129022 0.9068886
[225] 0.9338849 0.5399323 1.2580175 0.5728934 1.3942634 0.5776230 0.6070259
[232] 1.2590693 0.7783844 0.9332572 0.4554141 0.9844705 0.9177532 0.4897440
[239] 0.4974171 0.8884588 0.9048206 0.7620361 0.7492338 0.6355758 0.7745086
[246] 0.8988785 0.7004502 0.7701659 0.9155693 0.4622773 0.6532438 1.0345924
[253] 0.6919324 0.5744296 0.4194247 0.5983485 0.8423933 0.7309599 0.2878046
[260] 0.6489733 0.7660981 0.4850826 0.7223149 0.5451732 0.6056769 0.4190426
[267] 0.3951051 0.5736550 0.5338322 0.1005362 0.9479788 1.4806336 1.4733946
[274] 1.4417109 1.3969251 1.3673747 1.3036000 1.2796473 1.2419111 1.2410928
[281] 1.2339513 1.2262897 1.2107255 1.2102872 1.2102189 1.1984038 1.1869192
[288] 1.1853484 1.1771592 1.1646136 1.1600195 1.1433012 1.1420694 1.1319728
[295] 1.1250897 1.1080518 1.0994019 1.0897755 1.0816391 1.0244446 1.0119372
[302] 1.0113012 1.0095001 1.0067032 1.0064652 0.9722323 0.9688839 0.9648068
[309] 0.9609933 0.9596259 0.9585918 0.9546278 0.9426111 0.9397804 0.9299949
[316] 0.9189571 0.8796054 0.8327863 0.8110399 0.8104258 0.7959653 0.7249008
[323] 0.6964786 0.6559203 0.6386183 0.6123647 0.5120528 0.5110356 0.5070016
[330] 0.4752355 0.3181766 0.2744171 0.1409422

Unique values in apul_context_dependent :
[1] FALSE  TRUE    NA

Unique values in peve_context_strength :
  [1] 0.99626714 1.02449159 1.25263469 1.45076355 1.17445040 0.79956930
  [7] 1.16510795 0.92273479 1.35808525 0.77568342 1.09745363 1.24733457
 [13] 1.07489313 1.25458754 0.87769840 1.33917127 1.20400162 1.24427411
 [19] 0.61937307 1.04919890 0.93278295 0.93198699 1.07988651 1.00249447
 [25] 1.29297367 0.93128838 0.78095832 1.34095802 1.25945467 1.19836273
 [31] 0.70124053 1.01679781 1.00517251 1.13061201 0.64697208 0.95017198
 [37] 1.33873175 0.89646465 1.01422941 0.75206418 0.88021142 1.10525164
 [43] 0.94023458 1.29674550 0.73313502 0.97744503 0.84015171 1.33396632
 [49] 0.81615519 1.00176957 1.12296541 1.10928382 0.92776134 0.79022895
 [55] 1.17919774 0.86244003 0.85383851 0.38895441 0.92787545 0.61539113
 [61] 0.51660061 0.63759431 0.90798424 0.72352975 1.36931858 0.79297426
 [67] 0.48431794 0.72755745 0.76759946 1.14478775 0.44074229 0.45448232
 [73] 0.73371139 0.81632356 1.23352198 0.59271857 0.63161818 0.76520911
 [79] 1.08674775 0.46008801 1.17159121 0.64517903 0.84730111 0.27030874
 [85] 0.89826827 1.02940774 0.95378680 1.01525253 0.38930413 0.50313221
 [91] 0.67849890 1.08643819 1.08033786 0.75081575 0.74514370 0.30482527
 [97] 0.58757002 0.47010759 0.37384242 0.72698326         NA 1.52327334
[103] 1.38779722 1.33735922 1.31594426 1.16301358 1.27933673 1.22445709
[109] 1.22013655 1.19809717 1.10602494 1.24314076 1.06092097 1.22094345
[115] 1.19359519 1.24877790 1.19118337 1.20011511 1.10721771 1.07113952
[121] 0.91395409 0.90805196 1.17632432 1.20223857 1.29750449 1.13479927
[127] 0.85198973 1.23852454 0.97377659 1.05347390 1.19476862 1.17813712
[133] 1.21270746 1.13712496 1.02584020 1.14947264 1.28881281 1.03964051
[139] 0.84447591 1.06537942 1.08042306 1.17404962 0.89536276 1.03849217
[145] 1.32079036 1.04241187 1.19897496 1.12289152 0.88784861 1.13240293
[151] 0.31409834 0.76999260 0.92365284 1.20821553 0.86846732 1.14594258
[157] 1.12835920 1.12291005 0.44898009 0.88670657 0.96652306 1.20150543
[163] 1.17125385 0.64861834 0.92402810 1.05785827 0.91983090 0.79384044
[169] 0.64095444 0.61112882 0.67460424 0.41996047 0.76706124 0.80462681
[175] 0.80153122 0.53556279 1.02517709 1.11973541 1.08438686 0.54867661
[181] 0.83104740 0.69284560 1.15435790 1.23223587 0.32925325 0.66026316
[187] 1.15137039 0.57799979 0.86261148 0.91949035 0.73340498 1.00568694
[193] 0.78265509 1.13139543 0.68032691 0.60741037 0.50068704 0.77549221
[199] 0.25473459 0.37474014 0.89462849 0.64713383 0.53955424 0.29403184
[205] 0.37745822 0.44378433 0.52771055 0.48640639 0.60730233 0.84793002
[211] 0.71088690 0.55831453 0.99148566 0.79172468 0.49443793 0.79051981
[217] 0.61984580 0.67883407 0.53192178 0.16135035 0.50700535 0.47912362
[223] 0.39924622 0.64033772 0.53235119 0.42657142 0.50673299 0.06620029
[229] 0.58141529 1.34479192 1.31225025 1.24395644 1.24282007 1.22892159
[235] 1.19110334 1.15038711 1.14711174 1.14213707 1.11389655 1.05795908
[241] 1.04979512 1.03626541 1.01126897 0.99623873 0.92415085 0.92063887
[247] 0.91901129 0.91372043 0.89994438 0.89638919 0.89577838 0.84654246
[253] 0.83941076 0.80537758 0.75470095 0.75441620 0.73421825 0.64879956
[259] 0.60480241 0.59284432 0.54205873 0.50998914 0.48995108 0.48257571
[265] 0.31923068

Unique values in peve_context_dependent :
[1] FALSE  TRUE    NA

Unique values in ptua_context_strength :
  [1] 1.4149060 1.3928000 1.4860590 1.4495109 1.4849456 1.6114782 1.2455426
  [8] 1.1240720 1.1872014 1.0285770 1.2442072 1.2326676 1.3082528 1.0337898
 [15] 1.3727795 1.1966488 0.8942806 0.9918106 1.2619950 1.1633376 1.6073742
 [22] 0.5018816 1.1419919 1.5033297 0.9287888 1.2682021 1.2094349 0.9333425
 [29] 1.0695879 1.1997492 1.4397979 0.7956825 0.7937776 0.6864053 1.1802553
 [36] 0.9042285 1.1948569 1.1845484 0.8775869 1.1772377 1.3748497 0.7305533
 [43] 1.3469763 0.6150367 1.3548418 1.0083482 1.0963346 1.1526755 1.2417909
 [50] 1.1469886 1.0300454 0.6769978 0.8731906 1.1647562 1.0227259 1.1600103
 [57] 0.9502591 1.0905180 1.1079630 1.4151532 1.2664312 1.5436651 0.8465310
 [64] 1.0704609 0.6677958 1.1009586 0.9394141 0.9586694 1.1811509 1.0611002
 [71] 1.2700171 1.2162969 0.8191365 0.8195657 0.5305453 0.8098849 1.0241982
 [78] 0.8716133 1.1802553 1.0054205 0.7055174 1.0735794 0.6855617 1.2361614
 [85] 0.9842197 0.2731507 0.4617118 0.6375427 0.9505097 0.8526424 0.5741643
 [92] 0.5104113 0.3566194 0.8207341 1.0022485 1.0396646 0.8003053 0.9795584
 [99] 1.0833567 0.4288373 1.8332432 1.4989439 1.6018084 1.4029527 1.5130299
[106] 1.5486827 1.2597869 1.6304029 1.8811998 1.4730021 1.3153204 0.8883912
[113] 1.6724209 1.4900577 1.4325508 1.4445305 1.2442424        NA 1.3861076
[120] 1.3080677 1.5035696 1.2805398 1.1879140 1.3052371 1.2845079 1.3723581
[127] 1.2062798 1.5655934 1.4311872 1.2105032 0.8815807 1.1432456 1.5131892
[134] 1.1943152 1.2405243 1.1729523 1.1617436 1.1578730 1.2881734 1.5065581
[141] 1.2186678 1.4741894 1.3589648 1.2464541 1.3989783 1.4011298 1.1291728
[148] 1.1007402 1.0049497 1.1136929 0.7834103 1.1371853 1.1005468 1.4062393
[155] 0.6913917 0.9854329 0.9585385 1.2314569 1.4332412 1.2723484 1.0460129
[162] 1.0256640 0.9341771 1.5030831 1.0606414 0.8056531 1.0300096 1.0043698
[169] 1.1869582 1.0776977 1.1729453 1.0368704 1.2268789 1.0875312 1.0269248
[176] 1.3171084 1.0623263 1.2373284 1.0837157 1.2465059 1.2633611 0.7940659
[183] 0.7536408 0.9724537 0.8533388 1.1031084 0.8811798 0.9314258 1.1896435
[190] 0.7836523 1.2257648 1.7561475 1.2019104 1.3669221 0.4475605 1.2632253
[197] 1.5031122 1.0934763 0.8070476 1.2832160 0.9608722 1.1367811 0.8583550
[204] 0.8486230 0.7984237 1.0007019 0.8576449 1.0794262 1.5264451 1.1116005
[211] 1.0808658 0.9990668 1.2928985 0.7541299 0.7406954 1.2611974 1.2674468
[218] 0.9115597 1.0360718 0.9888021 0.9841479 1.1083526 0.6953924 0.7653758
[225] 0.6829402 0.7794138 0.4589443 0.9015460 0.9119556 1.2133065 0.9639233
[232] 1.0473459 1.3220820 1.1531126 1.0784330 1.0661141 1.3955691 1.0443655
[239] 0.9918902 0.9944223 0.8800967 1.2855380 0.7392336 1.1405782 1.2454831
[246] 0.7536202 0.6584946 1.0363671 1.2199974 0.8220159 0.9270068 1.0628450
[253] 0.9334491 1.0796553 0.9545088 0.5140624 1.4111704 1.0768536 1.0956281
[260] 0.6693501 1.1363274 0.7884408 0.7095188 0.8542455 0.6879702 0.5803579
[267] 0.9105391 0.7722967 0.7433553 0.9902182 0.4058612 1.0883539 1.0483941
[274] 1.1532600 0.8783116 1.0688953 0.8515742 0.7344220 0.6938705 1.1687729
[281] 0.9742204 1.0761619 0.6247228 0.6848879 1.3058337 1.1038883 1.0957076
[288] 0.6881710 0.6691165 0.8105987 1.1947330 0.8127902 1.0995842 1.0011162
[295] 0.8912026 1.0295142 0.7394939 0.6536069 0.7969532 0.4101780 0.7492877
[302] 0.9428848 0.8484580 0.9992072 0.5428510 1.0611807 0.6427285 0.6699944
[309] 0.7863228 1.1471034 0.7792173 0.8016012 0.5692416 0.7493976 0.7280138
[316] 0.6462151 0.9366575 0.3917691 1.8752375 1.8748981 1.8365720 1.6251565
[323] 1.5902256 1.5281611 1.5260327 1.5075041 1.4832541 1.4350747 1.4316854
[330] 1.3798743 1.3786324 1.3760189 1.3749854 1.3748288 1.3623577 1.3594297
[337] 1.3561613 1.3526470 1.3468568 1.3437551 1.3395873 1.3339894 1.3315317
[344] 1.3268290 1.3216576 1.3196924 1.3163854 1.3141797 1.3133541 1.3120328
[351] 1.2996034 1.2992670 1.2893918 1.2825196 1.2751756 1.2732174 1.2690494
[358] 1.2613364 1.2549526 1.2408649 1.2370585 1.2145719 1.2143128 1.2102717
[365] 1.2080838 1.2000141 1.1997900 1.1995144 1.1980875 1.1965785 1.1952378
[372] 1.1856142 1.1759676 1.1737987 1.1684083 1.1625975 1.1562130 1.1473461
[379] 1.1464274 1.1425534 1.1289598 1.1285714 1.1275336 1.1239780 1.1205354
[386] 1.1147864 1.1067151 1.1058399 1.0903008 1.0861473 1.0824998 1.0803419
[393] 1.0745229 1.0727695 1.0677412 1.0667246 1.0643131 1.0607802 1.0586900
[400] 1.0586081 1.0583729 1.0569797 1.0506318 1.0501887 1.0459287 1.0324508
[407] 1.0302358 1.0276669 1.0271985 1.0260012 1.0245751 1.0232299 1.0192004
[414] 1.0057687 1.0032600 1.0021127 0.9994402 0.9991554 0.9966063 0.9914873
[421] 0.9885497 0.9882614 0.9848329 0.9768307 0.9630503 0.9614221 0.9592008
[428] 0.9541078 0.9514773 0.9331183 0.9317241 0.9167096 0.8934545 0.8848535
[435] 0.8644039 0.8538326 0.8471971 0.8468868 0.8239513 0.8237795 0.8213319
[442] 0.8162255 0.7626210 0.7588087 0.7408166 0.7196337 0.7107158 0.6993804
[449] 0.6781055 0.6699571 0.6558806 0.6309829 0.6210306 0.6086447 0.6068133
[456] 0.5896320 0.5792348 0.5451309 0.5327403 0.4726444 0.3275282

Unique values in ptua_context_dependent :
[1] FALSE  TRUE    NA

Unique values in mean_context_strength :
  [1] 1.3107262 1.2953013 1.2368161 1.2363461 1.1995573 1.1992664 1.1922616
  [8] 1.1911464 1.1747705 1.1616497 1.1608175 1.1384074 1.1331370 1.1214286
 [15] 1.1137131 1.1105006 1.1104916 1.1002921 1.0764650 1.0730155 1.0729981
 [22] 1.0679487 1.0678866 1.0670465 1.0638239 1.0596912 1.0583627 1.0563844
 [29] 1.0555670 1.0506303 1.0381653 1.0370288 1.0343947 1.0245517 1.0115359
 [36] 1.0087714 1.0058096 0.9975428 0.9876930 0.9865250 0.9851720 0.9821881
 [43] 0.9669184 0.9652744 0.9635055 0.9611694 0.9610182 0.9585236 0.9573519
 [50] 0.9479809 0.9440299 0.9416616 0.9409040 0.9362459 0.9352405 0.9293086
 [57] 0.9273618 0.9251977 0.9199046 0.9192808 0.9183455 0.9155444 0.9139785
 [64] 0.9067108 0.9035943 0.8987960 0.8961757 0.8895426 0.8789537 0.8680981
 [71] 0.8653726 0.8610013 0.8609396 0.8593097 0.8585211 0.8431680 0.8431528
 [78] 0.8430319 0.8420848 0.8396590 0.8382724 0.8127164 0.8114878 0.7994878
 [85] 0.7985558 0.7865890 0.7781950 0.7651009 0.7624922 0.7545745 0.7471594
 [92] 0.7377219 0.7193289 0.7074034 0.6928555 0.6914667 0.6867064 0.6246519
 [99] 0.6188908 0.5809411 1.5516601 1.5111086 1.4902720 1.4821186 1.4504135
[106] 1.4430209 1.4212406 1.4067472 1.4006130 1.3944732 1.3821994 1.3468641
[113] 1.3424084 1.3131996 1.2977822 1.2973328 1.2913934 1.2824038 1.2677310
[120] 1.2662624 1.2621489 1.2503382 1.2489046 1.2481791 1.2413025 1.2391915
[127] 1.2247103 1.2204161 1.2189391 1.2146491 1.2044728 1.2020492 1.1963293
[134] 1.1960117 1.1955367 1.1861963 1.1837049 1.1820678 1.1809294 1.1792365
[141] 1.1716215 1.1685730 1.1672416 1.1666144 1.1645424 1.1607072 1.1587968
[148] 1.1564662 1.1545909 1.1527486 1.1514894 1.1512271 1.1498807 1.1381769
[155] 1.1359923 1.1354655 1.1291145 1.1283720 1.1265841 1.1265031 1.1250548
[162] 1.1250082 1.1245023 1.1230625 1.1227443 1.1203908 1.1192863 1.1186305
[169] 1.1166526 1.1165237 1.1153254 1.1127725 1.1121639 1.1085386 1.1083088
[176] 1.1074113 1.0993928 1.0993413 1.0975257 1.0902122 1.0881987 1.0860772
[183] 1.0832208 1.0807921 1.0733831 1.0713747 1.0680011 1.0663456 1.0611039
[190] 1.0602433 1.0581114 1.0574282 1.0569814 1.0497885 1.0496795 1.0449918
[197] 1.0440956 1.0400774 1.0394169 1.0388647 1.0387461 1.0383861 1.0363427
[204] 1.0359029 1.0351229 1.0317982 1.0287030 1.0240196 1.0166089 1.0149547
[211] 1.0085646 1.0076316 1.0075586 1.0047499 1.0026242 1.0021488 0.9992899
[218] 0.9977671 0.9970296 0.9930020 0.9927563 0.9889147 0.9877126 0.9852609
[225] 0.9837862 0.9827949 0.9764636 0.9758140 0.9711004 0.9636612 0.9598088
[232] 0.9580326 0.9578798 0.9577158 0.9564151 0.9534458 0.9519894 0.9510965
[239] 0.9484035 0.9450304 0.9391518 0.9388605 0.9320283 0.9284693 0.9255007
[246] 0.9252643 0.9180457 0.9177410 0.9171502 0.9168691 0.9141596 0.9138584
[253] 0.9138398 0.9086919 0.9077648 0.9057134 0.8982585 0.8979768 0.8970685
[260] 0.8962102 0.8946658 0.8922594 0.8905229 0.8893986 0.8891150 0.8890583
[267] 0.8886400 0.8843370 0.8827708 0.8790271 0.8778453 0.8770651 0.8743888
[274] 0.8741363 0.8739427 0.8731491 0.8702118 0.8685584 0.8581943 0.8579386
[281] 0.8575822 0.8571636 0.8557982 0.8539656 0.8534012 0.8513983 0.8478031
[288] 0.8468286 0.8466301 0.8465971 0.8395926 0.8386201 0.8356639 0.8352726
[295] 0.8319393 0.8306236 0.8279022 0.8269735 0.8269019 0.8244990 0.8232592
[302] 0.8215489 0.8169047 0.8149793 0.8145252 0.8135638 0.8120935 0.8106771
[309] 0.8078581 0.8045967 0.8013205 0.7999328 0.7968161 0.7965623 0.7883149
[316] 0.7869686 0.7863174 0.7860956 0.7810120 0.7716843 0.7644134 0.7633892
[323] 0.7579603 0.7570013 0.7530904 0.7507685 0.7487017 0.7405264 0.7369419
[330] 0.7268815 0.7224842 0.7223852 0.7206101 0.7186614 0.7114438 0.7093159
[337] 0.6944341 0.6926221 0.6754028 0.6744927 0.6607813 0.6594838 0.6591223
[344] 0.6542269 0.6365517 0.6321500 0.6007192 0.6004237 0.5927555 0.5874592
[351] 0.5842201 0.5615595 0.5530031 0.5363933 0.5202826 0.5185969 0.5070896
[358] 0.4865922 1.8752375 1.8748981 1.8365720 1.6251565 1.5902256 1.5281611
[365] 1.5260327 1.5075041 1.4832541 1.4806336 1.4733946 1.4417109 1.4350747
[372] 1.4316854 1.3969251 1.3861076 1.3798743 1.3786324 1.3760189 1.3749854
[379] 1.3748288 1.3673747 1.3623577 1.3594297 1.3561613 1.3526470 1.3468568
[386] 1.3447919 1.3437551 1.3395873 1.3339894 1.3315317 1.3268290 1.3216576
[393] 1.3196924 1.3163854 1.3141797 1.3133541 1.3122502 1.3120328 1.3036000
[400] 1.2996034 1.2992670 1.2893918 1.2825196 1.2796473 1.2751756 1.2732174
[407] 1.2690494 1.2613364 1.2549526 1.2439564 1.2428201 1.2419111 1.2410928
[414] 1.2408649 1.2370585 1.2339513 1.2289216 1.2262897 1.2145719 1.2143128
[421] 1.2107255 1.2102872 1.2102717 1.2102189 1.2080838 1.2000141 1.1997900
[428] 1.1995144 1.1984038 1.1980875 1.1965785 1.1952378 1.1911033 1.1869192
[435] 1.1856142 1.1853484 1.1771592 1.1759676 1.1737987 1.1729523 1.1684083
[442] 1.1646136 1.1625975 1.1600195 1.1562130 1.1503871 1.1473461 1.1471117
[449] 1.1464274 1.1433012 1.1425534 1.1421371 1.1420694 1.1319728 1.1289598
[456] 1.1285714 1.1275336 1.1250897 1.1239780 1.1205354 1.1147864 1.1138965
[463] 1.1080518 1.1067151 1.1058399 1.1005468 1.0994019 1.0903008 1.0897755
[470] 1.0861473 1.0824998 1.0816391 1.0803419 1.0745229 1.0727695 1.0677412
[477] 1.0667246 1.0643131 1.0607802 1.0586900 1.0586081 1.0583729 1.0579591
[484] 1.0569797 1.0506318 1.0501887 1.0497951 1.0459287 1.0368704 1.0362654
[491] 1.0324508 1.0302358 1.0276669 1.0271985 1.0260012 1.0245751 1.0244446
[498] 1.0232299 1.0192004 1.0119372 1.0113012 1.0112690 1.0095001 1.0067032
[505] 1.0064652 1.0057687 1.0032600 1.0021127 0.9994402 0.9991554 0.9966063
[512] 0.9962387 0.9914873 0.9885497 0.9882614 0.9854329 0.9848329 0.9768307
[519] 0.9724537 0.9722323 0.9688839 0.9648068 0.9630503 0.9614221 0.9609933
[526] 0.9596259 0.9592008 0.9585918 0.9585385 0.9546278 0.9541078 0.9514773
[533] 0.9428848 0.9426111 0.9397804 0.9331183 0.9317241 0.9299949 0.9241509
[540] 0.9206389 0.9190113 0.9189571 0.9167096 0.9137204 0.9105391 0.8999444
[547] 0.8963892 0.8957784 0.8934545 0.8848535 0.8796054 0.8644039 0.8538326
[554] 0.8471971 0.8468868 0.8465425 0.8394108 0.8327863 0.8239513 0.8237795
[561] 0.8213319 0.8162255 0.8110399 0.8104258 0.8053776 0.7959653 0.7626210
[568] 0.7588087 0.7547009 0.7544162 0.7408166 0.7342183 0.7249008 0.7196337
[575] 0.7107158 0.6993804 0.6964786 0.6781055 0.6699571 0.6559203 0.6558806
[582] 0.6487996 0.6386183 0.6309829 0.6210306 0.6123647 0.6086447 0.6068133
[589] 0.6048024 0.5928443 0.5896320 0.5792348 0.5451309 0.5420587 0.5327403
[596] 0.5120528 0.5110356 0.5099891 0.5070016 0.4899511 0.4825757 0.4752355
[603] 0.4726444 0.4589443 0.3275282 0.3192307 0.3181766 0.2744171 0.1409422</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter and summarize data where context dependent columns have TRUE values</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">length</span>(context_cols) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create filter for rows where any context dependent column is TRUE</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>  context_true_data <span class="ot">&lt;-</span> biomin_data <span class="sc">%&gt;%</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="fu">if_any</span>(<span class="fu">all_of</span>(context_cols), <span class="sc">~</span> . <span class="sc">==</span> <span class="cn">TRUE</span> <span class="sc">|</span> . <span class="sc">==</span> <span class="st">"TRUE"</span> <span class="sc">|</span> . <span class="sc">==</span> <span class="st">"true"</span>))</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"Number of rows with context dependent = TRUE:"</span>, <span class="fu">nrow</span>(context_true_data), <span class="st">"</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Summarize by og_id</span></span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="st">"og_id"</span> <span class="sc">%in%</span> <span class="fu">names</span>(context_true_data)) {</span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>    context_summary <span class="ot">&lt;-</span> context_true_data <span class="sc">%&gt;%</span></span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a>      <span class="fu">group_by</span>(og_id) <span class="sc">%&gt;%</span></span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>      <span class="fu">summarise</span>(</span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a>        <span class="at">n_records =</span> <span class="fu">n</span>(),</span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a>        <span class="fu">across</span>(<span class="fu">where</span>(is.numeric), <span class="sc">~</span> <span class="fu">mean</span>(.x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">.names =</span> <span class="st">"mean_{.col}"</span>),</span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a>        <span class="at">.groups =</span> <span class="st">"drop"</span></span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a>      )</span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(context_summary)</span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"Column 'og_id' not found in data</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">head</span>(context_true_data))</span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb13-24"><a href="#cb13-24" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Number of rows with context dependent = TRUE: 239 

# A tibble: 239 × 14
   og_id   n_records mean_n_species mean_apul_context_st…¹ mean_apul_improvement
   &lt;chr&gt;       &lt;int&gt;          &lt;dbl&gt;                  &lt;dbl&gt;                 &lt;dbl&gt;
 1 OG_000…         1              2                NaN                NaN       
 2 OG_001…         1              3                  0.259              0.000355
 3 OG_003…         1              1                NaN                NaN       
 4 OG_004…         1              3                  0.719              0.0805  
 5 OG_004…         1              3                  0.331              0.00357 
 6 OG_006…         1              2                  0.497              0.00731 
 7 OG_006…         1              2                  1.19               0.00175 
 8 OG_008…         1              2                NaN                NaN       
 9 OG_008…         1              1                NaN                NaN       
10 OG_011…         1              2                  1.51               0.0157  
# ℹ 229 more rows
# ℹ abbreviated name: ¹​mean_apul_context_strength
# ℹ 9 more variables: mean_apul_r2 &lt;dbl&gt;, mean_peve_context_strength &lt;dbl&gt;,
#   mean_peve_improvement &lt;dbl&gt;, mean_peve_r2 &lt;dbl&gt;,
#   mean_ptua_context_strength &lt;dbl&gt;, mean_ptua_improvement &lt;dbl&gt;,
#   mean_ptua_r2 &lt;dbl&gt;, mean_mean_context_strength &lt;dbl&gt;, mean_n_records &lt;dbl&gt;</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Display detailed view of context-dependent records</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">exists</span>(<span class="st">"context_true_data"</span>) <span class="sc">&amp;&amp;</span> <span class="fu">nrow</span>(context_true_data) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">Detailed view of context-dependent records:</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(context_true_data)</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Detailed view of context-dependent records:
# A tibble: 239 × 15
   og_id n_species apul_context_strength apul_context_dependent apul_improvement
   &lt;chr&gt;     &lt;dbl&gt;                 &lt;dbl&gt; &lt;lgl&gt;                             &lt;dbl&gt;
 1 OG_0…         3                 1.47  FALSE                          0.000408
 2 OG_0…         3                 0.972 FALSE                          0.00673 
 3 OG_0…         3                 1.17  FALSE                          0.00495 
 4 OG_0…         3                 1.53  TRUE                           0.265   
 5 OG_0…         3                 0.979 FALSE                          0.000131
 6 OG_0…         3                 1.68  FALSE                          0.0154  
 7 OG_0…         3                 1.14  FALSE                          0.0257  
 8 OG_0…         3                 0.935 FALSE                          0.00783 
 9 OG_0…         3                 1.02  FALSE                          0.00689 
10 OG_0…         3                 1.08  FALSE                          0.000887
# ℹ 229 more rows
# ℹ 10 more variables: apul_r2 &lt;dbl&gt;, peve_context_strength &lt;dbl&gt;,
#   peve_context_dependent &lt;lgl&gt;, peve_improvement &lt;dbl&gt;, peve_r2 &lt;dbl&gt;,
#   ptua_context_strength &lt;dbl&gt;, ptua_context_dependent &lt;lgl&gt;,
#   ptua_improvement &lt;dbl&gt;, ptua_r2 &lt;dbl&gt;, mean_context_strength &lt;dbl&gt;</code></pre>
</div>
</div>
</section>
<section id="species-specific-and-shared-context-dependent-ogs" class="level1">
<h1>Species-Specific and Shared Context-Dependent OGs</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify species-specific context dependent columns</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>species_context_cols <span class="ot">&lt;-</span> context_cols[<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"shared|common"</span>, context_cols, <span class="at">ignore.case =</span> <span class="cn">TRUE</span>)]</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Create summary of which OGs are TRUE for each species</span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>species_summary <span class="ot">&lt;-</span> biomin_data <span class="sc">%&gt;%</span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(og_id, <span class="fu">all_of</span>(context_cols), <span class="fu">everything</span>())</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Count how many species have TRUE for each OG</span></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">length</span>(context_cols) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>  og_species_count <span class="ot">&lt;-</span> biomin_data <span class="sc">%&gt;%</span></span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>      <span class="at">species_true_count =</span> <span class="fu">sum</span>(<span class="fu">c_across</span>(<span class="fu">all_of</span>(context_cols)) <span class="sc">==</span> <span class="cn">TRUE</span> <span class="sc">|</span> </span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>                               <span class="fu">c_across</span>(<span class="fu">all_of</span>(context_cols)) <span class="sc">==</span> <span class="st">"TRUE"</span> <span class="sc">|</span> </span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a>                               <span class="fu">c_across</span>(<span class="fu">all_of</span>(context_cols)) <span class="sc">==</span> <span class="st">"true"</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ungroup</span>()</span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a>  <span class="co"># OGs with TRUE in exactly one species (species-specific)</span></span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a>  single_species_ogs <span class="ot">&lt;-</span> og_species_count <span class="sc">%&gt;%</span></span>
<span id="cb17-21"><a href="#cb17-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(species_true_count <span class="sc">==</span> <span class="dv">1</span>)</span>
<span id="cb17-22"><a href="#cb17-22" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb17-23"><a href="#cb17-23" aria-hidden="true" tabindex="-1"></a>  <span class="co"># OGs shared across two species</span></span>
<span id="cb17-24"><a href="#cb17-24" aria-hidden="true" tabindex="-1"></a>  two_species_ogs <span class="ot">&lt;-</span> og_species_count <span class="sc">%&gt;%</span></span>
<span id="cb17-25"><a href="#cb17-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(species_true_count <span class="sc">==</span> <span class="dv">2</span>)</span>
<span id="cb17-26"><a href="#cb17-26" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb17-27"><a href="#cb17-27" aria-hidden="true" tabindex="-1"></a>  <span class="co"># OGs shared across three species</span></span>
<span id="cb17-28"><a href="#cb17-28" aria-hidden="true" tabindex="-1"></a>  three_species_ogs <span class="ot">&lt;-</span> og_species_count <span class="sc">%&gt;%</span></span>
<span id="cb17-29"><a href="#cb17-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(species_true_count <span class="sc">==</span> <span class="dv">3</span>)</span>
<span id="cb17-30"><a href="#cb17-30" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb17-31"><a href="#cb17-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"OGs with context-dependent = TRUE in exactly 1 species:"</span>, <span class="fu">nrow</span>(single_species_ogs), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb17-32"><a href="#cb17-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"OGs with context-dependent = TRUE in exactly 2 species:"</span>, <span class="fu">nrow</span>(two_species_ogs), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb17-33"><a href="#cb17-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"OGs with context-dependent = TRUE in all 3 species:"</span>, <span class="fu">nrow</span>(three_species_ogs), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb17-34"><a href="#cb17-34" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>OGs with context-dependent = TRUE in exactly 1 species: 205 
OGs with context-dependent = TRUE in exactly 2 species: 32 
OGs with context-dependent = TRUE in all 3 species: 2 </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Detailed breakdown by species</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (col <span class="cf">in</span> context_cols) {</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>  species_name <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"_context.*|context_dependent_|_methylation.*"</span>, <span class="st">""</span>, col, <span class="at">ignore.case =</span> <span class="cn">TRUE</span>)</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>  true_count <span class="ot">&lt;-</span> <span class="fu">sum</span>(biomin_data[[col]] <span class="sc">==</span> <span class="cn">TRUE</span> <span class="sc">|</span> biomin_data[[col]] <span class="sc">==</span> <span class="st">"TRUE"</span> <span class="sc">|</span> biomin_data[[col]] <span class="sc">==</span> <span class="st">"true"</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>, species_name, <span class="st">"- OGs with context-dependent = TRUE:"</span>, true_count, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>  species_ogs <span class="ot">&lt;-</span> biomin_data <span class="sc">%&gt;%</span></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(.data[[col]] <span class="sc">==</span> <span class="cn">TRUE</span> <span class="sc">|</span> .data[[col]] <span class="sc">==</span> <span class="st">"TRUE"</span> <span class="sc">|</span> .data[[col]] <span class="sc">==</span> <span class="st">"true"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(og_id, <span class="fu">all_of</span>(col), <span class="fu">where</span>(is.numeric))</span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">nrow</span>(species_ogs) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(species_ogs)</span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
 apul - OGs with context-dependent = TRUE: 0 

 apul - OGs with context-dependent = TRUE: 47 
# A tibble: 47 × 13
   og_id apul_context_dependent n_species apul_context_strength apul_improvement
   &lt;chr&gt; &lt;lgl&gt;                      &lt;dbl&gt;                 &lt;dbl&gt;            &lt;dbl&gt;
 1 OG_0… TRUE                           3                 1.53            0.265 
 2 OG_0… TRUE                           3                 1.23            0.0558
 3 OG_0… TRUE                           3                 0.695           0.170 
 4 OG_0… TRUE                           3                 0.973           0.0433
 5 OG_1… TRUE                           3                 1.30            0.0848
 6 OG_1… TRUE                           3                 1.17            0.0586
 7 OG_0… TRUE                           3                 0.389           0.0725
 8 OG_0… TRUE                           3                 1.04            0.111 
 9 OG_0… TRUE                           3                 0.982           0.0635
10 OG_0… TRUE                           3                 0.885           0.0849
# ℹ 37 more rows
# ℹ 8 more variables: apul_r2 &lt;dbl&gt;, peve_context_strength &lt;dbl&gt;,
#   peve_improvement &lt;dbl&gt;, peve_r2 &lt;dbl&gt;, ptua_context_strength &lt;dbl&gt;,
#   ptua_improvement &lt;dbl&gt;, ptua_r2 &lt;dbl&gt;, mean_context_strength &lt;dbl&gt;

 peve - OGs with context-dependent = TRUE: 0 

 peve - OGs with context-dependent = TRUE: 69 
# A tibble: 69 × 13
   og_id peve_context_dependent n_species apul_context_strength apul_improvement
   &lt;chr&gt; &lt;lgl&gt;                      &lt;dbl&gt;                 &lt;dbl&gt;            &lt;dbl&gt;
 1 OG_0… TRUE                           3                 0.972        0.00673  
 2 OG_0… TRUE                           3                 0.979        0.000131 
 3 OG_0… TRUE                           3                 1.14         0.0257   
 4 OG_0… TRUE                           3                 1.08         0.000887 
 5 OG_0… TRUE                           3                 0.796        0.00436  
 6 OG_0… TRUE                           3                 1.35         0.0291   
 7 OG_0… TRUE                           3                 1.01         0.000222 
 8 OG_0… TRUE                           3                 0.970        0.0000296
 9 OG_0… TRUE                           3                 0.895        0.00291  
10 OG_0… TRUE                           3                 0.838        0.00305  
# ℹ 59 more rows
# ℹ 8 more variables: apul_r2 &lt;dbl&gt;, peve_context_strength &lt;dbl&gt;,
#   peve_improvement &lt;dbl&gt;, peve_r2 &lt;dbl&gt;, ptua_context_strength &lt;dbl&gt;,
#   ptua_improvement &lt;dbl&gt;, ptua_r2 &lt;dbl&gt;, mean_context_strength &lt;dbl&gt;

 ptua - OGs with context-dependent = TRUE: 0 

 ptua - OGs with context-dependent = TRUE: 159 
# A tibble: 159 × 13
   og_id ptua_context_dependent n_species apul_context_strength apul_improvement
   &lt;chr&gt; &lt;lgl&gt;                      &lt;dbl&gt;                 &lt;dbl&gt;            &lt;dbl&gt;
 1 OG_0… TRUE                           3                 1.47          0.000408
 2 OG_0… TRUE                           3                 0.972         0.00673 
 3 OG_0… TRUE                           3                 1.17          0.00495 
 4 OG_0… TRUE                           3                 0.979         0.000131
 5 OG_0… TRUE                           3                 1.68          0.0154  
 6 OG_0… TRUE                           3                 0.935         0.00783 
 7 OG_0… TRUE                           3                 1.02          0.00689 
 8 OG_0… TRUE                           3                 1.08          0.000887
 9 OG_0… TRUE                           3                 0.796         0.00436 
10 OG_0… TRUE                           3                 1.23          0.0558  
# ℹ 149 more rows
# ℹ 8 more variables: apul_r2 &lt;dbl&gt;, peve_context_strength &lt;dbl&gt;,
#   peve_improvement &lt;dbl&gt;, peve_r2 &lt;dbl&gt;, ptua_context_strength &lt;dbl&gt;,
#   ptua_improvement &lt;dbl&gt;, ptua_r2 &lt;dbl&gt;, mean_context_strength &lt;dbl&gt;

 mean - OGs with context-dependent = TRUE: 0 </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Summary of shared OGs with quantitative metrics</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">exists</span>(<span class="st">"two_species_ogs"</span>) <span class="sc">&amp;&amp;</span> <span class="fu">nrow</span>(two_species_ogs) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">=== OGs Shared Across Two Species ===</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>  two_species_summary <span class="ot">&lt;-</span> two_species_ogs <span class="sc">%&gt;%</span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(og_id, <span class="fu">all_of</span>(context_cols), <span class="fu">where</span>(is.numeric))</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(two_species_summary)</span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
=== OGs Shared Across Two Species ===
# A tibble: 32 × 16
   og_id    apul_context_strength apul_context_dependent peve_context_strength
   &lt;chr&gt;                    &lt;dbl&gt; &lt;lgl&gt;                                  &lt;dbl&gt;
 1 OG_06119                 0.972 FALSE                                   1.25
 2 OG_07153                 0.979 FALSE                                   1.36
 3 OG_04467                 1.08  FALSE                                   1.25
 4 OG_07150                 0.796 FALSE                                   1.34
 5 OG_02190                 1.23  TRUE                                    1.20
 6 OG_08849                 1.01  FALSE                                   1.05
 7 OG_06616                 0.695 TRUE                                    1.00
 8 OG_01611                 1.26  FALSE                                   1.13
 9 OG_02666                 0.389 TRUE                                    1.33
10 OG_05666                 0.695 FALSE                                   1.00
# ℹ 22 more rows
# ℹ 12 more variables: peve_context_dependent &lt;lgl&gt;,
#   ptua_context_strength &lt;dbl&gt;, ptua_context_dependent &lt;lgl&gt;,
#   mean_context_strength &lt;dbl&gt;, n_species &lt;dbl&gt;, apul_improvement &lt;dbl&gt;,
#   apul_r2 &lt;dbl&gt;, peve_improvement &lt;dbl&gt;, peve_r2 &lt;dbl&gt;,
#   ptua_improvement &lt;dbl&gt;, ptua_r2 &lt;dbl&gt;, species_true_count &lt;int&gt;</code></pre>
</div>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">exists</span>(<span class="st">"three_species_ogs"</span>) <span class="sc">&amp;&amp;</span> <span class="fu">nrow</span>(three_species_ogs) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">=== OGs Shared Across All Three Species ===</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>  three_species_summary <span class="ot">&lt;-</span> three_species_ogs <span class="sc">%&gt;%</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(og_id, <span class="fu">all_of</span>(context_cols), <span class="fu">where</span>(is.numeric))</span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(three_species_summary)</span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
=== OGs Shared Across All Three Species ===
# A tibble: 2 × 16
  og_id    apul_context_strength apul_context_dependent peve_context_strength
  &lt;chr&gt;                    &lt;dbl&gt; &lt;lgl&gt;                                  &lt;dbl&gt;
1 OG_06592                 0.973 TRUE                                   0.701
2 OG_04243                 0.638 TRUE                                   1.17 
# ℹ 12 more variables: peve_context_dependent &lt;lgl&gt;,
#   ptua_context_strength &lt;dbl&gt;, ptua_context_dependent &lt;lgl&gt;,
#   mean_context_strength &lt;dbl&gt;, n_species &lt;dbl&gt;, apul_improvement &lt;dbl&gt;,
#   apul_r2 &lt;dbl&gt;, peve_improvement &lt;dbl&gt;, peve_r2 &lt;dbl&gt;,
#   ptua_improvement &lt;dbl&gt;, ptua_r2 &lt;dbl&gt;, species_true_count &lt;int&gt;</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate quantitative summary statistics for context-dependent OGs</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">exists</span>(<span class="st">"context_true_data"</span>) <span class="sc">&amp;&amp;</span> <span class="fu">nrow</span>(context_true_data) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>  numeric_cols <span class="ot">&lt;-</span> <span class="fu">names</span>(context_true_data)[<span class="fu">sapply</span>(context_true_data, is.numeric)]</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">length</span>(numeric_cols) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">=== Quantitative Summary of Context-Dependent OGs ===</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>    quant_summary <span class="ot">&lt;-</span> context_true_data <span class="sc">%&gt;%</span></span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>      <span class="fu">summarise</span>(<span class="fu">across</span>(<span class="fu">all_of</span>(numeric_cols), </span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">list</span>(<span class="at">mean =</span> <span class="sc">~</span><span class="fu">mean</span>(.x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>                            <span class="at">sd =</span> <span class="sc">~</span><span class="fu">sd</span>(.x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a>                            <span class="at">min =</span> <span class="sc">~</span><span class="fu">min</span>(.x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a>                            <span class="at">max =</span> <span class="sc">~</span><span class="fu">max</span>(.x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)),</span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a>                       <span class="at">.names =</span> <span class="st">"{.col}_{.fn}"</span>))</span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Transpose for better readability</span></span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a>    quant_summary_long <span class="ot">&lt;-</span> quant_summary <span class="sc">%&gt;%</span></span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a>      <span class="fu">pivot_longer</span>(<span class="fu">everything</span>(), <span class="at">names_to =</span> <span class="st">"metric"</span>, <span class="at">values_to =</span> <span class="st">"value"</span>)</span>
<span id="cb25-18"><a href="#cb25-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(quant_summary_long)</span>
<span id="cb25-19"><a href="#cb25-19" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb25-20"><a href="#cb25-20" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
=== Quantitative Summary of Context-Dependent OGs ===
# A tibble: 44 × 2
   metric                      value
   &lt;chr&gt;                       &lt;dbl&gt;
 1 n_species_mean             1.92  
 2 n_species_sd               0.738 
 3 n_species_min              1     
 4 n_species_max              3     
 5 apul_context_strength_mean 0.965 
 6 apul_context_strength_sd   0.313 
 7 apul_context_strength_min  0.141 
 8 apul_context_strength_max  1.81  
 9 apul_improvement_mean      0.0369
10 apul_improvement_sd        0.0440
# ℹ 34 more rows</code></pre>
</div>
</div>
</section>
<section id="summary-paragraph" class="level1">
<h1>Summary Paragraph</h1>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate dynamic summary paragraph</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">exists</span>(<span class="st">"context_cols"</span>) <span class="sc">&amp;&amp;</span> <span class="fu">length</span>(context_cols) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get counts per species</span></span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>  species_counts <span class="ot">&lt;-</span> <span class="fu">sapply</span>(context_cols, <span class="cf">function</span>(col) {</span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sum</span>(biomin_data[[col]] <span class="sc">==</span> <span class="cn">TRUE</span> <span class="sc">|</span> biomin_data[[col]] <span class="sc">==</span> <span class="st">"TRUE"</span> <span class="sc">|</span> biomin_data[[col]] <span class="sc">==</span> <span class="st">"true"</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get OG lists per species</span></span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a>  species_ogs_list <span class="ot">&lt;-</span> <span class="fu">lapply</span>(context_cols, <span class="cf">function</span>(col) {</span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a>    biomin_data <span class="sc">%&gt;%</span></span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(.data[[col]] <span class="sc">==</span> <span class="cn">TRUE</span> <span class="sc">|</span> .data[[col]] <span class="sc">==</span> <span class="st">"TRUE"</span> <span class="sc">|</span> .data[[col]] <span class="sc">==</span> <span class="st">"true"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a>      <span class="fu">pull</span>(og_id)</span>
<span id="cb27-14"><a href="#cb27-14" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb27-15"><a href="#cb27-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(species_ogs_list) <span class="ot">&lt;-</span> context_cols</span>
<span id="cb27-16"><a href="#cb27-16" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb27-17"><a href="#cb27-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Build paragraph</span></span>
<span id="cb27-18"><a href="#cb27-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">## Cross-Species Context-Dependent Methylation and miRNA Analysis</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb27-19"><a href="#cb27-19" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb27-20"><a href="#cb27-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"This analysis examined orthogroup (OG) patterns across species for context-dependent methylation and miRNA associations. "</span>)</span>
<span id="cb27-21"><a href="#cb27-21" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb27-22"><a href="#cb27-22" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Species-specific findings</span></span>
<span id="cb27-23"><a href="#cb27-23" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">seq_along</span>(context_cols)) {</span>
<span id="cb27-24"><a href="#cb27-24" aria-hidden="true" tabindex="-1"></a>    col <span class="ot">&lt;-</span> context_cols[i]</span>
<span id="cb27-25"><a href="#cb27-25" aria-hidden="true" tabindex="-1"></a>    species_name <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"_context.*|context_dependent_|_methylation.*"</span>, <span class="st">""</span>, col, <span class="at">ignore.case =</span> <span class="cn">TRUE</span>)</span>
<span id="cb27-26"><a href="#cb27-26" aria-hidden="true" tabindex="-1"></a>    ogs <span class="ot">&lt;-</span> species_ogs_list[[col]]</span>
<span id="cb27-27"><a href="#cb27-27" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">length</span>(ogs) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb27-28"><a href="#cb27-28" aria-hidden="true" tabindex="-1"></a>      <span class="fu">cat</span>(<span class="st">"**"</span>, species_name, <span class="st">"** showed context-dependent patterns in "</span>, <span class="fu">length</span>(ogs), <span class="st">" OGs ("</span>, </span>
<span id="cb27-29"><a href="#cb27-29" aria-hidden="true" tabindex="-1"></a>          <span class="fu">paste</span>(<span class="fu">head</span>(ogs, <span class="dv">5</span>), <span class="at">collapse =</span> <span class="st">", "</span>), <span class="cf">if</span>(<span class="fu">length</span>(ogs) <span class="sc">&gt;</span> <span class="dv">5</span>) <span class="st">", ..."</span> <span class="cf">else</span> <span class="st">""</span>, <span class="st">"). "</span>, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb27-30"><a href="#cb27-30" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb27-31"><a href="#cb27-31" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb27-32"><a href="#cb27-32" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb27-33"><a href="#cb27-33" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Shared patterns</span></span>
<span id="cb27-34"><a href="#cb27-34" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">exists</span>(<span class="st">"two_species_ogs"</span>) <span class="sc">&amp;&amp;</span> <span class="fu">nrow</span>(two_species_ogs) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb27-35"><a href="#cb27-35" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n\n</span><span class="st">A total of **"</span>, <span class="fu">nrow</span>(two_species_ogs), <span class="st">" OGs** exhibited context-dependent patterns shared across exactly two species"</span>, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb27-36"><a href="#cb27-36" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">nrow</span>(two_species_ogs) <span class="sc">&lt;=</span> <span class="dv">10</span>) {</span>
<span id="cb27-37"><a href="#cb27-37" aria-hidden="true" tabindex="-1"></a>      <span class="fu">cat</span>(<span class="st">" ("</span>, <span class="fu">paste</span>(two_species_ogs<span class="sc">$</span>og_id, <span class="at">collapse =</span> <span class="st">", "</span>), <span class="st">")"</span>, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb27-38"><a href="#cb27-38" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb27-39"><a href="#cb27-39" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">". "</span>)</span>
<span id="cb27-40"><a href="#cb27-40" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb27-41"><a href="#cb27-41" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb27-42"><a href="#cb27-42" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">exists</span>(<span class="st">"three_species_ogs"</span>) <span class="sc">&amp;&amp;</span> <span class="fu">nrow</span>(three_species_ogs) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb27-43"><a href="#cb27-43" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"Notably, **"</span>, <span class="fu">nrow</span>(three_species_ogs), <span class="st">" OGs** showed conserved context-dependent patterns across all three species"</span>, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb27-44"><a href="#cb27-44" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">nrow</span>(three_species_ogs) <span class="sc">&lt;=</span> <span class="dv">10</span>) {</span>
<span id="cb27-45"><a href="#cb27-45" aria-hidden="true" tabindex="-1"></a>      <span class="fu">cat</span>(<span class="st">" ("</span>, <span class="fu">paste</span>(three_species_ogs<span class="sc">$</span>og_id, <span class="at">collapse =</span> <span class="st">", "</span>), <span class="st">")"</span>, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb27-46"><a href="#cb27-46" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb27-47"><a href="#cb27-47" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">", suggesting evolutionary conservation of these regulatory mechanisms. "</span>)</span>
<span id="cb27-48"><a href="#cb27-48" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb27-49"><a href="#cb27-49" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb27-50"><a href="#cb27-50" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Quantitative highlights</span></span>
<span id="cb27-51"><a href="#cb27-51" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">exists</span>(<span class="st">"context_true_data"</span>) <span class="sc">&amp;&amp;</span> <span class="fu">nrow</span>(context_true_data) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb27-52"><a href="#cb27-52" aria-hidden="true" tabindex="-1"></a>    numeric_cols <span class="ot">&lt;-</span> <span class="fu">names</span>(context_true_data)[<span class="fu">sapply</span>(context_true_data, is.numeric)]</span>
<span id="cb27-53"><a href="#cb27-53" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">length</span>(numeric_cols) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb27-54"><a href="#cb27-54" aria-hidden="true" tabindex="-1"></a>      <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n\n</span><span class="st">Quantitatively, context-dependent OGs showed the following characteristics: "</span>)</span>
<span id="cb27-55"><a href="#cb27-55" aria-hidden="true" tabindex="-1"></a>      <span class="cf">for</span> (nc <span class="cf">in</span> <span class="fu">head</span>(numeric_cols, <span class="dv">3</span>)) {</span>
<span id="cb27-56"><a href="#cb27-56" aria-hidden="true" tabindex="-1"></a>        mean_val <span class="ot">&lt;-</span> <span class="fu">mean</span>(context_true_data[[nc]], <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb27-57"><a href="#cb27-57" aria-hidden="true" tabindex="-1"></a>        sd_val <span class="ot">&lt;-</span> <span class="fu">sd</span>(context_true_data[[nc]], <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb27-58"><a href="#cb27-58" aria-hidden="true" tabindex="-1"></a>        <span class="fu">cat</span>(<span class="st">"**"</span>, nc, <span class="st">"** (mean ± SD: "</span>, <span class="fu">round</span>(mean_val, <span class="dv">3</span>), <span class="st">" ± "</span>, <span class="fu">round</span>(sd_val, <span class="dv">3</span>), <span class="st">"); "</span>, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb27-59"><a href="#cb27-59" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb27-60"><a href="#cb27-60" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb27-61"><a href="#cb27-61" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb27-62"><a href="#cb27-62" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb27-63"><a href="#cb27-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb27-64"><a href="#cb27-64" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<section id="cross-species-context-dependent-methylation-and-mirna-analysis" class="level2">
<h2 class="anchored" data-anchor-id="cross-species-context-dependent-methylation-and-mirna-analysis">Cross-Species Context-Dependent Methylation and miRNA Analysis</h2>
<p>This analysis examined orthogroup (OG) patterns across species for context-dependent methylation and miRNA associations. <strong>apul</strong> showed context-dependent patterns in 47 OGs (OG_01452, OG_02190, OG_06616, OG_06592, OG_10121, …). <strong>peve</strong> showed context-dependent patterns in 69 OGs (OG_06119, OG_07153, OG_01753, OG_04467, OG_07150, …). <strong>ptua</strong> showed context-dependent patterns in 159 OGs (OG_02619, OG_06119, OG_05637, OG_07153, OG_01414, …).</p>
<p>A total of <strong>32 OGs</strong> exhibited context-dependent patterns shared across exactly two species. Notably, <strong>2 OGs</strong> showed conserved context-dependent patterns across all three species (OG_06592, OG_04243), suggesting evolutionary conservation of these regulatory mechanisms.</p>
<p>Quantitatively, context-dependent OGs showed the following characteristics: <strong>n_species</strong> (mean ± SD: 1.921 ± 0.738); <strong>apul_context_strength</strong> (mean ± SD: 0.965 ± 0.313); <strong>apul_improvement</strong> (mean ± SD: 0.037 ± 0.044);</p>


<!-- -->

</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
    const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
    const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
    let newTheme = '';
    if(darkModeDefault) {
      newTheme = isAlternate ? baseTheme : alternateTheme;
    } else {
      newTheme = isAlternate ? alternateTheme : baseTheme;
    }
    const changeGiscusTheme = () => {
      // From: https://github.com/giscus/giscus/issues/336
      const sendMessage = (message) => {
        const iframe = document.querySelector('iframe.giscus-frame');
        if (!iframe) return;
        iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
      }
      sendMessage({
        setConfig: {
          theme: newTheme
        }
      });
    }
    const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
    if (isGiscussLoaded) {
      changeGiscusTheme();
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  const darkModeDefault = true;
  let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
    toggleGiscusIfUsed(toAlternate, darkModeDefault);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/sr320\.github\.io\/tumbling-oysters");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb28" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Context dependent DNA methylation :: miRNA regulation of gene expression"</span></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a><span class="an">description:</span><span class="co"> "Contra"</span></span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a><span class="an">categories:</span><span class="co"> [e5, coral]</span></span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a><span class="co">#citation: </span></span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> 12-1-2025</span></span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a><span class="an">image:</span><span class="co"> http://gannet.fish.washington.edu/seashell/snaps/Monosnap_Image_2025-12-01_19-30-14.png # finding a good image</span></span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span></span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a><span class="co">  - name: Steven Roberts</span></span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a><span class="co">    url: </span></span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a><span class="co">    orcid: 0000-0001-8302-1138</span></span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a><span class="co">    affiliation: Professor, UW - School of Aquatic and Fishery Sciences</span></span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a><span class="co">    affiliation-url: https://robertslab.info</span></span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a><span class="co">  #url:  # self-defined</span></span>
<span id="cb28-15"><a href="#cb28-15" aria-hidden="true" tabindex="-1"></a><span class="an">draft:</span><span class="co"> false # setting this to `true` will prevent your post from appearing on your listing page until you're ready!</span></span>
<span id="cb28-16"><a href="#cb28-16" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb28-17"><a href="#cb28-17" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb28-18"><a href="#cb28-18" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: FALSE</span></span>
<span id="cb28-19"><a href="#cb28-19" aria-hidden="true" tabindex="-1"></a><span class="co">    code-tools: true</span></span>
<span id="cb28-20"><a href="#cb28-20" aria-hidden="true" tabindex="-1"></a><span class="co">    code-copy: true</span></span>
<span id="cb28-21"><a href="#cb28-21" aria-hidden="true" tabindex="-1"></a><span class="co">    highlight-style: github</span></span>
<span id="cb28-22"><a href="#cb28-22" aria-hidden="true" tabindex="-1"></a><span class="co">    code-overflow: wrap</span></span>
<span id="cb28-23"><a href="#cb28-23" aria-hidden="true" tabindex="-1"></a><span class="co">#runtime: shiny</span></span>
<span id="cb28-24"><a href="#cb28-24" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb28-25"><a href="#cb28-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-26"><a href="#cb28-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-27"><a href="#cb28-27" aria-hidden="true" tabindex="-1"></a><span class="in">```{r setup, include=FALSE}</span></span>
<span id="cb28-28"><a href="#cb28-28" aria-hidden="true" tabindex="-1"></a><span class="in">knitr::opts_chunk$set(echo = TRUE)</span></span>
<span id="cb28-29"><a href="#cb28-29" aria-hidden="true" tabindex="-1"></a><span class="in">library(tidyverse)</span></span>
<span id="cb28-30"><a href="#cb28-30" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-31"><a href="#cb28-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-32"><a href="#cb28-32" aria-hidden="true" tabindex="-1"></a><span class="fu"># Read in Cross-Species Methylation miRNA Comparison Data</span></span>
<span id="cb28-33"><a href="#cb28-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-34"><a href="#cb28-34" aria-hidden="true" tabindex="-1"></a><span class="in">```{r read-data}</span></span>
<span id="cb28-35"><a href="#cb28-35" aria-hidden="true" tabindex="-1"></a><span class="in"># Read CSV from remote URL</span></span>
<span id="cb28-36"><a href="#cb28-36" aria-hidden="true" tabindex="-1"></a><span class="in">biomin_data &lt;- read_csv("https://gannet.fish.washington.edu/v1_web/owlshell/bu-github/ConTra/output/biomin_comparison_20251130_052506/cross_species_methylation_mirna_comparison.csv")</span></span>
<span id="cb28-37"><a href="#cb28-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-38"><a href="#cb28-38" aria-hidden="true" tabindex="-1"></a><span class="in"># Display structure of the data</span></span>
<span id="cb28-39"><a href="#cb28-39" aria-hidden="true" tabindex="-1"></a><span class="in">glimpse(biomin_data)</span></span>
<span id="cb28-40"><a href="#cb28-40" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-41"><a href="#cb28-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-42"><a href="#cb28-42" aria-hidden="true" tabindex="-1"></a><span class="in">```{r view-columns}</span></span>
<span id="cb28-43"><a href="#cb28-43" aria-hidden="true" tabindex="-1"></a><span class="in"># View column names to identify context dependent columns</span></span>
<span id="cb28-44"><a href="#cb28-44" aria-hidden="true" tabindex="-1"></a><span class="in">colnames(biomin_data)</span></span>
<span id="cb28-45"><a href="#cb28-45" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-46"><a href="#cb28-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-47"><a href="#cb28-47" aria-hidden="true" tabindex="-1"></a><span class="fu"># Summary of Context Dependent Groups</span></span>
<span id="cb28-48"><a href="#cb28-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-49"><a href="#cb28-49" aria-hidden="true" tabindex="-1"></a><span class="in">```{r context-dependent-summary}</span></span>
<span id="cb28-50"><a href="#cb28-50" aria-hidden="true" tabindex="-1"></a><span class="in"># Identify columns that contain "context" in their name (case insensitive)</span></span>
<span id="cb28-51"><a href="#cb28-51" aria-hidden="true" tabindex="-1"></a><span class="in">context_cols &lt;- grep("context", names(biomin_data), ignore.case = TRUE, value = TRUE)</span></span>
<span id="cb28-52"><a href="#cb28-52" aria-hidden="true" tabindex="-1"></a><span class="in">cat("Context-related columns found:\n")</span></span>
<span id="cb28-53"><a href="#cb28-53" aria-hidden="true" tabindex="-1"></a><span class="in">print(context_cols)</span></span>
<span id="cb28-54"><a href="#cb28-54" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-55"><a href="#cb28-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-56"><a href="#cb28-56" aria-hidden="true" tabindex="-1"></a><span class="in">```{r filter-context-true}</span></span>
<span id="cb28-57"><a href="#cb28-57" aria-hidden="true" tabindex="-1"></a><span class="in"># Filter for rows where any context dependent column is TRUE</span></span>
<span id="cb28-58"><a href="#cb28-58" aria-hidden="true" tabindex="-1"></a><span class="in"># First, let's see unique values in context columns</span></span>
<span id="cb28-59"><a href="#cb28-59" aria-hidden="true" tabindex="-1"></a><span class="in">if (length(context_cols) &gt; 0) {</span></span>
<span id="cb28-60"><a href="#cb28-60" aria-hidden="true" tabindex="-1"></a><span class="in">  for (col in context_cols) {</span></span>
<span id="cb28-61"><a href="#cb28-61" aria-hidden="true" tabindex="-1"></a><span class="in">    cat("\nUnique values in", col, ":\n")</span></span>
<span id="cb28-62"><a href="#cb28-62" aria-hidden="true" tabindex="-1"></a><span class="in">    print(unique(biomin_data[[col]]))</span></span>
<span id="cb28-63"><a href="#cb28-63" aria-hidden="true" tabindex="-1"></a><span class="in">  }</span></span>
<span id="cb28-64"><a href="#cb28-64" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb28-65"><a href="#cb28-65" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-66"><a href="#cb28-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-67"><a href="#cb28-67" aria-hidden="true" tabindex="-1"></a><span class="in">```{r summarize-context-dependent}</span></span>
<span id="cb28-68"><a href="#cb28-68" aria-hidden="true" tabindex="-1"></a><span class="in"># Filter and summarize data where context dependent columns have TRUE values</span></span>
<span id="cb28-69"><a href="#cb28-69" aria-hidden="true" tabindex="-1"></a><span class="in">if (length(context_cols) &gt; 0) {</span></span>
<span id="cb28-70"><a href="#cb28-70" aria-hidden="true" tabindex="-1"></a><span class="in">  # Create filter for rows where any context dependent column is TRUE</span></span>
<span id="cb28-71"><a href="#cb28-71" aria-hidden="true" tabindex="-1"></a><span class="in">  context_true_data &lt;- biomin_data %&gt;%</span></span>
<span id="cb28-72"><a href="#cb28-72" aria-hidden="true" tabindex="-1"></a><span class="in">    filter(if_any(all_of(context_cols), ~ . == TRUE | . == "TRUE" | . == "true"))</span></span>
<span id="cb28-73"><a href="#cb28-73" aria-hidden="true" tabindex="-1"></a><span class="in">  </span></span>
<span id="cb28-74"><a href="#cb28-74" aria-hidden="true" tabindex="-1"></a><span class="in">  cat("Number of rows with context dependent = TRUE:", nrow(context_true_data), "\n\n")</span></span>
<span id="cb28-75"><a href="#cb28-75" aria-hidden="true" tabindex="-1"></a><span class="in">  </span></span>
<span id="cb28-76"><a href="#cb28-76" aria-hidden="true" tabindex="-1"></a><span class="in">  # Summarize by og_id</span></span>
<span id="cb28-77"><a href="#cb28-77" aria-hidden="true" tabindex="-1"></a><span class="in">  if ("og_id" %in% names(context_true_data)) {</span></span>
<span id="cb28-78"><a href="#cb28-78" aria-hidden="true" tabindex="-1"></a><span class="in">    context_summary &lt;- context_true_data %&gt;%</span></span>
<span id="cb28-79"><a href="#cb28-79" aria-hidden="true" tabindex="-1"></a><span class="in">      group_by(og_id) %&gt;%</span></span>
<span id="cb28-80"><a href="#cb28-80" aria-hidden="true" tabindex="-1"></a><span class="in">      summarise(</span></span>
<span id="cb28-81"><a href="#cb28-81" aria-hidden="true" tabindex="-1"></a><span class="in">        n_records = n(),</span></span>
<span id="cb28-82"><a href="#cb28-82" aria-hidden="true" tabindex="-1"></a><span class="in">        across(where(is.numeric), ~ mean(.x, na.rm = TRUE), .names = "mean_{.col}"),</span></span>
<span id="cb28-83"><a href="#cb28-83" aria-hidden="true" tabindex="-1"></a><span class="in">        .groups = "drop"</span></span>
<span id="cb28-84"><a href="#cb28-84" aria-hidden="true" tabindex="-1"></a><span class="in">      )</span></span>
<span id="cb28-85"><a href="#cb28-85" aria-hidden="true" tabindex="-1"></a><span class="in">    </span></span>
<span id="cb28-86"><a href="#cb28-86" aria-hidden="true" tabindex="-1"></a><span class="in">    print(context_summary)</span></span>
<span id="cb28-87"><a href="#cb28-87" aria-hidden="true" tabindex="-1"></a><span class="in">  } else {</span></span>
<span id="cb28-88"><a href="#cb28-88" aria-hidden="true" tabindex="-1"></a><span class="in">    cat("Column 'og_id' not found in data\n")</span></span>
<span id="cb28-89"><a href="#cb28-89" aria-hidden="true" tabindex="-1"></a><span class="in">    print(head(context_true_data))</span></span>
<span id="cb28-90"><a href="#cb28-90" aria-hidden="true" tabindex="-1"></a><span class="in">  }</span></span>
<span id="cb28-91"><a href="#cb28-91" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb28-92"><a href="#cb28-92" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-93"><a href="#cb28-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-94"><a href="#cb28-94" aria-hidden="true" tabindex="-1"></a><span class="in">```{r detailed-view}</span></span>
<span id="cb28-95"><a href="#cb28-95" aria-hidden="true" tabindex="-1"></a><span class="in"># Display detailed view of context-dependent records</span></span>
<span id="cb28-96"><a href="#cb28-96" aria-hidden="true" tabindex="-1"></a><span class="in">if (exists("context_true_data") &amp;&amp; nrow(context_true_data) &gt; 0) {</span></span>
<span id="cb28-97"><a href="#cb28-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-98"><a href="#cb28-98" aria-hidden="true" tabindex="-1"></a><span class="in">  cat("\nDetailed view of context-dependent records:\n")</span></span>
<span id="cb28-99"><a href="#cb28-99" aria-hidden="true" tabindex="-1"></a><span class="in">  print(context_true_data)</span></span>
<span id="cb28-100"><a href="#cb28-100" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb28-101"><a href="#cb28-101" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-102"><a href="#cb28-102" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-103"><a href="#cb28-103" aria-hidden="true" tabindex="-1"></a><span class="fu"># Species-Specific and Shared Context-Dependent OGs</span></span>
<span id="cb28-104"><a href="#cb28-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-105"><a href="#cb28-105" aria-hidden="true" tabindex="-1"></a><span class="in">```{r species-analysis}</span></span>
<span id="cb28-106"><a href="#cb28-106" aria-hidden="true" tabindex="-1"></a><span class="in"># Identify species-specific context dependent columns</span></span>
<span id="cb28-107"><a href="#cb28-107" aria-hidden="true" tabindex="-1"></a><span class="in">species_context_cols &lt;- context_cols[!grepl("shared|common", context_cols, ignore.case = TRUE)]</span></span>
<span id="cb28-108"><a href="#cb28-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-109"><a href="#cb28-109" aria-hidden="true" tabindex="-1"></a><span class="in"># Create summary of which OGs are TRUE for each species</span></span>
<span id="cb28-110"><a href="#cb28-110" aria-hidden="true" tabindex="-1"></a><span class="in">species_summary &lt;- biomin_data %&gt;%</span></span>
<span id="cb28-111"><a href="#cb28-111" aria-hidden="true" tabindex="-1"></a><span class="in">  select(og_id, all_of(context_cols), everything())</span></span>
<span id="cb28-112"><a href="#cb28-112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-113"><a href="#cb28-113" aria-hidden="true" tabindex="-1"></a><span class="in"># Count how many species have TRUE for each OG</span></span>
<span id="cb28-114"><a href="#cb28-114" aria-hidden="true" tabindex="-1"></a><span class="in">if (length(context_cols) &gt; 0) {</span></span>
<span id="cb28-115"><a href="#cb28-115" aria-hidden="true" tabindex="-1"></a><span class="in">  og_species_count &lt;- biomin_data %&gt;%</span></span>
<span id="cb28-116"><a href="#cb28-116" aria-hidden="true" tabindex="-1"></a><span class="in">    rowwise() %&gt;%</span></span>
<span id="cb28-117"><a href="#cb28-117" aria-hidden="true" tabindex="-1"></a><span class="in">    mutate(</span></span>
<span id="cb28-118"><a href="#cb28-118" aria-hidden="true" tabindex="-1"></a><span class="in">      species_true_count = sum(c_across(all_of(context_cols)) == TRUE | </span></span>
<span id="cb28-119"><a href="#cb28-119" aria-hidden="true" tabindex="-1"></a><span class="in">                               c_across(all_of(context_cols)) == "TRUE" | </span></span>
<span id="cb28-120"><a href="#cb28-120" aria-hidden="true" tabindex="-1"></a><span class="in">                               c_across(all_of(context_cols)) == "true", na.rm = TRUE)</span></span>
<span id="cb28-121"><a href="#cb28-121" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb28-122"><a href="#cb28-122" aria-hidden="true" tabindex="-1"></a><span class="in">    ungroup()</span></span>
<span id="cb28-123"><a href="#cb28-123" aria-hidden="true" tabindex="-1"></a><span class="in">  </span></span>
<span id="cb28-124"><a href="#cb28-124" aria-hidden="true" tabindex="-1"></a><span class="in">  # OGs with TRUE in exactly one species (species-specific)</span></span>
<span id="cb28-125"><a href="#cb28-125" aria-hidden="true" tabindex="-1"></a><span class="in">  single_species_ogs &lt;- og_species_count %&gt;%</span></span>
<span id="cb28-126"><a href="#cb28-126" aria-hidden="true" tabindex="-1"></a><span class="in">    filter(species_true_count == 1)</span></span>
<span id="cb28-127"><a href="#cb28-127" aria-hidden="true" tabindex="-1"></a><span class="in">  </span></span>
<span id="cb28-128"><a href="#cb28-128" aria-hidden="true" tabindex="-1"></a><span class="in">  # OGs shared across two species</span></span>
<span id="cb28-129"><a href="#cb28-129" aria-hidden="true" tabindex="-1"></a><span class="in">  two_species_ogs &lt;- og_species_count %&gt;%</span></span>
<span id="cb28-130"><a href="#cb28-130" aria-hidden="true" tabindex="-1"></a><span class="in">    filter(species_true_count == 2)</span></span>
<span id="cb28-131"><a href="#cb28-131" aria-hidden="true" tabindex="-1"></a><span class="in">  </span></span>
<span id="cb28-132"><a href="#cb28-132" aria-hidden="true" tabindex="-1"></a><span class="in">  # OGs shared across three species</span></span>
<span id="cb28-133"><a href="#cb28-133" aria-hidden="true" tabindex="-1"></a><span class="in">  three_species_ogs &lt;- og_species_count %&gt;%</span></span>
<span id="cb28-134"><a href="#cb28-134" aria-hidden="true" tabindex="-1"></a><span class="in">    filter(species_true_count == 3)</span></span>
<span id="cb28-135"><a href="#cb28-135" aria-hidden="true" tabindex="-1"></a><span class="in">  </span></span>
<span id="cb28-136"><a href="#cb28-136" aria-hidden="true" tabindex="-1"></a><span class="in">  cat("OGs with context-dependent = TRUE in exactly 1 species:", nrow(single_species_ogs), "\n")</span></span>
<span id="cb28-137"><a href="#cb28-137" aria-hidden="true" tabindex="-1"></a><span class="in">  cat("OGs with context-dependent = TRUE in exactly 2 species:", nrow(two_species_ogs), "\n")</span></span>
<span id="cb28-138"><a href="#cb28-138" aria-hidden="true" tabindex="-1"></a><span class="in">  cat("OGs with context-dependent = TRUE in all 3 species:", nrow(three_species_ogs), "\n")</span></span>
<span id="cb28-139"><a href="#cb28-139" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb28-140"><a href="#cb28-140" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-141"><a href="#cb28-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-142"><a href="#cb28-142" aria-hidden="true" tabindex="-1"></a><span class="in">```{r species-specific-details}</span></span>
<span id="cb28-143"><a href="#cb28-143" aria-hidden="true" tabindex="-1"></a><span class="in"># Detailed breakdown by species</span></span>
<span id="cb28-144"><a href="#cb28-144" aria-hidden="true" tabindex="-1"></a><span class="in">for (col in context_cols) {</span></span>
<span id="cb28-145"><a href="#cb28-145" aria-hidden="true" tabindex="-1"></a><span class="in">  species_name &lt;- gsub("_context.*|context_dependent_|_methylation.*", "", col, ignore.case = TRUE)</span></span>
<span id="cb28-146"><a href="#cb28-146" aria-hidden="true" tabindex="-1"></a><span class="in">  true_count &lt;- sum(biomin_data[[col]] == TRUE | biomin_data[[col]] == "TRUE" | biomin_data[[col]] == "true", na.rm = TRUE)</span></span>
<span id="cb28-147"><a href="#cb28-147" aria-hidden="true" tabindex="-1"></a><span class="in">  cat("\n", species_name, "- OGs with context-dependent = TRUE:", true_count, "\n")</span></span>
<span id="cb28-148"><a href="#cb28-148" aria-hidden="true" tabindex="-1"></a><span class="in">  </span></span>
<span id="cb28-149"><a href="#cb28-149" aria-hidden="true" tabindex="-1"></a><span class="in">  species_ogs &lt;- biomin_data %&gt;%</span></span>
<span id="cb28-150"><a href="#cb28-150" aria-hidden="true" tabindex="-1"></a><span class="in">    filter(.data[[col]] == TRUE | .data[[col]] == "TRUE" | .data[[col]] == "true") %&gt;%</span></span>
<span id="cb28-151"><a href="#cb28-151" aria-hidden="true" tabindex="-1"></a><span class="in">    select(og_id, all_of(col), where(is.numeric))</span></span>
<span id="cb28-152"><a href="#cb28-152" aria-hidden="true" tabindex="-1"></a><span class="in">  </span></span>
<span id="cb28-153"><a href="#cb28-153" aria-hidden="true" tabindex="-1"></a><span class="in">  if (nrow(species_ogs) &gt; 0) {</span></span>
<span id="cb28-154"><a href="#cb28-154" aria-hidden="true" tabindex="-1"></a><span class="in">    print(species_ogs)</span></span>
<span id="cb28-155"><a href="#cb28-155" aria-hidden="true" tabindex="-1"></a><span class="in">  }</span></span>
<span id="cb28-156"><a href="#cb28-156" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb28-157"><a href="#cb28-157" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-158"><a href="#cb28-158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-159"><a href="#cb28-159" aria-hidden="true" tabindex="-1"></a><span class="in">```{r shared-ogs-summary}</span></span>
<span id="cb28-160"><a href="#cb28-160" aria-hidden="true" tabindex="-1"></a><span class="in"># Summary of shared OGs with quantitative metrics</span></span>
<span id="cb28-161"><a href="#cb28-161" aria-hidden="true" tabindex="-1"></a><span class="in">if (exists("two_species_ogs") &amp;&amp; nrow(two_species_ogs) &gt; 0) {</span></span>
<span id="cb28-162"><a href="#cb28-162" aria-hidden="true" tabindex="-1"></a><span class="in">  cat("\n=== OGs Shared Across Two Species ===\n")</span></span>
<span id="cb28-163"><a href="#cb28-163" aria-hidden="true" tabindex="-1"></a><span class="in">  two_species_summary &lt;- two_species_ogs %&gt;%</span></span>
<span id="cb28-164"><a href="#cb28-164" aria-hidden="true" tabindex="-1"></a><span class="in">    select(og_id, all_of(context_cols), where(is.numeric))</span></span>
<span id="cb28-165"><a href="#cb28-165" aria-hidden="true" tabindex="-1"></a><span class="in">  print(two_species_summary)</span></span>
<span id="cb28-166"><a href="#cb28-166" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb28-167"><a href="#cb28-167" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-168"><a href="#cb28-168" aria-hidden="true" tabindex="-1"></a><span class="in">if (exists("three_species_ogs") &amp;&amp; nrow(three_species_ogs) &gt; 0) {</span></span>
<span id="cb28-169"><a href="#cb28-169" aria-hidden="true" tabindex="-1"></a><span class="in">  cat("\n=== OGs Shared Across All Three Species ===\n")</span></span>
<span id="cb28-170"><a href="#cb28-170" aria-hidden="true" tabindex="-1"></a><span class="in">  three_species_summary &lt;- three_species_ogs %&gt;%</span></span>
<span id="cb28-171"><a href="#cb28-171" aria-hidden="true" tabindex="-1"></a><span class="in">    select(og_id, all_of(context_cols), where(is.numeric))</span></span>
<span id="cb28-172"><a href="#cb28-172" aria-hidden="true" tabindex="-1"></a><span class="in">  print(three_species_summary)</span></span>
<span id="cb28-173"><a href="#cb28-173" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb28-174"><a href="#cb28-174" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-175"><a href="#cb28-175" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-176"><a href="#cb28-176" aria-hidden="true" tabindex="-1"></a><span class="in">```{r quantitative-summary}</span></span>
<span id="cb28-177"><a href="#cb28-177" aria-hidden="true" tabindex="-1"></a><span class="in"># Generate quantitative summary statistics for context-dependent OGs</span></span>
<span id="cb28-178"><a href="#cb28-178" aria-hidden="true" tabindex="-1"></a><span class="in">if (exists("context_true_data") &amp;&amp; nrow(context_true_data) &gt; 0) {</span></span>
<span id="cb28-179"><a href="#cb28-179" aria-hidden="true" tabindex="-1"></a><span class="in">  numeric_cols &lt;- names(context_true_data)[sapply(context_true_data, is.numeric)]</span></span>
<span id="cb28-180"><a href="#cb28-180" aria-hidden="true" tabindex="-1"></a><span class="in">  </span></span>
<span id="cb28-181"><a href="#cb28-181" aria-hidden="true" tabindex="-1"></a><span class="in">  if (length(numeric_cols) &gt; 0) {</span></span>
<span id="cb28-182"><a href="#cb28-182" aria-hidden="true" tabindex="-1"></a><span class="in">    cat("\n=== Quantitative Summary of Context-Dependent OGs ===\n")</span></span>
<span id="cb28-183"><a href="#cb28-183" aria-hidden="true" tabindex="-1"></a><span class="in">    quant_summary &lt;- context_true_data %&gt;%</span></span>
<span id="cb28-184"><a href="#cb28-184" aria-hidden="true" tabindex="-1"></a><span class="in">      summarise(across(all_of(numeric_cols), </span></span>
<span id="cb28-185"><a href="#cb28-185" aria-hidden="true" tabindex="-1"></a><span class="in">                       list(mean = ~mean(.x, na.rm = TRUE),</span></span>
<span id="cb28-186"><a href="#cb28-186" aria-hidden="true" tabindex="-1"></a><span class="in">                            sd = ~sd(.x, na.rm = TRUE),</span></span>
<span id="cb28-187"><a href="#cb28-187" aria-hidden="true" tabindex="-1"></a><span class="in">                            min = ~min(.x, na.rm = TRUE),</span></span>
<span id="cb28-188"><a href="#cb28-188" aria-hidden="true" tabindex="-1"></a><span class="in">                            max = ~max(.x, na.rm = TRUE)),</span></span>
<span id="cb28-189"><a href="#cb28-189" aria-hidden="true" tabindex="-1"></a><span class="in">                       .names = "{.col}_{.fn}"))</span></span>
<span id="cb28-190"><a href="#cb28-190" aria-hidden="true" tabindex="-1"></a><span class="in">    </span></span>
<span id="cb28-191"><a href="#cb28-191" aria-hidden="true" tabindex="-1"></a><span class="in">    # Transpose for better readability</span></span>
<span id="cb28-192"><a href="#cb28-192" aria-hidden="true" tabindex="-1"></a><span class="in">    quant_summary_long &lt;- quant_summary %&gt;%</span></span>
<span id="cb28-193"><a href="#cb28-193" aria-hidden="true" tabindex="-1"></a><span class="in">      pivot_longer(everything(), names_to = "metric", values_to = "value")</span></span>
<span id="cb28-194"><a href="#cb28-194" aria-hidden="true" tabindex="-1"></a><span class="in">    print(quant_summary_long)</span></span>
<span id="cb28-195"><a href="#cb28-195" aria-hidden="true" tabindex="-1"></a><span class="in">  }</span></span>
<span id="cb28-196"><a href="#cb28-196" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb28-197"><a href="#cb28-197" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-198"><a href="#cb28-198" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-199"><a href="#cb28-199" aria-hidden="true" tabindex="-1"></a><span class="fu"># Summary Paragraph</span></span>
<span id="cb28-200"><a href="#cb28-200" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-201"><a href="#cb28-201" aria-hidden="true" tabindex="-1"></a><span class="in">```{r generate-paragraph, results='asis'}</span></span>
<span id="cb28-202"><a href="#cb28-202" aria-hidden="true" tabindex="-1"></a><span class="in"># Generate dynamic summary paragraph</span></span>
<span id="cb28-203"><a href="#cb28-203" aria-hidden="true" tabindex="-1"></a><span class="in">if (exists("context_cols") &amp;&amp; length(context_cols) &gt; 0) {</span></span>
<span id="cb28-204"><a href="#cb28-204" aria-hidden="true" tabindex="-1"></a><span class="in">  </span></span>
<span id="cb28-205"><a href="#cb28-205" aria-hidden="true" tabindex="-1"></a><span class="in">  # Get counts per species</span></span>
<span id="cb28-206"><a href="#cb28-206" aria-hidden="true" tabindex="-1"></a><span class="in">  species_counts &lt;- sapply(context_cols, function(col) {</span></span>
<span id="cb28-207"><a href="#cb28-207" aria-hidden="true" tabindex="-1"></a><span class="in">    sum(biomin_data[[col]] == TRUE | biomin_data[[col]] == "TRUE" | biomin_data[[col]] == "true", na.rm = TRUE)</span></span>
<span id="cb28-208"><a href="#cb28-208" aria-hidden="true" tabindex="-1"></a><span class="in">  })</span></span>
<span id="cb28-209"><a href="#cb28-209" aria-hidden="true" tabindex="-1"></a><span class="in">  </span></span>
<span id="cb28-210"><a href="#cb28-210" aria-hidden="true" tabindex="-1"></a><span class="in">  # Get OG lists per species</span></span>
<span id="cb28-211"><a href="#cb28-211" aria-hidden="true" tabindex="-1"></a><span class="in">  species_ogs_list &lt;- lapply(context_cols, function(col) {</span></span>
<span id="cb28-212"><a href="#cb28-212" aria-hidden="true" tabindex="-1"></a><span class="in">    biomin_data %&gt;%</span></span>
<span id="cb28-213"><a href="#cb28-213" aria-hidden="true" tabindex="-1"></a><span class="in">      filter(.data[[col]] == TRUE | .data[[col]] == "TRUE" | .data[[col]] == "true") %&gt;%</span></span>
<span id="cb28-214"><a href="#cb28-214" aria-hidden="true" tabindex="-1"></a><span class="in">      pull(og_id)</span></span>
<span id="cb28-215"><a href="#cb28-215" aria-hidden="true" tabindex="-1"></a><span class="in">  })</span></span>
<span id="cb28-216"><a href="#cb28-216" aria-hidden="true" tabindex="-1"></a><span class="in">  names(species_ogs_list) &lt;- context_cols</span></span>
<span id="cb28-217"><a href="#cb28-217" aria-hidden="true" tabindex="-1"></a><span class="in">  </span></span>
<span id="cb28-218"><a href="#cb28-218" aria-hidden="true" tabindex="-1"></a><span class="in">  # Build paragraph</span></span>
<span id="cb28-219"><a href="#cb28-219" aria-hidden="true" tabindex="-1"></a><span class="in">  cat("\n## Cross-Species Context-Dependent Methylation and miRNA Analysis\n\n")</span></span>
<span id="cb28-220"><a href="#cb28-220" aria-hidden="true" tabindex="-1"></a><span class="in">  </span></span>
<span id="cb28-221"><a href="#cb28-221" aria-hidden="true" tabindex="-1"></a><span class="in">  cat("This analysis examined orthogroup (OG) patterns across species for context-dependent methylation and miRNA associations. ")</span></span>
<span id="cb28-222"><a href="#cb28-222" aria-hidden="true" tabindex="-1"></a><span class="in">  </span></span>
<span id="cb28-223"><a href="#cb28-223" aria-hidden="true" tabindex="-1"></a><span class="in">  # Species-specific findings</span></span>
<span id="cb28-224"><a href="#cb28-224" aria-hidden="true" tabindex="-1"></a><span class="in">  for (i in seq_along(context_cols)) {</span></span>
<span id="cb28-225"><a href="#cb28-225" aria-hidden="true" tabindex="-1"></a><span class="in">    col &lt;- context_cols[i]</span></span>
<span id="cb28-226"><a href="#cb28-226" aria-hidden="true" tabindex="-1"></a><span class="in">    species_name &lt;- gsub("_context.*|context_dependent_|_methylation.*", "", col, ignore.case = TRUE)</span></span>
<span id="cb28-227"><a href="#cb28-227" aria-hidden="true" tabindex="-1"></a><span class="in">    ogs &lt;- species_ogs_list[[col]]</span></span>
<span id="cb28-228"><a href="#cb28-228" aria-hidden="true" tabindex="-1"></a><span class="in">    if (length(ogs) &gt; 0) {</span></span>
<span id="cb28-229"><a href="#cb28-229" aria-hidden="true" tabindex="-1"></a><span class="in">      cat("**", species_name, "** showed context-dependent patterns in ", length(ogs), " OGs (", </span></span>
<span id="cb28-230"><a href="#cb28-230" aria-hidden="true" tabindex="-1"></a><span class="in">          paste(head(ogs, 5), collapse = ", "), if(length(ogs) &gt; 5) ", ..." else "", "). ", sep = "")</span></span>
<span id="cb28-231"><a href="#cb28-231" aria-hidden="true" tabindex="-1"></a><span class="in">    }</span></span>
<span id="cb28-232"><a href="#cb28-232" aria-hidden="true" tabindex="-1"></a><span class="in">  }</span></span>
<span id="cb28-233"><a href="#cb28-233" aria-hidden="true" tabindex="-1"></a><span class="in">  </span></span>
<span id="cb28-234"><a href="#cb28-234" aria-hidden="true" tabindex="-1"></a><span class="in">  # Shared patterns</span></span>
<span id="cb28-235"><a href="#cb28-235" aria-hidden="true" tabindex="-1"></a><span class="in">  if (exists("two_species_ogs") &amp;&amp; nrow(two_species_ogs) &gt; 0) {</span></span>
<span id="cb28-236"><a href="#cb28-236" aria-hidden="true" tabindex="-1"></a><span class="in">    cat("\n\nA total of **", nrow(two_species_ogs), " OGs** exhibited context-dependent patterns shared across exactly two species", sep = "")</span></span>
<span id="cb28-237"><a href="#cb28-237" aria-hidden="true" tabindex="-1"></a><span class="in">    if (nrow(two_species_ogs) &lt;= 10) {</span></span>
<span id="cb28-238"><a href="#cb28-238" aria-hidden="true" tabindex="-1"></a><span class="in">      cat(" (", paste(two_species_ogs$og_id, collapse = ", "), ")", sep = "")</span></span>
<span id="cb28-239"><a href="#cb28-239" aria-hidden="true" tabindex="-1"></a><span class="in">    }</span></span>
<span id="cb28-240"><a href="#cb28-240" aria-hidden="true" tabindex="-1"></a><span class="in">    cat(". ")</span></span>
<span id="cb28-241"><a href="#cb28-241" aria-hidden="true" tabindex="-1"></a><span class="in">  }</span></span>
<span id="cb28-242"><a href="#cb28-242" aria-hidden="true" tabindex="-1"></a><span class="in">  </span></span>
<span id="cb28-243"><a href="#cb28-243" aria-hidden="true" tabindex="-1"></a><span class="in">  if (exists("three_species_ogs") &amp;&amp; nrow(three_species_ogs) &gt; 0) {</span></span>
<span id="cb28-244"><a href="#cb28-244" aria-hidden="true" tabindex="-1"></a><span class="in">    cat("Notably, **", nrow(three_species_ogs), " OGs** showed conserved context-dependent patterns across all three species", sep = "")</span></span>
<span id="cb28-245"><a href="#cb28-245" aria-hidden="true" tabindex="-1"></a><span class="in">    if (nrow(three_species_ogs) &lt;= 10) {</span></span>
<span id="cb28-246"><a href="#cb28-246" aria-hidden="true" tabindex="-1"></a><span class="in">      cat(" (", paste(three_species_ogs$og_id, collapse = ", "), ")", sep = "")</span></span>
<span id="cb28-247"><a href="#cb28-247" aria-hidden="true" tabindex="-1"></a><span class="in">    }</span></span>
<span id="cb28-248"><a href="#cb28-248" aria-hidden="true" tabindex="-1"></a><span class="in">    cat(", suggesting evolutionary conservation of these regulatory mechanisms. ")</span></span>
<span id="cb28-249"><a href="#cb28-249" aria-hidden="true" tabindex="-1"></a><span class="in">  }</span></span>
<span id="cb28-250"><a href="#cb28-250" aria-hidden="true" tabindex="-1"></a><span class="in">  </span></span>
<span id="cb28-251"><a href="#cb28-251" aria-hidden="true" tabindex="-1"></a><span class="in">  # Quantitative highlights</span></span>
<span id="cb28-252"><a href="#cb28-252" aria-hidden="true" tabindex="-1"></a><span class="in">  if (exists("context_true_data") &amp;&amp; nrow(context_true_data) &gt; 0) {</span></span>
<span id="cb28-253"><a href="#cb28-253" aria-hidden="true" tabindex="-1"></a><span class="in">    numeric_cols &lt;- names(context_true_data)[sapply(context_true_data, is.numeric)]</span></span>
<span id="cb28-254"><a href="#cb28-254" aria-hidden="true" tabindex="-1"></a><span class="in">    if (length(numeric_cols) &gt; 0) {</span></span>
<span id="cb28-255"><a href="#cb28-255" aria-hidden="true" tabindex="-1"></a><span class="in">      cat("\n\nQuantitatively, context-dependent OGs showed the following characteristics: ")</span></span>
<span id="cb28-256"><a href="#cb28-256" aria-hidden="true" tabindex="-1"></a><span class="in">      for (nc in head(numeric_cols, 3)) {</span></span>
<span id="cb28-257"><a href="#cb28-257" aria-hidden="true" tabindex="-1"></a><span class="in">        mean_val &lt;- mean(context_true_data[[nc]], na.rm = TRUE)</span></span>
<span id="cb28-258"><a href="#cb28-258" aria-hidden="true" tabindex="-1"></a><span class="in">        sd_val &lt;- sd(context_true_data[[nc]], na.rm = TRUE)</span></span>
<span id="cb28-259"><a href="#cb28-259" aria-hidden="true" tabindex="-1"></a><span class="in">        cat("**", nc, "** (mean ± SD: ", round(mean_val, 3), " ± ", round(sd_val, 3), "); ", sep = "")</span></span>
<span id="cb28-260"><a href="#cb28-260" aria-hidden="true" tabindex="-1"></a><span class="in">      }</span></span>
<span id="cb28-261"><a href="#cb28-261" aria-hidden="true" tabindex="-1"></a><span class="in">    }</span></span>
<span id="cb28-262"><a href="#cb28-262" aria-hidden="true" tabindex="-1"></a><span class="in">  }</span></span>
<span id="cb28-263"><a href="#cb28-263" aria-hidden="true" tabindex="-1"></a><span class="in">  </span></span>
<span id="cb28-264"><a href="#cb28-264" aria-hidden="true" tabindex="-1"></a><span class="in">  cat("\n")</span></span>
<span id="cb28-265"><a href="#cb28-265" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb28-266"><a href="#cb28-266" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-267"><a href="#cb28-267" aria-hidden="true" tabindex="-1"></a></span></code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




</body></html>