<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Steven Roberts">
<meta name="dcterms.date" content="2023-11-03">

<title>Tumbling Oysters - Apul lncRNA Discovery</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../../site_libs/bootstrap/bootstrap-dark.min.css" rel="stylesheet" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<meta name="twitter:title" content="Tumbling Oysters - Apul lncRNA Discovery">
<meta name="twitter:description" content="">
<meta name="twitter:image" content="http://gannet.fish.washington.edu/seashell/snaps/Monosnap_deep-dive__RStudio_Server_2023-11-03_06-26-50.png">
<meta name="twitter:card" content="summary_large_image">
</head>

<body class="nav-fixed fullcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Tumbling Oysters</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../about.html" rel="" target="">
 <span class="menu-text">Me Elsewhere</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/sr320" rel="" target=""><i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com/sr320" rel="" target=""><i class="bi bi-twitter" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools">
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <div class="quarto-title-block"><div><h1 class="title">Apul lncRNA Discovery</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
                                <div class="quarto-categories">
                <div class="quarto-category">deepdive</div>
                <div class="quarto-category">R</div>
                <div class="quarto-category">lncrna</div>
                <div class="quarto-category">e5</div>
              </div>
                  </div>
  </div>
    
  <div class="quarto-title-meta-author">
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-heading">Affiliation</div>
    
      <div class="quarto-title-meta-contents">
      <p class="author">Steven Roberts <a href="https://orcid.org/0000-0001-8302-1138" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
    </div>
      <div class="quarto-title-meta-contents">
          <p class="affiliation">
              <a href="https://robertslab.info">
              Professor, UW - School of Aquatic and Fishery Sciences
              </a>
            </p>
        </div>
      </div>

  <div class="quarto-title-meta">

        
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">November 3, 2023</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">




<section id="apul-lncrna-discovery" class="level1">
<h1>05.3-Apul-lncRNA-discovery</h1>
<p>Steven Roberts 03 November, 2023</p>
<ul>
<li><a href="#1-run-hisat-on-rna-seq" id="toc-1-run-hisat-on-rna-seq">1 Run HiSat on RNA-seq</a>
<ul>
<li><a href="#11-grab-trimmed-rna-seq-reads" id="toc-11-grab-trimmed-rna-seq-reads">1.1 Grab Trimmed RNA-seq Reads</a></li>
<li><a href="#12-genome" id="toc-12-genome">1.2 Genome</a></li>
<li><a href="#13-hisat" id="toc-13-hisat">1.3 HiSat</a></li>
<li><a href="#14-convert-to-bams" id="toc-14-convert-to-bams">1.4 convert to bams</a></li>
</ul></li>
<li><a href="#2-stringtie" id="toc-2-stringtie">2 StringTie</a></li>
<li><a href="#3-gffcompare" id="toc-3-gffcompare">3 GFFcompare</a></li>
<li><a href="#4-filter" id="toc-4-filter">4 Filter</a></li>
<li><a href="#5-bedtools" id="toc-5-bedtools">5 Bedtools</a></li>
<li><a href="#6-cpc2" id="toc-6-cpc2">6 CPC2</a></li>
<li><a href="#7-de-duplicate-gtf" id="toc-7-de-duplicate-gtf">7 De-duplicate GTF</a></li>
<li><a href="#8-reformat-to-bed" id="toc-8-reformat-to-bed">8 Reformat to BED</a></li>
<li><a href="#9-bedtools" id="toc-9-bedtools">9 BEDTools</a></li>
</ul>
</section>
<section id="run-hisat-on-rna-seq" class="level1">
<h1>1 Run HiSat on RNA-seq</h1>
<section id="grab-trimmed-rna-seq-reads" class="level2">
<h2 class="anchored" data-anchor-id="grab-trimmed-rna-seq-reads">1.1 Grab Trimmed RNA-seq Reads</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-r</span> <span class="dt">\</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>--no-directories <span class="at">--no-parent</span> <span class="dt">\</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>-P ../data/fastq/ <span class="dt">\</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>-A <span class="st">"*fastq.gz"</span> https://gannet.fish.washington.edu/Atumefaciens/20230519-E5_coral-fastqc-fastp-multiqc-RNAseq/A_pulchra/trimmed/</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="genome" class="level2">
<h2 class="anchored" data-anchor-id="genome">1.2 Genome</h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> ../data</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-O</span> http://gannet.fish.washington.edu/seashell/snaps/GCF_013753865.1_Amil_v2.1_genomic.fna</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="hisat" class="level2">
<h2 class="anchored" data-anchor-id="hisat">1.3 HiSat</h2>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="ex">/home/shared/hisat2-2.2.1/hisat2_extract_exons.py</span> <span class="dt">\</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gtf <span class="dt">\</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span> ../output/05.32-lncRNA-discovery/m_exon.tab</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> ../output/05.32-lncRNA-discovery/m_exon.tab</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## NC_058066.1  1961    2118    -
## NC_058066.1  15360   15663   +
## NC_058066.1  18710   18774   -
## NC_058066.1  21000   21092   -
## NC_058066.1  22158   22442   +
## NC_058066.1  23084   23220   -
## NC_058066.1  24770   25066   +
## NC_058066.1  26652   26912   +
## NC_058066.1  27071   27358   +
## NC_058066.1  27658   27951   +</code></pre>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="ex">/home/shared/hisat2-2.2.1/hisat2_extract_splice_sites.py</span> <span class="dt">\</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gtf <span class="dt">\</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span> ../output/05.32-lncRNA-discovery/m_splice_sites.tab</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> ../output/05.32-lncRNA-discovery/m_splice_sites.tab</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## NC_058066.1  2118    18710   -
## NC_058066.1  15663   22158   +
## NC_058066.1  18774   21000   -
## NC_058066.1  21092   23084   -
## NC_058066.1  22442   24770   +
## NC_058066.1  25066   26652   +
## NC_058066.1  26912   27071   +
## NC_058066.1  27358   27658   +
## NC_058066.1  27951   28247   +
## NC_058066.1  28534   29196   +</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="ex">/home/shared/hisat2-2.2.1/hisat2-build</span> <span class="dt">\</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>../data/GCF_013753865.1_Amil_v2.1_genomic.fna <span class="dt">\</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>../data/GCF_013753865.1_Amil_v2.1.index <span class="dt">\</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>--exon ../output/05.32-lncRNA-discovery/m_exon.tab <span class="dt">\</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>--ss ../output/05.32-lncRNA-discovery/m_splice_sites.tab <span class="dt">\</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>-p 40 <span class="dt">\</span></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gtf <span class="dt">\</span></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>2<span class="op">&gt;</span> ../output/05.32-lncRNA-discovery/hisat2-build_stats.txt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">find</span> ../data/fastq/<span class="pp">*</span>gz</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## ../data/fastq/RNA-ACR-140-S1-TP2_R1_001.fastp-trim.20230519.fastq.gz
## ../data/fastq/RNA-ACR-140-S1-TP2_R2_001.fastp-trim.20230519.fastq.gz
## ../data/fastq/RNA-ACR-145-S1-TP2_R1_001.fastp-trim.20230519.fastq.gz
## ../data/fastq/RNA-ACR-145-S1-TP2_R2_001.fastp-trim.20230519.fastq.gz
## ../data/fastq/RNA-ACR-150-S1-TP2_R1_001.fastp-trim.20230519.fastq.gz
## ../data/fastq/RNA-ACR-150-S1-TP2_R2_001.fastp-trim.20230519.fastq.gz
## ../data/fastq/RNA-ACR-173-S1-TP2_R1_001.fastp-trim.20230519.fastq.gz
## ../data/fastq/RNA-ACR-173-S1-TP2_R2_001.fastp-trim.20230519.fastq.gz
## ../data/fastq/RNA-ACR-178-S1-TP2_R1_001.fastp-trim.20230519.fastq.gz
## ../data/fastq/RNA-ACR-178-S1-TP2_R2_001.fastp-trim.20230519.fastq.gz</code></pre>
<div class="sourceCode" id="cb12"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">find</span> ../data/fastq/<span class="pp">*</span>R2_001.fastp-trim.20230519.fastq.gz <span class="dt">\</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span> <span class="fu">xargs</span> basename <span class="at">-s</span> <span class="at">-S1-TP2_R2_001.fastp-trim.20230519.fastq.gz</span> <span class="kw">|</span> <span class="fu">xargs</span> <span class="at">-I{}</span> <span class="dt">\</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>echo {}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## RNA-ACR-140
## RNA-ACR-145
## RNA-ACR-150
## RNA-ACR-173
## RNA-ACR-178</code></pre>
<div class="sourceCode" id="cb14"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">find</span> ../data/fastq/<span class="pp">*</span>R2_001.fastp-trim.20230519.fastq.gz <span class="dt">\</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span> <span class="fu">xargs</span> basename <span class="at">-s</span> <span class="at">-S1-TP2_R2_001.fastp-trim.20230519.fastq.gz</span> <span class="kw">|</span> <span class="fu">xargs</span> <span class="at">-I{}</span> <span class="dt">\</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>/home/shared/hisat2-2.2.1/hisat2 <span class="dt">\</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>-x ../data/GCF_013753865.1_Amil_v2.1.index <span class="dt">\</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>-p 20 <span class="dt">\</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>-1 ../data/fastq/{}-S1-TP2_R1_001.fastp-trim.20230519.fastq.gz <span class="dt">\</span></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>-2 ../data/fastq/{}-S1-TP2_R2_001.fastp-trim.20230519.fastq.gz <span class="dt">\</span></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>-S ../output/05.32-lncRNA-discovery/{}.sam</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>47710408 reads; of these:
  47710408 (100.00%) were paired; of these:
    27056517 (56.71%) aligned concordantly 0 times
    19229695 (40.31%) aligned concordantly exactly 1 time
    1424196 (2.99%) aligned concordantly &gt;1 times
    ----
    27056517 pairs aligned concordantly 0 times; of these:
      1316579 (4.87%) aligned discordantly 1 time
    ----
    25739938 pairs aligned 0 times concordantly or discordantly; of these:
      51479876 mates make up the pairs; of these:
        41152624 (79.94%) aligned 0 times
        9290398 (18.05%) aligned exactly 1 time
        1036854 (2.01%) aligned &gt;1 times
56.87% overall alignment rate
42864294 reads; of these:
  42864294 (100.00%) were paired; of these:
    23649791 (55.17%) aligned concordantly 0 times
    17703533 (41.30%) aligned concordantly exactly 1 time
    1510970 (3.53%) aligned concordantly &gt;1 times
    ----
    23649791 pairs aligned concordantly 0 times; of these:
      1250605 (5.29%) aligned discordantly 1 time
    ----
    22399186 pairs aligned 0 times concordantly or discordantly; of these:
      44798372 mates make up the pairs; of these:
        35369621 (78.95%) aligned 0 times
        8218798 (18.35%) aligned exactly 1 time
        1209953 (2.70%) aligned &gt;1 times
58.74% overall alignment rate
43712298 reads; of these:
  43712298 (100.00%) were paired; of these:
    30355359 (69.44%) aligned concordantly 0 times
    12131465 (27.75%) aligned concordantly exactly 1 time
    1225474 (2.80%) aligned concordantly &gt;1 times
    ----
    30355359 pairs aligned concordantly 0 times; of these:
      848325 (2.79%) aligned discordantly 1 time
    ----
    29507034 pairs aligned 0 times concordantly or discordantly; of these:
      59014068 mates make up the pairs; of these:
        51031599 (86.47%) aligned 0 times
        6695839 (11.35%) aligned exactly 1 time
        1286630 (2.18%) aligned &gt;1 times
41.63% overall alignment rate
47501524 reads; of these:
  47501524 (100.00%) were paired; of these:
    27827915 (58.58%) aligned concordantly 0 times
    18298532 (38.52%) aligned concordantly exactly 1 time
    1375077 (2.89%) aligned concordantly &gt;1 times
    ----
    27827915 pairs aligned concordantly 0 times; of these:
      1321896 (4.75%) aligned discordantly 1 time
    ----
    26506019 pairs aligned 0 times concordantly or discordantly; of these:
      53012038 mates make up the pairs; of these:
        42654805 (80.46%) aligned 0 times
        9174277 (17.31%) aligned exactly 1 time
        1182956 (2.23%) aligned &gt;1 times
55.10% overall alignment rate
42677752 reads; of these:
  42677752 (100.00%) were paired; of these:
    25646587 (60.09%) aligned concordantly 0 times
    15654238 (36.68%) aligned concordantly exactly 1 time
    1376927 (3.23%) aligned concordantly &gt;1 times
    ----
    25646587 pairs aligned concordantly 0 times; of these:
      1105089 (4.31%) aligned discordantly 1 time
    ----
    24541498 pairs aligned 0 times concordantly or discordantly; of these:
      49082996 mates make up the pairs; of these:
        38229072 (77.89%) aligned 0 times
        9393012 (19.14%) aligned exactly 1 time
        1460912 (2.98%) aligned &gt;1 times
55.21% overall alignment rate</code></pre>
</section>
<section id="convert-to-bams" class="level2">
<h2 class="anchored" data-anchor-id="convert-to-bams">1.4 convert to bams</h2>
<div class="sourceCode" id="cb16"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> samfile <span class="kw">in</span> ../output/05.32-lncRNA-discovery/<span class="pp">*</span>.sam<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>  <span class="va">bamfile</span><span class="op">=</span><span class="st">"</span><span class="va">${samfile</span><span class="op">%</span>.sam<span class="va">}</span><span class="st">.bam"</span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>  <span class="va">sorted_bamfile</span><span class="op">=</span><span class="st">"</span><span class="va">${samfile</span><span class="op">%</span>.sam<span class="va">}</span><span class="st">.sorted.bam"</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Convert SAM to BAM</span></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>  <span class="ex">/home/shared/samtools-1.12/samtools</span> view <span class="at">-bS</span> <span class="st">"</span><span class="va">$samfile</span><span class="st">"</span> <span class="op">&gt;</span> <span class="st">"</span><span class="va">$bamfile</span><span class="st">"</span></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Sort BAM</span></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>  <span class="ex">/home/shared/samtools-1.12/samtools</span> sort <span class="st">"</span><span class="va">$bamfile</span><span class="st">"</span> <span class="at">-o</span> <span class="st">"</span><span class="va">$sorted_bamfile</span><span class="st">"</span></span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Index sorted BAM</span></span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>  <span class="ex">/home/shared/samtools-1.12/samtools</span> index <span class="st">"</span><span class="va">$sorted_bamfile</span><span class="st">"</span></span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
</section>
<section id="stringtie" class="level1">
<h1>2 StringTie</h1>
<p>StringTie uses the sorted BAM files to assemble transcripts for each sample, outputting them as GTF (Gene Transfer Format) files. And then merges all individual GTF assemblies into a single merged GTF file. This step extracts transcript information and merges GTFs from all samples–an important step in creating a canonical list of lncRNAs across all samples included in the pipeline.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">find</span> ../output/05.32-lncRNA-discovery/<span class="pp">*</span>sorted.bam <span class="dt">\</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span> <span class="fu">xargs</span> basename <span class="at">-s</span> .sorted.bam <span class="kw">|</span> <span class="fu">xargs</span> <span class="at">-I{}</span> <span class="dt">\</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>/home/shared/stringtie-2.2.1.Linux_x86_64/stringtie <span class="dt">\</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>-p 8 <span class="dt">\</span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>-G ../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gff <span class="dt">\</span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>-o ../output/05.32-lncRNA-discovery/{}.gtf <span class="dt">\</span></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>../output/05.32-lncRNA-discovery/{}.sorted.bam</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb18"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wc</span> <span class="at">-l</span> ../output/05.32-lncRNA-discovery/RNA<span class="pp">*</span>.gtf</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> ../output/05.32-lncRNA-discovery/RNA<span class="pp">*</span>.gtf</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>##    373617 ../output/05.32-lncRNA-discovery/RNA-ACR-140.gtf
##    322109 ../output/05.32-lncRNA-discovery/RNA-ACR-145.gtf
##    340380 ../output/05.32-lncRNA-discovery/RNA-ACR-150.gtf
##    367972 ../output/05.32-lncRNA-discovery/RNA-ACR-173.gtf
##    426804 ../output/05.32-lncRNA-discovery/RNA-ACR-178.gtf
##   1830882 total
## ../output/05.32-lncRNA-discovery/RNA-ACR-140.gtf
## ../output/05.32-lncRNA-discovery/RNA-ACR-145.gtf
## ../output/05.32-lncRNA-discovery/RNA-ACR-150.gtf
## ../output/05.32-lncRNA-discovery/RNA-ACR-173.gtf
## ../output/05.32-lncRNA-discovery/RNA-ACR-178.gtf</code></pre>
<p>Merges all individual GTF assemblies into a single merged GTF file.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="ex">/home/shared/stringtie-2.2.1.Linux_x86_64/stringtie</span> <span class="dt">\</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>--merge <span class="dt">\</span></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>-G ../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gff <span class="dt">\</span></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>-o ../output/05.32-lncRNA-discovery/stringtie_merged.gtf <span class="dt">\</span></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>../output/05.32-lncRNA-discovery/<span class="pp">*</span>.gtf</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb21"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="fu">wc</span> <span class="at">-l</span> ../output/05.32-lncRNA-discovery/stringtie_merged.gtf</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> ../output/05.32-lncRNA-discovery/stringtie_merged.gtf</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## 726445 ../output/05.32-lncRNA-discovery/stringtie_merged.gtf
## # /home/shared/stringtie-2.2.1.Linux_x86_64/stringtie --merge -G ../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gff -o ../output/05.32-lncRNA-discovery/stringtie_merged.gtf ../output/05.32-lncRNA-discovery/RNA-ACR-140.gtf ../output/05.32-lncRNA-discovery/RNA-ACR-145.gtf ../output/05.32-lncRNA-discovery/RNA-ACR-150.gtf ../output/05.32-lncRNA-discovery/RNA-ACR-173.gtf ../output/05.32-lncRNA-discovery/RNA-ACR-178.gtf
## # StringTie version 2.2.1
## NC_058066.1  StringTie   transcript  345 1190    1000    .   .   gene_id "MSTRG.1"; transcript_id "MSTRG.1.1"; 
## NC_058066.1  StringTie   exon    345 1190    1000    .   .   gene_id "MSTRG.1"; transcript_id "MSTRG.1.1"; exon_number "1"; 
## NC_058066.1  StringTie   transcript  1962    23221   1000    -   .   gene_id "MSTRG.2"; transcript_id "rna-XR_003825913.2"; gene_name "LOC114963522"; ref_gene_id "gene-LOC114963522"; 
## NC_058066.1  StringTie   exon    1962    2119    1000    -   .   gene_id "MSTRG.2"; transcript_id "rna-XR_003825913.2"; exon_number "1"; gene_name "LOC114963522"; ref_gene_id "gene-LOC114963522"; 
## NC_058066.1  StringTie   exon    18711   18775   1000    -   .   gene_id "MSTRG.2"; transcript_id "rna-XR_003825913.2"; exon_number "2"; gene_name "LOC114963522"; ref_gene_id "gene-LOC114963522"; 
## NC_058066.1  StringTie   exon    21001   21093   1000    -   .   gene_id "MSTRG.2"; transcript_id "rna-XR_003825913.2"; exon_number "3"; gene_name "LOC114963522"; ref_gene_id "gene-LOC114963522"; 
## NC_058066.1  StringTie   exon    23085   23221   1000    -   .   gene_id "MSTRG.2"; transcript_id "rna-XR_003825913.2"; exon_number "4"; gene_name "LOC114963522"; ref_gene_id "gene-LOC114963522"; 
## NC_058066.1  StringTie   transcript  3278    4416    1000    .   .   gene_id "MSTRG.3"; transcript_id "MSTRG.3.1";</code></pre>
</section>
<section id="gffcompare" class="level1">
<h1>3 GFFcompare</h1>
<div class="sourceCode" id="cb23"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="ex">/home/shared/gffcompare-0.12.6.Linux_x86_64/gffcompare</span> <span class="dt">\</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>-r ../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gff <span class="dt">\</span></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>-G <span class="dt">\</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>-o ../output/05.32-lncRNA-discovery/gffcompare_merged <span class="dt">\</span></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>../output/05.32-lncRNA-discovery/stringtie_merged.gtf</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb24"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> ../output/05.32-lncRNA-discovery/gffcompare_merged<span class="pp">*</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## ../output/05.32-lncRNA-discovery/gffcompare_merged.annotated.gtf
## ../output/05.32-lncRNA-discovery/gffcompare_merged.loci
## ../output/05.32-lncRNA-discovery/gffcompare_merged.stats
## ../output/05.32-lncRNA-discovery/gffcompare_merged.stringtie_merged.gtf.refmap
## ../output/05.32-lncRNA-discovery/gffcompare_merged.stringtie_merged.gtf.tmap
## ../output/05.32-lncRNA-discovery/gffcompare_merged.tracking</code></pre>
<div class="sourceCode" id="cb26"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> <span class="at">-3</span> ../output/05.32-lncRNA-discovery/gffcompare_merged.annotated.gtf</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## NC_058066.1  StringTie   transcript  9330    48114   .   +   .   transcript_id "MSTRG.5.2"; gene_id "MSTRG.5"; gene_name "LOC114963508"; xloc "XLOC_000001"; ref_gene_id "gene-LOC114963508"; cmp_ref "rna-XM_029342736.2"; class_code "j"; tss_id "TSS1";
## NC_058066.1  StringTie   exon    9330    9651    .   +   .   transcript_id "MSTRG.5.2"; gene_id "MSTRG.5"; exon_number "1";
## NC_058066.1  StringTie   exon    12840   12890   .   +   .   transcript_id "MSTRG.5.2"; gene_id "MSTRG.5"; exon_number "2";</code></pre>
</section>
<section id="filter" class="level1">
<h1>4 Filter</h1>
<p>Filters the combined GTF output from GFFcompare to select only the lines representing “transcripts” and excluding lines starting with “#” (these are lines in the output format from GFFcompare that don’t contain transcript information). This step further filters for a class code of “u”, and keep only those with lengths greater than 199 bases. The “u’ class code from the GFFcompare step is for”unknown” transcripts, that is those that are not previously annotated in our reference GFF as protein coding. The size filter of +200nt is a common filtering step for isolating lncRNAs.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">awk</span> <span class="st">'$3 == "transcript" &amp;&amp; $1 !~ /^#/ {print}'</span> <span class="dt">\</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>../output/05.32-lncRNA-discovery/gffcompare_merged.annotated.gtf <span class="kw">|</span> <span class="fu">grep</span> <span class="st">'class_code "u"'</span> <span class="kw">|</span> <span class="fu">awk</span> <span class="st">'$5 - $4 &gt; 199 {print}'</span> <span class="op">&gt;</span> ../output/05.32-lncRNA-discovery/Apul_lncRNA_candidates.gtf</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb29"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> ../output/05.32-lncRNA-discovery/Apul_lncRNA_candidates.gtf</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## NC_058066.1  StringTie   transcript  468619  469943  .   +   .   transcript_id "MSTRG.40.1"; gene_id "MSTRG.40"; xloc "XLOC_000020"; class_code "u"; tss_id "TSS38";
## NC_058066.1  StringTie   transcript  1153399 1165634 .   +   .   transcript_id "MSTRG.120.1"; gene_id "MSTRG.120"; xloc "XLOC_000058"; class_code "u"; tss_id "TSS95";
## NC_058066.1  StringTie   transcript  1153404 1165634 .   +   .   transcript_id "MSTRG.120.2"; gene_id "MSTRG.120"; xloc "XLOC_000058"; class_code "u"; tss_id "TSS95";
## NC_058066.1  StringTie   transcript  1153410 1165634 .   +   .   transcript_id "MSTRG.120.3"; gene_id "MSTRG.120"; xloc "XLOC_000058"; class_code "u"; tss_id "TSS95";
## NC_058066.1  StringTie   transcript  1154207 1155609 .   +   .   transcript_id "MSTRG.120.4"; gene_id "MSTRG.120"; xloc "XLOC_000058"; class_code "u"; tss_id "TSS96";
## NC_058066.1  StringTie   transcript  1155787 1165634 .   +   .   transcript_id "MSTRG.120.5"; gene_id "MSTRG.120"; xloc "XLOC_000058"; class_code "u"; tss_id "TSS97";
## NC_058066.1  StringTie   transcript  1222445 1227226 .   +   .   transcript_id "MSTRG.126.1"; gene_id "MSTRG.126"; xloc "XLOC_000059"; class_code "u"; tss_id "TSS98";
## NC_058066.1  StringTie   transcript  1222539 1227226 .   +   .   transcript_id "MSTRG.126.2"; gene_id "MSTRG.126"; xloc "XLOC_000059"; class_code "u"; tss_id "TSS98";
## NC_058066.1  StringTie   transcript  1256163 1257382 .   +   .   transcript_id "MSTRG.136.3"; gene_id "MSTRG.136"; xloc "XLOC_000062"; class_code "u"; tss_id "TSS103";
## NC_058066.1  StringTie   transcript  1256163 1264306 .   +   .   transcript_id "MSTRG.136.2"; gene_id "MSTRG.136"; xloc "XLOC_000062"; class_code "u"; tss_id "TSS103";</code></pre>
</section>
<section id="bedtools" class="level1">
<h1>5 Bedtools</h1>
<p>Extracts the sequence data from the <code>$FASTA</code> reference file based on the coordinates from the filtered GTF. The resulting sequences represent potential lncRNA candidates.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="ex">/home/shared/bedtools2/bin/fastaFromBed</span> <span class="dt">\</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>-fi ../data/GCF_013753865.1_Amil_v2.1_genomic.fna <span class="dt">\</span></span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>-bed ../output/05.32-lncRNA-discovery/Apul_lncRNA_candidates.gtf <span class="dt">\</span></span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>-fo ../output/05.32-lncRNA-discovery/Apul_lncRNA_candidates.fasta <span class="dt">\</span></span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>-name <span class="at">-split</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb32"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">fgrep</span> <span class="at">-c</span> <span class="st">"&gt;"</span> ../output/05.32-lncRNA-discovery/Apul_lncRNA_candidates.fasta</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> ../output/05.32-lncRNA-discovery/Apul_lncRNA_candidates.fasta</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="cpc2" class="level1">
<h1>6 CPC2</h1>
<p>Initializes a conda environment (Anaconda) and runs CPC2, a software to predict whether a transcript is coding or non-coding. The results are saved to the $OUTPUT_DIR. CPC2 uses ORF (Open Reading Frame) Analysis, Isometric Feature Mapping (Isomap), Sequence Homology, RNA Sequence Features, and Quality of Translation to assess coding potential and flag any transcripts we would want to exclude using the FASTA generated in the previous step.</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="bu">eval</span> <span class="st">"</span><span class="va">$(</span><span class="ex">/opt/anaconda/anaconda3/bin/conda</span> shell.bash hook<span class="va">)</span><span class="st">"</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="ex">python</span> /home/shared/CPC2_standalone-1.0.1/bin/CPC2.py <span class="dt">\</span></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>-i ../output/05.32-lncRNA-discovery/Apul_lncRNA_candidates.fasta <span class="dt">\</span></span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>-o ../output/05.32-lncRNA-discovery/Apul_CPC2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb34"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wc</span> <span class="at">-l</span> head ../output/05.32-lncRNA-discovery/Apul_CPC2.txt</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> ../output/05.32-lncRNA-discovery/Apul_CPC2.txt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## wc: head: No such file or directory
##   16695 ../output/05.32-lncRNA-discovery/Apul_CPC2.txt
##   16695 total
## #ID  transcript_length   peptide_length  Fickett_score   pI  ORF_integrity   coding_probability  label
## transcript::NC_058066.1:468618-469943    1325    73  0.3891  11.48389835357666   1   0.0882804   noncoding
## transcript::NC_058066.1:1153398-1165634  12236   134 0.3118  9.09901943206787    1   0.295457    noncoding
## transcript::NC_058066.1:1153403-1165634  12231   134 0.3118  9.09901943206787    1   0.295457    noncoding
## transcript::NC_058066.1:1153409-1165634  12225   134 0.3118  9.09901943206787    1   0.295457    noncoding
## transcript::NC_058066.1:1154206-1155609  1403    66  0.37373 8.802658271789554   1   0.0495107   noncoding
## transcript::NC_058066.1:1155786-1165634  9848    134 0.29479 9.09901943206787    1   0.241898    noncoding
## transcript::NC_058066.1:1222444-1227226  4782    62  0.32077 9.30048313140869    1   0.0302275   noncoding
## transcript::NC_058066.1:1222538-1227226  4688    62  0.32077 9.30048313140869    1   0.0302275   noncoding
## transcript::NC_058066.1:1256162-1257382  1220    41  0.34867000000000004 9.098761558532715   1   0.0140974   noncoding</code></pre>
<p>#Filter Filters the CPC2 results to get only noncoding transcripts (using the class “noncoding” from the CPC2 results) and extracts their IDs and matches these IDs with the sequences from the previous step to generate a GTF of long noncoding transcripts.</p>
<p>Matches these IDs with the sequences from the previous step to generate a GTF of noncoding transcripts.</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">awk</span> <span class="st">'$8 == "noncoding" {print $1}'</span> ../output/05.32-lncRNA-discovery/Apul_CPC2.txt <span class="op">&gt;</span> ../output/05.32-lncRNA-discovery/Apul_noncoding_transcripts_ids.txt</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a><span class="fu">grep</span> <span class="at">-Fwf</span> ../output/05.32-lncRNA-discovery/Apul_noncoding_transcripts_ids.txt ../output/05.32-lncRNA-discovery/Apul_lncRNA_candidates.fasta <span class="op">&gt;</span> ../output/05.32-lncRNA-discovery/Apul_lncRNAs.gtf</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb37"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wc</span> <span class="at">-l</span> ../output/05.32-lncRNA-discovery/Apul_lncRNAs.gtf</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> ../output/05.32-lncRNA-discovery/Apul_lncRNAs.gtf</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## 16122 ../output/05.32-lncRNA-discovery/Apul_lncRNAs.gtf
## &gt;transcript::NC_058066.1:468618-469943
## &gt;transcript::NC_058066.1:1153398-1165634
## &gt;transcript::NC_058066.1:1153403-1165634
## &gt;transcript::NC_058066.1:1153409-1165634
## &gt;transcript::NC_058066.1:1154206-1155609
## &gt;transcript::NC_058066.1:1155786-1165634
## &gt;transcript::NC_058066.1:1222444-1227226
## &gt;transcript::NC_058066.1:1222538-1227226
## &gt;transcript::NC_058066.1:1256162-1257382
## &gt;transcript::NC_058066.1:1256162-1264306</code></pre>
</section>
<section id="de-duplicate-gtf" class="level1">
<h1>7 De-duplicate GTF</h1>
<p>Removes duplicate entries from the GTF file. This results in a list of lncRNAs with transcript IDs that show scaffold and base range. awk ‘!seen<span class="math display">\[\$0\]</span>++’ $OUTPUT_DIR/apul_merged_final_lncRNAs.gtf &gt; $OUTPUT_DIR/apul_deduplicated_final_lncRNAs.gtf</p>
</section>
<section id="reformat-to-bed" class="level1">
<h1>8 Reformat to BED</h1>
<p>Converts the GTF format to BED (Browser Extensible Data) format, a commonly used format for representing genomic intervals. This allows us to use bedtools to generate a FASTA of lncRNAs. awk -F”:|-” ‘{print $3 “ $4” $5}’ $OUTPUT_DIR/apul_deduplicated_final_lncRNAs.gtf &gt; $OUTPUT_DIR/apul_deduplicated_final_lncRNAs.bed</p>
</section>
<section id="bedtools-1" class="level1">
<h1>9 BEDTools</h1>
<p>This extracts sequences from the reference FASTA based on the BED file’s coordinates. The resulting sequences represent the final set of non-redundant lncRNAs. $BEDTOOLS_PATH/bedtools<br>
getfasta -fi $FASTA -bed $OUTPUT_DIR/apul_deduplicated_final_lncRNAs.bed -fo $OUTPUT_DIR/apul_bedtools_lncRNAs.fasta -name ```</p>
<p>The results of this pipleine provide the user with GTF files containing transcript IDs (scaffold and sequence range), BED files displaying the same transcript IDs in BED format for use with BEDtools, and FASTA files containing the sequence of identified lncRNAs.</p>


<!-- -->

</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  let localAlternateSentinel = 'alternate';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb39" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Apul lncRNA Discovery"</span></span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a><span class="an">description:</span><span class="co"> ""</span></span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span></span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a><span class="co">  - name: Steven Roberts</span></span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a><span class="co">    url: </span></span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a><span class="co">    orcid: 0000-0001-8302-1138</span></span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a><span class="co">    affiliation: Professor, UW - School of Aquatic and Fishery Sciences</span></span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a><span class="co">    affiliation-url: https://robertslab.info</span></span>
<span id="cb39-10"><a href="#cb39-10" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> 11-03-2023</span></span>
<span id="cb39-11"><a href="#cb39-11" aria-hidden="true" tabindex="-1"></a><span class="an">categories:</span><span class="co"> [deepdive, R, lncrna, e5] # self-defined categories</span></span>
<span id="cb39-12"><a href="#cb39-12" aria-hidden="true" tabindex="-1"></a><span class="co">#citation: </span></span>
<span id="cb39-13"><a href="#cb39-13" aria-hidden="true" tabindex="-1"></a><span class="co">  #url:  # self-defined</span></span>
<span id="cb39-14"><a href="#cb39-14" aria-hidden="true" tabindex="-1"></a><span class="an">image:</span><span class="co"> http://gannet.fish.washington.edu/seashell/snaps/Monosnap_deep-dive__RStudio_Server_2023-11-03_06-26-50.png # finding a good image</span></span>
<span id="cb39-15"><a href="#cb39-15" aria-hidden="true" tabindex="-1"></a><span class="an">draft:</span><span class="co"> false # setting this to `true` will prevent your post from appearing on your listing page until you're ready!</span></span>
<span id="cb39-16"><a href="#cb39-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-17"><a href="#cb39-17" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb39-18"><a href="#cb39-18" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb39-19"><a href="#cb39-19" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: true</span></span>
<span id="cb39-20"><a href="#cb39-20" aria-hidden="true" tabindex="-1"></a><span class="co">    code-tools: true</span></span>
<span id="cb39-21"><a href="#cb39-21" aria-hidden="true" tabindex="-1"></a><span class="co">    code-copy: true</span></span>
<span id="cb39-22"><a href="#cb39-22" aria-hidden="true" tabindex="-1"></a><span class="co">    highlight-style: github</span></span>
<span id="cb39-23"><a href="#cb39-23" aria-hidden="true" tabindex="-1"></a><span class="co">    code-overflow: wrap</span></span>
<span id="cb39-24"><a href="#cb39-24" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb39-25"><a href="#cb39-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-26"><a href="#cb39-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-27"><a href="#cb39-27" aria-hidden="true" tabindex="-1"></a>05.3-Apul-lncRNA-discovery</span>
<span id="cb39-28"><a href="#cb39-28" aria-hidden="true" tabindex="-1"></a><span class="fu">================</span></span>
<span id="cb39-29"><a href="#cb39-29" aria-hidden="true" tabindex="-1"></a>Steven Roberts</span>
<span id="cb39-30"><a href="#cb39-30" aria-hidden="true" tabindex="-1"></a>03 November, 2023</span>
<span id="cb39-31"><a href="#cb39-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-32"><a href="#cb39-32" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span><span class="kw">&lt;a</span> <span class="er">href</span><span class="ot">=</span><span class="st">"#1-run-hisat-on-rna-seq"</span> <span class="er">id</span><span class="ot">=</span><span class="st">"toc-1-run-hisat-on-rna-seq"</span><span class="kw">&gt;</span>1 Run</span>
<span id="cb39-33"><a href="#cb39-33" aria-hidden="true" tabindex="-1"></a>  HiSat on RNA-seq<span class="kw">&lt;/a&gt;</span></span>
<span id="cb39-34"><a href="#cb39-34" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span><span class="kw">&lt;a</span> <span class="er">href</span><span class="ot">=</span><span class="st">"#11-grab-trimmed-rna-seq-reads"</span></span>
<span id="cb39-35"><a href="#cb39-35" aria-hidden="true" tabindex="-1"></a><span class="ot">    id=</span><span class="st">"toc-11-grab-trimmed-rna-seq-reads"</span><span class="kw">&gt;</span>1.1 Grab Trimmed RNA-seq</span>
<span id="cb39-36"><a href="#cb39-36" aria-hidden="true" tabindex="-1"></a>    Reads<span class="kw">&lt;/a&gt;</span></span>
<span id="cb39-37"><a href="#cb39-37" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span><span class="kw">&lt;a</span> <span class="er">href</span><span class="ot">=</span><span class="st">"#12-genome"</span> <span class="er">id</span><span class="ot">=</span><span class="st">"toc-12-genome"</span><span class="kw">&gt;</span>1.2 Genome<span class="kw">&lt;/a&gt;</span></span>
<span id="cb39-38"><a href="#cb39-38" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span><span class="kw">&lt;a</span> <span class="er">href</span><span class="ot">=</span><span class="st">"#13-hisat"</span> <span class="er">id</span><span class="ot">=</span><span class="st">"toc-13-hisat"</span><span class="kw">&gt;</span>1.3 HiSat<span class="kw">&lt;/a&gt;</span></span>
<span id="cb39-39"><a href="#cb39-39" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span><span class="kw">&lt;a</span> <span class="er">href</span><span class="ot">=</span><span class="st">"#14-convert-to-bams"</span> <span class="er">id</span><span class="ot">=</span><span class="st">"toc-14-convert-to-bams"</span><span class="kw">&gt;</span>1.4 convert to</span>
<span id="cb39-40"><a href="#cb39-40" aria-hidden="true" tabindex="-1"></a>    bams<span class="kw">&lt;/a&gt;</span></span>
<span id="cb39-41"><a href="#cb39-41" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span><span class="kw">&lt;a</span> <span class="er">href</span><span class="ot">=</span><span class="st">"#2-stringtie"</span> <span class="er">id</span><span class="ot">=</span><span class="st">"toc-2-stringtie"</span><span class="kw">&gt;</span>2 StringTie<span class="kw">&lt;/a&gt;</span></span>
<span id="cb39-42"><a href="#cb39-42" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span><span class="kw">&lt;a</span> <span class="er">href</span><span class="ot">=</span><span class="st">"#3-gffcompare"</span> <span class="er">id</span><span class="ot">=</span><span class="st">"toc-3-gffcompare"</span><span class="kw">&gt;</span>3 GFFcompare<span class="kw">&lt;/a&gt;</span></span>
<span id="cb39-43"><a href="#cb39-43" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span><span class="kw">&lt;a</span> <span class="er">href</span><span class="ot">=</span><span class="st">"#4-filter"</span> <span class="er">id</span><span class="ot">=</span><span class="st">"toc-4-filter"</span><span class="kw">&gt;</span>4 Filter<span class="kw">&lt;/a&gt;</span></span>
<span id="cb39-44"><a href="#cb39-44" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span><span class="kw">&lt;a</span> <span class="er">href</span><span class="ot">=</span><span class="st">"#5-bedtools"</span> <span class="er">id</span><span class="ot">=</span><span class="st">"toc-5-bedtools"</span><span class="kw">&gt;</span>5 Bedtools<span class="kw">&lt;/a&gt;</span></span>
<span id="cb39-45"><a href="#cb39-45" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span><span class="kw">&lt;a</span> <span class="er">href</span><span class="ot">=</span><span class="st">"#6-cpc2"</span> <span class="er">id</span><span class="ot">=</span><span class="st">"toc-6-cpc2"</span><span class="kw">&gt;</span>6 CPC2<span class="kw">&lt;/a&gt;</span></span>
<span id="cb39-46"><a href="#cb39-46" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span><span class="kw">&lt;a</span> <span class="er">href</span><span class="ot">=</span><span class="st">"#7-de-duplicate-gtf"</span> <span class="er">id</span><span class="ot">=</span><span class="st">"toc-7-de-duplicate-gtf"</span><span class="kw">&gt;</span>7 De-duplicate</span>
<span id="cb39-47"><a href="#cb39-47" aria-hidden="true" tabindex="-1"></a>  GTF<span class="kw">&lt;/a&gt;</span></span>
<span id="cb39-48"><a href="#cb39-48" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span><span class="kw">&lt;a</span> <span class="er">href</span><span class="ot">=</span><span class="st">"#8-reformat-to-bed"</span> <span class="er">id</span><span class="ot">=</span><span class="st">"toc-8-reformat-to-bed"</span><span class="kw">&gt;</span>8 Reformat to</span>
<span id="cb39-49"><a href="#cb39-49" aria-hidden="true" tabindex="-1"></a>  BED<span class="kw">&lt;/a&gt;</span></span>
<span id="cb39-50"><a href="#cb39-50" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span><span class="kw">&lt;a</span> <span class="er">href</span><span class="ot">=</span><span class="st">"#9-bedtools"</span> <span class="er">id</span><span class="ot">=</span><span class="st">"toc-9-bedtools"</span><span class="kw">&gt;</span>9 BEDTools<span class="kw">&lt;/a&gt;</span></span>
<span id="cb39-51"><a href="#cb39-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-52"><a href="#cb39-52" aria-hidden="true" tabindex="-1"></a><span class="fu"># 1 Run HiSat on RNA-seq</span></span>
<span id="cb39-53"><a href="#cb39-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-54"><a href="#cb39-54" aria-hidden="true" tabindex="-1"></a><span class="fu">## 1.1 Grab Trimmed RNA-seq Reads</span></span>
<span id="cb39-55"><a href="#cb39-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-56"><a href="#cb39-56" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb39-57"><a href="#cb39-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-58"><a href="#cb39-58" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-r</span> <span class="dt">\</span></span>
<span id="cb39-59"><a href="#cb39-59" aria-hidden="true" tabindex="-1"></a>--no-directories <span class="at">--no-parent</span> <span class="dt">\</span></span>
<span id="cb39-60"><a href="#cb39-60" aria-hidden="true" tabindex="-1"></a>-P ../data/fastq/ <span class="dt">\</span></span>
<span id="cb39-61"><a href="#cb39-61" aria-hidden="true" tabindex="-1"></a>-A <span class="st">"*fastq.gz"</span> https://gannet.fish.washington.edu/Atumefaciens/20230519-E5_coral-fastqc-fastp-multiqc-RNAseq/A_pulchra/trimmed/</span>
<span id="cb39-62"><a href="#cb39-62" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-63"><a href="#cb39-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-64"><a href="#cb39-64" aria-hidden="true" tabindex="-1"></a><span class="fu">## 1.2 Genome</span></span>
<span id="cb39-65"><a href="#cb39-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-66"><a href="#cb39-66" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb39-67"><a href="#cb39-67" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> ../data</span>
<span id="cb39-68"><a href="#cb39-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-69"><a href="#cb39-69" aria-hidden="true" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-O</span> http://gannet.fish.washington.edu/seashell/snaps/GCF_013753865.1_Amil_v2.1_genomic.fna</span>
<span id="cb39-70"><a href="#cb39-70" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-71"><a href="#cb39-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-72"><a href="#cb39-72" aria-hidden="true" tabindex="-1"></a><span class="fu">## 1.3 HiSat</span></span>
<span id="cb39-73"><a href="#cb39-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-74"><a href="#cb39-74" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb39-75"><a href="#cb39-75" aria-hidden="true" tabindex="-1"></a><span class="ex">/home/shared/hisat2-2.2.1/hisat2_extract_exons.py</span> <span class="dt">\</span></span>
<span id="cb39-76"><a href="#cb39-76" aria-hidden="true" tabindex="-1"></a>../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gtf <span class="dt">\</span></span>
<span id="cb39-77"><a href="#cb39-77" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span> ../output/05.32-lncRNA-discovery/m_exon.tab</span>
<span id="cb39-78"><a href="#cb39-78" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-79"><a href="#cb39-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-80"><a href="#cb39-80" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb39-81"><a href="#cb39-81" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> ../output/05.32-lncRNA-discovery/m_exon.tab</span>
<span id="cb39-82"><a href="#cb39-82" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-83"><a href="#cb39-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-84"><a href="#cb39-84" aria-hidden="true" tabindex="-1"></a><span class="in">    ## NC_058066.1  1961    2118    -</span></span>
<span id="cb39-85"><a href="#cb39-85" aria-hidden="true" tabindex="-1"></a><span class="in">    ## NC_058066.1  15360   15663   +</span></span>
<span id="cb39-86"><a href="#cb39-86" aria-hidden="true" tabindex="-1"></a><span class="in">    ## NC_058066.1  18710   18774   -</span></span>
<span id="cb39-87"><a href="#cb39-87" aria-hidden="true" tabindex="-1"></a><span class="in">    ## NC_058066.1  21000   21092   -</span></span>
<span id="cb39-88"><a href="#cb39-88" aria-hidden="true" tabindex="-1"></a><span class="in">    ## NC_058066.1  22158   22442   +</span></span>
<span id="cb39-89"><a href="#cb39-89" aria-hidden="true" tabindex="-1"></a><span class="in">    ## NC_058066.1  23084   23220   -</span></span>
<span id="cb39-90"><a href="#cb39-90" aria-hidden="true" tabindex="-1"></a><span class="in">    ## NC_058066.1  24770   25066   +</span></span>
<span id="cb39-91"><a href="#cb39-91" aria-hidden="true" tabindex="-1"></a><span class="in">    ## NC_058066.1  26652   26912   +</span></span>
<span id="cb39-92"><a href="#cb39-92" aria-hidden="true" tabindex="-1"></a><span class="in">    ## NC_058066.1  27071   27358   +</span></span>
<span id="cb39-93"><a href="#cb39-93" aria-hidden="true" tabindex="-1"></a><span class="in">    ## NC_058066.1  27658   27951   +</span></span>
<span id="cb39-94"><a href="#cb39-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-95"><a href="#cb39-95" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb39-96"><a href="#cb39-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-97"><a href="#cb39-97" aria-hidden="true" tabindex="-1"></a><span class="ex">/home/shared/hisat2-2.2.1/hisat2_extract_splice_sites.py</span> <span class="dt">\</span></span>
<span id="cb39-98"><a href="#cb39-98" aria-hidden="true" tabindex="-1"></a>../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gtf <span class="dt">\</span></span>
<span id="cb39-99"><a href="#cb39-99" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span> ../output/05.32-lncRNA-discovery/m_splice_sites.tab</span>
<span id="cb39-100"><a href="#cb39-100" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-101"><a href="#cb39-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-102"><a href="#cb39-102" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb39-103"><a href="#cb39-103" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> ../output/05.32-lncRNA-discovery/m_splice_sites.tab</span>
<span id="cb39-104"><a href="#cb39-104" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-105"><a href="#cb39-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-106"><a href="#cb39-106" aria-hidden="true" tabindex="-1"></a><span class="in">    ## NC_058066.1  2118    18710   -</span></span>
<span id="cb39-107"><a href="#cb39-107" aria-hidden="true" tabindex="-1"></a><span class="in">    ## NC_058066.1  15663   22158   +</span></span>
<span id="cb39-108"><a href="#cb39-108" aria-hidden="true" tabindex="-1"></a><span class="in">    ## NC_058066.1  18774   21000   -</span></span>
<span id="cb39-109"><a href="#cb39-109" aria-hidden="true" tabindex="-1"></a><span class="in">    ## NC_058066.1  21092   23084   -</span></span>
<span id="cb39-110"><a href="#cb39-110" aria-hidden="true" tabindex="-1"></a><span class="in">    ## NC_058066.1  22442   24770   +</span></span>
<span id="cb39-111"><a href="#cb39-111" aria-hidden="true" tabindex="-1"></a><span class="in">    ## NC_058066.1  25066   26652   +</span></span>
<span id="cb39-112"><a href="#cb39-112" aria-hidden="true" tabindex="-1"></a><span class="in">    ## NC_058066.1  26912   27071   +</span></span>
<span id="cb39-113"><a href="#cb39-113" aria-hidden="true" tabindex="-1"></a><span class="in">    ## NC_058066.1  27358   27658   +</span></span>
<span id="cb39-114"><a href="#cb39-114" aria-hidden="true" tabindex="-1"></a><span class="in">    ## NC_058066.1  27951   28247   +</span></span>
<span id="cb39-115"><a href="#cb39-115" aria-hidden="true" tabindex="-1"></a><span class="in">    ## NC_058066.1  28534   29196   +</span></span>
<span id="cb39-116"><a href="#cb39-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-117"><a href="#cb39-117" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb39-118"><a href="#cb39-118" aria-hidden="true" tabindex="-1"></a><span class="ex">/home/shared/hisat2-2.2.1/hisat2-build</span> <span class="dt">\</span></span>
<span id="cb39-119"><a href="#cb39-119" aria-hidden="true" tabindex="-1"></a>../data/GCF_013753865.1_Amil_v2.1_genomic.fna <span class="dt">\</span></span>
<span id="cb39-120"><a href="#cb39-120" aria-hidden="true" tabindex="-1"></a>../data/GCF_013753865.1_Amil_v2.1.index <span class="dt">\</span></span>
<span id="cb39-121"><a href="#cb39-121" aria-hidden="true" tabindex="-1"></a>--exon ../output/05.32-lncRNA-discovery/m_exon.tab <span class="dt">\</span></span>
<span id="cb39-122"><a href="#cb39-122" aria-hidden="true" tabindex="-1"></a>--ss ../output/05.32-lncRNA-discovery/m_splice_sites.tab <span class="dt">\</span></span>
<span id="cb39-123"><a href="#cb39-123" aria-hidden="true" tabindex="-1"></a>-p 40 <span class="dt">\</span></span>
<span id="cb39-124"><a href="#cb39-124" aria-hidden="true" tabindex="-1"></a>../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gtf <span class="dt">\</span></span>
<span id="cb39-125"><a href="#cb39-125" aria-hidden="true" tabindex="-1"></a>2<span class="op">&gt;</span> ../output/05.32-lncRNA-discovery/hisat2-build_stats.txt</span>
<span id="cb39-126"><a href="#cb39-126" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-127"><a href="#cb39-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-128"><a href="#cb39-128" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb39-129"><a href="#cb39-129" aria-hidden="true" tabindex="-1"></a><span class="fu">find</span> ../data/fastq/<span class="pp">*</span>gz</span>
<span id="cb39-130"><a href="#cb39-130" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-131"><a href="#cb39-131" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-132"><a href="#cb39-132" aria-hidden="true" tabindex="-1"></a><span class="in">    ## ../data/fastq/RNA-ACR-140-S1-TP2_R1_001.fastp-trim.20230519.fastq.gz</span></span>
<span id="cb39-133"><a href="#cb39-133" aria-hidden="true" tabindex="-1"></a><span class="in">    ## ../data/fastq/RNA-ACR-140-S1-TP2_R2_001.fastp-trim.20230519.fastq.gz</span></span>
<span id="cb39-134"><a href="#cb39-134" aria-hidden="true" tabindex="-1"></a><span class="in">    ## ../data/fastq/RNA-ACR-145-S1-TP2_R1_001.fastp-trim.20230519.fastq.gz</span></span>
<span id="cb39-135"><a href="#cb39-135" aria-hidden="true" tabindex="-1"></a><span class="in">    ## ../data/fastq/RNA-ACR-145-S1-TP2_R2_001.fastp-trim.20230519.fastq.gz</span></span>
<span id="cb39-136"><a href="#cb39-136" aria-hidden="true" tabindex="-1"></a><span class="in">    ## ../data/fastq/RNA-ACR-150-S1-TP2_R1_001.fastp-trim.20230519.fastq.gz</span></span>
<span id="cb39-137"><a href="#cb39-137" aria-hidden="true" tabindex="-1"></a><span class="in">    ## ../data/fastq/RNA-ACR-150-S1-TP2_R2_001.fastp-trim.20230519.fastq.gz</span></span>
<span id="cb39-138"><a href="#cb39-138" aria-hidden="true" tabindex="-1"></a><span class="in">    ## ../data/fastq/RNA-ACR-173-S1-TP2_R1_001.fastp-trim.20230519.fastq.gz</span></span>
<span id="cb39-139"><a href="#cb39-139" aria-hidden="true" tabindex="-1"></a><span class="in">    ## ../data/fastq/RNA-ACR-173-S1-TP2_R2_001.fastp-trim.20230519.fastq.gz</span></span>
<span id="cb39-140"><a href="#cb39-140" aria-hidden="true" tabindex="-1"></a><span class="in">    ## ../data/fastq/RNA-ACR-178-S1-TP2_R1_001.fastp-trim.20230519.fastq.gz</span></span>
<span id="cb39-141"><a href="#cb39-141" aria-hidden="true" tabindex="-1"></a><span class="in">    ## ../data/fastq/RNA-ACR-178-S1-TP2_R2_001.fastp-trim.20230519.fastq.gz</span></span>
<span id="cb39-142"><a href="#cb39-142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-143"><a href="#cb39-143" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb39-144"><a href="#cb39-144" aria-hidden="true" tabindex="-1"></a><span class="fu">find</span> ../data/fastq/<span class="pp">*</span>R2_001.fastp-trim.20230519.fastq.gz <span class="dt">\</span></span>
<span id="cb39-145"><a href="#cb39-145" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span> <span class="fu">xargs</span> basename <span class="at">-s</span> <span class="at">-S1-TP2_R2_001.fastp-trim.20230519.fastq.gz</span> <span class="kw">|</span> <span class="fu">xargs</span> <span class="at">-I{}</span> <span class="dt">\</span></span>
<span id="cb39-146"><a href="#cb39-146" aria-hidden="true" tabindex="-1"></a>echo {}</span>
<span id="cb39-147"><a href="#cb39-147" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-148"><a href="#cb39-148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-149"><a href="#cb39-149" aria-hidden="true" tabindex="-1"></a><span class="in">    ## RNA-ACR-140</span></span>
<span id="cb39-150"><a href="#cb39-150" aria-hidden="true" tabindex="-1"></a><span class="in">    ## RNA-ACR-145</span></span>
<span id="cb39-151"><a href="#cb39-151" aria-hidden="true" tabindex="-1"></a><span class="in">    ## RNA-ACR-150</span></span>
<span id="cb39-152"><a href="#cb39-152" aria-hidden="true" tabindex="-1"></a><span class="in">    ## RNA-ACR-173</span></span>
<span id="cb39-153"><a href="#cb39-153" aria-hidden="true" tabindex="-1"></a><span class="in">    ## RNA-ACR-178</span></span>
<span id="cb39-154"><a href="#cb39-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-155"><a href="#cb39-155" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb39-156"><a href="#cb39-156" aria-hidden="true" tabindex="-1"></a><span class="fu">find</span> ../data/fastq/<span class="pp">*</span>R2_001.fastp-trim.20230519.fastq.gz <span class="dt">\</span></span>
<span id="cb39-157"><a href="#cb39-157" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span> <span class="fu">xargs</span> basename <span class="at">-s</span> <span class="at">-S1-TP2_R2_001.fastp-trim.20230519.fastq.gz</span> <span class="kw">|</span> <span class="fu">xargs</span> <span class="at">-I{}</span> <span class="dt">\</span></span>
<span id="cb39-158"><a href="#cb39-158" aria-hidden="true" tabindex="-1"></a>/home/shared/hisat2-2.2.1/hisat2 <span class="dt">\</span></span>
<span id="cb39-159"><a href="#cb39-159" aria-hidden="true" tabindex="-1"></a>-x ../data/GCF_013753865.1_Amil_v2.1.index <span class="dt">\</span></span>
<span id="cb39-160"><a href="#cb39-160" aria-hidden="true" tabindex="-1"></a>-p 20 <span class="dt">\</span></span>
<span id="cb39-161"><a href="#cb39-161" aria-hidden="true" tabindex="-1"></a>-1 ../data/fastq/{}-S1-TP2_R1_001.fastp-trim.20230519.fastq.gz <span class="dt">\</span></span>
<span id="cb39-162"><a href="#cb39-162" aria-hidden="true" tabindex="-1"></a>-2 ../data/fastq/{}-S1-TP2_R2_001.fastp-trim.20230519.fastq.gz <span class="dt">\</span></span>
<span id="cb39-163"><a href="#cb39-163" aria-hidden="true" tabindex="-1"></a>-S ../output/05.32-lncRNA-discovery/{}.sam</span>
<span id="cb39-164"><a href="#cb39-164" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-165"><a href="#cb39-165" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-166"><a href="#cb39-166" aria-hidden="true" tabindex="-1"></a><span class="in">    47710408 reads; of these:</span></span>
<span id="cb39-167"><a href="#cb39-167" aria-hidden="true" tabindex="-1"></a><span class="in">      47710408 (100.00%) were paired; of these:</span></span>
<span id="cb39-168"><a href="#cb39-168" aria-hidden="true" tabindex="-1"></a><span class="in">        27056517 (56.71%) aligned concordantly 0 times</span></span>
<span id="cb39-169"><a href="#cb39-169" aria-hidden="true" tabindex="-1"></a><span class="in">        19229695 (40.31%) aligned concordantly exactly 1 time</span></span>
<span id="cb39-170"><a href="#cb39-170" aria-hidden="true" tabindex="-1"></a><span class="in">        1424196 (2.99%) aligned concordantly &gt;1 times</span></span>
<span id="cb39-171"><a href="#cb39-171" aria-hidden="true" tabindex="-1"></a><span class="in">        ----</span></span>
<span id="cb39-172"><a href="#cb39-172" aria-hidden="true" tabindex="-1"></a><span class="in">        27056517 pairs aligned concordantly 0 times; of these:</span></span>
<span id="cb39-173"><a href="#cb39-173" aria-hidden="true" tabindex="-1"></a><span class="in">          1316579 (4.87%) aligned discordantly 1 time</span></span>
<span id="cb39-174"><a href="#cb39-174" aria-hidden="true" tabindex="-1"></a><span class="in">        ----</span></span>
<span id="cb39-175"><a href="#cb39-175" aria-hidden="true" tabindex="-1"></a><span class="in">        25739938 pairs aligned 0 times concordantly or discordantly; of these:</span></span>
<span id="cb39-176"><a href="#cb39-176" aria-hidden="true" tabindex="-1"></a><span class="in">          51479876 mates make up the pairs; of these:</span></span>
<span id="cb39-177"><a href="#cb39-177" aria-hidden="true" tabindex="-1"></a><span class="in">            41152624 (79.94%) aligned 0 times</span></span>
<span id="cb39-178"><a href="#cb39-178" aria-hidden="true" tabindex="-1"></a><span class="in">            9290398 (18.05%) aligned exactly 1 time</span></span>
<span id="cb39-179"><a href="#cb39-179" aria-hidden="true" tabindex="-1"></a><span class="in">            1036854 (2.01%) aligned &gt;1 times</span></span>
<span id="cb39-180"><a href="#cb39-180" aria-hidden="true" tabindex="-1"></a><span class="in">    56.87% overall alignment rate</span></span>
<span id="cb39-181"><a href="#cb39-181" aria-hidden="true" tabindex="-1"></a><span class="in">    42864294 reads; of these:</span></span>
<span id="cb39-182"><a href="#cb39-182" aria-hidden="true" tabindex="-1"></a><span class="in">      42864294 (100.00%) were paired; of these:</span></span>
<span id="cb39-183"><a href="#cb39-183" aria-hidden="true" tabindex="-1"></a><span class="in">        23649791 (55.17%) aligned concordantly 0 times</span></span>
<span id="cb39-184"><a href="#cb39-184" aria-hidden="true" tabindex="-1"></a><span class="in">        17703533 (41.30%) aligned concordantly exactly 1 time</span></span>
<span id="cb39-185"><a href="#cb39-185" aria-hidden="true" tabindex="-1"></a><span class="in">        1510970 (3.53%) aligned concordantly &gt;1 times</span></span>
<span id="cb39-186"><a href="#cb39-186" aria-hidden="true" tabindex="-1"></a><span class="in">        ----</span></span>
<span id="cb39-187"><a href="#cb39-187" aria-hidden="true" tabindex="-1"></a><span class="in">        23649791 pairs aligned concordantly 0 times; of these:</span></span>
<span id="cb39-188"><a href="#cb39-188" aria-hidden="true" tabindex="-1"></a><span class="in">          1250605 (5.29%) aligned discordantly 1 time</span></span>
<span id="cb39-189"><a href="#cb39-189" aria-hidden="true" tabindex="-1"></a><span class="in">        ----</span></span>
<span id="cb39-190"><a href="#cb39-190" aria-hidden="true" tabindex="-1"></a><span class="in">        22399186 pairs aligned 0 times concordantly or discordantly; of these:</span></span>
<span id="cb39-191"><a href="#cb39-191" aria-hidden="true" tabindex="-1"></a><span class="in">          44798372 mates make up the pairs; of these:</span></span>
<span id="cb39-192"><a href="#cb39-192" aria-hidden="true" tabindex="-1"></a><span class="in">            35369621 (78.95%) aligned 0 times</span></span>
<span id="cb39-193"><a href="#cb39-193" aria-hidden="true" tabindex="-1"></a><span class="in">            8218798 (18.35%) aligned exactly 1 time</span></span>
<span id="cb39-194"><a href="#cb39-194" aria-hidden="true" tabindex="-1"></a><span class="in">            1209953 (2.70%) aligned &gt;1 times</span></span>
<span id="cb39-195"><a href="#cb39-195" aria-hidden="true" tabindex="-1"></a><span class="in">    58.74% overall alignment rate</span></span>
<span id="cb39-196"><a href="#cb39-196" aria-hidden="true" tabindex="-1"></a><span class="in">    43712298 reads; of these:</span></span>
<span id="cb39-197"><a href="#cb39-197" aria-hidden="true" tabindex="-1"></a><span class="in">      43712298 (100.00%) were paired; of these:</span></span>
<span id="cb39-198"><a href="#cb39-198" aria-hidden="true" tabindex="-1"></a><span class="in">        30355359 (69.44%) aligned concordantly 0 times</span></span>
<span id="cb39-199"><a href="#cb39-199" aria-hidden="true" tabindex="-1"></a><span class="in">        12131465 (27.75%) aligned concordantly exactly 1 time</span></span>
<span id="cb39-200"><a href="#cb39-200" aria-hidden="true" tabindex="-1"></a><span class="in">        1225474 (2.80%) aligned concordantly &gt;1 times</span></span>
<span id="cb39-201"><a href="#cb39-201" aria-hidden="true" tabindex="-1"></a><span class="in">        ----</span></span>
<span id="cb39-202"><a href="#cb39-202" aria-hidden="true" tabindex="-1"></a><span class="in">        30355359 pairs aligned concordantly 0 times; of these:</span></span>
<span id="cb39-203"><a href="#cb39-203" aria-hidden="true" tabindex="-1"></a><span class="in">          848325 (2.79%) aligned discordantly 1 time</span></span>
<span id="cb39-204"><a href="#cb39-204" aria-hidden="true" tabindex="-1"></a><span class="in">        ----</span></span>
<span id="cb39-205"><a href="#cb39-205" aria-hidden="true" tabindex="-1"></a><span class="in">        29507034 pairs aligned 0 times concordantly or discordantly; of these:</span></span>
<span id="cb39-206"><a href="#cb39-206" aria-hidden="true" tabindex="-1"></a><span class="in">          59014068 mates make up the pairs; of these:</span></span>
<span id="cb39-207"><a href="#cb39-207" aria-hidden="true" tabindex="-1"></a><span class="in">            51031599 (86.47%) aligned 0 times</span></span>
<span id="cb39-208"><a href="#cb39-208" aria-hidden="true" tabindex="-1"></a><span class="in">            6695839 (11.35%) aligned exactly 1 time</span></span>
<span id="cb39-209"><a href="#cb39-209" aria-hidden="true" tabindex="-1"></a><span class="in">            1286630 (2.18%) aligned &gt;1 times</span></span>
<span id="cb39-210"><a href="#cb39-210" aria-hidden="true" tabindex="-1"></a><span class="in">    41.63% overall alignment rate</span></span>
<span id="cb39-211"><a href="#cb39-211" aria-hidden="true" tabindex="-1"></a><span class="in">    47501524 reads; of these:</span></span>
<span id="cb39-212"><a href="#cb39-212" aria-hidden="true" tabindex="-1"></a><span class="in">      47501524 (100.00%) were paired; of these:</span></span>
<span id="cb39-213"><a href="#cb39-213" aria-hidden="true" tabindex="-1"></a><span class="in">        27827915 (58.58%) aligned concordantly 0 times</span></span>
<span id="cb39-214"><a href="#cb39-214" aria-hidden="true" tabindex="-1"></a><span class="in">        18298532 (38.52%) aligned concordantly exactly 1 time</span></span>
<span id="cb39-215"><a href="#cb39-215" aria-hidden="true" tabindex="-1"></a><span class="in">        1375077 (2.89%) aligned concordantly &gt;1 times</span></span>
<span id="cb39-216"><a href="#cb39-216" aria-hidden="true" tabindex="-1"></a><span class="in">        ----</span></span>
<span id="cb39-217"><a href="#cb39-217" aria-hidden="true" tabindex="-1"></a><span class="in">        27827915 pairs aligned concordantly 0 times; of these:</span></span>
<span id="cb39-218"><a href="#cb39-218" aria-hidden="true" tabindex="-1"></a><span class="in">          1321896 (4.75%) aligned discordantly 1 time</span></span>
<span id="cb39-219"><a href="#cb39-219" aria-hidden="true" tabindex="-1"></a><span class="in">        ----</span></span>
<span id="cb39-220"><a href="#cb39-220" aria-hidden="true" tabindex="-1"></a><span class="in">        26506019 pairs aligned 0 times concordantly or discordantly; of these:</span></span>
<span id="cb39-221"><a href="#cb39-221" aria-hidden="true" tabindex="-1"></a><span class="in">          53012038 mates make up the pairs; of these:</span></span>
<span id="cb39-222"><a href="#cb39-222" aria-hidden="true" tabindex="-1"></a><span class="in">            42654805 (80.46%) aligned 0 times</span></span>
<span id="cb39-223"><a href="#cb39-223" aria-hidden="true" tabindex="-1"></a><span class="in">            9174277 (17.31%) aligned exactly 1 time</span></span>
<span id="cb39-224"><a href="#cb39-224" aria-hidden="true" tabindex="-1"></a><span class="in">            1182956 (2.23%) aligned &gt;1 times</span></span>
<span id="cb39-225"><a href="#cb39-225" aria-hidden="true" tabindex="-1"></a><span class="in">    55.10% overall alignment rate</span></span>
<span id="cb39-226"><a href="#cb39-226" aria-hidden="true" tabindex="-1"></a><span class="in">    42677752 reads; of these:</span></span>
<span id="cb39-227"><a href="#cb39-227" aria-hidden="true" tabindex="-1"></a><span class="in">      42677752 (100.00%) were paired; of these:</span></span>
<span id="cb39-228"><a href="#cb39-228" aria-hidden="true" tabindex="-1"></a><span class="in">        25646587 (60.09%) aligned concordantly 0 times</span></span>
<span id="cb39-229"><a href="#cb39-229" aria-hidden="true" tabindex="-1"></a><span class="in">        15654238 (36.68%) aligned concordantly exactly 1 time</span></span>
<span id="cb39-230"><a href="#cb39-230" aria-hidden="true" tabindex="-1"></a><span class="in">        1376927 (3.23%) aligned concordantly &gt;1 times</span></span>
<span id="cb39-231"><a href="#cb39-231" aria-hidden="true" tabindex="-1"></a><span class="in">        ----</span></span>
<span id="cb39-232"><a href="#cb39-232" aria-hidden="true" tabindex="-1"></a><span class="in">        25646587 pairs aligned concordantly 0 times; of these:</span></span>
<span id="cb39-233"><a href="#cb39-233" aria-hidden="true" tabindex="-1"></a><span class="in">          1105089 (4.31%) aligned discordantly 1 time</span></span>
<span id="cb39-234"><a href="#cb39-234" aria-hidden="true" tabindex="-1"></a><span class="in">        ----</span></span>
<span id="cb39-235"><a href="#cb39-235" aria-hidden="true" tabindex="-1"></a><span class="in">        24541498 pairs aligned 0 times concordantly or discordantly; of these:</span></span>
<span id="cb39-236"><a href="#cb39-236" aria-hidden="true" tabindex="-1"></a><span class="in">          49082996 mates make up the pairs; of these:</span></span>
<span id="cb39-237"><a href="#cb39-237" aria-hidden="true" tabindex="-1"></a><span class="in">            38229072 (77.89%) aligned 0 times</span></span>
<span id="cb39-238"><a href="#cb39-238" aria-hidden="true" tabindex="-1"></a><span class="in">            9393012 (19.14%) aligned exactly 1 time</span></span>
<span id="cb39-239"><a href="#cb39-239" aria-hidden="true" tabindex="-1"></a><span class="in">            1460912 (2.98%) aligned &gt;1 times</span></span>
<span id="cb39-240"><a href="#cb39-240" aria-hidden="true" tabindex="-1"></a><span class="in">    55.21% overall alignment rate</span></span>
<span id="cb39-241"><a href="#cb39-241" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-242"><a href="#cb39-242" aria-hidden="true" tabindex="-1"></a><span class="fu">## 1.4 convert to bams</span></span>
<span id="cb39-243"><a href="#cb39-243" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-244"><a href="#cb39-244" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb39-245"><a href="#cb39-245" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> samfile <span class="kw">in</span> ../output/05.32-lncRNA-discovery/<span class="pp">*</span>.sam<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb39-246"><a href="#cb39-246" aria-hidden="true" tabindex="-1"></a>  <span class="va">bamfile</span><span class="op">=</span><span class="st">"</span><span class="va">${samfile</span><span class="op">%</span>.sam<span class="va">}</span><span class="st">.bam"</span></span>
<span id="cb39-247"><a href="#cb39-247" aria-hidden="true" tabindex="-1"></a>  <span class="va">sorted_bamfile</span><span class="op">=</span><span class="st">"</span><span class="va">${samfile</span><span class="op">%</span>.sam<span class="va">}</span><span class="st">.sorted.bam"</span></span>
<span id="cb39-248"><a href="#cb39-248" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb39-249"><a href="#cb39-249" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Convert SAM to BAM</span></span>
<span id="cb39-250"><a href="#cb39-250" aria-hidden="true" tabindex="-1"></a>  <span class="ex">/home/shared/samtools-1.12/samtools</span> view <span class="at">-bS</span> <span class="st">"</span><span class="va">$samfile</span><span class="st">"</span> <span class="op">&gt;</span> <span class="st">"</span><span class="va">$bamfile</span><span class="st">"</span></span>
<span id="cb39-251"><a href="#cb39-251" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb39-252"><a href="#cb39-252" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Sort BAM</span></span>
<span id="cb39-253"><a href="#cb39-253" aria-hidden="true" tabindex="-1"></a>  <span class="ex">/home/shared/samtools-1.12/samtools</span> sort <span class="st">"</span><span class="va">$bamfile</span><span class="st">"</span> <span class="at">-o</span> <span class="st">"</span><span class="va">$sorted_bamfile</span><span class="st">"</span></span>
<span id="cb39-254"><a href="#cb39-254" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb39-255"><a href="#cb39-255" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Index sorted BAM</span></span>
<span id="cb39-256"><a href="#cb39-256" aria-hidden="true" tabindex="-1"></a>  <span class="ex">/home/shared/samtools-1.12/samtools</span> index <span class="st">"</span><span class="va">$sorted_bamfile</span><span class="st">"</span></span>
<span id="cb39-257"><a href="#cb39-257" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb39-258"><a href="#cb39-258" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-259"><a href="#cb39-259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-260"><a href="#cb39-260" aria-hidden="true" tabindex="-1"></a><span class="fu"># 2 StringTie</span></span>
<span id="cb39-261"><a href="#cb39-261" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-262"><a href="#cb39-262" aria-hidden="true" tabindex="-1"></a>StringTie uses the sorted BAM files to assemble transcripts for each</span>
<span id="cb39-263"><a href="#cb39-263" aria-hidden="true" tabindex="-1"></a>sample, outputting them as GTF (Gene Transfer Format) files. And then</span>
<span id="cb39-264"><a href="#cb39-264" aria-hidden="true" tabindex="-1"></a>merges all individual GTF assemblies into a single merged GTF file. This</span>
<span id="cb39-265"><a href="#cb39-265" aria-hidden="true" tabindex="-1"></a>step extracts transcript information and merges GTFs from all samples–an</span>
<span id="cb39-266"><a href="#cb39-266" aria-hidden="true" tabindex="-1"></a>important step in creating a canonical list of lncRNAs across all</span>
<span id="cb39-267"><a href="#cb39-267" aria-hidden="true" tabindex="-1"></a>samples included in the pipeline.</span>
<span id="cb39-268"><a href="#cb39-268" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-269"><a href="#cb39-269" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb39-270"><a href="#cb39-270" aria-hidden="true" tabindex="-1"></a><span class="fu">find</span> ../output/05.32-lncRNA-discovery/<span class="pp">*</span>sorted.bam <span class="dt">\</span></span>
<span id="cb39-271"><a href="#cb39-271" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span> <span class="fu">xargs</span> basename <span class="at">-s</span> .sorted.bam <span class="kw">|</span> <span class="fu">xargs</span> <span class="at">-I{}</span> <span class="dt">\</span></span>
<span id="cb39-272"><a href="#cb39-272" aria-hidden="true" tabindex="-1"></a>/home/shared/stringtie-2.2.1.Linux_x86_64/stringtie <span class="dt">\</span></span>
<span id="cb39-273"><a href="#cb39-273" aria-hidden="true" tabindex="-1"></a>-p 8 <span class="dt">\</span></span>
<span id="cb39-274"><a href="#cb39-274" aria-hidden="true" tabindex="-1"></a>-G ../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gff <span class="dt">\</span></span>
<span id="cb39-275"><a href="#cb39-275" aria-hidden="true" tabindex="-1"></a>-o ../output/05.32-lncRNA-discovery/{}.gtf <span class="dt">\</span></span>
<span id="cb39-276"><a href="#cb39-276" aria-hidden="true" tabindex="-1"></a>../output/05.32-lncRNA-discovery/{}.sorted.bam</span>
<span id="cb39-277"><a href="#cb39-277" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-278"><a href="#cb39-278" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-279"><a href="#cb39-279" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb39-280"><a href="#cb39-280" aria-hidden="true" tabindex="-1"></a><span class="fu">wc</span> <span class="at">-l</span> ../output/05.32-lncRNA-discovery/RNA<span class="pp">*</span>.gtf</span>
<span id="cb39-281"><a href="#cb39-281" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> ../output/05.32-lncRNA-discovery/RNA<span class="pp">*</span>.gtf</span>
<span id="cb39-282"><a href="#cb39-282" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-283"><a href="#cb39-283" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-284"><a href="#cb39-284" aria-hidden="true" tabindex="-1"></a><span class="in">    ##    373617 ../output/05.32-lncRNA-discovery/RNA-ACR-140.gtf</span></span>
<span id="cb39-285"><a href="#cb39-285" aria-hidden="true" tabindex="-1"></a><span class="in">    ##    322109 ../output/05.32-lncRNA-discovery/RNA-ACR-145.gtf</span></span>
<span id="cb39-286"><a href="#cb39-286" aria-hidden="true" tabindex="-1"></a><span class="in">    ##    340380 ../output/05.32-lncRNA-discovery/RNA-ACR-150.gtf</span></span>
<span id="cb39-287"><a href="#cb39-287" aria-hidden="true" tabindex="-1"></a><span class="in">    ##    367972 ../output/05.32-lncRNA-discovery/RNA-ACR-173.gtf</span></span>
<span id="cb39-288"><a href="#cb39-288" aria-hidden="true" tabindex="-1"></a><span class="in">    ##    426804 ../output/05.32-lncRNA-discovery/RNA-ACR-178.gtf</span></span>
<span id="cb39-289"><a href="#cb39-289" aria-hidden="true" tabindex="-1"></a><span class="in">    ##   1830882 total</span></span>
<span id="cb39-290"><a href="#cb39-290" aria-hidden="true" tabindex="-1"></a><span class="in">    ## ../output/05.32-lncRNA-discovery/RNA-ACR-140.gtf</span></span>
<span id="cb39-291"><a href="#cb39-291" aria-hidden="true" tabindex="-1"></a><span class="in">    ## ../output/05.32-lncRNA-discovery/RNA-ACR-145.gtf</span></span>
<span id="cb39-292"><a href="#cb39-292" aria-hidden="true" tabindex="-1"></a><span class="in">    ## ../output/05.32-lncRNA-discovery/RNA-ACR-150.gtf</span></span>
<span id="cb39-293"><a href="#cb39-293" aria-hidden="true" tabindex="-1"></a><span class="in">    ## ../output/05.32-lncRNA-discovery/RNA-ACR-173.gtf</span></span>
<span id="cb39-294"><a href="#cb39-294" aria-hidden="true" tabindex="-1"></a><span class="in">    ## ../output/05.32-lncRNA-discovery/RNA-ACR-178.gtf</span></span>
<span id="cb39-295"><a href="#cb39-295" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-296"><a href="#cb39-296" aria-hidden="true" tabindex="-1"></a>Merges all individual GTF assemblies into a single merged GTF file.</span>
<span id="cb39-297"><a href="#cb39-297" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-298"><a href="#cb39-298" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb39-299"><a href="#cb39-299" aria-hidden="true" tabindex="-1"></a><span class="ex">/home/shared/stringtie-2.2.1.Linux_x86_64/stringtie</span> <span class="dt">\</span></span>
<span id="cb39-300"><a href="#cb39-300" aria-hidden="true" tabindex="-1"></a>--merge <span class="dt">\</span></span>
<span id="cb39-301"><a href="#cb39-301" aria-hidden="true" tabindex="-1"></a>-G ../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gff <span class="dt">\</span></span>
<span id="cb39-302"><a href="#cb39-302" aria-hidden="true" tabindex="-1"></a>-o ../output/05.32-lncRNA-discovery/stringtie_merged.gtf <span class="dt">\</span></span>
<span id="cb39-303"><a href="#cb39-303" aria-hidden="true" tabindex="-1"></a>../output/05.32-lncRNA-discovery/<span class="pp">*</span>.gtf</span>
<span id="cb39-304"><a href="#cb39-304" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-305"><a href="#cb39-305" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-306"><a href="#cb39-306" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb39-307"><a href="#cb39-307" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-308"><a href="#cb39-308" aria-hidden="true" tabindex="-1"></a><span class="fu">wc</span> <span class="at">-l</span> ../output/05.32-lncRNA-discovery/stringtie_merged.gtf</span>
<span id="cb39-309"><a href="#cb39-309" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> ../output/05.32-lncRNA-discovery/stringtie_merged.gtf</span>
<span id="cb39-310"><a href="#cb39-310" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-311"><a href="#cb39-311" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-312"><a href="#cb39-312" aria-hidden="true" tabindex="-1"></a><span class="in">    ## 726445 ../output/05.32-lncRNA-discovery/stringtie_merged.gtf</span></span>
<span id="cb39-313"><a href="#cb39-313" aria-hidden="true" tabindex="-1"></a><span class="in">    ## # /home/shared/stringtie-2.2.1.Linux_x86_64/stringtie --merge -G ../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gff -o ../output/05.32-lncRNA-discovery/stringtie_merged.gtf ../output/05.32-lncRNA-discovery/RNA-ACR-140.gtf ../output/05.32-lncRNA-discovery/RNA-ACR-145.gtf ../output/05.32-lncRNA-discovery/RNA-ACR-150.gtf ../output/05.32-lncRNA-discovery/RNA-ACR-173.gtf ../output/05.32-lncRNA-discovery/RNA-ACR-178.gtf</span></span>
<span id="cb39-314"><a href="#cb39-314" aria-hidden="true" tabindex="-1"></a><span class="in">    ## # StringTie version 2.2.1</span></span>
<span id="cb39-315"><a href="#cb39-315" aria-hidden="true" tabindex="-1"></a><span class="in">    ## NC_058066.1  StringTie   transcript  345 1190    1000    .   .   gene_id "MSTRG.1"; transcript_id "MSTRG.1.1"; </span></span>
<span id="cb39-316"><a href="#cb39-316" aria-hidden="true" tabindex="-1"></a><span class="in">    ## NC_058066.1  StringTie   exon    345 1190    1000    .   .   gene_id "MSTRG.1"; transcript_id "MSTRG.1.1"; exon_number "1"; </span></span>
<span id="cb39-317"><a href="#cb39-317" aria-hidden="true" tabindex="-1"></a><span class="in">    ## NC_058066.1  StringTie   transcript  1962    23221   1000    -   .   gene_id "MSTRG.2"; transcript_id "rna-XR_003825913.2"; gene_name "LOC114963522"; ref_gene_id "gene-LOC114963522"; </span></span>
<span id="cb39-318"><a href="#cb39-318" aria-hidden="true" tabindex="-1"></a><span class="in">    ## NC_058066.1  StringTie   exon    1962    2119    1000    -   .   gene_id "MSTRG.2"; transcript_id "rna-XR_003825913.2"; exon_number "1"; gene_name "LOC114963522"; ref_gene_id "gene-LOC114963522"; </span></span>
<span id="cb39-319"><a href="#cb39-319" aria-hidden="true" tabindex="-1"></a><span class="in">    ## NC_058066.1  StringTie   exon    18711   18775   1000    -   .   gene_id "MSTRG.2"; transcript_id "rna-XR_003825913.2"; exon_number "2"; gene_name "LOC114963522"; ref_gene_id "gene-LOC114963522"; </span></span>
<span id="cb39-320"><a href="#cb39-320" aria-hidden="true" tabindex="-1"></a><span class="in">    ## NC_058066.1  StringTie   exon    21001   21093   1000    -   .   gene_id "MSTRG.2"; transcript_id "rna-XR_003825913.2"; exon_number "3"; gene_name "LOC114963522"; ref_gene_id "gene-LOC114963522"; </span></span>
<span id="cb39-321"><a href="#cb39-321" aria-hidden="true" tabindex="-1"></a><span class="in">    ## NC_058066.1  StringTie   exon    23085   23221   1000    -   .   gene_id "MSTRG.2"; transcript_id "rna-XR_003825913.2"; exon_number "4"; gene_name "LOC114963522"; ref_gene_id "gene-LOC114963522"; </span></span>
<span id="cb39-322"><a href="#cb39-322" aria-hidden="true" tabindex="-1"></a><span class="in">    ## NC_058066.1  StringTie   transcript  3278    4416    1000    .   .   gene_id "MSTRG.3"; transcript_id "MSTRG.3.1";</span></span>
<span id="cb39-323"><a href="#cb39-323" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-324"><a href="#cb39-324" aria-hidden="true" tabindex="-1"></a><span class="fu"># 3 GFFcompare</span></span>
<span id="cb39-325"><a href="#cb39-325" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-326"><a href="#cb39-326" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb39-327"><a href="#cb39-327" aria-hidden="true" tabindex="-1"></a><span class="ex">/home/shared/gffcompare-0.12.6.Linux_x86_64/gffcompare</span> <span class="dt">\</span></span>
<span id="cb39-328"><a href="#cb39-328" aria-hidden="true" tabindex="-1"></a>-r ../data/Amil/ncbi_dataset/data/GCF_013753865.1/genomic.gff <span class="dt">\</span></span>
<span id="cb39-329"><a href="#cb39-329" aria-hidden="true" tabindex="-1"></a>-G <span class="dt">\</span></span>
<span id="cb39-330"><a href="#cb39-330" aria-hidden="true" tabindex="-1"></a>-o ../output/05.32-lncRNA-discovery/gffcompare_merged <span class="dt">\</span></span>
<span id="cb39-331"><a href="#cb39-331" aria-hidden="true" tabindex="-1"></a>../output/05.32-lncRNA-discovery/stringtie_merged.gtf</span>
<span id="cb39-332"><a href="#cb39-332" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-333"><a href="#cb39-333" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-334"><a href="#cb39-334" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb39-335"><a href="#cb39-335" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> ../output/05.32-lncRNA-discovery/gffcompare_merged<span class="pp">*</span></span>
<span id="cb39-336"><a href="#cb39-336" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-337"><a href="#cb39-337" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-338"><a href="#cb39-338" aria-hidden="true" tabindex="-1"></a><span class="in">    ## ../output/05.32-lncRNA-discovery/gffcompare_merged.annotated.gtf</span></span>
<span id="cb39-339"><a href="#cb39-339" aria-hidden="true" tabindex="-1"></a><span class="in">    ## ../output/05.32-lncRNA-discovery/gffcompare_merged.loci</span></span>
<span id="cb39-340"><a href="#cb39-340" aria-hidden="true" tabindex="-1"></a><span class="in">    ## ../output/05.32-lncRNA-discovery/gffcompare_merged.stats</span></span>
<span id="cb39-341"><a href="#cb39-341" aria-hidden="true" tabindex="-1"></a><span class="in">    ## ../output/05.32-lncRNA-discovery/gffcompare_merged.stringtie_merged.gtf.refmap</span></span>
<span id="cb39-342"><a href="#cb39-342" aria-hidden="true" tabindex="-1"></a><span class="in">    ## ../output/05.32-lncRNA-discovery/gffcompare_merged.stringtie_merged.gtf.tmap</span></span>
<span id="cb39-343"><a href="#cb39-343" aria-hidden="true" tabindex="-1"></a><span class="in">    ## ../output/05.32-lncRNA-discovery/gffcompare_merged.tracking</span></span>
<span id="cb39-344"><a href="#cb39-344" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-345"><a href="#cb39-345" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb39-346"><a href="#cb39-346" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> <span class="at">-3</span> ../output/05.32-lncRNA-discovery/gffcompare_merged.annotated.gtf</span>
<span id="cb39-347"><a href="#cb39-347" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-348"><a href="#cb39-348" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-349"><a href="#cb39-349" aria-hidden="true" tabindex="-1"></a><span class="in">    ## NC_058066.1  StringTie   transcript  9330    48114   .   +   .   transcript_id "MSTRG.5.2"; gene_id "MSTRG.5"; gene_name "LOC114963508"; xloc "XLOC_000001"; ref_gene_id "gene-LOC114963508"; cmp_ref "rna-XM_029342736.2"; class_code "j"; tss_id "TSS1";</span></span>
<span id="cb39-350"><a href="#cb39-350" aria-hidden="true" tabindex="-1"></a><span class="in">    ## NC_058066.1  StringTie   exon    9330    9651    .   +   .   transcript_id "MSTRG.5.2"; gene_id "MSTRG.5"; exon_number "1";</span></span>
<span id="cb39-351"><a href="#cb39-351" aria-hidden="true" tabindex="-1"></a><span class="in">    ## NC_058066.1  StringTie   exon    12840   12890   .   +   .   transcript_id "MSTRG.5.2"; gene_id "MSTRG.5"; exon_number "2";</span></span>
<span id="cb39-352"><a href="#cb39-352" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-353"><a href="#cb39-353" aria-hidden="true" tabindex="-1"></a><span class="fu"># 4 Filter</span></span>
<span id="cb39-354"><a href="#cb39-354" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-355"><a href="#cb39-355" aria-hidden="true" tabindex="-1"></a>Filters the combined GTF output from GFFcompare to select only the lines</span>
<span id="cb39-356"><a href="#cb39-356" aria-hidden="true" tabindex="-1"></a>representing “transcripts” and excluding lines starting with “<span class="sc">\#</span>” (these</span>
<span id="cb39-357"><a href="#cb39-357" aria-hidden="true" tabindex="-1"></a>are lines in the output format from GFFcompare that don’t contain</span>
<span id="cb39-358"><a href="#cb39-358" aria-hidden="true" tabindex="-1"></a>transcript information). This step further filters for a class code of</span>
<span id="cb39-359"><a href="#cb39-359" aria-hidden="true" tabindex="-1"></a>“u”, and keep only those with lengths greater than 199 bases. The “u’</span>
<span id="cb39-360"><a href="#cb39-360" aria-hidden="true" tabindex="-1"></a>class code from the GFFcompare step is for”unknown” transcripts, that is</span>
<span id="cb39-361"><a href="#cb39-361" aria-hidden="true" tabindex="-1"></a>those that are not previously annotated in our reference GFF as protein</span>
<span id="cb39-362"><a href="#cb39-362" aria-hidden="true" tabindex="-1"></a>coding. The size filter of +200nt is a common filtering step for</span>
<span id="cb39-363"><a href="#cb39-363" aria-hidden="true" tabindex="-1"></a>isolating lncRNAs.</span>
<span id="cb39-364"><a href="#cb39-364" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-365"><a href="#cb39-365" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb39-366"><a href="#cb39-366" aria-hidden="true" tabindex="-1"></a><span class="fu">awk</span> <span class="st">'$3 == "transcript" &amp;&amp; $1 !~ /^#/ {print}'</span> <span class="dt">\</span></span>
<span id="cb39-367"><a href="#cb39-367" aria-hidden="true" tabindex="-1"></a>../output/05.32-lncRNA-discovery/gffcompare_merged.annotated.gtf <span class="kw">|</span> <span class="fu">grep</span> <span class="st">'class_code "u"'</span> <span class="kw">|</span> <span class="fu">awk</span> <span class="st">'$5 - $4 &gt; 199 {print}'</span> <span class="op">&gt;</span> ../output/05.32-lncRNA-discovery/Apul_lncRNA_candidates.gtf</span>
<span id="cb39-368"><a href="#cb39-368" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-369"><a href="#cb39-369" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-370"><a href="#cb39-370" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb39-371"><a href="#cb39-371" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> ../output/05.32-lncRNA-discovery/Apul_lncRNA_candidates.gtf</span>
<span id="cb39-372"><a href="#cb39-372" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-373"><a href="#cb39-373" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-374"><a href="#cb39-374" aria-hidden="true" tabindex="-1"></a><span class="in">    ## NC_058066.1  StringTie   transcript  468619  469943  .   +   .   transcript_id "MSTRG.40.1"; gene_id "MSTRG.40"; xloc "XLOC_000020"; class_code "u"; tss_id "TSS38";</span></span>
<span id="cb39-375"><a href="#cb39-375" aria-hidden="true" tabindex="-1"></a><span class="in">    ## NC_058066.1  StringTie   transcript  1153399 1165634 .   +   .   transcript_id "MSTRG.120.1"; gene_id "MSTRG.120"; xloc "XLOC_000058"; class_code "u"; tss_id "TSS95";</span></span>
<span id="cb39-376"><a href="#cb39-376" aria-hidden="true" tabindex="-1"></a><span class="in">    ## NC_058066.1  StringTie   transcript  1153404 1165634 .   +   .   transcript_id "MSTRG.120.2"; gene_id "MSTRG.120"; xloc "XLOC_000058"; class_code "u"; tss_id "TSS95";</span></span>
<span id="cb39-377"><a href="#cb39-377" aria-hidden="true" tabindex="-1"></a><span class="in">    ## NC_058066.1  StringTie   transcript  1153410 1165634 .   +   .   transcript_id "MSTRG.120.3"; gene_id "MSTRG.120"; xloc "XLOC_000058"; class_code "u"; tss_id "TSS95";</span></span>
<span id="cb39-378"><a href="#cb39-378" aria-hidden="true" tabindex="-1"></a><span class="in">    ## NC_058066.1  StringTie   transcript  1154207 1155609 .   +   .   transcript_id "MSTRG.120.4"; gene_id "MSTRG.120"; xloc "XLOC_000058"; class_code "u"; tss_id "TSS96";</span></span>
<span id="cb39-379"><a href="#cb39-379" aria-hidden="true" tabindex="-1"></a><span class="in">    ## NC_058066.1  StringTie   transcript  1155787 1165634 .   +   .   transcript_id "MSTRG.120.5"; gene_id "MSTRG.120"; xloc "XLOC_000058"; class_code "u"; tss_id "TSS97";</span></span>
<span id="cb39-380"><a href="#cb39-380" aria-hidden="true" tabindex="-1"></a><span class="in">    ## NC_058066.1  StringTie   transcript  1222445 1227226 .   +   .   transcript_id "MSTRG.126.1"; gene_id "MSTRG.126"; xloc "XLOC_000059"; class_code "u"; tss_id "TSS98";</span></span>
<span id="cb39-381"><a href="#cb39-381" aria-hidden="true" tabindex="-1"></a><span class="in">    ## NC_058066.1  StringTie   transcript  1222539 1227226 .   +   .   transcript_id "MSTRG.126.2"; gene_id "MSTRG.126"; xloc "XLOC_000059"; class_code "u"; tss_id "TSS98";</span></span>
<span id="cb39-382"><a href="#cb39-382" aria-hidden="true" tabindex="-1"></a><span class="in">    ## NC_058066.1  StringTie   transcript  1256163 1257382 .   +   .   transcript_id "MSTRG.136.3"; gene_id "MSTRG.136"; xloc "XLOC_000062"; class_code "u"; tss_id "TSS103";</span></span>
<span id="cb39-383"><a href="#cb39-383" aria-hidden="true" tabindex="-1"></a><span class="in">    ## NC_058066.1  StringTie   transcript  1256163 1264306 .   +   .   transcript_id "MSTRG.136.2"; gene_id "MSTRG.136"; xloc "XLOC_000062"; class_code "u"; tss_id "TSS103";</span></span>
<span id="cb39-384"><a href="#cb39-384" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-385"><a href="#cb39-385" aria-hidden="true" tabindex="-1"></a><span class="fu"># 5 Bedtools</span></span>
<span id="cb39-386"><a href="#cb39-386" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-387"><a href="#cb39-387" aria-hidden="true" tabindex="-1"></a>Extracts the sequence data from the <span class="in">`$FASTA`</span> reference file based on the</span>
<span id="cb39-388"><a href="#cb39-388" aria-hidden="true" tabindex="-1"></a>coordinates from the filtered GTF. The resulting sequences represent</span>
<span id="cb39-389"><a href="#cb39-389" aria-hidden="true" tabindex="-1"></a>potential lncRNA candidates.</span>
<span id="cb39-390"><a href="#cb39-390" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-391"><a href="#cb39-391" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb39-392"><a href="#cb39-392" aria-hidden="true" tabindex="-1"></a><span class="ex">/home/shared/bedtools2/bin/fastaFromBed</span> <span class="dt">\</span></span>
<span id="cb39-393"><a href="#cb39-393" aria-hidden="true" tabindex="-1"></a>-fi ../data/GCF_013753865.1_Amil_v2.1_genomic.fna <span class="dt">\</span></span>
<span id="cb39-394"><a href="#cb39-394" aria-hidden="true" tabindex="-1"></a>-bed ../output/05.32-lncRNA-discovery/Apul_lncRNA_candidates.gtf <span class="dt">\</span></span>
<span id="cb39-395"><a href="#cb39-395" aria-hidden="true" tabindex="-1"></a>-fo ../output/05.32-lncRNA-discovery/Apul_lncRNA_candidates.fasta <span class="dt">\</span></span>
<span id="cb39-396"><a href="#cb39-396" aria-hidden="true" tabindex="-1"></a>-name <span class="at">-split</span></span>
<span id="cb39-397"><a href="#cb39-397" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-398"><a href="#cb39-398" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-399"><a href="#cb39-399" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb39-400"><a href="#cb39-400" aria-hidden="true" tabindex="-1"></a><span class="fu">fgrep</span> <span class="at">-c</span> <span class="st">"&gt;"</span> ../output/05.32-lncRNA-discovery/Apul_lncRNA_candidates.fasta</span>
<span id="cb39-401"><a href="#cb39-401" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> ../output/05.32-lncRNA-discovery/Apul_lncRNA_candidates.fasta</span>
<span id="cb39-402"><a href="#cb39-402" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-403"><a href="#cb39-403" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-404"><a href="#cb39-404" aria-hidden="true" tabindex="-1"></a><span class="fu"># 6 CPC2</span></span>
<span id="cb39-405"><a href="#cb39-405" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-406"><a href="#cb39-406" aria-hidden="true" tabindex="-1"></a>Initializes a conda environment (Anaconda) and runs CPC2, a software to</span>
<span id="cb39-407"><a href="#cb39-407" aria-hidden="true" tabindex="-1"></a>predict whether a transcript is coding or non-coding. The results are</span>
<span id="cb39-408"><a href="#cb39-408" aria-hidden="true" tabindex="-1"></a>saved to the \$OUTPUT_DIR. CPC2 uses ORF (Open Reading Frame) Analysis,</span>
<span id="cb39-409"><a href="#cb39-409" aria-hidden="true" tabindex="-1"></a>Isometric Feature Mapping (Isomap), Sequence Homology, RNA Sequence</span>
<span id="cb39-410"><a href="#cb39-410" aria-hidden="true" tabindex="-1"></a>Features, and Quality of Translation to assess coding potential and flag</span>
<span id="cb39-411"><a href="#cb39-411" aria-hidden="true" tabindex="-1"></a>any transcripts we would want to exclude using the FASTA generated in</span>
<span id="cb39-412"><a href="#cb39-412" aria-hidden="true" tabindex="-1"></a>the previous step.</span>
<span id="cb39-413"><a href="#cb39-413" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-414"><a href="#cb39-414" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb39-415"><a href="#cb39-415" aria-hidden="true" tabindex="-1"></a><span class="bu">eval</span> <span class="st">"</span><span class="va">$(</span><span class="ex">/opt/anaconda/anaconda3/bin/conda</span> shell.bash hook<span class="va">)</span><span class="st">"</span></span>
<span id="cb39-416"><a href="#cb39-416" aria-hidden="true" tabindex="-1"></a><span class="ex">python</span> /home/shared/CPC2_standalone-1.0.1/bin/CPC2.py <span class="dt">\</span></span>
<span id="cb39-417"><a href="#cb39-417" aria-hidden="true" tabindex="-1"></a>-i ../output/05.32-lncRNA-discovery/Apul_lncRNA_candidates.fasta <span class="dt">\</span></span>
<span id="cb39-418"><a href="#cb39-418" aria-hidden="true" tabindex="-1"></a>-o ../output/05.32-lncRNA-discovery/Apul_CPC2</span>
<span id="cb39-419"><a href="#cb39-419" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-420"><a href="#cb39-420" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-421"><a href="#cb39-421" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb39-422"><a href="#cb39-422" aria-hidden="true" tabindex="-1"></a><span class="fu">wc</span> <span class="at">-l</span> head ../output/05.32-lncRNA-discovery/Apul_CPC2.txt</span>
<span id="cb39-423"><a href="#cb39-423" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> ../output/05.32-lncRNA-discovery/Apul_CPC2.txt</span>
<span id="cb39-424"><a href="#cb39-424" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-425"><a href="#cb39-425" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-426"><a href="#cb39-426" aria-hidden="true" tabindex="-1"></a><span class="in">    ## wc: head: No such file or directory</span></span>
<span id="cb39-427"><a href="#cb39-427" aria-hidden="true" tabindex="-1"></a><span class="in">    ##   16695 ../output/05.32-lncRNA-discovery/Apul_CPC2.txt</span></span>
<span id="cb39-428"><a href="#cb39-428" aria-hidden="true" tabindex="-1"></a><span class="in">    ##   16695 total</span></span>
<span id="cb39-429"><a href="#cb39-429" aria-hidden="true" tabindex="-1"></a><span class="in">    ## #ID  transcript_length   peptide_length  Fickett_score   pI  ORF_integrity   coding_probability  label</span></span>
<span id="cb39-430"><a href="#cb39-430" aria-hidden="true" tabindex="-1"></a><span class="in">    ## transcript::NC_058066.1:468618-469943    1325    73  0.3891  11.48389835357666   1   0.0882804   noncoding</span></span>
<span id="cb39-431"><a href="#cb39-431" aria-hidden="true" tabindex="-1"></a><span class="in">    ## transcript::NC_058066.1:1153398-1165634  12236   134 0.3118  9.09901943206787    1   0.295457    noncoding</span></span>
<span id="cb39-432"><a href="#cb39-432" aria-hidden="true" tabindex="-1"></a><span class="in">    ## transcript::NC_058066.1:1153403-1165634  12231   134 0.3118  9.09901943206787    1   0.295457    noncoding</span></span>
<span id="cb39-433"><a href="#cb39-433" aria-hidden="true" tabindex="-1"></a><span class="in">    ## transcript::NC_058066.1:1153409-1165634  12225   134 0.3118  9.09901943206787    1   0.295457    noncoding</span></span>
<span id="cb39-434"><a href="#cb39-434" aria-hidden="true" tabindex="-1"></a><span class="in">    ## transcript::NC_058066.1:1154206-1155609  1403    66  0.37373 8.802658271789554   1   0.0495107   noncoding</span></span>
<span id="cb39-435"><a href="#cb39-435" aria-hidden="true" tabindex="-1"></a><span class="in">    ## transcript::NC_058066.1:1155786-1165634  9848    134 0.29479 9.09901943206787    1   0.241898    noncoding</span></span>
<span id="cb39-436"><a href="#cb39-436" aria-hidden="true" tabindex="-1"></a><span class="in">    ## transcript::NC_058066.1:1222444-1227226  4782    62  0.32077 9.30048313140869    1   0.0302275   noncoding</span></span>
<span id="cb39-437"><a href="#cb39-437" aria-hidden="true" tabindex="-1"></a><span class="in">    ## transcript::NC_058066.1:1222538-1227226  4688    62  0.32077 9.30048313140869    1   0.0302275   noncoding</span></span>
<span id="cb39-438"><a href="#cb39-438" aria-hidden="true" tabindex="-1"></a><span class="in">    ## transcript::NC_058066.1:1256162-1257382  1220    41  0.34867000000000004 9.098761558532715   1   0.0140974   noncoding</span></span>
<span id="cb39-439"><a href="#cb39-439" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-440"><a href="#cb39-440" aria-hidden="true" tabindex="-1"></a><span class="sc">\#</span>Filter Filters the CPC2 results to get only noncoding transcripts</span>
<span id="cb39-441"><a href="#cb39-441" aria-hidden="true" tabindex="-1"></a>(using the class “noncoding” from the CPC2 results) and extracts their</span>
<span id="cb39-442"><a href="#cb39-442" aria-hidden="true" tabindex="-1"></a>IDs and matches these IDs with the sequences from the previous step to</span>
<span id="cb39-443"><a href="#cb39-443" aria-hidden="true" tabindex="-1"></a>generate a GTF of long noncoding transcripts.</span>
<span id="cb39-444"><a href="#cb39-444" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-445"><a href="#cb39-445" aria-hidden="true" tabindex="-1"></a>Matches these IDs with the sequences from the previous step to generate</span>
<span id="cb39-446"><a href="#cb39-446" aria-hidden="true" tabindex="-1"></a>a GTF of noncoding transcripts.</span>
<span id="cb39-447"><a href="#cb39-447" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-448"><a href="#cb39-448" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb39-449"><a href="#cb39-449" aria-hidden="true" tabindex="-1"></a><span class="fu">awk</span> <span class="st">'$8 == "noncoding" {print $1}'</span> ../output/05.32-lncRNA-discovery/Apul_CPC2.txt <span class="op">&gt;</span> ../output/05.32-lncRNA-discovery/Apul_noncoding_transcripts_ids.txt</span>
<span id="cb39-450"><a href="#cb39-450" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-451"><a href="#cb39-451" aria-hidden="true" tabindex="-1"></a><span class="fu">grep</span> <span class="at">-Fwf</span> ../output/05.32-lncRNA-discovery/Apul_noncoding_transcripts_ids.txt ../output/05.32-lncRNA-discovery/Apul_lncRNA_candidates.fasta <span class="op">&gt;</span> ../output/05.32-lncRNA-discovery/Apul_lncRNAs.gtf</span>
<span id="cb39-452"><a href="#cb39-452" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-453"><a href="#cb39-453" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-454"><a href="#cb39-454" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb39-455"><a href="#cb39-455" aria-hidden="true" tabindex="-1"></a><span class="fu">wc</span> <span class="at">-l</span> ../output/05.32-lncRNA-discovery/Apul_lncRNAs.gtf</span>
<span id="cb39-456"><a href="#cb39-456" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> ../output/05.32-lncRNA-discovery/Apul_lncRNAs.gtf</span>
<span id="cb39-457"><a href="#cb39-457" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-458"><a href="#cb39-458" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-459"><a href="#cb39-459" aria-hidden="true" tabindex="-1"></a><span class="in">    ## 16122 ../output/05.32-lncRNA-discovery/Apul_lncRNAs.gtf</span></span>
<span id="cb39-460"><a href="#cb39-460" aria-hidden="true" tabindex="-1"></a><span class="in">    ## &gt;transcript::NC_058066.1:468618-469943</span></span>
<span id="cb39-461"><a href="#cb39-461" aria-hidden="true" tabindex="-1"></a><span class="in">    ## &gt;transcript::NC_058066.1:1153398-1165634</span></span>
<span id="cb39-462"><a href="#cb39-462" aria-hidden="true" tabindex="-1"></a><span class="in">    ## &gt;transcript::NC_058066.1:1153403-1165634</span></span>
<span id="cb39-463"><a href="#cb39-463" aria-hidden="true" tabindex="-1"></a><span class="in">    ## &gt;transcript::NC_058066.1:1153409-1165634</span></span>
<span id="cb39-464"><a href="#cb39-464" aria-hidden="true" tabindex="-1"></a><span class="in">    ## &gt;transcript::NC_058066.1:1154206-1155609</span></span>
<span id="cb39-465"><a href="#cb39-465" aria-hidden="true" tabindex="-1"></a><span class="in">    ## &gt;transcript::NC_058066.1:1155786-1165634</span></span>
<span id="cb39-466"><a href="#cb39-466" aria-hidden="true" tabindex="-1"></a><span class="in">    ## &gt;transcript::NC_058066.1:1222444-1227226</span></span>
<span id="cb39-467"><a href="#cb39-467" aria-hidden="true" tabindex="-1"></a><span class="in">    ## &gt;transcript::NC_058066.1:1222538-1227226</span></span>
<span id="cb39-468"><a href="#cb39-468" aria-hidden="true" tabindex="-1"></a><span class="in">    ## &gt;transcript::NC_058066.1:1256162-1257382</span></span>
<span id="cb39-469"><a href="#cb39-469" aria-hidden="true" tabindex="-1"></a><span class="in">    ## &gt;transcript::NC_058066.1:1256162-1264306</span></span>
<span id="cb39-470"><a href="#cb39-470" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-471"><a href="#cb39-471" aria-hidden="true" tabindex="-1"></a><span class="fu"># 7 De-duplicate GTF</span></span>
<span id="cb39-472"><a href="#cb39-472" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-473"><a href="#cb39-473" aria-hidden="true" tabindex="-1"></a>Removes duplicate entries from the GTF file. This results in a list of</span>
<span id="cb39-474"><a href="#cb39-474" aria-hidden="true" tabindex="-1"></a>lncRNAs with transcript IDs that show scaffold and base range. awk</span>
<span id="cb39-475"><a href="#cb39-475" aria-hidden="true" tabindex="-1"></a>‘!seen$$\$0$$++’ \$OUTPUT_DIR/apul_merged_final_lncRNAs.gtf <span class="sc">\&gt;</span></span>
<span id="cb39-476"><a href="#cb39-476" aria-hidden="true" tabindex="-1"></a>\$OUTPUT_DIR/apul_deduplicated_final_lncRNAs.gtf</span>
<span id="cb39-477"><a href="#cb39-477" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-478"><a href="#cb39-478" aria-hidden="true" tabindex="-1"></a><span class="fu"># 8 Reformat to BED</span></span>
<span id="cb39-479"><a href="#cb39-479" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-480"><a href="#cb39-480" aria-hidden="true" tabindex="-1"></a>Converts the GTF format to BED (Browser Extensible Data) format, a</span>
<span id="cb39-481"><a href="#cb39-481" aria-hidden="true" tabindex="-1"></a>commonly used format for representing genomic intervals. This allows us</span>
<span id="cb39-482"><a href="#cb39-482" aria-hidden="true" tabindex="-1"></a>to use bedtools to generate a FASTA of lncRNAs. awk -F”:\|-” ‘{print \$3</span>
<span id="cb39-483"><a href="#cb39-483" aria-hidden="true" tabindex="-1"></a>“ \$4” \$5}’ \$OUTPUT_DIR/apul_deduplicated_final_lncRNAs.gtf <span class="sc">\&gt;</span></span>
<span id="cb39-484"><a href="#cb39-484" aria-hidden="true" tabindex="-1"></a>\$OUTPUT_DIR/apul_deduplicated_final_lncRNAs.bed</span>
<span id="cb39-485"><a href="#cb39-485" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-486"><a href="#cb39-486" aria-hidden="true" tabindex="-1"></a><span class="fu"># 9 BEDTools</span></span>
<span id="cb39-487"><a href="#cb39-487" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-488"><a href="#cb39-488" aria-hidden="true" tabindex="-1"></a>This extracts sequences from the reference FASTA based on the BED file’s</span>
<span id="cb39-489"><a href="#cb39-489" aria-hidden="true" tabindex="-1"></a>coordinates. The resulting sequences represent the final set of</span>
<span id="cb39-490"><a href="#cb39-490" aria-hidden="true" tabindex="-1"></a>non-redundant lncRNAs. \$BEDTOOLS_PATH/bedtools  </span>
<span id="cb39-491"><a href="#cb39-491" aria-hidden="true" tabindex="-1"></a>getfasta -fi \$FASTA -bed</span>
<span id="cb39-492"><a href="#cb39-492" aria-hidden="true" tabindex="-1"></a>\$OUTPUT_DIR/apul_deduplicated_final_lncRNAs.bed -fo</span>
<span id="cb39-493"><a href="#cb39-493" aria-hidden="true" tabindex="-1"></a>\$OUTPUT_DIR/apul_bedtools_lncRNAs.fasta -name <span class="sc">\`\`\`</span></span>
<span id="cb39-494"><a href="#cb39-494" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-495"><a href="#cb39-495" aria-hidden="true" tabindex="-1"></a>The results of this pipleine provide the user with GTF files containing</span>
<span id="cb39-496"><a href="#cb39-496" aria-hidden="true" tabindex="-1"></a>transcript IDs (scaffold and sequence range), BED files displaying the</span>
<span id="cb39-497"><a href="#cb39-497" aria-hidden="true" tabindex="-1"></a>same transcript IDs in BED format for use with BEDtools, and FASTA files</span>
<span id="cb39-498"><a href="#cb39-498" aria-hidden="true" tabindex="-1"></a>containing the sequence of identified lncRNAs.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>