---
title: "DNA Methylation Synthesis"
subtitle: "Fine Tuning Our Understanding of the Epigenetic Mechanism"
listing:
  contents: posts
  sort: "date desc"
  type: default
  categories: true
  sort-ui: false
  filter-ui: false
page-layout: full
title-block-banner: true
image: "epigeneticPath.png"

---
Quite candidly, I have been conflating types of DNA methylation and this week’s readings helped me get to the bottom of why I was struggling to grasp the conceptual understanding of different types of DNA methylation. But first, let's back up and get on the same page. DNA methylation is an epigenetic mechanism that allows an organism to alter its genetic response to the environment without making permanent changes to its DNA.

DNA methylation is one of the ways we understand the role of epigenetic response in organisms, but there are a few more; histone modification and non-coding RNA modification are the other mechanisms. While much work has been done to explain this temporary response to  an environmental stressor, there is still plenty of work to be done with DNA methylation leading the way in research and understanding. With that said, DNA methylation manifests in either transcriptional regulation or in gene silencing and the [Roberts & Gavery 2012](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3249382/) addresses both. While this paper was published in 2012, the role of DNA methylation as an epigenetic mechanism hasn’t changed.

![epigeneticPath]([epigenticPath.png](https://github.com/sr320/tumbling-oysters/blob/main/posts/ChrisPosts/epigenticPath.png?raw=true))
Image Caption: This graphic is from the [Pham et al. 2023](https://doi.org/10.1016/j.envpol.2023.121737) review paper on epigenetic tool usage in elucidating the impact of contaminants in the aquatic environment.

## Roberts & Gavery 2012
* Main Claim: Evidence that the absence of DNA methylation could contribute to phenotypic plasticity by increasing the number of transcriptional opportunities.
** Limited methylation might passively facilitate specific transcriptional opportunities
* Main Claim: The absence of germline methylation facilitates random variation that contributes to phenotypic plasticity and thus could increase adaptive potential.
-- Genes predicted to be methylated at the germline have less genetic diversity compared to genes lacking germline methylation.
- Main Claim: Methylation may not be functioning as an “on/off” switch but rather function as a fine-tuning of transcriptional control of conserved genes.

## Pham et al. 2023
- Methods based review paper that discusses how each method can be used to prove how epigenetic analytical approaches allow measuring the long term environmental responses of organisms in aquatic ecosystems subjected to harmful effects of climate change and water pollution
- Reviewed 78 papers from 2016 - 2021 that applied these methods to investigate the responses of aquatic organisms to environmental changes and pollution and then categorized them
- Main Claim: From most to least used, DNA methylation, histone modification, and non-coding RNAs are still the key methods to understanding epigenetic response
-- Next generation sequencing (NGS) has gained significant popularity and subsequently more established protocols than in 2012
- Main Claim: DNA methylation analysis falls into three main categories
-- Enzyme restriction digestion-based 
-- Enzyme restricted methyl-specific immunoprecipitation
-- Bisulfite (in)dependent sequencing strategies
- R&G Supporting Claim: MSAP & MeDIP-seq methods of exploring DNA methylation patterns and heritability
- Methylation Sensitive Amplification Polymorphism
-- Genomic  DNA is first digested  with a methylation-sensitive restriction enzyme.  The DNA fragments are then ligated to adaptors and selectively amplified by labeled  primers. PCR products  from different individuals are  compared  to detect  targeted  epi-polymorphic loci 
-- “In case of other Mollusca species, M. galloprovincialis and X. securis, under  invasion  and salinity  stresses  as well as habitat  degradation,  higher  demethylation was recognized in more stressed populations, suggesting  the involvement of environmental epigenetics in the plasticity process
- Methylated  DNA  immunoprecipitation
-- Utilizes anti-5mC antibodies for affinity-based enrichment of methylated DNA regions
-- “A study using  MeDIP-seq revealed a potential link between transgenerational  phenotypic  plasticity and DNA methylation in S. purpuratus under stress conditions

## Research Application
  DNA methylation is an important variable in exploring Mytilus’ response to contaminants as the prevalence of larger and more frequent weather events pushes them to their energetic threshold of survival. The first exploration would be to quantify the percent cover of methylation across genes or physiological processes of interest and follow-up with lab experiments to determine adaptability to single- stressor and multiple- stressor environments. 
	I am particularly interested in evaluating Mytilus spp. physiological response to PPCPs in coastal marine waterways through evaluating epigenetic response, heritability of response, and potential trade-offs in response to a challenging environment. An example of relevance to my work would be to quantify percent methylation coverage via BS-sequencing and follow-up with single contaminant exposure trials in mature mussels to tease out the trade-offs required to survive in a multiple-stressor environment. A control group would be exposed to standard environmental conditions prior to contaminant exposure. Treatment groups could include acute thermal stress, chronic thermal stress with a finite end, and long-term thermal stress exposures at temporal scales that mimic high temperature fluctuations related to seasonality. The next step would be to expose each treatment group to the same contaminant to determine the organism’s success in offloading the contaminant. Sampling of tissues would take place across a temporal scale that would yield the most complete picture of exposure response. 

## Conclusion
- Germline methylation and heritability are still big questions 
- Pham et al. takes the findings of Roberts & Gavery as standing truths
- Accessible methods and molecular corroboration are the focus of the epigenetic conversation now
-- Current  studies  of  epigenetics  mainly  focus  on  chromatin  remodeling  through  interplay  of  DNA  methylation,  histone  posttranslational modifications, and noncoding RNAs (ncRNAs)
- Reference libraries are useful but not necessary with new techniques aimed at using more non-model species to test methods
- Next steps/ Third generation 
-- DNA methylation analysis  has come to a third stage –  sequencing-based methylation profiling, including nanopore sequencing  and  single-molecule  real-time  sequencing  technologies,  which  have  been created and already adapted to DNA methylation analysis by direct  (bisulfite  independent)  detection  of  base  modifications
