---
title: "Energy staging"
description: "Can we predict where they are on the curve?"
categories: [e5, methylation]
#citation: 
date: 04-10-2025
image: http://gannet.fish.washington.edu/seashell/snaps/Monosnap_23_GO_Annoations_2025-04-10_12-36-18.png # finding a good image

author:
  - name: Steven Roberts
    url: 
    orcid: 0000-0001-8302-1138
    affiliation: Professor, UW - School of Aquatic and Fishery Sciences
    affiliation-url: https://robertslab.info
  #url:  # self-defined
draft: false # setting this to `true` will prevent your post from appearing on your listing page until you're ready!
format:
  html:
    code-fold: FALSE
    code-tools: true
    code-copy: true
    highlight-style: github
    code-overflow: wrap
---

```{r esetup, include=FALSE}
knitr::opts_chunk$set(
  echo = TRUE,         # Display code chunks
  eval = FALSE,        # Avoid automatic execution
  warning = FALSE,     # Hide warnings
  message = FALSE,     # Hide messages
  fig.width = 6,       # Set default plot width
  fig.height = 4,      # Set default plot height
  fig.align = "center", # Center align plots
  comment = ""         # Prevents '##' in output
)
```

Tackled two primary tasks

Identified proteins involved in key energy utilization pathways and identified proteins corresponding to epigenetic machinery.

# For the energy utilization pathways, 
we focused on the following:

```
- Glycolysis GO:0006096
- Gluconeogenesis GO:0006094
- Lipolysis/lipid catabolism GO:0016042
- Fatty acid beta oxidation GO:0006635
- Starvation GO:0042594
- Lipid biosynthesis GO:0008610
- Protein catabolic process GO:0030163
```

and ran something on order of.

# Variables

```{r}
# Global R options
knitr::opts_chunk$set(echo = TRUE)

# Define key paths and tool directories
 
OUT_DIR <- "../output/23-Apul-energy-GO/"
evalue <- "1E-20"
fasta <- "../data/Apulchra-genome.pep.faa"

# Export these as environment variables for bash chunks.
Sys.setenv(
  OUT_DIR = OUT_DIR,
  evalue = evalue,
  fasta =fasta
)
```


# Glycolysis GO:0006096

```{bash}
GO="0006096"

curl -H "Accept: text/plain" "https://rest.uniprot.org/uniprotkb/stream?format=fasta&query=%28%28go%3A"${GO}"%29%29+AND+%28reviewed%3Atrue%29" -o "${OUT_DIR}"SwissProt-GO:"${GO}".fa

head "${OUT_DIR}"SwissProt-GO:"${GO}".fa

echo "Number of Proteins"
grep -c ">" "${OUT_DIR}"SwissProt-GO:"${GO}".fa

/home/shared/ncbi-blast-2.15.0+/bin/makeblastdb \
-in "${OUT_DIR}"SwissProt-GO:"${GO}".fa \
-dbtype prot \
-out "${OUT_DIR}"SwissProt-GO:"${GO}"


/home/shared/ncbi-blast-2.15.0+/bin/blastp \
-query $fasta \
-db "${OUT_DIR}"SwissProt-GO:"${GO}" \
-out "${OUT_DIR}"Apul_blastp-GO:"${GO}"_out.tab \
-evalue "${evalue}" \
-num_threads 42 \
-max_target_seqs 1 \
-max_hsps 1 \
-outfmt 6 \
2> "${OUT_DIR}"blast_warnings"${GO}".txt

head "${OUT_DIR}"Apul_blastp-GO:"${GO}"_out.tab

echo "Number of hits"

wc -l "${OUT_DIR}"Apul_blastp-GO:"${GO}"_out.tab
```

# For Machinery...

more along the lines of 

Apul 

https://raw.githubusercontent.com/urol-e5/deep-dive-expression/main/D-Apul/data/Apulchra-genome.pep.faa



```{bash}
cd ../data
curl -o Apulchra-genome.pep.faa https://raw.githubusercontent.com/urol-e5/deep-dive-expression/main/D-Apul/data/Apulchra-genome.pep.faa

```


```{bash}
head ../data/Apulchra-genome.pep.faa
```


```{bash}

/home/shared/ncbi-blast-2.15.0+/bin/makeblastdb \
-in ../data/Apulchra-genome.pep.faa \
-dbtype prot \
-out ../output/25-Apul-epimods-blast/Apul-proteins
```

```{bash}
head ../data/Machinery.fasta
```


```{bash}
fasta="../data/Machinery.fasta"

/home/shared/ncbi-blast-2.15.0+/bin/blastp \
-query $fasta \
-db ../output/25-Apul-epimods-blast/Apul-proteins \
-out ../output/25-Apul-epimods-blast/Mach-blastp-Apul_out.tab \
-evalue 1E-05 \
-num_threads 48 \
-max_target_seqs 1 \
-max_hsps 1 \
-outfmt 6
```

```{r, engine='bash'}       
wc -l ../output/25-Apul-epimods-blast/Mach-blastp-Apul_out.tab
```

```{r, engine='bash'}        
head ../output/25-Apul-epimods-blast/Mach-blastp-Apul_out.tab
```


