---
layout: post
title: August Bits
date: ‘2022-08-01’
categories:
tags:
---

This is a running daily of all the stuff done in August.
Or just some thoughts


**August 17**
Now that we have some bare data in ceabigr how do we go forward beyond the simple correlation with DEG and DMG.

Lets take each of 26 libraries take as many indices as possible. To do this lets determine a mean for each metric then determine the variance / outlier. Then we look for correlation of metrics that deviate simultaneously in a single library. If we see this correlation across several libraries then we maybe have something.  

Within a given library - here are a list of things we need to obtain mean values (across all libraries)

This is really a list of genome features with some form of quantification

-[ ] gene expression     
-[ ] transcript expression    
-[ ] isoform count     
-[ ] Something that describes isoform expression preponderance. (see issue)    
-[ ] lncRNA expression    
-[ ] copy number variation  
-[ ]     


it seems like we could get a 1:1 correlation but there is also a need look for multiple correlations, (eg if x = up AND y = down THEN z = down)


From methylation perspective we can look at loci or gene methylation, using the same framework of identifying outliers of mean.

We could also look at methylation level of other features.
-[ ] sliding windows    
-[ ] exons    
-[ ] introns    
-[ ] te    
