<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.40">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Steven Roberts">
<meta name="dcterms.date" content="2025-06-21">
<meta name="description" content="that’s hit">

<title>The lncRNA repertoire of Ptuh – Tumbling Oysters</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-86daaaaad7353f9cc0c554efc1dd6d94.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-dark-f8dc6eab18fde03278982b0b35885446.css" rel="stylesheet" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap-0b0971fe3c507452e01567f62a81532a.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../../site_libs/bootstrap/bootstrap-dark-e76cdc2a28bf56c993f4e6ffb14f81c0.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="nav-fixed fullcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top quarto-banner">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Tumbling Oysters</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/sr320"> <i class="bi bi-github" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../notes.html"> <i class="bi bi-newspaper" role="img">
</i> 
<span class="menu-text">Daily Notes</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="../../notes.txt"> <i class="bi bi-📝 Notes (TXT)" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <div class="quarto-title-block"><div><h1 class="title">The lncRNA repertoire of Ptuh</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
                  <div>
        <div class="description">
          that’s hit
        </div>
      </div>
                          <div class="quarto-categories">
                <div class="quarto-category">lncRNA</div>
                <div class="quarto-category">e5</div>
              </div>
                  </div>
  </div>
    
  <div class="quarto-title-meta-author">
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-heading">Affiliation</div>
    
      <div class="quarto-title-meta-contents">
      <p class="author">Steven Roberts <a href="https://orcid.org/0000-0001-8302-1138" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
    </div>
    <div class="quarto-title-meta-contents">
          <p class="affiliation">
              <a href="https://robertslab.info">
              Professor, UW - School of Aquatic and Fishery Sciences
              </a>
            </p>
        </div>
    </div>

  <div class="quarto-title-meta">

        
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">June 21, 2025</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">





<section id="tldr" class="level1">
<h1>TLDR</h1>
<p>fasta - https://raw.githubusercontent.com/urol-e5/deep-dive-expression/refs/heads/main/F-Ptuh/output/17-Ptuh-lncRNA/Ptuh-lncRNA.fasta</p>
<p>gtf - https://raw.githubusercontent.com/urol-e5/deep-dive-expression/refs/heads/main/F-Ptuh/output/17-Ptuh-lncRNA/Ptuh-lncRNA.gtf</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># URLs</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>bed_url <span class="ot">&lt;-</span> <span class="st">"https://raw.githubusercontent.com/urol-e5/deep-dive-expression/refs/heads/main/F-Ptuh/output/17-Ptuh-lncRNA/lncRNA.bed"</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>gtf_url <span class="ot">&lt;-</span> <span class="st">"https://raw.githubusercontent.com/urol-e5/deep-dive-expression/refs/heads/main/F-Ptuh/output/17-Ptuh-lncRNA/lncRNA.gtf"</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>fasta_url <span class="ot">&lt;-</span> <span class="st">"https://raw.githubusercontent.com/urol-e5/deep-dive-expression/refs/heads/main/F-Ptuh/output/17-Ptuh-lncRNA/lncRNA.fasta"</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="do">### BED file</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>bed <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(bed_url, <span class="at">col_names =</span> <span class="fu">c</span>(<span class="st">"chrom"</span>, <span class="st">"start"</span>, <span class="st">"end"</span>))</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"BED file head:</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>BED file head:</code></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">head</span>(bed))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 3
  chrom                                   start    end
  &lt;chr&gt;                                   &lt;dbl&gt;  &lt;dbl&gt;
1 Pocillopora_meandrina_HIv1___Sc0000000 130006 130942
2 Pocillopora_meandrina_HIv1___Sc0000000 164394 165221
3 Pocillopora_meandrina_HIv1___Sc0000000 164596 165221
4 Pocillopora_meandrina_HIv1___Sc0000000 168916 182502
5 Pocillopora_meandrina_HIv1___Sc0000000 245808 248612
6 Pocillopora_meandrina_HIv1___Sc0000000 282742 283165</code></pre>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Basic stats for BED</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>bed<span class="sc">$</span>length <span class="ot">&lt;-</span> bed<span class="sc">$</span>end <span class="sc">-</span> bed<span class="sc">$</span>start <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">BED file stats:</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
BED file stats:</code></pre>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Number of features:"</span>, <span class="fu">nrow</span>(bed), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Number of features: 16153 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Mean length:"</span>, <span class="fu">mean</span>(bed<span class="sc">$</span>length), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Mean length: 3125.615 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Median length:"</span>, <span class="fu">median</span>(bed<span class="sc">$</span>length), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Median length: 702 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="do">### GTF file</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>gtf <span class="ot">&lt;-</span> <span class="fu">import</span>(gtf_url, <span class="at">format =</span> <span class="st">"gtf"</span>)</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">GTF file head:</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
GTF file head:</code></pre>
</div>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">head</span>(gtf))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 6 ranges and 5 metadata columns:
                    seqnames        ranges strand |   source     type     score
                       &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; | &lt;factor&gt; &lt;factor&gt; &lt;numeric&gt;
  [1] Pocillopora_meandrin.. 130006-130942      + |       NA   lncRNA        NA
  [2] Pocillopora_meandrin.. 164394-165221      + |       NA   lncRNA        NA
  [3] Pocillopora_meandrin.. 164596-165221      + |       NA   lncRNA        NA
  [4] Pocillopora_meandrin.. 168916-182502      + |       NA   lncRNA        NA
  [5] Pocillopora_meandrin.. 245808-248612      + |       NA   lncRNA        NA
  [6] Pocillopora_meandrin.. 282742-283165      + |       NA   lncRNA        NA
          phase     gene_id
      &lt;integer&gt; &lt;character&gt;
  [1]      &lt;NA&gt;  lncRNA_001
  [2]      &lt;NA&gt;  lncRNA_002
  [3]      &lt;NA&gt;  lncRNA_003
  [4]      &lt;NA&gt;  lncRNA_004
  [5]      &lt;NA&gt;  lncRNA_005
  [6]      &lt;NA&gt;  lncRNA_006
  -------
  seqinfo: 175 sequences from an unspecified genome; no seqlengths</code></pre>
</div>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Basic stats for GTF</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">GTF file stats:</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
GTF file stats:</code></pre>
</div>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Number of entries:"</span>, <span class="fu">length</span>(gtf), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Number of entries: 16153 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Unique feature types:"</span>, <span class="fu">paste</span>(<span class="fu">unique</span>(gtf<span class="sc">$</span>type), <span class="at">collapse =</span> <span class="st">", "</span>), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Unique feature types: lncRNA </code></pre>
</div>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="do">### FASTA file</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>fasta <span class="ot">&lt;-</span> <span class="fu">readDNAStringSet</span>(fasta_url)</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">FASTA file head:</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
FASTA file head:</code></pre>
</div>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">head</span>(<span class="fu">names</span>(fasta)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "transcript::Pocillopora_meandrina_HIv1___Sc0000000:130007-130942"
[2] "transcript::Pocillopora_meandrina_HIv1___Sc0000000:164395-165221"
[3] "transcript::Pocillopora_meandrina_HIv1___Sc0000000:164597-165221"
[4] "transcript::Pocillopora_meandrina_HIv1___Sc0000000:168917-182502"
[5] "transcript::Pocillopora_meandrina_HIv1___Sc0000000:245809-248612"
[6] "transcript::Pocillopora_meandrina_HIv1___Sc0000000:282743-283165"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">head</span>(fasta[[<span class="dv">1</span>]]))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>6-letter DNAString object
seq: GCCTTG</code></pre>
</div>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Basic stats for FASTA</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>seq_lengths <span class="ot">&lt;-</span> <span class="fu">width</span>(fasta)</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">FASTA file stats:</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
FASTA file stats:</code></pre>
</div>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Number of sequences:"</span>, <span class="fu">length</span>(fasta), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Number of sequences: 16153 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Mean sequence length:"</span>, <span class="fu">mean</span>(seq_lengths), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Mean sequence length: 3123.615 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Median sequence length:"</span>, <span class="fu">median</span>(seq_lengths), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Median sequence length: 700 </code></pre>
</div>
</div>
</section>
<section id="renaming-fasta" class="level1">
<h1>Renaming fasta</h1>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="co"># File paths</span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>fasta_url <span class="ot">&lt;-</span> <span class="st">"https://raw.githubusercontent.com/urol-e5/deep-dive-expression/refs/heads/main/F-Ptuh/output/17-Ptuh-lncRNA/lncRNA.fasta"</span></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>gtf_url <span class="ot">&lt;-</span> <span class="st">"https://raw.githubusercontent.com/urol-e5/deep-dive-expression/refs/heads/main/F-Ptuh/output/17-Ptuh-lncRNA/lncRNA.gtf"</span></span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Read in FASTA and GTF</span></span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a>fasta <span class="ot">&lt;-</span> <span class="fu">readDNAStringSet</span>(fasta_url)</span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a>gtf <span class="ot">&lt;-</span> <span class="fu">import</span>(gtf_url, <span class="at">format =</span> <span class="st">"gtf"</span>)</span>
<span id="cb37-8"><a href="#cb37-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-9"><a href="#cb37-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract gene_id from GTF</span></span>
<span id="cb37-10"><a href="#cb37-10" aria-hidden="true" tabindex="-1"></a>gene_ids <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">mcols</span>(gtf)<span class="sc">$</span>gene_id, <span class="cf">function</span>(x) <span class="fu">gsub</span>(<span class="st">'"'</span>, <span class="st">''</span>, x))  <span class="co"># remove quotes</span></span>
<span id="cb37-11"><a href="#cb37-11" aria-hidden="true" tabindex="-1"></a>unique_gene_ids <span class="ot">&lt;-</span> <span class="fu">unique</span>(gene_ids)</span>
<span id="cb37-12"><a href="#cb37-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-13"><a href="#cb37-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate new FASTA headers</span></span>
<span id="cb37-14"><a href="#cb37-14" aria-hidden="true" tabindex="-1"></a>new_headers <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"Ptuh_"</span>, unique_gene_ids)</span>
<span id="cb37-15"><a href="#cb37-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-16"><a href="#cb37-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Confirm lengths match</span></span>
<span id="cb37-17"><a href="#cb37-17" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">length</span>(fasta) <span class="sc">!=</span> <span class="fu">length</span>(new_headers)) {</span>
<span id="cb37-18"><a href="#cb37-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stop</span>(<span class="st">"Mismatch between number of FASTA sequences and GTF gene IDs."</span>)</span>
<span id="cb37-19"><a href="#cb37-19" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb37-20"><a href="#cb37-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-21"><a href="#cb37-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Rename sequences</span></span>
<span id="cb37-22"><a href="#cb37-22" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(fasta) <span class="ot">&lt;-</span> new_headers</span>
<span id="cb37-23"><a href="#cb37-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-24"><a href="#cb37-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Write new FASTA</span></span>
<span id="cb37-25"><a href="#cb37-25" aria-hidden="true" tabindex="-1"></a><span class="fu">writeXStringSet</span>(fasta, <span class="at">filepath =</span> <span class="st">"lncRNA_renamed.fasta"</span>, <span class="at">format =</span> <span class="st">"fasta"</span>)</span>
<span id="cb37-26"><a href="#cb37-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-27"><a href="#cb37-27" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"FASTA headers renamed and saved to lncRNA_renamed.fasta</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="fu">grep</span> <span class="st">"&gt;"</span> lncRNA_renamed.fasta <span class="kw">|</span> <span class="fu">head</span> </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Read renamed FASTA and original GTF</span></span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>fasta <span class="ot">&lt;-</span> <span class="fu">readDNAStringSet</span>(<span class="st">"lncRNA_renamed.fasta"</span>)</span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>gtf <span class="ot">&lt;-</span> <span class="fu">import</span>(<span class="st">"https://raw.githubusercontent.com/urol-e5/deep-dive-expression/refs/heads/main/F-Ptuh/output/17-Ptuh-lncRNA/lncRNA.gtf"</span>, <span class="at">format =</span> <span class="st">"gtf"</span>)</span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract gene IDs from GTF</span></span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a>gtf_gene_ids <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">gsub</span>(<span class="st">'"'</span>, <span class="st">''</span>, <span class="fu">mcols</span>(gtf)<span class="sc">$</span>gene_id))  <span class="co"># remove quotes</span></span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract IDs from FASTA headers</span></span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a>fasta_headers <span class="ot">&lt;-</span> <span class="fu">names</span>(fasta)</span>
<span id="cb39-10"><a href="#cb39-10" aria-hidden="true" tabindex="-1"></a>fasta_gene_ids <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"^Ptuh_"</span>, <span class="st">""</span>, fasta_headers)</span>
<span id="cb39-11"><a href="#cb39-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-12"><a href="#cb39-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Compare</span></span>
<span id="cb39-13"><a href="#cb39-13" aria-hidden="true" tabindex="-1"></a>all_match <span class="ot">&lt;-</span> <span class="fu">all</span>(fasta_gene_ids <span class="sc">%in%</span> gtf_gene_ids) <span class="sc">&amp;&amp;</span> <span class="fu">all</span>(gtf_gene_ids <span class="sc">%in%</span> fasta_gene_ids)</span>
<span id="cb39-14"><a href="#cb39-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-15"><a href="#cb39-15" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (all_match) {</span>
<span id="cb39-16"><a href="#cb39-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"✅ FASTA headers match gene_ids in GTF exactly.</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb39-17"><a href="#cb39-17" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb39-18"><a href="#cb39-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"❌ Mismatch found between FASTA headers and GTF gene_ids.</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb39-19"><a href="#cb39-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-20"><a href="#cb39-20" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Optional: show what's missing</span></span>
<span id="cb39-21"><a href="#cb39-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">In FASTA but not GTF:</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb39-22"><a href="#cb39-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">setdiff</span>(fasta_gene_ids, gtf_gene_ids))</span>
<span id="cb39-23"><a href="#cb39-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-24"><a href="#cb39-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">In GTF but not FASTA:</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb39-25"><a href="#cb39-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">setdiff</span>(gtf_gene_ids, fasta_gene_ids))</span>
<span id="cb39-26"><a href="#cb39-26" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>✅ FASTA headers match gene_ids in GTF exactly.</code></pre>
</div>
</div>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Read files</span></span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>fasta <span class="ot">&lt;-</span> <span class="fu">readDNAStringSet</span>(<span class="st">"lncRNA_renamed.fasta"</span>)</span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>gtf <span class="ot">&lt;-</span> <span class="fu">import</span>(<span class="st">"https://raw.githubusercontent.com/urol-e5/deep-dive-expression/refs/heads/main/F-Ptuh/output/17-Ptuh-lncRNA/lncRNA.gtf"</span>, <span class="at">format =</span> <span class="st">"gtf"</span>)</span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Clean gene_id and extract relevant fields</span></span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a>gtf_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(gtf)</span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a>gtf_df<span class="sc">$</span>gene_id <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">'"'</span>, <span class="st">''</span>, gtf_df<span class="sc">$</span>gene_id)</span>
<span id="cb41-8"><a href="#cb41-8" aria-hidden="true" tabindex="-1"></a>gtf_summary <span class="ot">&lt;-</span> gtf_df <span class="sc">%&gt;%</span></span>
<span id="cb41-9"><a href="#cb41-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(seqnames, start, end, gene_id) <span class="sc">%&gt;%</span></span>
<span id="cb41-10"><a href="#cb41-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(gene_id) <span class="sc">%&gt;%</span></span>
<span id="cb41-11"><a href="#cb41-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">gtf_length =</span> <span class="fu">sum</span>(end <span class="sc">-</span> start <span class="sc">+</span> <span class="dv">1</span>), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb41-12"><a href="#cb41-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-13"><a href="#cb41-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract FASTA lengths</span></span>
<span id="cb41-14"><a href="#cb41-14" aria-hidden="true" tabindex="-1"></a>fasta_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb41-15"><a href="#cb41-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">gene_id =</span> <span class="fu">gsub</span>(<span class="st">"^Ptuh_"</span>, <span class="st">""</span>, <span class="fu">names</span>(fasta)),</span>
<span id="cb41-16"><a href="#cb41-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">fasta_length =</span> <span class="fu">width</span>(fasta),</span>
<span id="cb41-17"><a href="#cb41-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">fasta_header =</span> <span class="fu">names</span>(fasta)</span>
<span id="cb41-18"><a href="#cb41-18" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb41-19"><a href="#cb41-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-20"><a href="#cb41-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Join and compare</span></span>
<span id="cb41-21"><a href="#cb41-21" aria-hidden="true" tabindex="-1"></a>merged <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">inner_join</span>(gtf_summary, fasta_df, <span class="at">by =</span> <span class="st">"gene_id"</span>)</span>
<span id="cb41-22"><a href="#cb41-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-23"><a href="#cb41-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Check consistency</span></span>
<span id="cb41-24"><a href="#cb41-24" aria-hidden="true" tabindex="-1"></a>merged<span class="sc">$</span>length_match <span class="ot">&lt;-</span> merged<span class="sc">$</span>gtf_length <span class="sc">==</span> merged<span class="sc">$</span>fasta_length</span>
<span id="cb41-25"><a href="#cb41-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-26"><a href="#cb41-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Display a few examples</span></span>
<span id="cb41-27"><a href="#cb41-27" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"🔍 Sample GTF vs FASTA length comparison:</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>🔍 Sample GTF vs FASTA length comparison:</code></pre>
</div>
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">head</span>(merged[, <span class="fu">c</span>(<span class="st">"fasta_header"</span>, <span class="st">"gtf_length"</span>, <span class="st">"fasta_length"</span>, <span class="st">"length_match"</span>)]), <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 4
  fasta_header    gtf_length fasta_length length_match
  &lt;chr&gt;                &lt;dbl&gt;        &lt;int&gt; &lt;lgl&gt;       
1 Ptuh_lncRNA_001        937          935 FALSE       
2 Ptuh_lncRNA_002        828          826 FALSE       
3 Ptuh_lncRNA_003        626          624 FALSE       
4 Ptuh_lncRNA_004      13587        13585 FALSE       
5 Ptuh_lncRNA_005       2805         2803 FALSE       
6 Ptuh_lncRNA_006        424          422 FALSE       </code></pre>
</div>
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Overall match rate</span></span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>match_rate <span class="ot">&lt;-</span> <span class="fu">mean</span>(merged<span class="sc">$</span>length_match)</span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">✅ Percent of exact matches: %.1f%%</span><span class="sc">\n</span><span class="st">"</span>, <span class="dv">100</span> <span class="sc">*</span> match_rate))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
✅ Percent of exact matches: 0.0%</code></pre>
</div>
</div>
<hr>
</section>
<section id="code" class="level1">
<h1>Code</h1>
<div class="sourceCode" id="cb47"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Global R options</span></span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a><span class="ex">knitr::opts_chunk</span><span class="va">$set</span><span class="er">(</span><span class="bu">echo</span> = TRUE<span class="kw">)</span></span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Define key paths and tool directories</span></span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb47-6"><a href="#cb47-6" aria-hidden="true" tabindex="-1"></a><span class="ex">DATA_DIR</span> <span class="op">&lt;</span>- <span class="st">"../data/17-Ptuh-lncRNA"</span></span>
<span id="cb47-7"><a href="#cb47-7" aria-hidden="true" tabindex="-1"></a><span class="ex">OUTPUT_DIR</span> <span class="op">&lt;</span>- <span class="st">"../output/17-Ptuh-lncRNA"</span></span>
<span id="cb47-8"><a href="#cb47-8" aria-hidden="true" tabindex="-1"></a><span class="ex">THREADS</span> <span class="op">&lt;</span>- <span class="st">"24"</span></span>
<span id="cb47-9"><a href="#cb47-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb47-10"><a href="#cb47-10" aria-hidden="true" tabindex="-1"></a><span class="ex">FASTQ_SOURCE</span> <span class="op">&lt;</span>- <span class="st">"https://gannet.fish.washington.edu/Atumefaciens/20230519-E5_coral-fastqc-fastp-multiqc-RNAseq/P_meandrina/trimmed/"</span></span>
<span id="cb47-11"><a href="#cb47-11" aria-hidden="true" tabindex="-1"></a><span class="ex">FASTQ_SUFFIX</span> <span class="op">&lt;</span>- <span class="st">"fastq.gz"</span></span>
<span id="cb47-12"><a href="#cb47-12" aria-hidden="true" tabindex="-1"></a><span class="ex">GENOME_SOURCE</span> <span class="op">&lt;</span>- <span class="st">"https://owl.fish.washington.edu/halfshell/genomic-databank/Pocillopora_meandrina_HIv1.assembly.fasta"</span></span>
<span id="cb47-13"><a href="#cb47-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-14"><a href="#cb47-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-15"><a href="#cb47-15" aria-hidden="true" tabindex="-1"></a><span class="ex">GTF_SOURCE</span> <span class="op">&lt;</span>- <span class="st">"https://raw.githubusercontent.com/urol-e5/timeseries_molecular/d5f546705e3df40558eeaa5c18b122c79d2f4453/F-Ptua/data/Pocillopora_meandrina_HIv1.genes-validated.gtf"</span></span>
<span id="cb47-16"><a href="#cb47-16" aria-hidden="true" tabindex="-1"></a><span class="ex">GFF_SOURCE</span> <span class="op">&lt;</span>- <span class="st">"https://gannet.fish.washington.edu/seashell/bu-github/deep-dive-expression/F-Ptuh/data/Pocillopora_meandrina_HIv1.genes-validated.gff3"</span></span>
<span id="cb47-17"><a href="#cb47-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-18"><a href="#cb47-18" aria-hidden="true" tabindex="-1"></a><span class="ex">GFFPATTERN</span> <span class="op">&lt;</span>- <span class="st">'class_code "u"|class_code "x"|class_code "o"|class_code "i"'</span></span>
<span id="cb47-19"><a href="#cb47-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-20"><a href="#cb47-20" aria-hidden="true" tabindex="-1"></a><span class="co">#RAVEN</span></span>
<span id="cb47-21"><a href="#cb47-21" aria-hidden="true" tabindex="-1"></a><span class="co"># HISAT2_DIR &lt;- "/home/shared/hisat2-2.2.1/"</span></span>
<span id="cb47-22"><a href="#cb47-22" aria-hidden="true" tabindex="-1"></a><span class="co"># SAMTOOLS_DIR &lt;- "/home/shared/samtools-1.12/"</span></span>
<span id="cb47-23"><a href="#cb47-23" aria-hidden="true" tabindex="-1"></a><span class="co"># STRINGTIE_DIR &lt;- "/home/shared/stringtie-2.2.1.Linux_x86_64"</span></span>
<span id="cb47-24"><a href="#cb47-24" aria-hidden="true" tabindex="-1"></a><span class="co"># GFFCOMPARE_DIR &lt;- "/home/shared/gffcompare-0.12.6.Linux_x86_64"</span></span>
<span id="cb47-25"><a href="#cb47-25" aria-hidden="true" tabindex="-1"></a><span class="co"># BEDTOOLS_DIR &lt;- "/home/shared/bedtools2/bin"</span></span>
<span id="cb47-26"><a href="#cb47-26" aria-hidden="true" tabindex="-1"></a><span class="co"># CPC2_DIR &lt;- "/home/shared/CPC2_standalone-1.0.1"</span></span>
<span id="cb47-27"><a href="#cb47-27" aria-hidden="true" tabindex="-1"></a><span class="co"># CONDA_PATH &lt;- "/opt/anaconda/anaconda3/bin/conda"</span></span>
<span id="cb47-28"><a href="#cb47-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-29"><a href="#cb47-29" aria-hidden="true" tabindex="-1"></a><span class="co">#KLONE</span></span>
<span id="cb47-30"><a href="#cb47-30" aria-hidden="true" tabindex="-1"></a><span class="ex">HISAT2_DIR</span> <span class="op">&lt;</span>- <span class="st">""</span></span>
<span id="cb47-31"><a href="#cb47-31" aria-hidden="true" tabindex="-1"></a><span class="ex">SAMTOOLS_DIR</span> <span class="op">&lt;</span>- <span class="st">""</span></span>
<span id="cb47-32"><a href="#cb47-32" aria-hidden="true" tabindex="-1"></a><span class="ex">STRINGTIE_DIR</span> <span class="op">&lt;</span>- <span class="st">""</span></span>
<span id="cb47-33"><a href="#cb47-33" aria-hidden="true" tabindex="-1"></a><span class="ex">GFFCOMPARE_DIR</span> <span class="op">&lt;</span>- <span class="st">"/srlab/programs/gffcompare-0.12.6.Linux_x86_64/"</span></span>
<span id="cb47-34"><a href="#cb47-34" aria-hidden="true" tabindex="-1"></a><span class="ex">BEDTOOLS_DIR</span> <span class="op">&lt;</span>- <span class="st">""</span></span>
<span id="cb47-35"><a href="#cb47-35" aria-hidden="true" tabindex="-1"></a><span class="ex">CPC2_DIR</span> <span class="op">&lt;</span>- <span class="st">"/srlab/programs/CPC2_standalone-1.0.1/bin/"</span></span>
<span id="cb47-36"><a href="#cb47-36" aria-hidden="true" tabindex="-1"></a><span class="ex">CONDA_PATH</span> <span class="op">&lt;</span>- <span class="st">"/mmfs1/gscratch/srlab/nextflow/bin/miniforge/bin/conda"</span></span>
<span id="cb47-37"><a href="#cb47-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-38"><a href="#cb47-38" aria-hidden="true" tabindex="-1"></a><span class="ex">GENOME_FASTA</span> <span class="op">&lt;</span>- file.path<span class="er">(</span><span class="ex">DATA_DIR,</span> <span class="st">"genome.fasta"</span><span class="kw">)</span></span>
<span id="cb47-39"><a href="#cb47-39" aria-hidden="true" tabindex="-1"></a><span class="ex">GENOME_GTF</span> <span class="op">&lt;</span>- file.path<span class="er">(</span><span class="ex">DATA_DIR,</span> <span class="st">"genome.gtf"</span><span class="kw">)</span></span>
<span id="cb47-40"><a href="#cb47-40" aria-hidden="true" tabindex="-1"></a><span class="ex">GENOME_GFF</span> <span class="op">&lt;</span>- file.path<span class="er">(</span><span class="ex">DATA_DIR,</span> <span class="st">"genome.gff"</span><span class="kw">)</span></span>
<span id="cb47-41"><a href="#cb47-41" aria-hidden="true" tabindex="-1"></a><span class="ex">FASTQ_DIR</span> <span class="op">&lt;</span>- file.path<span class="er">(</span><span class="ex">DATA_DIR,</span> <span class="st">"fastq"</span><span class="kw">)</span></span>
<span id="cb47-42"><a href="#cb47-42" aria-hidden="true" tabindex="-1"></a><span class="ex">GENOME_INDEX</span> <span class="op">&lt;</span>- file.path<span class="er">(</span><span class="ex">OUTPUT_DIR,</span> <span class="st">"genome.index"</span><span class="kw">)</span></span>
<span id="cb47-43"><a href="#cb47-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-44"><a href="#cb47-44" aria-hidden="true" tabindex="-1"></a><span class="co"># Export these as environment variables for bash chunks.</span></span>
<span id="cb47-45"><a href="#cb47-45" aria-hidden="true" tabindex="-1"></a><span class="ex">Sys.setenv</span><span class="er">(</span></span>
<span id="cb47-46"><a href="#cb47-46" aria-hidden="true" tabindex="-1"></a>  <span class="ex">THREADS</span> = THREADS,</span>
<span id="cb47-47"><a href="#cb47-47" aria-hidden="true" tabindex="-1"></a>  <span class="ex">DATA_DIR</span> = DATA_DIR,</span>
<span id="cb47-48"><a href="#cb47-48" aria-hidden="true" tabindex="-1"></a>  <span class="ex">FASTQ_SOURCE</span> = FASTQ_SOURCE,</span>
<span id="cb47-49"><a href="#cb47-49" aria-hidden="true" tabindex="-1"></a>  <span class="ex">FASTQ_SUFFIX</span> = FASTQ_SUFFIX,</span>
<span id="cb47-50"><a href="#cb47-50" aria-hidden="true" tabindex="-1"></a>  <span class="ex">OUTPUT_DIR</span> = OUTPUT_DIR,</span>
<span id="cb47-51"><a href="#cb47-51" aria-hidden="true" tabindex="-1"></a>  <span class="ex">GENOME_SOURCE</span> = GENOME_SOURCE,</span>
<span id="cb47-52"><a href="#cb47-52" aria-hidden="true" tabindex="-1"></a>  <span class="ex">GTF_SOURCE</span> = GTF_SOURCE,</span>
<span id="cb47-53"><a href="#cb47-53" aria-hidden="true" tabindex="-1"></a>  <span class="ex">GFF_SOURCE</span> = GFF_SOURCE,</span>
<span id="cb47-54"><a href="#cb47-54" aria-hidden="true" tabindex="-1"></a>  <span class="ex">HISAT2_DIR</span> = HISAT2_DIR,</span>
<span id="cb47-55"><a href="#cb47-55" aria-hidden="true" tabindex="-1"></a>  <span class="ex">SAMTOOLS_DIR</span> = SAMTOOLS_DIR,</span>
<span id="cb47-56"><a href="#cb47-56" aria-hidden="true" tabindex="-1"></a>  <span class="ex">STRINGTIE_DIR</span> = STRINGTIE_DIR,</span>
<span id="cb47-57"><a href="#cb47-57" aria-hidden="true" tabindex="-1"></a>  <span class="ex">GFFCOMPARE_DIR</span> = GFFCOMPARE_DIR,</span>
<span id="cb47-58"><a href="#cb47-58" aria-hidden="true" tabindex="-1"></a>  <span class="ex">BEDTOOLS_DIR</span> = BEDTOOLS_DIR,</span>
<span id="cb47-59"><a href="#cb47-59" aria-hidden="true" tabindex="-1"></a>  <span class="ex">CPC2_DIR</span> = CPC2_DIR,</span>
<span id="cb47-60"><a href="#cb47-60" aria-hidden="true" tabindex="-1"></a>  <span class="ex">CONDA_PATH</span> = CONDA_PATH,</span>
<span id="cb47-61"><a href="#cb47-61" aria-hidden="true" tabindex="-1"></a>  <span class="ex">GENOME_FASTA</span> = GENOME_FASTA,</span>
<span id="cb47-62"><a href="#cb47-62" aria-hidden="true" tabindex="-1"></a>  <span class="ex">GENOME_GTF</span> = GENOME_GTF,</span>
<span id="cb47-63"><a href="#cb47-63" aria-hidden="true" tabindex="-1"></a>  <span class="ex">GENOME_GFF</span> = GENOME_GFF,</span>
<span id="cb47-64"><a href="#cb47-64" aria-hidden="true" tabindex="-1"></a>  <span class="ex">FASTQ_DIR</span> = FASTQ_DIR,</span>
<span id="cb47-65"><a href="#cb47-65" aria-hidden="true" tabindex="-1"></a>  <span class="ex">GENOME_INDEX</span> = GENOME_INDEX,</span>
<span id="cb47-66"><a href="#cb47-66" aria-hidden="true" tabindex="-1"></a>  <span class="ex">GFFPATTERN</span> = GFFPATTERN</span>
<span id="cb47-67"><a href="#cb47-67" aria-hidden="true" tabindex="-1"></a><span class="kw">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="st">"</span><span class="va">${DATA_DIR}</span><span class="st">"</span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-nv</span> <span class="at">-r</span> <span class="dt">\</span></span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a>--no-directories <span class="at">--no-parent</span> <span class="dt">\</span></span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a>-P <span class="va">${FASTQ_DIR}</span> <span class="dt">\</span></span>
<span id="cb49-4"><a href="#cb49-4" aria-hidden="true" tabindex="-1"></a>-A <span class="st">"*</span><span class="va">${FASTQ_SUFFIX}</span><span class="st">"</span> <span class="va">${FASTQ_SOURCE}</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> <span class="va">${FASTQ_DIR}</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>RNA-POC-47-S1-TP2_R1_001.fastp-trim.20230519.fastq.gz
RNA-POC-47-S1-TP2_R2_001.fastp-trim.20230519.fastq.gz
RNA-POC-48-S1-TP2_R1_001.fastp-trim.20230519.fastq.gz
RNA-POC-48-S1-TP2_R2_001.fastp-trim.20230519.fastq.gz
RNA-POC-50-S1-TP2_R1_001.fastp-trim.20230519.fastq.gz
RNA-POC-50-S1-TP2_R2_001.fastp-trim.20230519.fastq.gz
RNA-POC-53-S1-TP2_R1_001.fastp-trim.20230519.fastq.gz
RNA-POC-53-S1-TP2_R2_001.fastp-trim.20230519.fastq.gz
RNA-POC-57-S1-TP2_R1_001.fastp-trim.20230519.fastq.gz
RNA-POC-57-S1-TP2_R2_001.fastp-trim.20230519.fastq.gz</code></pre>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-o</span> <span class="st">"</span><span class="va">${GENOME_FASTA}</span><span class="st">"</span> <span class="st">"</span><span class="va">${GENOME_SOURCE}</span><span class="st">"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-o</span> <span class="st">"</span><span class="va">${GENOME_GTF}</span><span class="st">"</span> <span class="st">"</span><span class="va">${GTF_SOURCE}</span><span class="st">"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-o</span> <span class="st">"</span><span class="va">${GENOME_GFF}</span><span class="st">"</span> <span class="st">"</span><span class="va">${GFF_SOURCE}</span><span class="st">"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="va">output_fasta</span><span class="op">=</span><span class="va">$(</span><span class="fu">head</span> <span class="at">-1</span> <span class="st">"</span><span class="va">${GENOME_FASTA}</span><span class="st">"</span><span class="va">)</span></span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a><span class="va">output_gff</span><span class="op">=</span><span class="va">$(</span><span class="fu">head</span> <span class="at">-2</span> <span class="st">"</span><span class="va">${GENOME_GFF}</span><span class="st">"</span><span class="va">)</span></span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a><span class="va">output_gtf</span><span class="op">=</span><span class="va">$(</span><span class="fu">head</span> <span class="at">-1</span> <span class="st">"</span><span class="va">${GENOME_GTF}</span><span class="st">"</span><span class="va">)</span></span>
<span id="cb55-4"><a href="#cb55-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-5"><a href="#cb55-5" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> <span class="kw">[[</span> <span class="st">"</span><span class="va">$output_fasta</span><span class="st">"</span> <span class="ot">==</span> <span class="pp">*</span>html<span class="pp">*</span> <span class="kw">||</span> <span class="st">"</span><span class="va">$output_gff</span><span class="st">"</span> <span class="ot">==</span> <span class="pp">*</span>html<span class="pp">*</span> <span class="kw">||</span> <span class="st">"</span><span class="va">$output_gtf</span><span class="st">"</span> <span class="ot">==</span> <span class="pp">*</span>html<span class="pp">*</span> <span class="kw">]];</span> <span class="cf">then</span></span>
<span id="cb55-6"><a href="#cb55-6" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">"FAIL - FFS you downloaded a HTML not and genome feature file!"</span></span>
<span id="cb55-7"><a href="#cb55-7" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span></span>
<span id="cb55-8"><a href="#cb55-8" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">"</span><span class="va">$output_fasta</span><span class="st">"</span></span>
<span id="cb55-9"><a href="#cb55-9" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">"</span><span class="va">$output_gff</span><span class="st">"</span></span>
<span id="cb55-10"><a href="#cb55-10" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">"</span><span class="va">$output_gtf</span><span class="st">"</span></span>
<span id="cb55-11"><a href="#cb55-11" aria-hidden="true" tabindex="-1"></a><span class="cf">fi</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="hisat" class="level1">
<h1>HISAT</h1>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="st">"</span><span class="va">${HISAT2_DIR}</span><span class="st">hisat2_extract_exons.py"</span> <span class="st">"</span><span class="va">${GENOME_GTF}</span><span class="st">"</span> <span class="op">&gt;</span> <span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">/exon.txt"</span></span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a><span class="st">"</span><span class="va">${HISAT2_DIR}</span><span class="st">hisat2_extract_splice_sites.py"</span> <span class="st">"</span><span class="va">${GENOME_GTF}</span><span class="st">"</span> <span class="op">&gt;</span> <span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">/splice_sites.txt"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a><span class="st">"</span><span class="va">${HISAT2_DIR}</span><span class="st">hisat2-build"</span> <span class="dt">\</span></span>
<span id="cb57-3"><a href="#cb57-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">-p</span> <span class="st">"</span><span class="va">${THREADS}</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb57-4"><a href="#cb57-4" aria-hidden="true" tabindex="-1"></a>  <span class="st">"</span><span class="va">${GENOME_FASTA}</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb57-5"><a href="#cb57-5" aria-hidden="true" tabindex="-1"></a>  <span class="st">"</span><span class="va">${GENOME_INDEX}</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb57-6"><a href="#cb57-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">--exon</span> <span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">/exon.txt"</span> <span class="dt">\</span></span>
<span id="cb57-7"><a href="#cb57-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">--ss</span> <span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">/splice_sites.txt"</span> <span class="dt">\</span></span>
<span id="cb57-8"><a href="#cb57-8" aria-hidden="true" tabindex="-1"></a>  <span class="dv">2</span><span class="op">&gt;</span> <span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">/hisat2-build_stats.txt"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Loop over every file ending in .fastq.gz that contains "_R2_"</span></span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> r2 <span class="kw">in</span> <span class="st">"</span><span class="va">${FASTQ_DIR}</span><span class="st">"</span>/<span class="pp">*</span>_R2_<span class="pp">*</span>.<span class="st">"</span><span class="va">${FASTQ_SUFFIX}</span><span class="st">"</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Get the basename (filename without path)</span></span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a>    <span class="va">base</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="st">"</span><span class="va">$r2</span><span class="st">"</span><span class="va">)</span></span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb58-6"><a href="#cb58-6" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Derive a sample name by taking everything before "_R2_"</span></span>
<span id="cb58-7"><a href="#cb58-7" aria-hidden="true" tabindex="-1"></a>    <span class="va">sample</span><span class="op">=</span><span class="st">"</span><span class="va">${base</span><span class="op">%%</span>_R2_<span class="pp">*</span><span class="va">}</span><span class="st">"</span></span>
<span id="cb58-8"><a href="#cb58-8" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb58-9"><a href="#cb58-9" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Construct the corresponding R1 filename by replacing "_R2_" with "_R1_"</span></span>
<span id="cb58-10"><a href="#cb58-10" aria-hidden="true" tabindex="-1"></a>    <span class="va">r1</span><span class="op">=</span><span class="st">"</span><span class="va">${r2</span><span class="op">/</span><span class="ss">_R2_</span><span class="op">/</span>_R1_<span class="va">}</span><span class="st">"</span></span>
<span id="cb58-11"><a href="#cb58-11" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb58-12"><a href="#cb58-12" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Define the output SAM file name using the sample name</span></span>
<span id="cb58-13"><a href="#cb58-13" aria-hidden="true" tabindex="-1"></a>    <span class="va">output</span><span class="op">=</span><span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">/</span><span class="va">${sample}</span><span class="st">.sam"</span></span>
<span id="cb58-14"><a href="#cb58-14" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb58-15"><a href="#cb58-15" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Run hisat2 with the paired-end files</span></span>
<span id="cb58-16"><a href="#cb58-16" aria-hidden="true" tabindex="-1"></a>    <span class="st">"</span><span class="va">${HISAT2_DIR}</span><span class="st">hisat2"</span> <span class="dt">\</span></span>
<span id="cb58-17"><a href="#cb58-17" aria-hidden="true" tabindex="-1"></a>      <span class="at">-x</span> <span class="st">"</span><span class="va">${GENOME_INDEX}</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb58-18"><a href="#cb58-18" aria-hidden="true" tabindex="-1"></a>      <span class="at">-p</span> <span class="st">"</span><span class="va">${THREADS}</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb58-19"><a href="#cb58-19" aria-hidden="true" tabindex="-1"></a>      <span class="at">-1</span> <span class="st">"</span><span class="va">$r1</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb58-20"><a href="#cb58-20" aria-hidden="true" tabindex="-1"></a>      <span class="at">-2</span> <span class="st">"</span><span class="va">$r2</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb58-21"><a href="#cb58-21" aria-hidden="true" tabindex="-1"></a>      <span class="at">-S</span> <span class="st">"</span><span class="va">$output</span><span class="st">"</span></span>
<span id="cb58-22"><a href="#cb58-22" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="convert-sam-to-bam" class="level2">
<h2 class="anchored" data-anchor-id="convert-sam-to-bam">convert SAM to BAM</h2>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> samfile <span class="kw">in</span> <span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">/</span><span class="va">${sample}</span><span class="st">"</span><span class="pp">*</span>.sam<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a>  <span class="va">bamfile</span><span class="op">=</span><span class="st">"</span><span class="va">${samfile</span><span class="op">%</span>.sam<span class="va">}</span><span class="st">.bam"</span></span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a>  <span class="va">sorted_bamfile</span><span class="op">=</span><span class="st">"</span><span class="va">${samfile</span><span class="op">%</span>.sam<span class="va">}</span><span class="st">.sorted.bam"</span></span>
<span id="cb59-4"><a href="#cb59-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb59-5"><a href="#cb59-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Convert SAM to BAM</span></span>
<span id="cb59-6"><a href="#cb59-6" aria-hidden="true" tabindex="-1"></a>  <span class="st">"</span><span class="va">${SAMTOOLS_DIR}</span><span class="st">samtools"</span> view <span class="at">-bS</span> <span class="at">-@</span> <span class="st">"</span><span class="va">${THREADS}</span><span class="st">"</span> <span class="st">"</span><span class="va">$samfile</span><span class="st">"</span> <span class="op">&gt;</span> <span class="st">"</span><span class="va">$bamfile</span><span class="st">"</span></span>
<span id="cb59-7"><a href="#cb59-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb59-8"><a href="#cb59-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Sort BAM</span></span>
<span id="cb59-9"><a href="#cb59-9" aria-hidden="true" tabindex="-1"></a>  <span class="st">"</span><span class="va">${SAMTOOLS_DIR}</span><span class="st">samtools"</span> sort <span class="at">-@</span> <span class="st">"</span><span class="va">${THREADS}</span><span class="st">"</span> <span class="st">"</span><span class="va">$bamfile</span><span class="st">"</span> <span class="at">-o</span> <span class="st">"</span><span class="va">$sorted_bamfile</span><span class="st">"</span></span>
<span id="cb59-10"><a href="#cb59-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb59-11"><a href="#cb59-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Index sorted BAM</span></span>
<span id="cb59-12"><a href="#cb59-12" aria-hidden="true" tabindex="-1"></a>  <span class="st">"</span><span class="va">${SAMTOOLS_DIR}</span><span class="st">samtools"</span> index <span class="at">-@</span> <span class="st">"</span><span class="va">${THREADS}</span><span class="st">"</span> <span class="st">"</span><span class="va">$sorted_bamfile</span><span class="st">"</span></span>
<span id="cb59-13"><a href="#cb59-13" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="stringtie" class="level1">
<h1>StringTie</h1>
<p>StringTie uses the sorted BAM files to assemble transcripts for each sample, outputting them as GTF (Gene Transfer Format) files. And then merges all individual GTF assemblies into a single merged GTF file. This step extracts transcript information and merges GTFs from all samples–an important step in creating a canonical list of lncRNAs across all samples included in the pipeline.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="fu">find</span> <span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">"</span> <span class="at">-name</span> <span class="st">"*sorted.bam"</span> <span class="dt">\</span></span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span> <span class="fu">xargs</span> <span class="at">-n</span> 1 basename <span class="at">-s</span> .sorted.bam <span class="kw">|</span> <span class="fu">xargs</span> <span class="at">-I{}</span> <span class="dt">\</span></span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a><span class="st">"</span><span class="va">${STRINGTIE_DIR}</span><span class="st">stringtie"</span> <span class="dt">\</span></span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a>-p <span class="st">"</span><span class="va">${THREADS}</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb60-5"><a href="#cb60-5" aria-hidden="true" tabindex="-1"></a>-G <span class="st">"</span><span class="va">${GENOME_GFF}</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb60-6"><a href="#cb60-6" aria-hidden="true" tabindex="-1"></a>-o <span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">/{}.gtf"</span> <span class="dt">\</span></span>
<span id="cb60-7"><a href="#cb60-7" aria-hidden="true" tabindex="-1"></a><span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">/{}.sorted.bam"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> <span class="va">${OUTPUT_DIR}</span>/<span class="pp">*</span>.gtf</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wc</span> <span class="at">-l</span> <span class="va">${OUTPUT_DIR}</span>/<span class="pp">*</span>.gtf</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb63"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="st">"</span><span class="va">${STRINGTIE_DIR}</span><span class="st">stringtie"</span> <span class="dt">\</span></span>
<span id="cb63-2"><a href="#cb63-2" aria-hidden="true" tabindex="-1"></a>--merge <span class="dt">\</span></span>
<span id="cb63-3"><a href="#cb63-3" aria-hidden="true" tabindex="-1"></a>-G <span class="st">"</span><span class="va">${GENOME_GFF}</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb63-4"><a href="#cb63-4" aria-hidden="true" tabindex="-1"></a>-o <span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">/stringtie_merged.gtf"</span> <span class="dt">\</span></span>
<span id="cb63-5"><a href="#cb63-5" aria-hidden="true" tabindex="-1"></a><span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">/"</span><span class="pp">*</span>.gtf</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wc</span> <span class="at">-l</span> <span class="va">${OUTPUT_DIR}</span>/stringtie_merged.gtf</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> <span class="va">${OUTPUT_DIR}</span>/stringtie_merged.gtf</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>#GFFCOMPARE</p>
<p><img src="http://gannet.fish.washington.edu/seashell/snaps/Monosnap_687474703a2f2f67616e6e65742e666973682e77617368696e67746f6e2e6564752f7365617368656c6c2f736e6170732f323032332d31312d30335f30392d3_2024-12-20_04-02-37.png" class="img-fluid"></p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="st">"</span><span class="va">${GFFCOMPARE_DIR}</span><span class="st">gffcompare"</span> <span class="dt">\</span></span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a>-r <span class="st">"</span><span class="va">${GENOME_GFF}</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb66-3"><a href="#cb66-3" aria-hidden="true" tabindex="-1"></a>-o <span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">/gffcompare_merged"</span> <span class="dt">\</span></span>
<span id="cb66-4"><a href="#cb66-4" aria-hidden="true" tabindex="-1"></a><span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">/stringtie_merged.gtf"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb67"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> <span class="at">-4</span> <span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">"</span>/gffcompare_merged<span class="pp">*</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wc</span> <span class="at">-l</span> <span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">"</span>/gffcompare_merged<span class="pp">*</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"</span><span class="va">${GFFPATTERN}</span><span class="st">"</span></span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="fu">awk</span> <span class="st">'$3 == "transcript" &amp;&amp; $1 !~ /^#/'</span> <span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">/gffcompare_merged.annotated.gtf"</span> <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a><span class="fu">grep</span> <span class="at">-E</span> <span class="st">"</span><span class="va">${GFFPATTERN}</span><span class="st">"</span> <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb70-3"><a href="#cb70-3" aria-hidden="true" tabindex="-1"></a><span class="fu">awk</span> <span class="st">'($5 - $4 &gt; 199) || ($4 - $5 &gt; 199)'</span> <span class="op">&gt;</span> <span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">/lncRNA_candidates.gtf"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> <span class="va">${OUTPUT_DIR}</span>/lncRNA_candidates.gtf</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wc</span> <span class="at">-l</span> <span class="va">${OUTPUT_DIR}</span>/lncRNA_candidates.gtf</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sort</span> ../output/17-Ptuh-lncRNA/lncRNA_candidates.gtf <span class="kw">|</span> <span class="fu">uniq</span> <span class="at">-d</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="bedtools" class="level1">
<h1>Bedtools</h1>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="st">"</span><span class="va">${BEDTOOLS_DIR}</span><span class="st">"</span><span class="ex">bedtools</span> getfasta <span class="dt">\</span></span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a>-fi <span class="st">"</span><span class="va">${GENOME_FASTA}</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb74-3"><a href="#cb74-3" aria-hidden="true" tabindex="-1"></a>-bed <span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">/lncRNA_candidates.gtf"</span> <span class="dt">\</span></span>
<span id="cb74-4"><a href="#cb74-4" aria-hidden="true" tabindex="-1"></a>-fo <span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">/lncRNA_candidates.fasta"</span> <span class="dt">\</span></span>
<span id="cb74-5"><a href="#cb74-5" aria-hidden="true" tabindex="-1"></a>-name <span class="at">-split</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb75"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> <span class="va">${OUTPUT_DIR}</span>/lncRNA_candidates.fasta</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>#CPC2</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> https://github.com/gao-lab/CPC2_standalone/archive/refs/tags/v1.0.1.zip</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb77"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unzip</span> v1.0.1.zip</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="bu">eval</span> <span class="st">"</span><span class="va">$(</span><span class="ex">/mmfs1/gscratch/srlab/nextflow/bin/miniforge/bin/conda</span> shell.bash hook<span class="va">)</span><span class="st">"</span></span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate /mmfs1/gscratch/srlab/nextflow/bin/miniforge/envs/nextflow  <span class="co"># replace with your actual env name</span></span>
<span id="cb78-3"><a href="#cb78-3" aria-hidden="true" tabindex="-1"></a><span class="ex">python</span> /mmfs1/gscratch/scrubbed/sr320/github/deep-dive-expression/F-Ptuh/code/CPC2_standalone-1.0.1/bin/CPC2.py <span class="dt">\</span></span>
<span id="cb78-4"><a href="#cb78-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">-i</span> <span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">/lncRNA_candidates.fasta"</span> <span class="dt">\</span></span>
<span id="cb78-5"><a href="#cb78-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">-o</span> <span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">/CPC2"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Filter</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a><span class="fu">awk</span> <span class="st">'$8 == "noncoding" {print $1}'</span> <span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">/CPC2.txt"</span> <span class="op">&gt;</span> <span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">/noncoding_transcripts_ids.txt"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> <span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">/CPC2.txt"</span></span>
<span id="cb80-2"><a href="#cb80-2" aria-hidden="true" tabindex="-1"></a><span class="fu">wc</span> <span class="at">-l</span> <span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">/CPC2.txt"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb81"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> <span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">/noncoding_transcripts_ids.txt"</span></span>
<span id="cb81-2"><a href="#cb81-2" aria-hidden="true" tabindex="-1"></a><span class="fu">wc</span> <span class="at">-l</span> <span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">/noncoding_transcripts_ids.txt"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Subsetting fasta</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="st">"</span><span class="va">${SAMTOOLS_DIR}</span><span class="st">samtools"</span> faidx <span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">/lncRNA_candidates.fasta"</span> <span class="dt">\</span></span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a>-r <span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">/noncoding_transcripts_ids.txt"</span> <span class="dt">\</span></span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span> <span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">/lncRNA.fasta"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb83"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> <span class="at">-2</span> <span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">/lncRNA.fasta"</span></span>
<span id="cb83-2"><a href="#cb83-2" aria-hidden="true" tabindex="-1"></a><span class="fu">grep</span> <span class="at">-c</span> <span class="st">"&gt;"</span> <span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">/lncRNA.fasta"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define input and output file paths using the OUTPUT_DIR variable</span></span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a><span class="va">input</span><span class="op">=</span><span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">/noncoding_transcripts_ids.txt"</span></span>
<span id="cb84-3"><a href="#cb84-3" aria-hidden="true" tabindex="-1"></a><span class="va">output</span><span class="op">=</span><span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">/lncRNA.bed"</span></span>
<span id="cb84-4"><a href="#cb84-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-5"><a href="#cb84-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Process each line of the input file</span></span>
<span id="cb84-6"><a href="#cb84-6" aria-hidden="true" tabindex="-1"></a><span class="cf">while</span> <span class="va">IFS</span><span class="op">=</span> <span class="bu">read</span> <span class="at">-r</span> <span class="va">line</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb84-7"><a href="#cb84-7" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Remove "transcript::" from the line</span></span>
<span id="cb84-8"><a href="#cb84-8" aria-hidden="true" tabindex="-1"></a>    <span class="va">line</span><span class="op">=</span><span class="st">"</span><span class="va">${line</span><span class="op">//</span><span class="ss">transcript::</span><span class="op">/</span><span class="va">}</span><span class="st">"</span></span>
<span id="cb84-9"><a href="#cb84-9" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb84-10"><a href="#cb84-10" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Split the line by ':' to get the chromosome and position string</span></span>
<span id="cb84-11"><a href="#cb84-11" aria-hidden="true" tabindex="-1"></a>    <span class="va">IFS</span><span class="op">=</span><span class="st">':'</span> <span class="bu">read</span> <span class="at">-r</span> <span class="va">chromosome</span> <span class="va">pos</span> <span class="op">&lt;&lt;&lt;</span> <span class="st">"</span><span class="va">$line</span><span class="st">"</span></span>
<span id="cb84-12"><a href="#cb84-12" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb84-13"><a href="#cb84-13" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Split the position string by '-' to separate start and end positions</span></span>
<span id="cb84-14"><a href="#cb84-14" aria-hidden="true" tabindex="-1"></a>    <span class="va">IFS</span><span class="op">=</span><span class="st">'-'</span> <span class="bu">read</span> <span class="at">-r</span> <span class="va">start</span> <span class="va">end</span> <span class="op">&lt;&lt;&lt;</span> <span class="st">"</span><span class="va">$pos</span><span class="st">"</span></span>
<span id="cb84-15"><a href="#cb84-15" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb84-16"><a href="#cb84-16" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Convert the start position to 0-based by subtracting 1</span></span>
<span id="cb84-17"><a href="#cb84-17" aria-hidden="true" tabindex="-1"></a>    <span class="va">start</span><span class="op">=</span><span class="va">$((start</span> <span class="op">-</span> <span class="dv">1</span><span class="va">))</span></span>
<span id="cb84-18"><a href="#cb84-18" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb84-19"><a href="#cb84-19" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Write the chromosome, updated start, and end positions to the output file (tab-separated)</span></span>
<span id="cb84-20"><a href="#cb84-20" aria-hidden="true" tabindex="-1"></a>    <span class="bu">printf</span> <span class="st">"%s\t%s\t%s\n"</span> <span class="st">"</span><span class="va">$chromosome</span><span class="st">"</span> <span class="st">"</span><span class="va">$start</span><span class="st">"</span> <span class="st">"</span><span class="va">$end</span><span class="st">"</span></span>
<span id="cb84-21"><a href="#cb84-21" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span> <span class="op">&lt;</span> <span class="st">"</span><span class="va">$input</span><span class="st">"</span> <span class="op">&gt;</span> <span class="st">"</span><span class="va">$output</span><span class="st">"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb85"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> <span class="at">-1</span> <span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">/lncRNA.bed"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a><span class="fu">awk</span> <span class="st">'BEGIN{OFS="\t"; count=1} {printf "%s\t.\tlncRNA\t%d\t%d\t.\t+\t.\tgene_id \"lncRNA_%03d\";\n", $1, $2, $3, count++;}'</span> <span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">/lncRNA.bed"</span> <span class="dt">\</span></span>
<span id="cb86-2"><a href="#cb86-2" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span> <span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">/lncRNA.gtf"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb87"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb87-1"><a href="#cb87-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> <span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">/lncRNA.gtf"</span></span>
<span id="cb87-2"><a href="#cb87-2" aria-hidden="true" tabindex="-1"></a><span class="fu">wc</span> <span class="at">-l</span> <span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">/lncRNA.gtf"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="summary-table" class="level1">
<h1>Summary Table</h1>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="va">tf_file</span><span class="op">=</span><span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">/lncRNA.gtf"</span></span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a><span class="fu">awk</span> <span class="st">'</span></span>
<span id="cb88-4"><a href="#cb88-4" aria-hidden="true" tabindex="-1"></a><span class="st">BEGIN {</span></span>
<span id="cb88-5"><a href="#cb88-5" aria-hidden="true" tabindex="-1"></a><span class="st">    total_entries = 0;</span></span>
<span id="cb88-6"><a href="#cb88-6" aria-hidden="true" tabindex="-1"></a><span class="st">    min_length = 1e9;</span></span>
<span id="cb88-7"><a href="#cb88-7" aria-hidden="true" tabindex="-1"></a><span class="st">    max_length = 0;</span></span>
<span id="cb88-8"><a href="#cb88-8" aria-hidden="true" tabindex="-1"></a><span class="st">    sum_length = 0;</span></span>
<span id="cb88-9"><a href="#cb88-9" aria-hidden="true" tabindex="-1"></a><span class="st">}</span></span>
<span id="cb88-10"><a href="#cb88-10" aria-hidden="true" tabindex="-1"></a><span class="st"># Skip comment lines</span></span>
<span id="cb88-11"><a href="#cb88-11" aria-hidden="true" tabindex="-1"></a><span class="st">/^#/ { next }</span></span>
<span id="cb88-12"><a href="#cb88-12" aria-hidden="true" tabindex="-1"></a><span class="st">{</span></span>
<span id="cb88-13"><a href="#cb88-13" aria-hidden="true" tabindex="-1"></a><span class="st">    if (NF &lt; 9) next;</span></span>
<span id="cb88-14"><a href="#cb88-14" aria-hidden="true" tabindex="-1"></a><span class="st">    total_entries++;</span></span>
<span id="cb88-15"><a href="#cb88-15" aria-hidden="true" tabindex="-1"></a><span class="st">    start = $4;</span></span>
<span id="cb88-16"><a href="#cb88-16" aria-hidden="true" tabindex="-1"></a><span class="st">    end = $5;</span></span>
<span id="cb88-17"><a href="#cb88-17" aria-hidden="true" tabindex="-1"></a><span class="st">    gene_length = end - start + 1;</span></span>
<span id="cb88-18"><a href="#cb88-18" aria-hidden="true" tabindex="-1"></a><span class="st">    if (gene_length &lt; min_length) min_length = gene_length;</span></span>
<span id="cb88-19"><a href="#cb88-19" aria-hidden="true" tabindex="-1"></a><span class="st">    if (gene_length &gt; max_length) max_length = gene_length;</span></span>
<span id="cb88-20"><a href="#cb88-20" aria-hidden="true" tabindex="-1"></a><span class="st">    sum_length += gene_length;</span></span>
<span id="cb88-21"><a href="#cb88-21" aria-hidden="true" tabindex="-1"></a><span class="st">    feature[$3]++;</span></span>
<span id="cb88-22"><a href="#cb88-22" aria-hidden="true" tabindex="-1"></a><span class="st">    chrom[$1]++;</span></span>
<span id="cb88-23"><a href="#cb88-23" aria-hidden="true" tabindex="-1"></a><span class="st">    # Use two-argument match() and then extract the gene_id manually.</span></span>
<span id="cb88-24"><a href="#cb88-24" aria-hidden="true" tabindex="-1"></a><span class="st">    if (match($9, /gene_id "[^"]+"/)) {</span></span>
<span id="cb88-25"><a href="#cb88-25" aria-hidden="true" tabindex="-1"></a><span class="st">        gene_str = substr($9, RSTART, RLENGTH);</span></span>
<span id="cb88-26"><a href="#cb88-26" aria-hidden="true" tabindex="-1"></a><span class="st">        # Remove the "gene_id " prefix and the quotes.</span></span>
<span id="cb88-27"><a href="#cb88-27" aria-hidden="true" tabindex="-1"></a><span class="st">        gsub(/gene_id "/, "", gene_str);</span></span>
<span id="cb88-28"><a href="#cb88-28" aria-hidden="true" tabindex="-1"></a><span class="st">        gsub(/"/, "", gene_str);</span></span>
<span id="cb88-29"><a href="#cb88-29" aria-hidden="true" tabindex="-1"></a><span class="st">        genes[gene_str] = 1;</span></span>
<span id="cb88-30"><a href="#cb88-30" aria-hidden="true" tabindex="-1"></a><span class="st">    }</span></span>
<span id="cb88-31"><a href="#cb88-31" aria-hidden="true" tabindex="-1"></a><span class="st">}</span></span>
<span id="cb88-32"><a href="#cb88-32" aria-hidden="true" tabindex="-1"></a><span class="st">END {</span></span>
<span id="cb88-33"><a href="#cb88-33" aria-hidden="true" tabindex="-1"></a><span class="st">    avg_length = (total_entries &gt; 0) ? sum_length / total_entries : 0;</span></span>
<span id="cb88-34"><a href="#cb88-34" aria-hidden="true" tabindex="-1"></a><span class="st">    unique_gene_count = 0;</span></span>
<span id="cb88-35"><a href="#cb88-35" aria-hidden="true" tabindex="-1"></a><span class="st">    for (g in genes)</span></span>
<span id="cb88-36"><a href="#cb88-36" aria-hidden="true" tabindex="-1"></a><span class="st">        unique_gene_count++;</span></span>
<span id="cb88-37"><a href="#cb88-37" aria-hidden="true" tabindex="-1"></a><span class="st">    print "Basic GTF File Statistics:";</span></span>
<span id="cb88-38"><a href="#cb88-38" aria-hidden="true" tabindex="-1"></a><span class="st">    print "--------------------------";</span></span>
<span id="cb88-39"><a href="#cb88-39" aria-hidden="true" tabindex="-1"></a><span class="st">    print "Total entries:      " total_entries;</span></span>
<span id="cb88-40"><a href="#cb88-40" aria-hidden="true" tabindex="-1"></a><span class="st">    print "Unique genes:       " unique_gene_count;</span></span>
<span id="cb88-41"><a href="#cb88-41" aria-hidden="true" tabindex="-1"></a><span class="st">    print "Min gene length:    " min_length;</span></span>
<span id="cb88-42"><a href="#cb88-42" aria-hidden="true" tabindex="-1"></a><span class="st">    print "Max gene length:    " max_length;</span></span>
<span id="cb88-43"><a href="#cb88-43" aria-hidden="true" tabindex="-1"></a><span class="st">    printf("Average gene length: %.2f\n", avg_length);</span></span>
<span id="cb88-44"><a href="#cb88-44" aria-hidden="true" tabindex="-1"></a><span class="st">    print "\nFeature counts:";</span></span>
<span id="cb88-45"><a href="#cb88-45" aria-hidden="true" tabindex="-1"></a><span class="st">    for (f in feature) {</span></span>
<span id="cb88-46"><a href="#cb88-46" aria-hidden="true" tabindex="-1"></a><span class="st">        print "  " f ": " feature[f];</span></span>
<span id="cb88-47"><a href="#cb88-47" aria-hidden="true" tabindex="-1"></a><span class="st">    }</span></span>
<span id="cb88-48"><a href="#cb88-48" aria-hidden="true" tabindex="-1"></a><span class="st">    print "\nChromosome counts:";</span></span>
<span id="cb88-49"><a href="#cb88-49" aria-hidden="true" tabindex="-1"></a><span class="st">    for (c in chrom) {</span></span>
<span id="cb88-50"><a href="#cb88-50" aria-hidden="true" tabindex="-1"></a><span class="st">        print "  " c ": " chrom[c];</span></span>
<span id="cb88-51"><a href="#cb88-51" aria-hidden="true" tabindex="-1"></a><span class="st">    }</span></span>
<span id="cb88-52"><a href="#cb88-52" aria-hidden="true" tabindex="-1"></a><span class="st">}</span></span>
<span id="cb88-53"><a href="#cb88-53" aria-hidden="true" tabindex="-1"></a><span class="st">'</span> <span class="st">"</span><span class="va">$tf_file</span><span class="st">"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>Basic GTF File Statistics:
--------------------------
Total entries:      16153
Unique genes:       0
Min gene length:    203
Max gene length:    227016
Average gene length: 3125.62

Feature counts:
  lncRNA: 16153

Chromosome counts:
  Pocillopora_meandrina_HIv1___xfSc0000447: 7
  Pocillopora_meandrina_HIv1___xfSc0000094: 2
  Pocillopora_meandrina_HIv1___Sc0000041: 71
  Pocillopora_meandrina_HIv1___xfSc0000343: 7
  Pocillopora_meandrina_HIv1___xfSc0000004: 3
  Pocillopora_meandrina_HIv1___Sc0000011: 362
  Pocillopora_meandrina_HIv1___xfSc0000812: 1
  Pocillopora_meandrina_HIv1___xfSc0000081: 2
  Pocillopora_meandrina_HIv1___Sc0000032: 132
  Pocillopora_meandrina_HIv1___xpSc0001344: 1
  Pocillopora_meandrina_HIv1___xpSc0001331: 1
  Pocillopora_meandrina_HIv1___xfSc0001179: 1
  Pocillopora_meandrina_HIv1___xfSc0000284: 2
  Pocillopora_meandrina_HIv1___xfSc0000692: 3
  Pocillopora_meandrina_HIv1___xfSc0000029: 2
  Pocillopora_meandrina_HIv1___Sc0000052: 3
  Pocillopora_meandrina_HIv1___xfSc0000145: 2
  Pocillopora_meandrina_HIv1___Sc0000046: 28
  Pocillopora_meandrina_HIv1___xfSc0000948: 1
  Pocillopora_meandrina_HIv1___Sc0000025: 274
  Pocillopora_meandrina_HIv1___xfSc0000477: 2
  Pocillopora_meandrina_HIv1___Sc0000066: 1
  Pocillopora_meandrina_HIv1___Sc0000039: 96
  Pocillopora_meandrina_HIv1___xpSc0001280: 5
  Pocillopora_meandrina_HIv1___xfSc0001170: 1
  Pocillopora_meandrina_HIv1___xfSc0000212: 7
  Pocillopora_meandrina_HIv1___xfSc0000007: 36
  Pocillopora_meandrina_HIv1___Sc0000012: 457
  Pocillopora_meandrina_HIv1___xfSc0000974: 1
  Pocillopora_meandrina_HIv1___xfSc0000817: 1
  Pocillopora_meandrina_HIv1___xfSc0000570: 1
  Pocillopora_meandrina_HIv1___xfSc0000479: 2
  Pocillopora_meandrina_HIv1___xfSc0000237: 2
  Pocillopora_meandrina_HIv1___Sc0000068: 2
  Pocillopora_meandrina_HIv1___Sc0000037: 101
  Pocillopora_meandrina_HIv1___Sc0000003: 803
  Pocillopora_meandrina_HIv1___xfSc0000428: 4
  Pocillopora_meandrina_HIv1___xfSc0000059: 6
  Pocillopora_meandrina_HIv1___Sc0000057: 5
  Pocillopora_meandrina_HIv1___xfSc0000146: 5
  Pocillopora_meandrina_HIv1___Sc0000045: 52
  Pocillopora_meandrina_HIv1___Sc0000009: 729
  Pocillopora_meandrina_HIv1___xfSc0000952: 1
  Pocillopora_meandrina_HIv1___xfSc0000835: 2
  Pocillopora_meandrina_HIv1___xfSc0000000: 71
  Pocillopora_meandrina_HIv1___Sc0000015: 328
  Pocillopora_meandrina_HIv1___xfSc0000716: 1
  Pocillopora_meandrina_HIv1___Sc0000080: 2
  Pocillopora_meandrina_HIv1___xfSc0000616: 1
  Pocillopora_meandrina_HIv1___Sc0000034: 118
  Pocillopora_meandrina_HIv1___xpSc0001290: 1
  Pocillopora_meandrina_HIv1___xfSc0000527: 2
  Pocillopora_meandrina_HIv1___xfSc0000206: 6
  Pocillopora_meandrina_HIv1___Sc0000006: 561
  Pocillopora_meandrina_HIv1___xpSc0001273: 29
  Pocillopora_meandrina_HIv1___xfSc0000540: 4
  Pocillopora_meandrina_HIv1___Sc0000058: 5
  Pocillopora_meandrina_HIv1___Sc0000021: 290
  Pocillopora_meandrina_HIv1___xfSc0000859: 2
  Pocillopora_meandrina_HIv1___Sc0000040: 123
  Pocillopora_meandrina_HIv1___xfSc0000890: 1
  Pocillopora_meandrina_HIv1___xfSc0000836: 1
  Pocillopora_meandrina_HIv1___xfSc0000555: 2
  Pocillopora_meandrina_HIv1___xfSc0000003: 15
  Pocillopora_meandrina_HIv1___Sc0000016: 448
  Pocillopora_meandrina_HIv1___xfSc0000765: 5
  Pocillopora_meandrina_HIv1___Sc0000033: 112
  Pocillopora_meandrina_HIv1___Sc0000018: 348
  Pocillopora_meandrina_HIv1___xfSc0000415: 2
  Pocillopora_meandrina_HIv1___Sc0000053: 14
  Pocillopora_meandrina_HIv1___xfSc0000436: 4
  Pocillopora_meandrina_HIv1___xfSc0000205: 13
  Pocillopora_meandrina_HIv1___Sc0000070: 2
  Pocillopora_meandrina_HIv1___Sc0000049: 4
  Pocillopora_meandrina_HIv1___Sc0000005: 551
  Pocillopora_meandrina_HIv1___xpSc0001276: 1
  Pocillopora_meandrina_HIv1___Sc0000024: 245
  Pocillopora_meandrina_HIv1___Sc0000065: 2
  Pocillopora_meandrina_HIv1___xpSc0001281: 4
  Pocillopora_meandrina_HIv1___xfSc0000006: 13
  Pocillopora_meandrina_HIv1___Sc0000013: 555
  Pocillopora_meandrina_HIv1___xfSc0001237: 2
  Pocillopora_meandrina_HIv1___xfSc0000612: 2
  Pocillopora_meandrina_HIv1___xfSc0000168: 3
  Pocillopora_meandrina_HIv1___xfSc0000083: 2
  Pocillopora_meandrina_HIv1___Sc0000030: 198
  Pocillopora_meandrina_HIv1___Sc0000002: 725
  Pocillopora_meandrina_HIv1___xfSc0000175: 3
  Pocillopora_meandrina_HIv1___Sc0000054: 1
  Pocillopora_meandrina_HIv1___xfSc0000565: 2
  Pocillopora_meandrina_HIv1___Sc0000044: 51
  Pocillopora_meandrina_HIv1___Sc0000008: 538
  Pocillopora_meandrina_HIv1___Sc0000027: 291
  Pocillopora_meandrina_HIv1___xfSc0000840: 1
  Pocillopora_meandrina_HIv1___xfSc0000596: 2
  Pocillopora_meandrina_HIv1___Sc0000060: 1
  Pocillopora_meandrina_HIv1___xfSc0000763: 1
  Pocillopora_meandrina_HIv1___Sc0000035: 227
  Pocillopora_meandrina_HIv1___xfSc0000968: 2
  Pocillopora_meandrina_HIv1___Sc0000001: 883
  Pocillopora_meandrina_HIv1___xpSc0001355: 1
  Pocillopora_meandrina_HIv1___xfSc0000699: 2
  Pocillopora_meandrina_HIv1___xfSc0000655: 4
  Pocillopora_meandrina_HIv1___xfSc0000426: 7
  Pocillopora_meandrina_HIv1___Sc0000020: 294
  Pocillopora_meandrina_HIv1___xfSc0000621: 2
  Pocillopora_meandrina_HIv1___xfSc0000445: 2
  Pocillopora_meandrina_HIv1___xfSc0000074: 1
  Pocillopora_meandrina_HIv1___Sc0000043: 60
  Pocillopora_meandrina_HIv1___xfSc0000837: 1
  Pocillopora_meandrina_HIv1___xfSc0000288: 5
  Pocillopora_meandrina_HIv1___xfSc0000217: 5
  Pocillopora_meandrina_HIv1___xfSc0000002: 34
  Pocillopora_meandrina_HIv1___Sc0000017: 382
  Pocillopora_meandrina_HIv1___Sc0000019: 291
  Pocillopora_meandrina_HIv1___xfSc0000012: 7
  Pocillopora_meandrina_HIv1___Sc0000029: 178
  Pocillopora_meandrina_HIv1___xfSc0000886: 3
  Pocillopora_meandrina_HIv1___xfSc0000868: 3
  Pocillopora_meandrina_HIv1___xfSc0000824: 8
  Pocillopora_meandrina_HIv1___xfSc0000583: 1
  Pocillopora_meandrina_HIv1___Sc0000071: 2
  Pocillopora_meandrina_HIv1___Sc0000004: 652
  Pocillopora_meandrina_HIv1___xfSc0000704: 4
  Pocillopora_meandrina_HIv1___xfSc0000469: 3
  Pocillopora_meandrina_HIv1___xfSc0000223: 2
  Pocillopora_meandrina_HIv1___xfSc0000195: 1
  Pocillopora_meandrina_HIv1___xfSc0000021: 2
  Pocillopora_meandrina_HIv1___Sc0000023: 213
  Pocillopora_meandrina_HIv1___xfSc0000264: 3
  Pocillopora_meandrina_HIv1___Sc0000010: 615
  Pocillopora_meandrina_HIv1___xfSc0001122: 3
  Pocillopora_meandrina_HIv1___xfSc0000811: 1
  Pocillopora_meandrina_HIv1___Sc0000031: 185
  Pocillopora_meandrina_HIv1___xfSc0000788: 5
  Pocillopora_meandrina_HIv1___xfSc0000875: 1
  Pocillopora_meandrina_HIv1___xfSc0000488: 7
  Pocillopora_meandrina_HIv1___xfSc0000017: 2
  Pocillopora_meandrina_HIv1___Sc0000055: 12
  Pocillopora_meandrina_HIv1___xfSc0001011: 2
  Pocillopora_meandrina_HIv1___xfSc0000092: 6
  Pocillopora_meandrina_HIv1___Sc0000047: 22
  Pocillopora_meandrina_HIv1___xfSc0000482: 4
  Pocillopora_meandrina_HIv1___xfSc0000132: 4
  Pocillopora_meandrina_HIv1___xfSc0000024: 2
  Pocillopora_meandrina_HIv1___Sc0000026: 271
  Pocillopora_meandrina_HIv1___xfSc0000995: 1
  Pocillopora_meandrina_HIv1___Sc0000067: 2
  Pocillopora_meandrina_HIv1___Sc0000038: 93
  Pocillopora_meandrina_HIv1___xfSc0000642: 1
  Pocillopora_meandrina_HIv1___xfSc0000376: 1
  Pocillopora_meandrina_HIv1___Sc0000082: 3
  Pocillopora_meandrina_HIv1___xfSc0000975: 3
  Pocillopora_meandrina_HIv1___xfSc0000760: 2
  Pocillopora_meandrina_HIv1___Sc0000069: 1
  Pocillopora_meandrina_HIv1___Sc0000036: 124
  Pocillopora_meandrina_HIv1___xfSc0000275: 1
  Pocillopora_meandrina_HIv1___Sc0000000: 922
  Pocillopora_meandrina_HIv1___xfSc0000199: 2
  Pocillopora_meandrina_HIv1___xfSc0000058: 1
  Pocillopora_meandrina_HIv1___xfSc0000014: 1
  Pocillopora_meandrina_HIv1___xfSc0000725: 5
  Pocillopora_meandrina_HIv1___Sc0000042: 68
  Pocillopora_meandrina_HIv1___xfSc0000892: 1
  Pocillopora_meandrina_HIv1___xfSc0000001: 67
  Pocillopora_meandrina_HIv1___Sc0000014: 378
  Pocillopora_meandrina_HIv1___xfSc0000262: 3
  Pocillopora_meandrina_HIv1___xfSc0000228: 4
  Pocillopora_meandrina_HIv1___Sc0000051: 13
  Pocillopora_meandrina_HIv1___Sc0000028: 208
  Pocillopora_meandrina_HIv1___xfSc0000885: 5
  Pocillopora_meandrina_HIv1___xfSc0000272: 5
  Pocillopora_meandrina_HIv1___Sc0000007: 643
  Pocillopora_meandrina_HIv1___xfSc0000705: 1
  Pocillopora_meandrina_HIv1___Sc0000022: 231</code></pre>


<!-- -->

</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
    const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
    const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
    let newTheme = '';
    if(darkModeDefault) {
      newTheme = isAlternate ? baseTheme : alternateTheme;
    } else {
      newTheme = isAlternate ? alternateTheme : baseTheme;
    }
    const changeGiscusTheme = () => {
      // From: https://github.com/giscus/giscus/issues/336
      const sendMessage = (message) => {
        const iframe = document.querySelector('iframe.giscus-frame');
        if (!iframe) return;
        iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
      }
      sendMessage({
        setConfig: {
          theme: newTheme
        }
      });
    }
    const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
    if (isGiscussLoaded) {
      changeGiscusTheme();
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  const darkModeDefault = true;
  let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
    toggleGiscusIfUsed(toAlternate, darkModeDefault);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/sr320\.github\.io\/tumbling-oysters");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb90" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb90-2"><a href="#cb90-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "The lncRNA repertoire of Ptuh"</span></span>
<span id="cb90-3"><a href="#cb90-3" aria-hidden="true" tabindex="-1"></a><span class="an">description:</span><span class="co"> "that's hit"</span></span>
<span id="cb90-4"><a href="#cb90-4" aria-hidden="true" tabindex="-1"></a><span class="an">categories:</span><span class="co"> [lncRNA, e5]</span></span>
<span id="cb90-5"><a href="#cb90-5" aria-hidden="true" tabindex="-1"></a><span class="co">#citation: </span></span>
<span id="cb90-6"><a href="#cb90-6" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> 06-21-2025</span></span>
<span id="cb90-7"><a href="#cb90-7" aria-hidden="true" tabindex="-1"></a><span class="an">image:</span><span class="co"> http://gannet.fish.washington.edu/seashell/snaps/Monosnap_deep-dive-expression__RStudio_Server_2025-06-21_17-48-43.png # finding a good image</span></span>
<span id="cb90-8"><a href="#cb90-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-9"><a href="#cb90-9" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span></span>
<span id="cb90-10"><a href="#cb90-10" aria-hidden="true" tabindex="-1"></a><span class="co">  - name: Steven Roberts</span></span>
<span id="cb90-11"><a href="#cb90-11" aria-hidden="true" tabindex="-1"></a><span class="co">    url: </span></span>
<span id="cb90-12"><a href="#cb90-12" aria-hidden="true" tabindex="-1"></a><span class="co">    orcid: 0000-0001-8302-1138</span></span>
<span id="cb90-13"><a href="#cb90-13" aria-hidden="true" tabindex="-1"></a><span class="co">    affiliation: Professor, UW - School of Aquatic and Fishery Sciences</span></span>
<span id="cb90-14"><a href="#cb90-14" aria-hidden="true" tabindex="-1"></a><span class="co">    affiliation-url: https://robertslab.info</span></span>
<span id="cb90-15"><a href="#cb90-15" aria-hidden="true" tabindex="-1"></a><span class="co">  #url:  # self-defined</span></span>
<span id="cb90-16"><a href="#cb90-16" aria-hidden="true" tabindex="-1"></a><span class="an">draft:</span><span class="co"> false # setting this to `true` will prevent your post from appearing on your listing page until you're ready!</span></span>
<span id="cb90-17"><a href="#cb90-17" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb90-18"><a href="#cb90-18" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb90-19"><a href="#cb90-19" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: FALSE</span></span>
<span id="cb90-20"><a href="#cb90-20" aria-hidden="true" tabindex="-1"></a><span class="co">    code-tools: true</span></span>
<span id="cb90-21"><a href="#cb90-21" aria-hidden="true" tabindex="-1"></a><span class="co">    code-copy: true</span></span>
<span id="cb90-22"><a href="#cb90-22" aria-hidden="true" tabindex="-1"></a><span class="co">    highlight-style: github</span></span>
<span id="cb90-23"><a href="#cb90-23" aria-hidden="true" tabindex="-1"></a><span class="co">    code-overflow: wrap</span></span>
<span id="cb90-24"><a href="#cb90-24" aria-hidden="true" tabindex="-1"></a><span class="co">#runtime: shiny</span></span>
<span id="cb90-25"><a href="#cb90-25" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb90-26"><a href="#cb90-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-27"><a href="#cb90-27" aria-hidden="true" tabindex="-1"></a><span class="in">```{r setup, include=FALSE}</span></span>
<span id="cb90-28"><a href="#cb90-28" aria-hidden="true" tabindex="-1"></a><span class="in">library(knitr)</span></span>
<span id="cb90-29"><a href="#cb90-29" aria-hidden="true" tabindex="-1"></a><span class="in">library(readr)</span></span>
<span id="cb90-30"><a href="#cb90-30" aria-hidden="true" tabindex="-1"></a><span class="in">library(DT)</span></span>
<span id="cb90-31"><a href="#cb90-31" aria-hidden="true" tabindex="-1"></a><span class="in">library(shiny)</span></span>
<span id="cb90-32"><a href="#cb90-32" aria-hidden="true" tabindex="-1"></a><span class="in">library(plotly)</span></span>
<span id="cb90-33"><a href="#cb90-33" aria-hidden="true" tabindex="-1"></a><span class="in">if (!requireNamespace("BiocManager", quietly = TRUE))</span></span>
<span id="cb90-34"><a href="#cb90-34" aria-hidden="true" tabindex="-1"></a><span class="in">    install.packages("BiocManager")</span></span>
<span id="cb90-35"><a href="#cb90-35" aria-hidden="true" tabindex="-1"></a><span class="in">BiocManager::install(c("Biostrings", "rtracklayer"))</span></span>
<span id="cb90-36"><a href="#cb90-36" aria-hidden="true" tabindex="-1"></a><span class="in">library(tidyverse)</span></span>
<span id="cb90-37"><a href="#cb90-37" aria-hidden="true" tabindex="-1"></a><span class="in">library(Biostrings)</span></span>
<span id="cb90-38"><a href="#cb90-38" aria-hidden="true" tabindex="-1"></a><span class="in">library(rtracklayer)</span></span>
<span id="cb90-39"><a href="#cb90-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-40"><a href="#cb90-40" aria-hidden="true" tabindex="-1"></a><span class="in">knitr::opts_chunk$set(</span></span>
<span id="cb90-41"><a href="#cb90-41" aria-hidden="true" tabindex="-1"></a><span class="in">  echo = TRUE,         # Display code chunks</span></span>
<span id="cb90-42"><a href="#cb90-42" aria-hidden="true" tabindex="-1"></a><span class="in">  eval = FALSE,         # Evaluate code chunks</span></span>
<span id="cb90-43"><a href="#cb90-43" aria-hidden="true" tabindex="-1"></a><span class="in">  warning = FALSE,     # Hide warnings</span></span>
<span id="cb90-44"><a href="#cb90-44" aria-hidden="true" tabindex="-1"></a><span class="in">  message = FALSE,     # Hide messages</span></span>
<span id="cb90-45"><a href="#cb90-45" aria-hidden="true" tabindex="-1"></a><span class="in">  fig.width = 6,       # Set plot width in inches</span></span>
<span id="cb90-46"><a href="#cb90-46" aria-hidden="true" tabindex="-1"></a><span class="in">  fig.height = 4,      # Set plot height in inches</span></span>
<span id="cb90-47"><a href="#cb90-47" aria-hidden="true" tabindex="-1"></a><span class="in">  fig.align = "center", # Align plots to the center</span></span>
<span id="cb90-48"><a href="#cb90-48" aria-hidden="true" tabindex="-1"></a><span class="in">  comment = ""         # Prevents appending '##' to beginning of lines in code output</span></span>
<span id="cb90-49"><a href="#cb90-49" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb90-50"><a href="#cb90-50" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-51"><a href="#cb90-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-52"><a href="#cb90-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-53"><a href="#cb90-53" aria-hidden="true" tabindex="-1"></a><span class="fu"># TLDR </span></span>
<span id="cb90-54"><a href="#cb90-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-55"><a href="#cb90-55" aria-hidden="true" tabindex="-1"></a>fasta - https://raw.githubusercontent.com/urol-e5/deep-dive-expression/refs/heads/main/F-Ptuh/output/17-Ptuh-lncRNA/Ptuh-lncRNA.fasta</span>
<span id="cb90-56"><a href="#cb90-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-57"><a href="#cb90-57" aria-hidden="true" tabindex="-1"></a>gtf - https://raw.githubusercontent.com/urol-e5/deep-dive-expression/refs/heads/main/F-Ptuh/output/17-Ptuh-lncRNA/Ptuh-lncRNA.gtf</span>
<span id="cb90-58"><a href="#cb90-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-59"><a href="#cb90-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-60"><a href="#cb90-60" aria-hidden="true" tabindex="-1"></a><span class="fu"># file checks </span></span>
<span id="cb90-61"><a href="#cb90-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-62"><a href="#cb90-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-63"><a href="#cb90-63" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval=TRUE}</span></span>
<span id="cb90-64"><a href="#cb90-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-65"><a href="#cb90-65" aria-hidden="true" tabindex="-1"></a><span class="in"># URLs</span></span>
<span id="cb90-66"><a href="#cb90-66" aria-hidden="true" tabindex="-1"></a><span class="in">bed_url &lt;- "https://raw.githubusercontent.com/urol-e5/deep-dive-expression/refs/heads/main/F-Ptuh/output/17-Ptuh-lncRNA/lncRNA.bed"</span></span>
<span id="cb90-67"><a href="#cb90-67" aria-hidden="true" tabindex="-1"></a><span class="in">gtf_url &lt;- "https://raw.githubusercontent.com/urol-e5/deep-dive-expression/refs/heads/main/F-Ptuh/output/17-Ptuh-lncRNA/lncRNA.gtf"</span></span>
<span id="cb90-68"><a href="#cb90-68" aria-hidden="true" tabindex="-1"></a><span class="in">fasta_url &lt;- "https://raw.githubusercontent.com/urol-e5/deep-dive-expression/refs/heads/main/F-Ptuh/output/17-Ptuh-lncRNA/lncRNA.fasta"</span></span>
<span id="cb90-69"><a href="#cb90-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-70"><a href="#cb90-70" aria-hidden="true" tabindex="-1"></a><span class="in">### BED file</span></span>
<span id="cb90-71"><a href="#cb90-71" aria-hidden="true" tabindex="-1"></a><span class="in">bed &lt;- read_tsv(bed_url, col_names = c("chrom", "start", "end"))</span></span>
<span id="cb90-72"><a href="#cb90-72" aria-hidden="true" tabindex="-1"></a><span class="in">cat("BED file head:\n")</span></span>
<span id="cb90-73"><a href="#cb90-73" aria-hidden="true" tabindex="-1"></a><span class="in">print(head(bed))</span></span>
<span id="cb90-74"><a href="#cb90-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-75"><a href="#cb90-75" aria-hidden="true" tabindex="-1"></a><span class="in"># Basic stats for BED</span></span>
<span id="cb90-76"><a href="#cb90-76" aria-hidden="true" tabindex="-1"></a><span class="in">bed$length &lt;- bed$end - bed$start + 1</span></span>
<span id="cb90-77"><a href="#cb90-77" aria-hidden="true" tabindex="-1"></a><span class="in">cat("\nBED file stats:\n")</span></span>
<span id="cb90-78"><a href="#cb90-78" aria-hidden="true" tabindex="-1"></a><span class="in">cat("Number of features:", nrow(bed), "\n")</span></span>
<span id="cb90-79"><a href="#cb90-79" aria-hidden="true" tabindex="-1"></a><span class="in">cat("Mean length:", mean(bed$length), "\n")</span></span>
<span id="cb90-80"><a href="#cb90-80" aria-hidden="true" tabindex="-1"></a><span class="in">cat("Median length:", median(bed$length), "\n")</span></span>
<span id="cb90-81"><a href="#cb90-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-82"><a href="#cb90-82" aria-hidden="true" tabindex="-1"></a><span class="in">### GTF file</span></span>
<span id="cb90-83"><a href="#cb90-83" aria-hidden="true" tabindex="-1"></a><span class="in">gtf &lt;- import(gtf_url, format = "gtf")</span></span>
<span id="cb90-84"><a href="#cb90-84" aria-hidden="true" tabindex="-1"></a><span class="in">cat("\nGTF file head:\n")</span></span>
<span id="cb90-85"><a href="#cb90-85" aria-hidden="true" tabindex="-1"></a><span class="in">print(head(gtf))</span></span>
<span id="cb90-86"><a href="#cb90-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-87"><a href="#cb90-87" aria-hidden="true" tabindex="-1"></a><span class="in"># Basic stats for GTF</span></span>
<span id="cb90-88"><a href="#cb90-88" aria-hidden="true" tabindex="-1"></a><span class="in">cat("\nGTF file stats:\n")</span></span>
<span id="cb90-89"><a href="#cb90-89" aria-hidden="true" tabindex="-1"></a><span class="in">cat("Number of entries:", length(gtf), "\n")</span></span>
<span id="cb90-90"><a href="#cb90-90" aria-hidden="true" tabindex="-1"></a><span class="in">cat("Unique feature types:", paste(unique(gtf$type), collapse = ", "), "\n")</span></span>
<span id="cb90-91"><a href="#cb90-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-92"><a href="#cb90-92" aria-hidden="true" tabindex="-1"></a><span class="in">### FASTA file</span></span>
<span id="cb90-93"><a href="#cb90-93" aria-hidden="true" tabindex="-1"></a><span class="in">fasta &lt;- readDNAStringSet(fasta_url)</span></span>
<span id="cb90-94"><a href="#cb90-94" aria-hidden="true" tabindex="-1"></a><span class="in">cat("\nFASTA file head:\n")</span></span>
<span id="cb90-95"><a href="#cb90-95" aria-hidden="true" tabindex="-1"></a><span class="in">print(head(names(fasta)))</span></span>
<span id="cb90-96"><a href="#cb90-96" aria-hidden="true" tabindex="-1"></a><span class="in">print(head(fasta[[1]]))</span></span>
<span id="cb90-97"><a href="#cb90-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-98"><a href="#cb90-98" aria-hidden="true" tabindex="-1"></a><span class="in"># Basic stats for FASTA</span></span>
<span id="cb90-99"><a href="#cb90-99" aria-hidden="true" tabindex="-1"></a><span class="in">seq_lengths &lt;- width(fasta)</span></span>
<span id="cb90-100"><a href="#cb90-100" aria-hidden="true" tabindex="-1"></a><span class="in">cat("\nFASTA file stats:\n")</span></span>
<span id="cb90-101"><a href="#cb90-101" aria-hidden="true" tabindex="-1"></a><span class="in">cat("Number of sequences:", length(fasta), "\n")</span></span>
<span id="cb90-102"><a href="#cb90-102" aria-hidden="true" tabindex="-1"></a><span class="in">cat("Mean sequence length:", mean(seq_lengths), "\n")</span></span>
<span id="cb90-103"><a href="#cb90-103" aria-hidden="true" tabindex="-1"></a><span class="in">cat("Median sequence length:", median(seq_lengths), "\n")</span></span>
<span id="cb90-104"><a href="#cb90-104" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-105"><a href="#cb90-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-106"><a href="#cb90-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-107"><a href="#cb90-107" aria-hidden="true" tabindex="-1"></a><span class="fu"># Renaming fasta</span></span>
<span id="cb90-108"><a href="#cb90-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-111"><a href="#cb90-111" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb90-112"><a href="#cb90-112" aria-hidden="true" tabindex="-1"></a><span class="co"># File paths</span></span>
<span id="cb90-113"><a href="#cb90-113" aria-hidden="true" tabindex="-1"></a>fasta_url <span class="ot">&lt;-</span> <span class="st">"https://raw.githubusercontent.com/urol-e5/deep-dive-expression/refs/heads/main/F-Ptuh/output/17-Ptuh-lncRNA/lncRNA.fasta"</span></span>
<span id="cb90-114"><a href="#cb90-114" aria-hidden="true" tabindex="-1"></a>gtf_url <span class="ot">&lt;-</span> <span class="st">"https://raw.githubusercontent.com/urol-e5/deep-dive-expression/refs/heads/main/F-Ptuh/output/17-Ptuh-lncRNA/lncRNA.gtf"</span></span>
<span id="cb90-115"><a href="#cb90-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-116"><a href="#cb90-116" aria-hidden="true" tabindex="-1"></a><span class="co"># Read in FASTA and GTF</span></span>
<span id="cb90-117"><a href="#cb90-117" aria-hidden="true" tabindex="-1"></a>fasta <span class="ot">&lt;-</span> <span class="fu">readDNAStringSet</span>(fasta_url)</span>
<span id="cb90-118"><a href="#cb90-118" aria-hidden="true" tabindex="-1"></a>gtf <span class="ot">&lt;-</span> <span class="fu">import</span>(gtf_url, <span class="at">format =</span> <span class="st">"gtf"</span>)</span>
<span id="cb90-119"><a href="#cb90-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-120"><a href="#cb90-120" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract gene_id from GTF</span></span>
<span id="cb90-121"><a href="#cb90-121" aria-hidden="true" tabindex="-1"></a>gene_ids <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">mcols</span>(gtf)<span class="sc">$</span>gene_id, <span class="cf">function</span>(x) <span class="fu">gsub</span>(<span class="st">'"'</span>, <span class="st">''</span>, x))  <span class="co"># remove quotes</span></span>
<span id="cb90-122"><a href="#cb90-122" aria-hidden="true" tabindex="-1"></a>unique_gene_ids <span class="ot">&lt;-</span> <span class="fu">unique</span>(gene_ids)</span>
<span id="cb90-123"><a href="#cb90-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-124"><a href="#cb90-124" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate new FASTA headers</span></span>
<span id="cb90-125"><a href="#cb90-125" aria-hidden="true" tabindex="-1"></a>new_headers <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"Ptuh_"</span>, unique_gene_ids)</span>
<span id="cb90-126"><a href="#cb90-126" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-127"><a href="#cb90-127" aria-hidden="true" tabindex="-1"></a><span class="co"># Confirm lengths match</span></span>
<span id="cb90-128"><a href="#cb90-128" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">length</span>(fasta) <span class="sc">!=</span> <span class="fu">length</span>(new_headers)) {</span>
<span id="cb90-129"><a href="#cb90-129" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stop</span>(<span class="st">"Mismatch between number of FASTA sequences and GTF gene IDs."</span>)</span>
<span id="cb90-130"><a href="#cb90-130" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb90-131"><a href="#cb90-131" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-132"><a href="#cb90-132" aria-hidden="true" tabindex="-1"></a><span class="co"># Rename sequences</span></span>
<span id="cb90-133"><a href="#cb90-133" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(fasta) <span class="ot">&lt;-</span> new_headers</span>
<span id="cb90-134"><a href="#cb90-134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-135"><a href="#cb90-135" aria-hidden="true" tabindex="-1"></a><span class="co"># Write new FASTA</span></span>
<span id="cb90-136"><a href="#cb90-136" aria-hidden="true" tabindex="-1"></a><span class="fu">writeXStringSet</span>(fasta, <span class="at">filepath =</span> <span class="st">"lncRNA_renamed.fasta"</span>, <span class="at">format =</span> <span class="st">"fasta"</span>)</span>
<span id="cb90-137"><a href="#cb90-137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-138"><a href="#cb90-138" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"FASTA headers renamed and saved to lncRNA_renamed.fasta</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb90-139"><a href="#cb90-139" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-142"><a href="#cb90-142" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb90-143"><a href="#cb90-143" aria-hidden="true" tabindex="-1"></a><span class="fu">grep</span> <span class="st">"&gt;"</span> lncRNA_renamed.fasta <span class="kw">|</span> <span class="fu">head</span> </span>
<span id="cb90-144"><a href="#cb90-144" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-145"><a href="#cb90-145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-146"><a href="#cb90-146" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval=TRUE}</span></span>
<span id="cb90-147"><a href="#cb90-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-148"><a href="#cb90-148" aria-hidden="true" tabindex="-1"></a><span class="in"># Read renamed FASTA and original GTF</span></span>
<span id="cb90-149"><a href="#cb90-149" aria-hidden="true" tabindex="-1"></a><span class="in">fasta &lt;- readDNAStringSet("lncRNA_renamed.fasta")</span></span>
<span id="cb90-150"><a href="#cb90-150" aria-hidden="true" tabindex="-1"></a><span class="in">gtf &lt;- import("https://raw.githubusercontent.com/urol-e5/deep-dive-expression/refs/heads/main/F-Ptuh/output/17-Ptuh-lncRNA/lncRNA.gtf", format = "gtf")</span></span>
<span id="cb90-151"><a href="#cb90-151" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-152"><a href="#cb90-152" aria-hidden="true" tabindex="-1"></a><span class="in"># Extract gene IDs from GTF</span></span>
<span id="cb90-153"><a href="#cb90-153" aria-hidden="true" tabindex="-1"></a><span class="in">gtf_gene_ids &lt;- unique(gsub('"', '', mcols(gtf)$gene_id))  # remove quotes</span></span>
<span id="cb90-154"><a href="#cb90-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-155"><a href="#cb90-155" aria-hidden="true" tabindex="-1"></a><span class="in"># Extract IDs from FASTA headers</span></span>
<span id="cb90-156"><a href="#cb90-156" aria-hidden="true" tabindex="-1"></a><span class="in">fasta_headers &lt;- names(fasta)</span></span>
<span id="cb90-157"><a href="#cb90-157" aria-hidden="true" tabindex="-1"></a><span class="in">fasta_gene_ids &lt;- gsub("^Ptuh_", "", fasta_headers)</span></span>
<span id="cb90-158"><a href="#cb90-158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-159"><a href="#cb90-159" aria-hidden="true" tabindex="-1"></a><span class="in"># Compare</span></span>
<span id="cb90-160"><a href="#cb90-160" aria-hidden="true" tabindex="-1"></a><span class="in">all_match &lt;- all(fasta_gene_ids %in% gtf_gene_ids) &amp;&amp; all(gtf_gene_ids %in% fasta_gene_ids)</span></span>
<span id="cb90-161"><a href="#cb90-161" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-162"><a href="#cb90-162" aria-hidden="true" tabindex="-1"></a><span class="in">if (all_match) {</span></span>
<span id="cb90-163"><a href="#cb90-163" aria-hidden="true" tabindex="-1"></a><span class="in">  cat("✅ FASTA headers match gene_ids in GTF exactly.\n")</span></span>
<span id="cb90-164"><a href="#cb90-164" aria-hidden="true" tabindex="-1"></a><span class="in">} else {</span></span>
<span id="cb90-165"><a href="#cb90-165" aria-hidden="true" tabindex="-1"></a><span class="in">  cat("❌ Mismatch found between FASTA headers and GTF gene_ids.\n")</span></span>
<span id="cb90-166"><a href="#cb90-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-167"><a href="#cb90-167" aria-hidden="true" tabindex="-1"></a><span class="in">  # Optional: show what's missing</span></span>
<span id="cb90-168"><a href="#cb90-168" aria-hidden="true" tabindex="-1"></a><span class="in">  cat("\nIn FASTA but not GTF:\n")</span></span>
<span id="cb90-169"><a href="#cb90-169" aria-hidden="true" tabindex="-1"></a><span class="in">  print(setdiff(fasta_gene_ids, gtf_gene_ids))</span></span>
<span id="cb90-170"><a href="#cb90-170" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-171"><a href="#cb90-171" aria-hidden="true" tabindex="-1"></a><span class="in">  cat("\nIn GTF but not FASTA:\n")</span></span>
<span id="cb90-172"><a href="#cb90-172" aria-hidden="true" tabindex="-1"></a><span class="in">  print(setdiff(gtf_gene_ids, fasta_gene_ids))</span></span>
<span id="cb90-173"><a href="#cb90-173" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb90-174"><a href="#cb90-174" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-175"><a href="#cb90-175" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-176"><a href="#cb90-176" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-177"><a href="#cb90-177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-178"><a href="#cb90-178" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval=TRUE}</span></span>
<span id="cb90-179"><a href="#cb90-179" aria-hidden="true" tabindex="-1"></a><span class="in"># Read files</span></span>
<span id="cb90-180"><a href="#cb90-180" aria-hidden="true" tabindex="-1"></a><span class="in">fasta &lt;- readDNAStringSet("lncRNA_renamed.fasta")</span></span>
<span id="cb90-181"><a href="#cb90-181" aria-hidden="true" tabindex="-1"></a><span class="in">gtf &lt;- import("https://raw.githubusercontent.com/urol-e5/deep-dive-expression/refs/heads/main/F-Ptuh/output/17-Ptuh-lncRNA/lncRNA.gtf", format = "gtf")</span></span>
<span id="cb90-182"><a href="#cb90-182" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-183"><a href="#cb90-183" aria-hidden="true" tabindex="-1"></a><span class="in"># Clean gene_id and extract relevant fields</span></span>
<span id="cb90-184"><a href="#cb90-184" aria-hidden="true" tabindex="-1"></a><span class="in">gtf_df &lt;- as.data.frame(gtf)</span></span>
<span id="cb90-185"><a href="#cb90-185" aria-hidden="true" tabindex="-1"></a><span class="in">gtf_df$gene_id &lt;- gsub('"', '', gtf_df$gene_id)</span></span>
<span id="cb90-186"><a href="#cb90-186" aria-hidden="true" tabindex="-1"></a><span class="in">gtf_summary &lt;- gtf_df %&gt;%</span></span>
<span id="cb90-187"><a href="#cb90-187" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::select(seqnames, start, end, gene_id) %&gt;%</span></span>
<span id="cb90-188"><a href="#cb90-188" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::group_by(gene_id) %&gt;%</span></span>
<span id="cb90-189"><a href="#cb90-189" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::summarise(gtf_length = sum(end - start + 1), .groups = "drop")</span></span>
<span id="cb90-190"><a href="#cb90-190" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-191"><a href="#cb90-191" aria-hidden="true" tabindex="-1"></a><span class="in"># Extract FASTA lengths</span></span>
<span id="cb90-192"><a href="#cb90-192" aria-hidden="true" tabindex="-1"></a><span class="in">fasta_df &lt;- data.frame(</span></span>
<span id="cb90-193"><a href="#cb90-193" aria-hidden="true" tabindex="-1"></a><span class="in">  gene_id = gsub("^Ptuh_", "", names(fasta)),</span></span>
<span id="cb90-194"><a href="#cb90-194" aria-hidden="true" tabindex="-1"></a><span class="in">  fasta_length = width(fasta),</span></span>
<span id="cb90-195"><a href="#cb90-195" aria-hidden="true" tabindex="-1"></a><span class="in">  fasta_header = names(fasta)</span></span>
<span id="cb90-196"><a href="#cb90-196" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb90-197"><a href="#cb90-197" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-198"><a href="#cb90-198" aria-hidden="true" tabindex="-1"></a><span class="in"># Join and compare</span></span>
<span id="cb90-199"><a href="#cb90-199" aria-hidden="true" tabindex="-1"></a><span class="in">merged &lt;- dplyr::inner_join(gtf_summary, fasta_df, by = "gene_id")</span></span>
<span id="cb90-200"><a href="#cb90-200" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-201"><a href="#cb90-201" aria-hidden="true" tabindex="-1"></a><span class="in"># Check consistency</span></span>
<span id="cb90-202"><a href="#cb90-202" aria-hidden="true" tabindex="-1"></a><span class="in">merged$length_match &lt;- merged$gtf_length == merged$fasta_length</span></span>
<span id="cb90-203"><a href="#cb90-203" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-204"><a href="#cb90-204" aria-hidden="true" tabindex="-1"></a><span class="in"># Display a few examples</span></span>
<span id="cb90-205"><a href="#cb90-205" aria-hidden="true" tabindex="-1"></a><span class="in">cat("🔍 Sample GTF vs FASTA length comparison:\n")</span></span>
<span id="cb90-206"><a href="#cb90-206" aria-hidden="true" tabindex="-1"></a><span class="in">print(head(merged[, c("fasta_header", "gtf_length", "fasta_length", "length_match")]), row.names = FALSE)</span></span>
<span id="cb90-207"><a href="#cb90-207" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-208"><a href="#cb90-208" aria-hidden="true" tabindex="-1"></a><span class="in"># Overall match rate</span></span>
<span id="cb90-209"><a href="#cb90-209" aria-hidden="true" tabindex="-1"></a><span class="in">match_rate &lt;- mean(merged$length_match)</span></span>
<span id="cb90-210"><a href="#cb90-210" aria-hidden="true" tabindex="-1"></a><span class="in">cat(sprintf("\n✅ Percent of exact matches: %.1f%%\n", 100 * match_rate))</span></span>
<span id="cb90-211"><a href="#cb90-211" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-212"><a href="#cb90-212" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-213"><a href="#cb90-213" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-214"><a href="#cb90-214" aria-hidden="true" tabindex="-1"></a>---</span>
<span id="cb90-215"><a href="#cb90-215" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-216"><a href="#cb90-216" aria-hidden="true" tabindex="-1"></a><span class="fu"># Code</span></span>
<span id="cb90-217"><a href="#cb90-217" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-218"><a href="#cb90-218" aria-hidden="true" tabindex="-1"></a><span class="in">``` bash</span></span>
<span id="cb90-219"><a href="#cb90-219" aria-hidden="true" tabindex="-1"></a><span class="co"># Global R options</span></span>
<span id="cb90-220"><a href="#cb90-220" aria-hidden="true" tabindex="-1"></a><span class="ex">knitr::opts_chunk</span><span class="va">$set</span><span class="er">(</span><span class="bu">echo</span> = TRUE<span class="kw">)</span></span>
<span id="cb90-221"><a href="#cb90-221" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-222"><a href="#cb90-222" aria-hidden="true" tabindex="-1"></a><span class="co"># Define key paths and tool directories</span></span>
<span id="cb90-223"><a href="#cb90-223" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb90-224"><a href="#cb90-224" aria-hidden="true" tabindex="-1"></a><span class="ex">DATA_DIR</span> <span class="op">&lt;</span>- <span class="st">"../data/17-Ptuh-lncRNA"</span></span>
<span id="cb90-225"><a href="#cb90-225" aria-hidden="true" tabindex="-1"></a><span class="ex">OUTPUT_DIR</span> <span class="op">&lt;</span>- <span class="st">"../output/17-Ptuh-lncRNA"</span></span>
<span id="cb90-226"><a href="#cb90-226" aria-hidden="true" tabindex="-1"></a><span class="ex">THREADS</span> <span class="op">&lt;</span>- <span class="st">"24"</span></span>
<span id="cb90-227"><a href="#cb90-227" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb90-228"><a href="#cb90-228" aria-hidden="true" tabindex="-1"></a><span class="ex">FASTQ_SOURCE</span> <span class="op">&lt;</span>- <span class="st">"https://gannet.fish.washington.edu/Atumefaciens/20230519-E5_coral-fastqc-fastp-multiqc-RNAseq/P_meandrina/trimmed/"</span></span>
<span id="cb90-229"><a href="#cb90-229" aria-hidden="true" tabindex="-1"></a><span class="ex">FASTQ_SUFFIX</span> <span class="op">&lt;</span>- <span class="st">"fastq.gz"</span></span>
<span id="cb90-230"><a href="#cb90-230" aria-hidden="true" tabindex="-1"></a><span class="ex">GENOME_SOURCE</span> <span class="op">&lt;</span>- <span class="st">"https://owl.fish.washington.edu/halfshell/genomic-databank/Pocillopora_meandrina_HIv1.assembly.fasta"</span></span>
<span id="cb90-231"><a href="#cb90-231" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-232"><a href="#cb90-232" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-233"><a href="#cb90-233" aria-hidden="true" tabindex="-1"></a><span class="ex">GTF_SOURCE</span> <span class="op">&lt;</span>- <span class="st">"https://raw.githubusercontent.com/urol-e5/timeseries_molecular/d5f546705e3df40558eeaa5c18b122c79d2f4453/F-Ptua/data/Pocillopora_meandrina_HIv1.genes-validated.gtf"</span></span>
<span id="cb90-234"><a href="#cb90-234" aria-hidden="true" tabindex="-1"></a><span class="ex">GFF_SOURCE</span> <span class="op">&lt;</span>- <span class="st">"https://gannet.fish.washington.edu/seashell/bu-github/deep-dive-expression/F-Ptuh/data/Pocillopora_meandrina_HIv1.genes-validated.gff3"</span></span>
<span id="cb90-235"><a href="#cb90-235" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-236"><a href="#cb90-236" aria-hidden="true" tabindex="-1"></a><span class="ex">GFFPATTERN</span> <span class="op">&lt;</span>- <span class="st">'class_code "u"|class_code "x"|class_code "o"|class_code "i"'</span></span>
<span id="cb90-237"><a href="#cb90-237" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-238"><a href="#cb90-238" aria-hidden="true" tabindex="-1"></a><span class="co">#RAVEN</span></span>
<span id="cb90-239"><a href="#cb90-239" aria-hidden="true" tabindex="-1"></a><span class="co"># HISAT2_DIR &lt;- "/home/shared/hisat2-2.2.1/"</span></span>
<span id="cb90-240"><a href="#cb90-240" aria-hidden="true" tabindex="-1"></a><span class="co"># SAMTOOLS_DIR &lt;- "/home/shared/samtools-1.12/"</span></span>
<span id="cb90-241"><a href="#cb90-241" aria-hidden="true" tabindex="-1"></a><span class="co"># STRINGTIE_DIR &lt;- "/home/shared/stringtie-2.2.1.Linux_x86_64"</span></span>
<span id="cb90-242"><a href="#cb90-242" aria-hidden="true" tabindex="-1"></a><span class="co"># GFFCOMPARE_DIR &lt;- "/home/shared/gffcompare-0.12.6.Linux_x86_64"</span></span>
<span id="cb90-243"><a href="#cb90-243" aria-hidden="true" tabindex="-1"></a><span class="co"># BEDTOOLS_DIR &lt;- "/home/shared/bedtools2/bin"</span></span>
<span id="cb90-244"><a href="#cb90-244" aria-hidden="true" tabindex="-1"></a><span class="co"># CPC2_DIR &lt;- "/home/shared/CPC2_standalone-1.0.1"</span></span>
<span id="cb90-245"><a href="#cb90-245" aria-hidden="true" tabindex="-1"></a><span class="co"># CONDA_PATH &lt;- "/opt/anaconda/anaconda3/bin/conda"</span></span>
<span id="cb90-246"><a href="#cb90-246" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-247"><a href="#cb90-247" aria-hidden="true" tabindex="-1"></a><span class="co">#KLONE</span></span>
<span id="cb90-248"><a href="#cb90-248" aria-hidden="true" tabindex="-1"></a><span class="ex">HISAT2_DIR</span> <span class="op">&lt;</span>- <span class="st">""</span></span>
<span id="cb90-249"><a href="#cb90-249" aria-hidden="true" tabindex="-1"></a><span class="ex">SAMTOOLS_DIR</span> <span class="op">&lt;</span>- <span class="st">""</span></span>
<span id="cb90-250"><a href="#cb90-250" aria-hidden="true" tabindex="-1"></a><span class="ex">STRINGTIE_DIR</span> <span class="op">&lt;</span>- <span class="st">""</span></span>
<span id="cb90-251"><a href="#cb90-251" aria-hidden="true" tabindex="-1"></a><span class="ex">GFFCOMPARE_DIR</span> <span class="op">&lt;</span>- <span class="st">"/srlab/programs/gffcompare-0.12.6.Linux_x86_64/"</span></span>
<span id="cb90-252"><a href="#cb90-252" aria-hidden="true" tabindex="-1"></a><span class="ex">BEDTOOLS_DIR</span> <span class="op">&lt;</span>- <span class="st">""</span></span>
<span id="cb90-253"><a href="#cb90-253" aria-hidden="true" tabindex="-1"></a><span class="ex">CPC2_DIR</span> <span class="op">&lt;</span>- <span class="st">"/srlab/programs/CPC2_standalone-1.0.1/bin/"</span></span>
<span id="cb90-254"><a href="#cb90-254" aria-hidden="true" tabindex="-1"></a><span class="ex">CONDA_PATH</span> <span class="op">&lt;</span>- <span class="st">"/mmfs1/gscratch/srlab/nextflow/bin/miniforge/bin/conda"</span></span>
<span id="cb90-255"><a href="#cb90-255" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-256"><a href="#cb90-256" aria-hidden="true" tabindex="-1"></a><span class="ex">GENOME_FASTA</span> <span class="op">&lt;</span>- file.path<span class="er">(</span><span class="ex">DATA_DIR,</span> <span class="st">"genome.fasta"</span><span class="kw">)</span></span>
<span id="cb90-257"><a href="#cb90-257" aria-hidden="true" tabindex="-1"></a><span class="ex">GENOME_GTF</span> <span class="op">&lt;</span>- file.path<span class="er">(</span><span class="ex">DATA_DIR,</span> <span class="st">"genome.gtf"</span><span class="kw">)</span></span>
<span id="cb90-258"><a href="#cb90-258" aria-hidden="true" tabindex="-1"></a><span class="ex">GENOME_GFF</span> <span class="op">&lt;</span>- file.path<span class="er">(</span><span class="ex">DATA_DIR,</span> <span class="st">"genome.gff"</span><span class="kw">)</span></span>
<span id="cb90-259"><a href="#cb90-259" aria-hidden="true" tabindex="-1"></a><span class="ex">FASTQ_DIR</span> <span class="op">&lt;</span>- file.path<span class="er">(</span><span class="ex">DATA_DIR,</span> <span class="st">"fastq"</span><span class="kw">)</span></span>
<span id="cb90-260"><a href="#cb90-260" aria-hidden="true" tabindex="-1"></a><span class="ex">GENOME_INDEX</span> <span class="op">&lt;</span>- file.path<span class="er">(</span><span class="ex">OUTPUT_DIR,</span> <span class="st">"genome.index"</span><span class="kw">)</span></span>
<span id="cb90-261"><a href="#cb90-261" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-262"><a href="#cb90-262" aria-hidden="true" tabindex="-1"></a><span class="co"># Export these as environment variables for bash chunks.</span></span>
<span id="cb90-263"><a href="#cb90-263" aria-hidden="true" tabindex="-1"></a><span class="ex">Sys.setenv</span><span class="er">(</span></span>
<span id="cb90-264"><a href="#cb90-264" aria-hidden="true" tabindex="-1"></a>  <span class="ex">THREADS</span> = THREADS,</span>
<span id="cb90-265"><a href="#cb90-265" aria-hidden="true" tabindex="-1"></a>  <span class="ex">DATA_DIR</span> = DATA_DIR,</span>
<span id="cb90-266"><a href="#cb90-266" aria-hidden="true" tabindex="-1"></a>  <span class="ex">FASTQ_SOURCE</span> = FASTQ_SOURCE,</span>
<span id="cb90-267"><a href="#cb90-267" aria-hidden="true" tabindex="-1"></a>  <span class="ex">FASTQ_SUFFIX</span> = FASTQ_SUFFIX,</span>
<span id="cb90-268"><a href="#cb90-268" aria-hidden="true" tabindex="-1"></a>  <span class="ex">OUTPUT_DIR</span> = OUTPUT_DIR,</span>
<span id="cb90-269"><a href="#cb90-269" aria-hidden="true" tabindex="-1"></a>  <span class="ex">GENOME_SOURCE</span> = GENOME_SOURCE,</span>
<span id="cb90-270"><a href="#cb90-270" aria-hidden="true" tabindex="-1"></a>  <span class="ex">GTF_SOURCE</span> = GTF_SOURCE,</span>
<span id="cb90-271"><a href="#cb90-271" aria-hidden="true" tabindex="-1"></a>  <span class="ex">GFF_SOURCE</span> = GFF_SOURCE,</span>
<span id="cb90-272"><a href="#cb90-272" aria-hidden="true" tabindex="-1"></a>  <span class="ex">HISAT2_DIR</span> = HISAT2_DIR,</span>
<span id="cb90-273"><a href="#cb90-273" aria-hidden="true" tabindex="-1"></a>  <span class="ex">SAMTOOLS_DIR</span> = SAMTOOLS_DIR,</span>
<span id="cb90-274"><a href="#cb90-274" aria-hidden="true" tabindex="-1"></a>  <span class="ex">STRINGTIE_DIR</span> = STRINGTIE_DIR,</span>
<span id="cb90-275"><a href="#cb90-275" aria-hidden="true" tabindex="-1"></a>  <span class="ex">GFFCOMPARE_DIR</span> = GFFCOMPARE_DIR,</span>
<span id="cb90-276"><a href="#cb90-276" aria-hidden="true" tabindex="-1"></a>  <span class="ex">BEDTOOLS_DIR</span> = BEDTOOLS_DIR,</span>
<span id="cb90-277"><a href="#cb90-277" aria-hidden="true" tabindex="-1"></a>  <span class="ex">CPC2_DIR</span> = CPC2_DIR,</span>
<span id="cb90-278"><a href="#cb90-278" aria-hidden="true" tabindex="-1"></a>  <span class="ex">CONDA_PATH</span> = CONDA_PATH,</span>
<span id="cb90-279"><a href="#cb90-279" aria-hidden="true" tabindex="-1"></a>  <span class="ex">GENOME_FASTA</span> = GENOME_FASTA,</span>
<span id="cb90-280"><a href="#cb90-280" aria-hidden="true" tabindex="-1"></a>  <span class="ex">GENOME_GTF</span> = GENOME_GTF,</span>
<span id="cb90-281"><a href="#cb90-281" aria-hidden="true" tabindex="-1"></a>  <span class="ex">GENOME_GFF</span> = GENOME_GFF,</span>
<span id="cb90-282"><a href="#cb90-282" aria-hidden="true" tabindex="-1"></a>  <span class="ex">FASTQ_DIR</span> = FASTQ_DIR,</span>
<span id="cb90-283"><a href="#cb90-283" aria-hidden="true" tabindex="-1"></a>  <span class="ex">GENOME_INDEX</span> = GENOME_INDEX,</span>
<span id="cb90-284"><a href="#cb90-284" aria-hidden="true" tabindex="-1"></a>  <span class="ex">GFFPATTERN</span> = GFFPATTERN</span>
<span id="cb90-285"><a href="#cb90-285" aria-hidden="true" tabindex="-1"></a><span class="kw">)</span></span>
<span id="cb90-286"><a href="#cb90-286" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-287"><a href="#cb90-287" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-288"><a href="#cb90-288" aria-hidden="true" tabindex="-1"></a><span class="in">```{r,engine='bash'}</span></span>
<span id="cb90-289"><a href="#cb90-289" aria-hidden="true" tabindex="-1"></a><span class="in">mkdir -p "${DATA_DIR}"</span></span>
<span id="cb90-290"><a href="#cb90-290" aria-hidden="true" tabindex="-1"></a><span class="in">mkdir -p "${OUTPUT_DIR}"</span></span>
<span id="cb90-291"><a href="#cb90-291" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-292"><a href="#cb90-292" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-293"><a href="#cb90-293" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-294"><a href="#cb90-294" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, engine='bash'}</span></span>
<span id="cb90-295"><a href="#cb90-295" aria-hidden="true" tabindex="-1"></a><span class="in">wget -nv -r \</span></span>
<span id="cb90-296"><a href="#cb90-296" aria-hidden="true" tabindex="-1"></a><span class="in">--no-directories --no-parent \</span></span>
<span id="cb90-297"><a href="#cb90-297" aria-hidden="true" tabindex="-1"></a><span class="in">-P ${FASTQ_DIR} \</span></span>
<span id="cb90-298"><a href="#cb90-298" aria-hidden="true" tabindex="-1"></a><span class="in">-A "*${FASTQ_SUFFIX}" ${FASTQ_SOURCE}</span></span>
<span id="cb90-299"><a href="#cb90-299" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-300"><a href="#cb90-300" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-301"><a href="#cb90-301" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-302"><a href="#cb90-302" aria-hidden="true" tabindex="-1"></a><span class="in">```{r ls, engine='bash'}</span></span>
<span id="cb90-303"><a href="#cb90-303" aria-hidden="true" tabindex="-1"></a><span class="in">ls ${FASTQ_DIR}</span></span>
<span id="cb90-304"><a href="#cb90-304" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-305"><a href="#cb90-305" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-306"><a href="#cb90-306" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-307"><a href="#cb90-307" aria-hidden="true" tabindex="-1"></a><span class="in">RNA-POC-47-S1-TP2_R1_001.fastp-trim.20230519.fastq.gz</span></span>
<span id="cb90-308"><a href="#cb90-308" aria-hidden="true" tabindex="-1"></a><span class="in">RNA-POC-47-S1-TP2_R2_001.fastp-trim.20230519.fastq.gz</span></span>
<span id="cb90-309"><a href="#cb90-309" aria-hidden="true" tabindex="-1"></a><span class="in">RNA-POC-48-S1-TP2_R1_001.fastp-trim.20230519.fastq.gz</span></span>
<span id="cb90-310"><a href="#cb90-310" aria-hidden="true" tabindex="-1"></a><span class="in">RNA-POC-48-S1-TP2_R2_001.fastp-trim.20230519.fastq.gz</span></span>
<span id="cb90-311"><a href="#cb90-311" aria-hidden="true" tabindex="-1"></a><span class="in">RNA-POC-50-S1-TP2_R1_001.fastp-trim.20230519.fastq.gz</span></span>
<span id="cb90-312"><a href="#cb90-312" aria-hidden="true" tabindex="-1"></a><span class="in">RNA-POC-50-S1-TP2_R2_001.fastp-trim.20230519.fastq.gz</span></span>
<span id="cb90-313"><a href="#cb90-313" aria-hidden="true" tabindex="-1"></a><span class="in">RNA-POC-53-S1-TP2_R1_001.fastp-trim.20230519.fastq.gz</span></span>
<span id="cb90-314"><a href="#cb90-314" aria-hidden="true" tabindex="-1"></a><span class="in">RNA-POC-53-S1-TP2_R2_001.fastp-trim.20230519.fastq.gz</span></span>
<span id="cb90-315"><a href="#cb90-315" aria-hidden="true" tabindex="-1"></a><span class="in">RNA-POC-57-S1-TP2_R1_001.fastp-trim.20230519.fastq.gz</span></span>
<span id="cb90-316"><a href="#cb90-316" aria-hidden="true" tabindex="-1"></a><span class="in">RNA-POC-57-S1-TP2_R2_001.fastp-trim.20230519.fastq.gz</span></span>
<span id="cb90-317"><a href="#cb90-317" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-318"><a href="#cb90-318" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-319"><a href="#cb90-319" aria-hidden="true" tabindex="-1"></a><span class="in">```{r fasta, engine='bash'}</span></span>
<span id="cb90-320"><a href="#cb90-320" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-321"><a href="#cb90-321" aria-hidden="true" tabindex="-1"></a><span class="in">curl -o "${GENOME_FASTA}" "${GENOME_SOURCE}"</span></span>
<span id="cb90-322"><a href="#cb90-322" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-323"><a href="#cb90-323" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-324"><a href="#cb90-324" aria-hidden="true" tabindex="-1"></a><span class="in">```{r gtf, engine='bash'}</span></span>
<span id="cb90-325"><a href="#cb90-325" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-326"><a href="#cb90-326" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-327"><a href="#cb90-327" aria-hidden="true" tabindex="-1"></a><span class="in">curl -o "${GENOME_GTF}" "${GTF_SOURCE}"</span></span>
<span id="cb90-328"><a href="#cb90-328" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-329"><a href="#cb90-329" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-330"><a href="#cb90-330" aria-hidden="true" tabindex="-1"></a><span class="in">```{r gff, engine='bash'}</span></span>
<span id="cb90-331"><a href="#cb90-331" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-332"><a href="#cb90-332" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-333"><a href="#cb90-333" aria-hidden="true" tabindex="-1"></a><span class="in">curl -o "${GENOME_GFF}" "${GFF_SOURCE}"</span></span>
<span id="cb90-334"><a href="#cb90-334" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-335"><a href="#cb90-335" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-336"><a href="#cb90-336" aria-hidden="true" tabindex="-1"></a><span class="in">```{r file check, engine='bash'}</span></span>
<span id="cb90-337"><a href="#cb90-337" aria-hidden="true" tabindex="-1"></a><span class="in">output_fasta=$(head -1 "${GENOME_FASTA}")</span></span>
<span id="cb90-338"><a href="#cb90-338" aria-hidden="true" tabindex="-1"></a><span class="in">output_gff=$(head -2 "${GENOME_GFF}")</span></span>
<span id="cb90-339"><a href="#cb90-339" aria-hidden="true" tabindex="-1"></a><span class="in">output_gtf=$(head -1 "${GENOME_GTF}")</span></span>
<span id="cb90-340"><a href="#cb90-340" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-341"><a href="#cb90-341" aria-hidden="true" tabindex="-1"></a><span class="in">if [[ "$output_fasta" == *html* || "$output_gff" == *html* || "$output_gtf" == *html* ]]; then</span></span>
<span id="cb90-342"><a href="#cb90-342" aria-hidden="true" tabindex="-1"></a><span class="in">    echo "FAIL - FFS you downloaded a HTML not and genome feature file!"</span></span>
<span id="cb90-343"><a href="#cb90-343" aria-hidden="true" tabindex="-1"></a><span class="in">else</span></span>
<span id="cb90-344"><a href="#cb90-344" aria-hidden="true" tabindex="-1"></a><span class="in">    echo "$output_fasta"</span></span>
<span id="cb90-345"><a href="#cb90-345" aria-hidden="true" tabindex="-1"></a><span class="in">    echo "$output_gff"</span></span>
<span id="cb90-346"><a href="#cb90-346" aria-hidden="true" tabindex="-1"></a><span class="in">    echo "$output_gtf"</span></span>
<span id="cb90-347"><a href="#cb90-347" aria-hidden="true" tabindex="-1"></a><span class="in">fi</span></span>
<span id="cb90-348"><a href="#cb90-348" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-349"><a href="#cb90-349" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-350"><a href="#cb90-350" aria-hidden="true" tabindex="-1"></a><span class="fu"># HISAT</span></span>
<span id="cb90-351"><a href="#cb90-351" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-352"><a href="#cb90-352" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, engine='bash'}</span></span>
<span id="cb90-353"><a href="#cb90-353" aria-hidden="true" tabindex="-1"></a><span class="in">"${HISAT2_DIR}hisat2_extract_exons.py" "${GENOME_GTF}" &gt; "${OUTPUT_DIR}/exon.txt"</span></span>
<span id="cb90-354"><a href="#cb90-354" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-355"><a href="#cb90-355" aria-hidden="true" tabindex="-1"></a><span class="in">"${HISAT2_DIR}hisat2_extract_splice_sites.py" "${GENOME_GTF}" &gt; "${OUTPUT_DIR}/splice_sites.txt"</span></span>
<span id="cb90-356"><a href="#cb90-356" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-357"><a href="#cb90-357" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-358"><a href="#cb90-358" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-359"><a href="#cb90-359" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, engine='bash'}</span></span>
<span id="cb90-360"><a href="#cb90-360" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-361"><a href="#cb90-361" aria-hidden="true" tabindex="-1"></a><span class="in">"${HISAT2_DIR}hisat2-build" \</span></span>
<span id="cb90-362"><a href="#cb90-362" aria-hidden="true" tabindex="-1"></a><span class="in">  -p "${THREADS}" \</span></span>
<span id="cb90-363"><a href="#cb90-363" aria-hidden="true" tabindex="-1"></a><span class="in">  "${GENOME_FASTA}" \</span></span>
<span id="cb90-364"><a href="#cb90-364" aria-hidden="true" tabindex="-1"></a><span class="in">  "${GENOME_INDEX}" \</span></span>
<span id="cb90-365"><a href="#cb90-365" aria-hidden="true" tabindex="-1"></a><span class="in">  --exon "${OUTPUT_DIR}/exon.txt" \</span></span>
<span id="cb90-366"><a href="#cb90-366" aria-hidden="true" tabindex="-1"></a><span class="in">  --ss "${OUTPUT_DIR}/splice_sites.txt" \</span></span>
<span id="cb90-367"><a href="#cb90-367" aria-hidden="true" tabindex="-1"></a><span class="in">  2&gt; "${OUTPUT_DIR}/hisat2-build_stats.txt"</span></span>
<span id="cb90-368"><a href="#cb90-368" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-369"><a href="#cb90-369" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-370"><a href="#cb90-370" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-373"><a href="#cb90-373" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb90-374"><a href="#cb90-374" aria-hidden="true" tabindex="-1"></a><span class="co"># Loop over every file ending in .fastq.gz that contains "_R2_"</span></span>
<span id="cb90-375"><a href="#cb90-375" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> r2 <span class="kw">in</span> <span class="st">"</span><span class="va">${FASTQ_DIR}</span><span class="st">"</span>/<span class="pp">*</span>_R2_<span class="pp">*</span>.<span class="st">"</span><span class="va">${FASTQ_SUFFIX}</span><span class="st">"</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb90-376"><a href="#cb90-376" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Get the basename (filename without path)</span></span>
<span id="cb90-377"><a href="#cb90-377" aria-hidden="true" tabindex="-1"></a>    <span class="va">base</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="st">"</span><span class="va">$r2</span><span class="st">"</span><span class="va">)</span></span>
<span id="cb90-378"><a href="#cb90-378" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb90-379"><a href="#cb90-379" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Derive a sample name by taking everything before "_R2_"</span></span>
<span id="cb90-380"><a href="#cb90-380" aria-hidden="true" tabindex="-1"></a>    <span class="va">sample</span><span class="op">=</span><span class="st">"</span><span class="va">${base</span><span class="op">%%</span>_R2_<span class="pp">*</span><span class="va">}</span><span class="st">"</span></span>
<span id="cb90-381"><a href="#cb90-381" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb90-382"><a href="#cb90-382" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Construct the corresponding R1 filename by replacing "_R2_" with "_R1_"</span></span>
<span id="cb90-383"><a href="#cb90-383" aria-hidden="true" tabindex="-1"></a>    <span class="va">r1</span><span class="op">=</span><span class="st">"</span><span class="va">${r2</span><span class="op">/</span><span class="ss">_R2_</span><span class="op">/</span>_R1_<span class="va">}</span><span class="st">"</span></span>
<span id="cb90-384"><a href="#cb90-384" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb90-385"><a href="#cb90-385" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Define the output SAM file name using the sample name</span></span>
<span id="cb90-386"><a href="#cb90-386" aria-hidden="true" tabindex="-1"></a>    <span class="va">output</span><span class="op">=</span><span class="st">"</span><span class="va">${OUTPUT_DIR}</span><span class="st">/</span><span class="va">${sample}</span><span class="st">.sam"</span></span>
<span id="cb90-387"><a href="#cb90-387" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb90-388"><a href="#cb90-388" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Run hisat2 with the paired-end files</span></span>
<span id="cb90-389"><a href="#cb90-389" aria-hidden="true" tabindex="-1"></a>    <span class="st">"</span><span class="va">${HISAT2_DIR}</span><span class="st">hisat2"</span> <span class="dt">\</span></span>
<span id="cb90-390"><a href="#cb90-390" aria-hidden="true" tabindex="-1"></a>      <span class="at">-x</span> <span class="st">"</span><span class="va">${GENOME_INDEX}</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb90-391"><a href="#cb90-391" aria-hidden="true" tabindex="-1"></a>      <span class="at">-p</span> <span class="st">"</span><span class="va">${THREADS}</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb90-392"><a href="#cb90-392" aria-hidden="true" tabindex="-1"></a>      <span class="at">-1</span> <span class="st">"</span><span class="va">$r1</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb90-393"><a href="#cb90-393" aria-hidden="true" tabindex="-1"></a>      <span class="at">-2</span> <span class="st">"</span><span class="va">$r2</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb90-394"><a href="#cb90-394" aria-hidden="true" tabindex="-1"></a>      <span class="at">-S</span> <span class="st">"</span><span class="va">$output</span><span class="st">"</span></span>
<span id="cb90-395"><a href="#cb90-395" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb90-396"><a href="#cb90-396" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-397"><a href="#cb90-397" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-398"><a href="#cb90-398" aria-hidden="true" tabindex="-1"></a><span class="fu">## convert SAM to BAM</span></span>
<span id="cb90-399"><a href="#cb90-399" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-400"><a href="#cb90-400" aria-hidden="true" tabindex="-1"></a><span class="in">```{r convertSAM, engine='bash'}</span></span>
<span id="cb90-401"><a href="#cb90-401" aria-hidden="true" tabindex="-1"></a><span class="in">for samfile in "${OUTPUT_DIR}/${sample}"*.sam; do</span></span>
<span id="cb90-402"><a href="#cb90-402" aria-hidden="true" tabindex="-1"></a><span class="in">  bamfile="${samfile%.sam}.bam"</span></span>
<span id="cb90-403"><a href="#cb90-403" aria-hidden="true" tabindex="-1"></a><span class="in">  sorted_bamfile="${samfile%.sam}.sorted.bam"</span></span>
<span id="cb90-404"><a href="#cb90-404" aria-hidden="true" tabindex="-1"></a><span class="in">  </span></span>
<span id="cb90-405"><a href="#cb90-405" aria-hidden="true" tabindex="-1"></a><span class="in">  # Convert SAM to BAM</span></span>
<span id="cb90-406"><a href="#cb90-406" aria-hidden="true" tabindex="-1"></a><span class="in">  "${SAMTOOLS_DIR}samtools" view -bS -@ "${THREADS}" "$samfile" &gt; "$bamfile"</span></span>
<span id="cb90-407"><a href="#cb90-407" aria-hidden="true" tabindex="-1"></a><span class="in">  </span></span>
<span id="cb90-408"><a href="#cb90-408" aria-hidden="true" tabindex="-1"></a><span class="in">  # Sort BAM</span></span>
<span id="cb90-409"><a href="#cb90-409" aria-hidden="true" tabindex="-1"></a><span class="in">  "${SAMTOOLS_DIR}samtools" sort -@ "${THREADS}" "$bamfile" -o "$sorted_bamfile"</span></span>
<span id="cb90-410"><a href="#cb90-410" aria-hidden="true" tabindex="-1"></a><span class="in">  </span></span>
<span id="cb90-411"><a href="#cb90-411" aria-hidden="true" tabindex="-1"></a><span class="in">  # Index sorted BAM</span></span>
<span id="cb90-412"><a href="#cb90-412" aria-hidden="true" tabindex="-1"></a><span class="in">  "${SAMTOOLS_DIR}samtools" index -@ "${THREADS}" "$sorted_bamfile"</span></span>
<span id="cb90-413"><a href="#cb90-413" aria-hidden="true" tabindex="-1"></a><span class="in">done</span></span>
<span id="cb90-414"><a href="#cb90-414" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-415"><a href="#cb90-415" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-416"><a href="#cb90-416" aria-hidden="true" tabindex="-1"></a><span class="fu"># StringTie</span></span>
<span id="cb90-417"><a href="#cb90-417" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-418"><a href="#cb90-418" aria-hidden="true" tabindex="-1"></a>StringTie uses the sorted BAM files to assemble transcripts for each sample, outputting them as GTF (Gene Transfer Format) files. And then merges all individual GTF assemblies into a single merged GTF file. This step extracts transcript information and merges GTFs from all samples–an important step in creating a canonical list of lncRNAs across all samples included in the pipeline.</span>
<span id="cb90-419"><a href="#cb90-419" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-420"><a href="#cb90-420" aria-hidden="true" tabindex="-1"></a><span class="in">```{r stringtie, engine='bash'}</span></span>
<span id="cb90-421"><a href="#cb90-421" aria-hidden="true" tabindex="-1"></a><span class="in">find "${OUTPUT_DIR}" -name "*sorted.bam" \</span></span>
<span id="cb90-422"><a href="#cb90-422" aria-hidden="true" tabindex="-1"></a><span class="in">| xargs -n 1 basename -s .sorted.bam | xargs -I{} \</span></span>
<span id="cb90-423"><a href="#cb90-423" aria-hidden="true" tabindex="-1"></a><span class="in">"${STRINGTIE_DIR}stringtie" \</span></span>
<span id="cb90-424"><a href="#cb90-424" aria-hidden="true" tabindex="-1"></a><span class="in">-p "${THREADS}" \</span></span>
<span id="cb90-425"><a href="#cb90-425" aria-hidden="true" tabindex="-1"></a><span class="in">-G "${GENOME_GFF}" \</span></span>
<span id="cb90-426"><a href="#cb90-426" aria-hidden="true" tabindex="-1"></a><span class="in">-o "${OUTPUT_DIR}/{}.gtf" \</span></span>
<span id="cb90-427"><a href="#cb90-427" aria-hidden="true" tabindex="-1"></a><span class="in">"${OUTPUT_DIR}/{}.sorted.bam"</span></span>
<span id="cb90-428"><a href="#cb90-428" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-429"><a href="#cb90-429" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-430"><a href="#cb90-430" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, engine='bash'}</span></span>
<span id="cb90-431"><a href="#cb90-431" aria-hidden="true" tabindex="-1"></a><span class="in">head ${OUTPUT_DIR}/*.gtf</span></span>
<span id="cb90-432"><a href="#cb90-432" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-433"><a href="#cb90-433" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-434"><a href="#cb90-434" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, engine='bash'}</span></span>
<span id="cb90-435"><a href="#cb90-435" aria-hidden="true" tabindex="-1"></a><span class="in">wc -l ${OUTPUT_DIR}/*.gtf</span></span>
<span id="cb90-436"><a href="#cb90-436" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-437"><a href="#cb90-437" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-438"><a href="#cb90-438" aria-hidden="true" tabindex="-1"></a><span class="in">```{r merge, engine='bash'}</span></span>
<span id="cb90-439"><a href="#cb90-439" aria-hidden="true" tabindex="-1"></a><span class="in">"${STRINGTIE_DIR}stringtie" \</span></span>
<span id="cb90-440"><a href="#cb90-440" aria-hidden="true" tabindex="-1"></a><span class="in">--merge \</span></span>
<span id="cb90-441"><a href="#cb90-441" aria-hidden="true" tabindex="-1"></a><span class="in">-G "${GENOME_GFF}" \</span></span>
<span id="cb90-442"><a href="#cb90-442" aria-hidden="true" tabindex="-1"></a><span class="in">-o "${OUTPUT_DIR}/stringtie_merged.gtf" \</span></span>
<span id="cb90-443"><a href="#cb90-443" aria-hidden="true" tabindex="-1"></a><span class="in">"${OUTPUT_DIR}/"*.gtf</span></span>
<span id="cb90-444"><a href="#cb90-444" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-445"><a href="#cb90-445" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-446"><a href="#cb90-446" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, engine='bash'}</span></span>
<span id="cb90-447"><a href="#cb90-447" aria-hidden="true" tabindex="-1"></a><span class="in">wc -l ${OUTPUT_DIR}/stringtie_merged.gtf</span></span>
<span id="cb90-448"><a href="#cb90-448" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-449"><a href="#cb90-449" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-450"><a href="#cb90-450" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, engine='bash'}</span></span>
<span id="cb90-451"><a href="#cb90-451" aria-hidden="true" tabindex="-1"></a><span class="in">head ${OUTPUT_DIR}/stringtie_merged.gtf</span></span>
<span id="cb90-452"><a href="#cb90-452" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-453"><a href="#cb90-453" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-454"><a href="#cb90-454" aria-hidden="true" tabindex="-1"></a>#GFFCOMPARE</span>
<span id="cb90-455"><a href="#cb90-455" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-456"><a href="#cb90-456" aria-hidden="true" tabindex="-1"></a><span class="al">![](http://gannet.fish.washington.edu/seashell/snaps/Monosnap_687474703a2f2f67616e6e65742e666973682e77617368696e67746f6e2e6564752f7365617368656c6c2f736e6170732f323032332d31312d30335f30392d3_2024-12-20_04-02-37.png)</span></span>
<span id="cb90-457"><a href="#cb90-457" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-458"><a href="#cb90-458" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-459"><a href="#cb90-459" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-460"><a href="#cb90-460" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-461"><a href="#cb90-461" aria-hidden="true" tabindex="-1"></a><span class="in">```{r gffcomp, engine='bash'}</span></span>
<span id="cb90-462"><a href="#cb90-462" aria-hidden="true" tabindex="-1"></a><span class="in">"${GFFCOMPARE_DIR}gffcompare" \</span></span>
<span id="cb90-463"><a href="#cb90-463" aria-hidden="true" tabindex="-1"></a><span class="in">-r "${GENOME_GFF}" \</span></span>
<span id="cb90-464"><a href="#cb90-464" aria-hidden="true" tabindex="-1"></a><span class="in">-o "${OUTPUT_DIR}/gffcompare_merged" \</span></span>
<span id="cb90-465"><a href="#cb90-465" aria-hidden="true" tabindex="-1"></a><span class="in">"${OUTPUT_DIR}/stringtie_merged.gtf"</span></span>
<span id="cb90-466"><a href="#cb90-466" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-467"><a href="#cb90-467" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-468"><a href="#cb90-468" aria-hidden="true" tabindex="-1"></a><span class="in">```{r viewgff, engine='bash'}</span></span>
<span id="cb90-469"><a href="#cb90-469" aria-hidden="true" tabindex="-1"></a><span class="in">head -4 "${OUTPUT_DIR}"/gffcompare_merged*</span></span>
<span id="cb90-470"><a href="#cb90-470" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-471"><a href="#cb90-471" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-472"><a href="#cb90-472" aria-hidden="true" tabindex="-1"></a><span class="in">```{r viewgf, engine='bash'}</span></span>
<span id="cb90-473"><a href="#cb90-473" aria-hidden="true" tabindex="-1"></a><span class="in">wc -l "${OUTPUT_DIR}"/gffcompare_merged*</span></span>
<span id="cb90-474"><a href="#cb90-474" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-475"><a href="#cb90-475" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-476"><a href="#cb90-476" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-477"><a href="#cb90-477" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, engine='bash'}</span></span>
<span id="cb90-478"><a href="#cb90-478" aria-hidden="true" tabindex="-1"></a><span class="in">echo "${GFFPATTERN}"</span></span>
<span id="cb90-479"><a href="#cb90-479" aria-hidden="true" tabindex="-1"></a><span class="in">echo "${OUTPUT_DIR}"</span></span>
<span id="cb90-480"><a href="#cb90-480" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-481"><a href="#cb90-481" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-482"><a href="#cb90-482" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-483"><a href="#cb90-483" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-484"><a href="#cb90-484" aria-hidden="true" tabindex="-1"></a><span class="in">```{r gff filter, engine='bash'}</span></span>
<span id="cb90-485"><a href="#cb90-485" aria-hidden="true" tabindex="-1"></a><span class="in">awk '$3 == "transcript" &amp;&amp; $1 !~ /^#/' "${OUTPUT_DIR}/gffcompare_merged.annotated.gtf" | \</span></span>
<span id="cb90-486"><a href="#cb90-486" aria-hidden="true" tabindex="-1"></a><span class="in">grep -E "${GFFPATTERN}" | \</span></span>
<span id="cb90-487"><a href="#cb90-487" aria-hidden="true" tabindex="-1"></a><span class="in">awk '($5 - $4 &gt; 199) || ($4 - $5 &gt; 199)' &gt; "${OUTPUT_DIR}/lncRNA_candidates.gtf"</span></span>
<span id="cb90-488"><a href="#cb90-488" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-489"><a href="#cb90-489" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-490"><a href="#cb90-490" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-491"><a href="#cb90-491" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, engine='bash'}</span></span>
<span id="cb90-492"><a href="#cb90-492" aria-hidden="true" tabindex="-1"></a><span class="in">head ${OUTPUT_DIR}/lncRNA_candidates.gtf</span></span>
<span id="cb90-493"><a href="#cb90-493" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-494"><a href="#cb90-494" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-495"><a href="#cb90-495" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, engine='bash'}</span></span>
<span id="cb90-496"><a href="#cb90-496" aria-hidden="true" tabindex="-1"></a><span class="in">wc -l ${OUTPUT_DIR}/lncRNA_candidates.gtf</span></span>
<span id="cb90-497"><a href="#cb90-497" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-498"><a href="#cb90-498" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-499"><a href="#cb90-499" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, engine='bash'}</span></span>
<span id="cb90-500"><a href="#cb90-500" aria-hidden="true" tabindex="-1"></a><span class="in">sort ../output/17-Ptuh-lncRNA/lncRNA_candidates.gtf | uniq -d</span></span>
<span id="cb90-501"><a href="#cb90-501" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-502"><a href="#cb90-502" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-503"><a href="#cb90-503" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-504"><a href="#cb90-504" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-505"><a href="#cb90-505" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-506"><a href="#cb90-506" aria-hidden="true" tabindex="-1"></a><span class="fu"># Bedtools</span></span>
<span id="cb90-507"><a href="#cb90-507" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-508"><a href="#cb90-508" aria-hidden="true" tabindex="-1"></a><span class="in">```{r fasta2, engine='bash'}</span></span>
<span id="cb90-509"><a href="#cb90-509" aria-hidden="true" tabindex="-1"></a><span class="in">"${BEDTOOLS_DIR}"bedtools getfasta \</span></span>
<span id="cb90-510"><a href="#cb90-510" aria-hidden="true" tabindex="-1"></a><span class="in">-fi "${GENOME_FASTA}" \</span></span>
<span id="cb90-511"><a href="#cb90-511" aria-hidden="true" tabindex="-1"></a><span class="in">-bed "${OUTPUT_DIR}/lncRNA_candidates.gtf" \</span></span>
<span id="cb90-512"><a href="#cb90-512" aria-hidden="true" tabindex="-1"></a><span class="in">-fo "${OUTPUT_DIR}/lncRNA_candidates.fasta" \</span></span>
<span id="cb90-513"><a href="#cb90-513" aria-hidden="true" tabindex="-1"></a><span class="in">-name -split</span></span>
<span id="cb90-514"><a href="#cb90-514" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-515"><a href="#cb90-515" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-516"><a href="#cb90-516" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, engine='bash'}</span></span>
<span id="cb90-517"><a href="#cb90-517" aria-hidden="true" tabindex="-1"></a><span class="in">head ${OUTPUT_DIR}/lncRNA_candidates.fasta</span></span>
<span id="cb90-518"><a href="#cb90-518" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-519"><a href="#cb90-519" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-520"><a href="#cb90-520" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-521"><a href="#cb90-521" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-522"><a href="#cb90-522" aria-hidden="true" tabindex="-1"></a>#CPC2</span>
<span id="cb90-523"><a href="#cb90-523" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-526"><a href="#cb90-526" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb90-527"><a href="#cb90-527" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> https://github.com/gao-lab/CPC2_standalone/archive/refs/tags/v1.0.1.zip</span>
<span id="cb90-528"><a href="#cb90-528" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-531"><a href="#cb90-531" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb90-532"><a href="#cb90-532" aria-hidden="true" tabindex="-1"></a><span class="fu">unzip</span> v1.0.1.zip</span>
<span id="cb90-533"><a href="#cb90-533" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-534"><a href="#cb90-534" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-535"><a href="#cb90-535" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-536"><a href="#cb90-536" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-537"><a href="#cb90-537" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-538"><a href="#cb90-538" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-539"><a href="#cb90-539" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, engine='bash'}</span></span>
<span id="cb90-540"><a href="#cb90-540" aria-hidden="true" tabindex="-1"></a><span class="in">eval "$(/mmfs1/gscratch/srlab/nextflow/bin/miniforge/bin/conda shell.bash hook)"</span></span>
<span id="cb90-541"><a href="#cb90-541" aria-hidden="true" tabindex="-1"></a><span class="in">conda activate /mmfs1/gscratch/srlab/nextflow/bin/miniforge/envs/nextflow  # replace with your actual env name</span></span>
<span id="cb90-542"><a href="#cb90-542" aria-hidden="true" tabindex="-1"></a><span class="in">python /mmfs1/gscratch/scrubbed/sr320/github/deep-dive-expression/F-Ptuh/code/CPC2_standalone-1.0.1/bin/CPC2.py \</span></span>
<span id="cb90-543"><a href="#cb90-543" aria-hidden="true" tabindex="-1"></a><span class="in">  -i "${OUTPUT_DIR}/lncRNA_candidates.fasta" \</span></span>
<span id="cb90-544"><a href="#cb90-544" aria-hidden="true" tabindex="-1"></a><span class="in">  -o "${OUTPUT_DIR}/CPC2"</span></span>
<span id="cb90-545"><a href="#cb90-545" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-546"><a href="#cb90-546" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-547"><a href="#cb90-547" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-548"><a href="#cb90-548" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-549"><a href="#cb90-549" aria-hidden="true" tabindex="-1"></a>Filter</span>
<span id="cb90-550"><a href="#cb90-550" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-551"><a href="#cb90-551" aria-hidden="true" tabindex="-1"></a><span class="in">```{r filCPC, engine='bash'}</span></span>
<span id="cb90-552"><a href="#cb90-552" aria-hidden="true" tabindex="-1"></a><span class="in">awk '$8 == "noncoding" {print $1}' "${OUTPUT_DIR}/CPC2.txt" &gt; "${OUTPUT_DIR}/noncoding_transcripts_ids.txt"</span></span>
<span id="cb90-553"><a href="#cb90-553" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-554"><a href="#cb90-554" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-555"><a href="#cb90-555" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, engine='bash'}</span></span>
<span id="cb90-556"><a href="#cb90-556" aria-hidden="true" tabindex="-1"></a><span class="in">head "${OUTPUT_DIR}/CPC2.txt"</span></span>
<span id="cb90-557"><a href="#cb90-557" aria-hidden="true" tabindex="-1"></a><span class="in">wc -l "${OUTPUT_DIR}/CPC2.txt"</span></span>
<span id="cb90-558"><a href="#cb90-558" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-559"><a href="#cb90-559" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-560"><a href="#cb90-560" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, engine='bash'}</span></span>
<span id="cb90-561"><a href="#cb90-561" aria-hidden="true" tabindex="-1"></a><span class="in">head "${OUTPUT_DIR}/noncoding_transcripts_ids.txt"</span></span>
<span id="cb90-562"><a href="#cb90-562" aria-hidden="true" tabindex="-1"></a><span class="in">wc -l "${OUTPUT_DIR}/noncoding_transcripts_ids.txt"</span></span>
<span id="cb90-563"><a href="#cb90-563" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-564"><a href="#cb90-564" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-565"><a href="#cb90-565" aria-hidden="true" tabindex="-1"></a>Subsetting fasta</span>
<span id="cb90-566"><a href="#cb90-566" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-567"><a href="#cb90-567" aria-hidden="true" tabindex="-1"></a><span class="in">```{r ssfasta, engine='bash'}</span></span>
<span id="cb90-568"><a href="#cb90-568" aria-hidden="true" tabindex="-1"></a><span class="in">"${SAMTOOLS_DIR}samtools" faidx "${OUTPUT_DIR}/lncRNA_candidates.fasta" \</span></span>
<span id="cb90-569"><a href="#cb90-569" aria-hidden="true" tabindex="-1"></a><span class="in">-r "${OUTPUT_DIR}/noncoding_transcripts_ids.txt" \</span></span>
<span id="cb90-570"><a href="#cb90-570" aria-hidden="true" tabindex="-1"></a><span class="in">&gt; "${OUTPUT_DIR}/lncRNA.fasta"</span></span>
<span id="cb90-571"><a href="#cb90-571" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-572"><a href="#cb90-572" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-573"><a href="#cb90-573" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, engine='bash'}</span></span>
<span id="cb90-574"><a href="#cb90-574" aria-hidden="true" tabindex="-1"></a><span class="in">head -2 "${OUTPUT_DIR}/lncRNA.fasta"</span></span>
<span id="cb90-575"><a href="#cb90-575" aria-hidden="true" tabindex="-1"></a><span class="in">grep -c "&gt;" "${OUTPUT_DIR}/lncRNA.fasta"</span></span>
<span id="cb90-576"><a href="#cb90-576" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-577"><a href="#cb90-577" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-578"><a href="#cb90-578" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-579"><a href="#cb90-579" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-580"><a href="#cb90-580" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-581"><a href="#cb90-581" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-582"><a href="#cb90-582" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-583"><a href="#cb90-583" aria-hidden="true" tabindex="-1"></a><span class="in">```{r lncRNAgtf, engine='bash'}</span></span>
<span id="cb90-584"><a href="#cb90-584" aria-hidden="true" tabindex="-1"></a><span class="in"># Define input and output file paths using the OUTPUT_DIR variable</span></span>
<span id="cb90-585"><a href="#cb90-585" aria-hidden="true" tabindex="-1"></a><span class="in">input="${OUTPUT_DIR}/noncoding_transcripts_ids.txt"</span></span>
<span id="cb90-586"><a href="#cb90-586" aria-hidden="true" tabindex="-1"></a><span class="in">output="${OUTPUT_DIR}/lncRNA.bed"</span></span>
<span id="cb90-587"><a href="#cb90-587" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-588"><a href="#cb90-588" aria-hidden="true" tabindex="-1"></a><span class="in"># Process each line of the input file</span></span>
<span id="cb90-589"><a href="#cb90-589" aria-hidden="true" tabindex="-1"></a><span class="in">while IFS= read -r line; do</span></span>
<span id="cb90-590"><a href="#cb90-590" aria-hidden="true" tabindex="-1"></a><span class="in">    # Remove "transcript::" from the line</span></span>
<span id="cb90-591"><a href="#cb90-591" aria-hidden="true" tabindex="-1"></a><span class="in">    line="${line//transcript::/}"</span></span>
<span id="cb90-592"><a href="#cb90-592" aria-hidden="true" tabindex="-1"></a><span class="in">    </span></span>
<span id="cb90-593"><a href="#cb90-593" aria-hidden="true" tabindex="-1"></a><span class="in">    # Split the line by ':' to get the chromosome and position string</span></span>
<span id="cb90-594"><a href="#cb90-594" aria-hidden="true" tabindex="-1"></a><span class="in">    IFS=':' read -r chromosome pos &lt;&lt;&lt; "$line"</span></span>
<span id="cb90-595"><a href="#cb90-595" aria-hidden="true" tabindex="-1"></a><span class="in">    </span></span>
<span id="cb90-596"><a href="#cb90-596" aria-hidden="true" tabindex="-1"></a><span class="in">    # Split the position string by '-' to separate start and end positions</span></span>
<span id="cb90-597"><a href="#cb90-597" aria-hidden="true" tabindex="-1"></a><span class="in">    IFS='-' read -r start end &lt;&lt;&lt; "$pos"</span></span>
<span id="cb90-598"><a href="#cb90-598" aria-hidden="true" tabindex="-1"></a><span class="in">    </span></span>
<span id="cb90-599"><a href="#cb90-599" aria-hidden="true" tabindex="-1"></a><span class="in">    # Convert the start position to 0-based by subtracting 1</span></span>
<span id="cb90-600"><a href="#cb90-600" aria-hidden="true" tabindex="-1"></a><span class="in">    start=$((start - 1))</span></span>
<span id="cb90-601"><a href="#cb90-601" aria-hidden="true" tabindex="-1"></a><span class="in">    </span></span>
<span id="cb90-602"><a href="#cb90-602" aria-hidden="true" tabindex="-1"></a><span class="in">    # Write the chromosome, updated start, and end positions to the output file (tab-separated)</span></span>
<span id="cb90-603"><a href="#cb90-603" aria-hidden="true" tabindex="-1"></a><span class="in">    printf "%s\t%s\t%s\n" "$chromosome" "$start" "$end"</span></span>
<span id="cb90-604"><a href="#cb90-604" aria-hidden="true" tabindex="-1"></a><span class="in">done &lt; "$input" &gt; "$output"</span></span>
<span id="cb90-605"><a href="#cb90-605" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-606"><a href="#cb90-606" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-607"><a href="#cb90-607" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, engine='bash'}</span></span>
<span id="cb90-608"><a href="#cb90-608" aria-hidden="true" tabindex="-1"></a><span class="in">head -1 "${OUTPUT_DIR}/lncRNA.bed"</span></span>
<span id="cb90-609"><a href="#cb90-609" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-610"><a href="#cb90-610" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-611"><a href="#cb90-611" aria-hidden="true" tabindex="-1"></a><span class="in">```{r renamegtf, engine='bash'}</span></span>
<span id="cb90-612"><a href="#cb90-612" aria-hidden="true" tabindex="-1"></a><span class="in">awk 'BEGIN{OFS="\t"; count=1} {printf "%s\t.\tlncRNA\t%d\t%d\t.\t+\t.\tgene_id \"lncRNA_%03d\";\n", $1, $2, $3, count++;}' "${OUTPUT_DIR}/lncRNA.bed" \</span></span>
<span id="cb90-613"><a href="#cb90-613" aria-hidden="true" tabindex="-1"></a><span class="in">&gt; "${OUTPUT_DIR}/lncRNA.gtf"</span></span>
<span id="cb90-614"><a href="#cb90-614" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-615"><a href="#cb90-615" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-616"><a href="#cb90-616" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, engine='bash'}</span></span>
<span id="cb90-617"><a href="#cb90-617" aria-hidden="true" tabindex="-1"></a><span class="in">head "${OUTPUT_DIR}/lncRNA.gtf"</span></span>
<span id="cb90-618"><a href="#cb90-618" aria-hidden="true" tabindex="-1"></a><span class="in">wc -l "${OUTPUT_DIR}/lncRNA.gtf"</span></span>
<span id="cb90-619"><a href="#cb90-619" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-620"><a href="#cb90-620" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-621"><a href="#cb90-621" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-622"><a href="#cb90-622" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-623"><a href="#cb90-623" aria-hidden="true" tabindex="-1"></a><span class="fu"># Summary Table</span></span>
<span id="cb90-624"><a href="#cb90-624" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-625"><a href="#cb90-625" aria-hidden="true" tabindex="-1"></a><span class="in">```{r summary, engine='bash'}</span></span>
<span id="cb90-626"><a href="#cb90-626" aria-hidden="true" tabindex="-1"></a><span class="in">tf_file="${OUTPUT_DIR}/lncRNA.gtf"</span></span>
<span id="cb90-627"><a href="#cb90-627" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-628"><a href="#cb90-628" aria-hidden="true" tabindex="-1"></a><span class="in">awk '</span></span>
<span id="cb90-629"><a href="#cb90-629" aria-hidden="true" tabindex="-1"></a><span class="in">BEGIN {</span></span>
<span id="cb90-630"><a href="#cb90-630" aria-hidden="true" tabindex="-1"></a><span class="in">    total_entries = 0;</span></span>
<span id="cb90-631"><a href="#cb90-631" aria-hidden="true" tabindex="-1"></a><span class="in">    min_length = 1e9;</span></span>
<span id="cb90-632"><a href="#cb90-632" aria-hidden="true" tabindex="-1"></a><span class="in">    max_length = 0;</span></span>
<span id="cb90-633"><a href="#cb90-633" aria-hidden="true" tabindex="-1"></a><span class="in">    sum_length = 0;</span></span>
<span id="cb90-634"><a href="#cb90-634" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb90-635"><a href="#cb90-635" aria-hidden="true" tabindex="-1"></a><span class="in"># Skip comment lines</span></span>
<span id="cb90-636"><a href="#cb90-636" aria-hidden="true" tabindex="-1"></a><span class="in">/^#/ { next }</span></span>
<span id="cb90-637"><a href="#cb90-637" aria-hidden="true" tabindex="-1"></a><span class="in">{</span></span>
<span id="cb90-638"><a href="#cb90-638" aria-hidden="true" tabindex="-1"></a><span class="in">    if (NF &lt; 9) next;</span></span>
<span id="cb90-639"><a href="#cb90-639" aria-hidden="true" tabindex="-1"></a><span class="in">    total_entries++;</span></span>
<span id="cb90-640"><a href="#cb90-640" aria-hidden="true" tabindex="-1"></a><span class="in">    start = $4;</span></span>
<span id="cb90-641"><a href="#cb90-641" aria-hidden="true" tabindex="-1"></a><span class="in">    end = $5;</span></span>
<span id="cb90-642"><a href="#cb90-642" aria-hidden="true" tabindex="-1"></a><span class="in">    gene_length = end - start + 1;</span></span>
<span id="cb90-643"><a href="#cb90-643" aria-hidden="true" tabindex="-1"></a><span class="in">    if (gene_length &lt; min_length) min_length = gene_length;</span></span>
<span id="cb90-644"><a href="#cb90-644" aria-hidden="true" tabindex="-1"></a><span class="in">    if (gene_length &gt; max_length) max_length = gene_length;</span></span>
<span id="cb90-645"><a href="#cb90-645" aria-hidden="true" tabindex="-1"></a><span class="in">    sum_length += gene_length;</span></span>
<span id="cb90-646"><a href="#cb90-646" aria-hidden="true" tabindex="-1"></a><span class="in">    feature[$3]++;</span></span>
<span id="cb90-647"><a href="#cb90-647" aria-hidden="true" tabindex="-1"></a><span class="in">    chrom[$1]++;</span></span>
<span id="cb90-648"><a href="#cb90-648" aria-hidden="true" tabindex="-1"></a><span class="in">    # Use two-argument match() and then extract the gene_id manually.</span></span>
<span id="cb90-649"><a href="#cb90-649" aria-hidden="true" tabindex="-1"></a><span class="in">    if (match($9, /gene_id "[^"]+"/)) {</span></span>
<span id="cb90-650"><a href="#cb90-650" aria-hidden="true" tabindex="-1"></a><span class="in">        gene_str = substr($9, RSTART, RLENGTH);</span></span>
<span id="cb90-651"><a href="#cb90-651" aria-hidden="true" tabindex="-1"></a><span class="in">        # Remove the "gene_id " prefix and the quotes.</span></span>
<span id="cb90-652"><a href="#cb90-652" aria-hidden="true" tabindex="-1"></a><span class="in">        gsub(/gene_id "/, "", gene_str);</span></span>
<span id="cb90-653"><a href="#cb90-653" aria-hidden="true" tabindex="-1"></a><span class="in">        gsub(/"/, "", gene_str);</span></span>
<span id="cb90-654"><a href="#cb90-654" aria-hidden="true" tabindex="-1"></a><span class="in">        genes[gene_str] = 1;</span></span>
<span id="cb90-655"><a href="#cb90-655" aria-hidden="true" tabindex="-1"></a><span class="in">    }</span></span>
<span id="cb90-656"><a href="#cb90-656" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb90-657"><a href="#cb90-657" aria-hidden="true" tabindex="-1"></a><span class="in">END {</span></span>
<span id="cb90-658"><a href="#cb90-658" aria-hidden="true" tabindex="-1"></a><span class="in">    avg_length = (total_entries &gt; 0) ? sum_length / total_entries : 0;</span></span>
<span id="cb90-659"><a href="#cb90-659" aria-hidden="true" tabindex="-1"></a><span class="in">    unique_gene_count = 0;</span></span>
<span id="cb90-660"><a href="#cb90-660" aria-hidden="true" tabindex="-1"></a><span class="in">    for (g in genes)</span></span>
<span id="cb90-661"><a href="#cb90-661" aria-hidden="true" tabindex="-1"></a><span class="in">        unique_gene_count++;</span></span>
<span id="cb90-662"><a href="#cb90-662" aria-hidden="true" tabindex="-1"></a><span class="in">    print "Basic GTF File Statistics:";</span></span>
<span id="cb90-663"><a href="#cb90-663" aria-hidden="true" tabindex="-1"></a><span class="in">    print "--------------------------";</span></span>
<span id="cb90-664"><a href="#cb90-664" aria-hidden="true" tabindex="-1"></a><span class="in">    print "Total entries:      " total_entries;</span></span>
<span id="cb90-665"><a href="#cb90-665" aria-hidden="true" tabindex="-1"></a><span class="in">    print "Unique genes:       " unique_gene_count;</span></span>
<span id="cb90-666"><a href="#cb90-666" aria-hidden="true" tabindex="-1"></a><span class="in">    print "Min gene length:    " min_length;</span></span>
<span id="cb90-667"><a href="#cb90-667" aria-hidden="true" tabindex="-1"></a><span class="in">    print "Max gene length:    " max_length;</span></span>
<span id="cb90-668"><a href="#cb90-668" aria-hidden="true" tabindex="-1"></a><span class="in">    printf("Average gene length: %.2f\n", avg_length);</span></span>
<span id="cb90-669"><a href="#cb90-669" aria-hidden="true" tabindex="-1"></a><span class="in">    print "\nFeature counts:";</span></span>
<span id="cb90-670"><a href="#cb90-670" aria-hidden="true" tabindex="-1"></a><span class="in">    for (f in feature) {</span></span>
<span id="cb90-671"><a href="#cb90-671" aria-hidden="true" tabindex="-1"></a><span class="in">        print "  " f ": " feature[f];</span></span>
<span id="cb90-672"><a href="#cb90-672" aria-hidden="true" tabindex="-1"></a><span class="in">    }</span></span>
<span id="cb90-673"><a href="#cb90-673" aria-hidden="true" tabindex="-1"></a><span class="in">    print "\nChromosome counts:";</span></span>
<span id="cb90-674"><a href="#cb90-674" aria-hidden="true" tabindex="-1"></a><span class="in">    for (c in chrom) {</span></span>
<span id="cb90-675"><a href="#cb90-675" aria-hidden="true" tabindex="-1"></a><span class="in">        print "  " c ": " chrom[c];</span></span>
<span id="cb90-676"><a href="#cb90-676" aria-hidden="true" tabindex="-1"></a><span class="in">    }</span></span>
<span id="cb90-677"><a href="#cb90-677" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb90-678"><a href="#cb90-678" aria-hidden="true" tabindex="-1"></a><span class="in">' "$tf_file"</span></span>
<span id="cb90-679"><a href="#cb90-679" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-680"><a href="#cb90-680" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-681"><a href="#cb90-681" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-682"><a href="#cb90-682" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb90-683"><a href="#cb90-683" aria-hidden="true" tabindex="-1"></a><span class="in">Basic GTF File Statistics:</span></span>
<span id="cb90-684"><a href="#cb90-684" aria-hidden="true" tabindex="-1"></a><span class="in">--------------------------</span></span>
<span id="cb90-685"><a href="#cb90-685" aria-hidden="true" tabindex="-1"></a><span class="in">Total entries:      16153</span></span>
<span id="cb90-686"><a href="#cb90-686" aria-hidden="true" tabindex="-1"></a><span class="in">Unique genes:       0</span></span>
<span id="cb90-687"><a href="#cb90-687" aria-hidden="true" tabindex="-1"></a><span class="in">Min gene length:    203</span></span>
<span id="cb90-688"><a href="#cb90-688" aria-hidden="true" tabindex="-1"></a><span class="in">Max gene length:    227016</span></span>
<span id="cb90-689"><a href="#cb90-689" aria-hidden="true" tabindex="-1"></a><span class="in">Average gene length: 3125.62</span></span>
<span id="cb90-690"><a href="#cb90-690" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-691"><a href="#cb90-691" aria-hidden="true" tabindex="-1"></a><span class="in">Feature counts:</span></span>
<span id="cb90-692"><a href="#cb90-692" aria-hidden="true" tabindex="-1"></a><span class="in">  lncRNA: 16153</span></span>
<span id="cb90-693"><a href="#cb90-693" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-694"><a href="#cb90-694" aria-hidden="true" tabindex="-1"></a><span class="in">Chromosome counts:</span></span>
<span id="cb90-695"><a href="#cb90-695" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000447: 7</span></span>
<span id="cb90-696"><a href="#cb90-696" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000094: 2</span></span>
<span id="cb90-697"><a href="#cb90-697" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000041: 71</span></span>
<span id="cb90-698"><a href="#cb90-698" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000343: 7</span></span>
<span id="cb90-699"><a href="#cb90-699" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000004: 3</span></span>
<span id="cb90-700"><a href="#cb90-700" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000011: 362</span></span>
<span id="cb90-701"><a href="#cb90-701" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000812: 1</span></span>
<span id="cb90-702"><a href="#cb90-702" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000081: 2</span></span>
<span id="cb90-703"><a href="#cb90-703" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000032: 132</span></span>
<span id="cb90-704"><a href="#cb90-704" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xpSc0001344: 1</span></span>
<span id="cb90-705"><a href="#cb90-705" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xpSc0001331: 1</span></span>
<span id="cb90-706"><a href="#cb90-706" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0001179: 1</span></span>
<span id="cb90-707"><a href="#cb90-707" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000284: 2</span></span>
<span id="cb90-708"><a href="#cb90-708" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000692: 3</span></span>
<span id="cb90-709"><a href="#cb90-709" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000029: 2</span></span>
<span id="cb90-710"><a href="#cb90-710" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000052: 3</span></span>
<span id="cb90-711"><a href="#cb90-711" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000145: 2</span></span>
<span id="cb90-712"><a href="#cb90-712" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000046: 28</span></span>
<span id="cb90-713"><a href="#cb90-713" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000948: 1</span></span>
<span id="cb90-714"><a href="#cb90-714" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000025: 274</span></span>
<span id="cb90-715"><a href="#cb90-715" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000477: 2</span></span>
<span id="cb90-716"><a href="#cb90-716" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000066: 1</span></span>
<span id="cb90-717"><a href="#cb90-717" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000039: 96</span></span>
<span id="cb90-718"><a href="#cb90-718" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xpSc0001280: 5</span></span>
<span id="cb90-719"><a href="#cb90-719" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0001170: 1</span></span>
<span id="cb90-720"><a href="#cb90-720" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000212: 7</span></span>
<span id="cb90-721"><a href="#cb90-721" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000007: 36</span></span>
<span id="cb90-722"><a href="#cb90-722" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000012: 457</span></span>
<span id="cb90-723"><a href="#cb90-723" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000974: 1</span></span>
<span id="cb90-724"><a href="#cb90-724" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000817: 1</span></span>
<span id="cb90-725"><a href="#cb90-725" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000570: 1</span></span>
<span id="cb90-726"><a href="#cb90-726" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000479: 2</span></span>
<span id="cb90-727"><a href="#cb90-727" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000237: 2</span></span>
<span id="cb90-728"><a href="#cb90-728" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000068: 2</span></span>
<span id="cb90-729"><a href="#cb90-729" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000037: 101</span></span>
<span id="cb90-730"><a href="#cb90-730" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000003: 803</span></span>
<span id="cb90-731"><a href="#cb90-731" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000428: 4</span></span>
<span id="cb90-732"><a href="#cb90-732" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000059: 6</span></span>
<span id="cb90-733"><a href="#cb90-733" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000057: 5</span></span>
<span id="cb90-734"><a href="#cb90-734" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000146: 5</span></span>
<span id="cb90-735"><a href="#cb90-735" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000045: 52</span></span>
<span id="cb90-736"><a href="#cb90-736" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000009: 729</span></span>
<span id="cb90-737"><a href="#cb90-737" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000952: 1</span></span>
<span id="cb90-738"><a href="#cb90-738" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000835: 2</span></span>
<span id="cb90-739"><a href="#cb90-739" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000000: 71</span></span>
<span id="cb90-740"><a href="#cb90-740" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000015: 328</span></span>
<span id="cb90-741"><a href="#cb90-741" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000716: 1</span></span>
<span id="cb90-742"><a href="#cb90-742" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000080: 2</span></span>
<span id="cb90-743"><a href="#cb90-743" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000616: 1</span></span>
<span id="cb90-744"><a href="#cb90-744" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000034: 118</span></span>
<span id="cb90-745"><a href="#cb90-745" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xpSc0001290: 1</span></span>
<span id="cb90-746"><a href="#cb90-746" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000527: 2</span></span>
<span id="cb90-747"><a href="#cb90-747" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000206: 6</span></span>
<span id="cb90-748"><a href="#cb90-748" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000006: 561</span></span>
<span id="cb90-749"><a href="#cb90-749" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xpSc0001273: 29</span></span>
<span id="cb90-750"><a href="#cb90-750" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000540: 4</span></span>
<span id="cb90-751"><a href="#cb90-751" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000058: 5</span></span>
<span id="cb90-752"><a href="#cb90-752" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000021: 290</span></span>
<span id="cb90-753"><a href="#cb90-753" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000859: 2</span></span>
<span id="cb90-754"><a href="#cb90-754" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000040: 123</span></span>
<span id="cb90-755"><a href="#cb90-755" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000890: 1</span></span>
<span id="cb90-756"><a href="#cb90-756" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000836: 1</span></span>
<span id="cb90-757"><a href="#cb90-757" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000555: 2</span></span>
<span id="cb90-758"><a href="#cb90-758" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000003: 15</span></span>
<span id="cb90-759"><a href="#cb90-759" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000016: 448</span></span>
<span id="cb90-760"><a href="#cb90-760" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000765: 5</span></span>
<span id="cb90-761"><a href="#cb90-761" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000033: 112</span></span>
<span id="cb90-762"><a href="#cb90-762" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000018: 348</span></span>
<span id="cb90-763"><a href="#cb90-763" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000415: 2</span></span>
<span id="cb90-764"><a href="#cb90-764" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000053: 14</span></span>
<span id="cb90-765"><a href="#cb90-765" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000436: 4</span></span>
<span id="cb90-766"><a href="#cb90-766" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000205: 13</span></span>
<span id="cb90-767"><a href="#cb90-767" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000070: 2</span></span>
<span id="cb90-768"><a href="#cb90-768" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000049: 4</span></span>
<span id="cb90-769"><a href="#cb90-769" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000005: 551</span></span>
<span id="cb90-770"><a href="#cb90-770" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xpSc0001276: 1</span></span>
<span id="cb90-771"><a href="#cb90-771" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000024: 245</span></span>
<span id="cb90-772"><a href="#cb90-772" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000065: 2</span></span>
<span id="cb90-773"><a href="#cb90-773" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xpSc0001281: 4</span></span>
<span id="cb90-774"><a href="#cb90-774" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000006: 13</span></span>
<span id="cb90-775"><a href="#cb90-775" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000013: 555</span></span>
<span id="cb90-776"><a href="#cb90-776" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0001237: 2</span></span>
<span id="cb90-777"><a href="#cb90-777" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000612: 2</span></span>
<span id="cb90-778"><a href="#cb90-778" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000168: 3</span></span>
<span id="cb90-779"><a href="#cb90-779" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000083: 2</span></span>
<span id="cb90-780"><a href="#cb90-780" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000030: 198</span></span>
<span id="cb90-781"><a href="#cb90-781" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000002: 725</span></span>
<span id="cb90-782"><a href="#cb90-782" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000175: 3</span></span>
<span id="cb90-783"><a href="#cb90-783" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000054: 1</span></span>
<span id="cb90-784"><a href="#cb90-784" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000565: 2</span></span>
<span id="cb90-785"><a href="#cb90-785" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000044: 51</span></span>
<span id="cb90-786"><a href="#cb90-786" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000008: 538</span></span>
<span id="cb90-787"><a href="#cb90-787" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000027: 291</span></span>
<span id="cb90-788"><a href="#cb90-788" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000840: 1</span></span>
<span id="cb90-789"><a href="#cb90-789" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000596: 2</span></span>
<span id="cb90-790"><a href="#cb90-790" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000060: 1</span></span>
<span id="cb90-791"><a href="#cb90-791" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000763: 1</span></span>
<span id="cb90-792"><a href="#cb90-792" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000035: 227</span></span>
<span id="cb90-793"><a href="#cb90-793" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000968: 2</span></span>
<span id="cb90-794"><a href="#cb90-794" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000001: 883</span></span>
<span id="cb90-795"><a href="#cb90-795" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xpSc0001355: 1</span></span>
<span id="cb90-796"><a href="#cb90-796" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000699: 2</span></span>
<span id="cb90-797"><a href="#cb90-797" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000655: 4</span></span>
<span id="cb90-798"><a href="#cb90-798" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000426: 7</span></span>
<span id="cb90-799"><a href="#cb90-799" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000020: 294</span></span>
<span id="cb90-800"><a href="#cb90-800" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000621: 2</span></span>
<span id="cb90-801"><a href="#cb90-801" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000445: 2</span></span>
<span id="cb90-802"><a href="#cb90-802" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000074: 1</span></span>
<span id="cb90-803"><a href="#cb90-803" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000043: 60</span></span>
<span id="cb90-804"><a href="#cb90-804" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000837: 1</span></span>
<span id="cb90-805"><a href="#cb90-805" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000288: 5</span></span>
<span id="cb90-806"><a href="#cb90-806" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000217: 5</span></span>
<span id="cb90-807"><a href="#cb90-807" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000002: 34</span></span>
<span id="cb90-808"><a href="#cb90-808" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000017: 382</span></span>
<span id="cb90-809"><a href="#cb90-809" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000019: 291</span></span>
<span id="cb90-810"><a href="#cb90-810" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000012: 7</span></span>
<span id="cb90-811"><a href="#cb90-811" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000029: 178</span></span>
<span id="cb90-812"><a href="#cb90-812" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000886: 3</span></span>
<span id="cb90-813"><a href="#cb90-813" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000868: 3</span></span>
<span id="cb90-814"><a href="#cb90-814" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000824: 8</span></span>
<span id="cb90-815"><a href="#cb90-815" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000583: 1</span></span>
<span id="cb90-816"><a href="#cb90-816" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000071: 2</span></span>
<span id="cb90-817"><a href="#cb90-817" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000004: 652</span></span>
<span id="cb90-818"><a href="#cb90-818" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000704: 4</span></span>
<span id="cb90-819"><a href="#cb90-819" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000469: 3</span></span>
<span id="cb90-820"><a href="#cb90-820" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000223: 2</span></span>
<span id="cb90-821"><a href="#cb90-821" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000195: 1</span></span>
<span id="cb90-822"><a href="#cb90-822" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000021: 2</span></span>
<span id="cb90-823"><a href="#cb90-823" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000023: 213</span></span>
<span id="cb90-824"><a href="#cb90-824" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000264: 3</span></span>
<span id="cb90-825"><a href="#cb90-825" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000010: 615</span></span>
<span id="cb90-826"><a href="#cb90-826" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0001122: 3</span></span>
<span id="cb90-827"><a href="#cb90-827" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000811: 1</span></span>
<span id="cb90-828"><a href="#cb90-828" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000031: 185</span></span>
<span id="cb90-829"><a href="#cb90-829" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000788: 5</span></span>
<span id="cb90-830"><a href="#cb90-830" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000875: 1</span></span>
<span id="cb90-831"><a href="#cb90-831" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000488: 7</span></span>
<span id="cb90-832"><a href="#cb90-832" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000017: 2</span></span>
<span id="cb90-833"><a href="#cb90-833" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000055: 12</span></span>
<span id="cb90-834"><a href="#cb90-834" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0001011: 2</span></span>
<span id="cb90-835"><a href="#cb90-835" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000092: 6</span></span>
<span id="cb90-836"><a href="#cb90-836" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000047: 22</span></span>
<span id="cb90-837"><a href="#cb90-837" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000482: 4</span></span>
<span id="cb90-838"><a href="#cb90-838" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000132: 4</span></span>
<span id="cb90-839"><a href="#cb90-839" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000024: 2</span></span>
<span id="cb90-840"><a href="#cb90-840" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000026: 271</span></span>
<span id="cb90-841"><a href="#cb90-841" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000995: 1</span></span>
<span id="cb90-842"><a href="#cb90-842" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000067: 2</span></span>
<span id="cb90-843"><a href="#cb90-843" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000038: 93</span></span>
<span id="cb90-844"><a href="#cb90-844" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000642: 1</span></span>
<span id="cb90-845"><a href="#cb90-845" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000376: 1</span></span>
<span id="cb90-846"><a href="#cb90-846" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000082: 3</span></span>
<span id="cb90-847"><a href="#cb90-847" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000975: 3</span></span>
<span id="cb90-848"><a href="#cb90-848" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000760: 2</span></span>
<span id="cb90-849"><a href="#cb90-849" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000069: 1</span></span>
<span id="cb90-850"><a href="#cb90-850" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000036: 124</span></span>
<span id="cb90-851"><a href="#cb90-851" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000275: 1</span></span>
<span id="cb90-852"><a href="#cb90-852" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000000: 922</span></span>
<span id="cb90-853"><a href="#cb90-853" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000199: 2</span></span>
<span id="cb90-854"><a href="#cb90-854" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000058: 1</span></span>
<span id="cb90-855"><a href="#cb90-855" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000014: 1</span></span>
<span id="cb90-856"><a href="#cb90-856" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000725: 5</span></span>
<span id="cb90-857"><a href="#cb90-857" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000042: 68</span></span>
<span id="cb90-858"><a href="#cb90-858" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000892: 1</span></span>
<span id="cb90-859"><a href="#cb90-859" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000001: 67</span></span>
<span id="cb90-860"><a href="#cb90-860" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000014: 378</span></span>
<span id="cb90-861"><a href="#cb90-861" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000262: 3</span></span>
<span id="cb90-862"><a href="#cb90-862" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000228: 4</span></span>
<span id="cb90-863"><a href="#cb90-863" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000051: 13</span></span>
<span id="cb90-864"><a href="#cb90-864" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000028: 208</span></span>
<span id="cb90-865"><a href="#cb90-865" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000885: 5</span></span>
<span id="cb90-866"><a href="#cb90-866" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000272: 5</span></span>
<span id="cb90-867"><a href="#cb90-867" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000007: 643</span></span>
<span id="cb90-868"><a href="#cb90-868" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___xfSc0000705: 1</span></span>
<span id="cb90-869"><a href="#cb90-869" aria-hidden="true" tabindex="-1"></a><span class="in">  Pocillopora_meandrina_HIv1___Sc0000022: 231</span></span>
<span id="cb90-870"><a href="#cb90-870" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




</body></html>